From b22afbc35fc288f1baeab7acfe11aa5a770ad144 Mon Sep 17 00:00:00 2001
From: Andrius da Costa Ribas <andriusmao@gmail.com>
Date: Fri, 16 Mar 2012 00:34:49 -0300
Subject: [PATCH 1/2] INCLUDE(CheckFunctionExists) before using check_function_exists macro.

---
 kexi/CMakeLists.txt |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/kexi/CMakeLists.txt b/kexi/CMakeLists.txt
index af9f3f0..a724a34 100644
--- a/kexi/CMakeLists.txt
+++ b/kexi/CMakeLists.txt
@@ -1,5 +1,7 @@
 project(kexi)
 
+INCLUDE(CheckFunctionExists)
+
 SET(CMAKE_ALLOW_LOOSE_LOOP_CONSTRUCTS true)
 
 option(KEXI_MOBILE "compile a mobile Kexi" OFF)
-- 
1.7.4.msysgit.0

