diff --git a/CMakeLists.txt b/CMakeLists.txt
index c4181f9..613bbf9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -294,10 +294,16 @@ if(STATIC AND CMAKE_COMPILER_IS_GNUCXX)
 endif(STATIC AND CMAKE_COMPILER_IS_GNUCXX)
 
 if(WIN32)
-  link_libraries(imm32 winmm dbghelp Secur32)  # missing by default :/
-
+  link_libraries(imm32 winmm Secur32)  # missing by default :/
+  if(NOT WITH_KDE) 
+     link_libraries(dbghelp)
+  endif(NOT WITH_KDE)
   if(HAVE_SSL)
+    if( WITH_KDE )
+        find_package(OpenSSL)
+    else( WITH_KDE )
      link_libraries(${OPENSSL_LIBRARIES} libeay32MD)
+   endif( WITH_KDE )
   endif(HAVE_SSL)
 endif(WIN32)
 
diff --git a/src/common/CMakeLists.txt b/src/common/CMakeLists.txt
index f8337cb..f691d8a 100644
--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -28,7 +28,7 @@ set(SOURCES
     util.cpp)
 
 if(CMAKE_HOST_WIN32)
-    set(SOURCES ${SOURCES} logbacktrace_win.cpp)
+    #set(SOURCES ${SOURCES} logbacktrace_win.cpp)
 endif(CMAKE_HOST_WIN32)
 if(CMAKE_HOST_UNIX)
     set(SOURCES ${SOURCES} logbacktrace_unix.cpp)
diff --git a/src/common/identity.cpp b/src/common/identity.cpp
index 275afcc..d2b483d 100644
--- a/src/common/identity.cpp
+++ b/src/common/identity.cpp
@@ -102,7 +102,7 @@ QString Identity::defaultNick() {
   DWORD  bufCharCount = 128;
   //if(GetUserNameEx(/* NameSamCompatible */ 1, infoBuf, &bufCharCount))
   if(GetUserNameEx(NameSamCompatible, infoBuf, &bufCharCount)) {
-    QString nickName(infoBuf);
+    QString nickName( QString::fromUtf16( reinterpret_cast<ushort*>(infoBuf)));
     int lastBs = nickName.lastIndexOf('\\');
     if(lastBs != -1) {
       nickName = nickName.mid(lastBs + 1);
@@ -138,7 +138,7 @@ QString Identity::defaultRealName() {
   TCHAR  infoBuf[128];
   DWORD  bufCharCount = 128;
   if(GetUserName(infoBuf, &bufCharCount))
-    return QString(infoBuf);
+    return QString::fromUtf16( reinterpret_cast<ushort*>( infoBuf));
   else
     return generalDefault;
 #else
diff --git a/src/common/quassel.cpp b/src/common/quassel.cpp
index e6e2072..e8c66b7 100644
--- a/src/common/quassel.cpp
+++ b/src/common/quassel.cpp
@@ -224,7 +224,9 @@ void Quassel::handleSignal(int sig) {
 #ifndef Q_OS_WIN32
   case SIGBUS:
 #endif
+#ifndef Q_OS_WIN32
     logBacktrace(coreDumpFileName());
+#endif
     exit(EXIT_FAILURE);
     break;
   default:
