diff -Nru -x '*~' nmealib.orig/CMakeLists.txt nmealib/CMakeLists.txt
--- nmealib.orig/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ nmealib/CMakeLists.txt	2012-04-24 09:48:35.291101700 +0200
@@ -0,0 +1,39 @@
+PROJECT(nmealib)
+
+cmake_minimum_required(VERSION 2.8)
+
+set (NMEALIB_SOURCES
+    src/context.c
+    src/generate.c
+    src/generator.c
+    src/gmath.c
+    src/info.c
+    src/parse.c
+    src/parser.c
+    src/sentence.c
+    src/time.c
+    src/tok.c
+)
+
+set(NMEALIB_HEADERS
+    include/nmea/config.h
+    include/nmea/context.h
+    include/nmea/generate.h
+    include/nmea/generator.h
+    include/nmea/gmath.h
+    include/nmea/info.h
+    include/nmea/nmea.h
+    include/nmea/parse.h
+    include/nmea/parser.h
+    include/nmea/sentence.h
+    include/nmea/time.h
+    include/nmea/tok.h
+    include/nmea/units.h
+)
+
+include_directories(src include)
+
+add_library(nmealib ${NMEALIB_SOURCES})
+target_link_libraries(nmealib)
+install(TARGETS nmealib RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+install(FILES ${NMEALIB_HEADERS} DESTINATION include/nmea)
diff -Nru -x '*~' nmealib.orig/include/nmea/units.h nmealib/include/nmea/units.h
--- nmealib.orig/include/nmea/units.h	2012-04-26 10:24:37.516368600 +0200
+++ nmealib/include/nmea/units.h	2012-04-25 08:14:09.498834200 +0200
@@ -20,6 +20,7 @@
 #define NMEA_TUD_YARDS      (1.0936)        /**< Yeards, meter * NMEA_TUD_YARDS = yard */
 #define NMEA_TUD_KNOTS      (1.852)         /**< Knots, kilometer / NMEA_TUD_KNOTS = knot */
 #define NMEA_TUD_MILES      (1.609)         /**< Miles, kilometer / NMEA_TUD_MILES = mile */
+#define NMEA_TUD_FEED       (1/0.3048)      /**< Feet, meter / NMEA_TUD_FEED = feet */
 
 /*
  * Speed units
diff -Nru -x '*~' nmealib.orig/src/parse.c nmealib/src/parse.c
--- nmealib.orig/src/parse.c	2012-04-26 10:24:37.612374000 +0200
+++ nmealib/src/parse.c	2012-04-25 08:06:04.196076500 +0200
@@ -382,7 +382,7 @@
     info->utc.hsec = pack->utc.hsec;
     info->sig = pack->sig;
     info->HDOP = pack->HDOP;
-    info->elv = pack->elv;
+    info->elv = (pack->elv_units == 'M')?pack->elv:pack->elv / NMEA_TUD_FEED;
     info->lat = ((pack->ns == 'N')?pack->lat:-(pack->lat));
     info->lon = ((pack->ew == 'E')?pack->lon:-(pack->lon));
     info->smask |= GPGGA;
