diff -Nru -x '*~' pairs-4.10.2.orig/pairseditor/CMakeLists.txt pairs-4.10.2/pairseditor/CMakeLists.txt
--- pairs-4.10.2.orig/pairseditor/CMakeLists.txt	2013-03-01 07:50:00.149201000 +0100
+++ pairs-4.10.2/pairseditor/CMakeLists.txt	2013-06-02 23:14:56.570873300 +0200
@@ -5,7 +5,7 @@
 
 kde4_add_ui_files( pairseditor_SOURCES mainwindowview.ui )
 
-kde4_add_app_icon(pairs_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../icons/hi*-apps-pairseditor*")
+kde4_add_app_icon(pairs_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../icons/hi*-apps-pairseditor.png")
 
 kde4_add_executable( pairseditor ${pairseditor_SOURCES} )
 
