Index: libgcrypt-1.5.3/random/rndw32.c
===================================================================
--- libgcrypt-1.5.3.orig/random/rndw32.c
+++ libgcrypt-1.5.3/random/rndw32.c
@@ -829,10 +829,10 @@ _gcry_rndw32_gather_random_fast (void (*
      and milliseconds since Windows was started.  */
 
   {
-    byte buffer[20*sizeof(ulong)], *bufptr;
+    byte buffer[20*sizeof(intptr_t)], *bufptr;
 
     bufptr = buffer;
-#define ADD(f)  do { ulong along = (ulong)(f);                  \
+#define ADD(f)  do { intptr_t along = (intptr_t)(f);            \
                      memcpy (bufptr, &along, sizeof (along) );  \
                      bufptr += sizeof (along);                  \
                    } while (0)
@@ -888,7 +888,7 @@ _gcry_rndw32_gather_random_fast (void (*
   {
     HANDLE handle;
     FILETIME creationTime, exitTime, kernelTime, userTime;
-    DWORD minimumWorkingSetSize, maximumWorkingSetSize;
+    ULONG_PTR minimumWorkingSetSize, maximumWorkingSetSize;
 
     handle = GetCurrentThread ();
     GetThreadTimes (handle, &creationTime, &exitTime,
Index: libgcrypt-1.5.3/src/ath.c
===================================================================
--- libgcrypt-1.5.3.orig/src/ath.c
+++ libgcrypt-1.5.3/src/ath.c
@@ -41,7 +41,11 @@
 #else
 #include <winsock2.h>
 #define ssize_t SSIZE_T
+#ifndef _WIN64
 typedef int pid_t;
+#else
+typedef __int64 pid_t;
+#endif
 #endif
 #include <errno.h>
 
Index: libgcrypt-1.5.3/src/secmem.c
===================================================================
--- libgcrypt-1.5.3.orig/src/secmem.c
+++ libgcrypt-1.5.3/src/secmem.c
@@ -31,7 +31,11 @@
 #ifdef _WIN32
 #include <winsock2.h>
 #define ssize_t SSIZE_T
+#ifndef _WIN64
 typedef int pid_t;
+#else
+typedef __int64 pid_t;
+#endif
 #endif
 
 #if defined(HAVE_MLOCK) || defined(HAVE_MMAP)
