diff -Nru -x '*~' libgcrypt-1.5.3.orig/random/rndw32.c libgcrypt-1.5.3/random/rndw32.c
--- libgcrypt-1.5.3.orig/random/rndw32.c	2014-05-02 01:02:35.796423700 +0200
+++ libgcrypt-1.5.3/random/rndw32.c	2014-05-02 01:00:36.148606200 +0200
@@ -829,10 +829,10 @@
      and milliseconds since Windows was started.  */
 
   {
-    byte buffer[20*sizeof(ulong)], *bufptr;
+    byte buffer[20*sizeof(intptr_t)], *bufptr;
 
     bufptr = buffer;
-#define ADD(f)  do { ulong along = (ulong)(f);                  \
+#define ADD(f)  do { intptr_t along = (intptr_t)(f);            \
                      memcpy (bufptr, &along, sizeof (along) );  \
                      bufptr += sizeof (along);                  \
                    } while (0)
@@ -888,7 +888,7 @@
   {
     HANDLE handle;
     FILETIME creationTime, exitTime, kernelTime, userTime;
-    DWORD minimumWorkingSetSize, maximumWorkingSetSize;
+    ULONG_PTR minimumWorkingSetSize, maximumWorkingSetSize;
 
     handle = GetCurrentThread ();
     GetThreadTimes (handle, &creationTime, &exitTime,
diff -Nru -x '*~' libgcrypt-1.5.3.orig/src/ath.c libgcrypt-1.5.3/src/ath.c
--- libgcrypt-1.5.3.orig/src/ath.c	2014-05-02 01:02:35.797423800 +0200
+++ libgcrypt-1.5.3/src/ath.c	2014-05-02 01:00:36.150606300 +0200
@@ -40,7 +40,7 @@
 #include <sys/wait.h>
 #else
 #include <winsock2.h>
-#define ssize_t SSIZE_T
+typedef SSIZE_T ssize_t;
 typedef int pid_t;
 #endif
 #include <errno.h>
diff -Nru -x '*~' libgcrypt-1.5.3.orig/src/gcrypt.h libgcrypt-1.5.3/src/gcrypt.h
--- libgcrypt-1.5.3.orig/src/gcrypt.h	2013-07-25 11:23:38.000000000 +0200
+++ libgcrypt-1.5.3/src/gcrypt.h	2014-05-02 01:02:21.396005300 +0200
@@ -37,7 +37,7 @@
 # include <ws2tcpip.h>
 # include <time.h>
 # ifndef __GNUC__
-  typedef long ssize_t;
+  typedef SSIZE_T ssize_t;
   typedef int  pid_t;
 # endif /*!__GNUC__*/
 #else
diff -Nru -x '*~' libgcrypt-1.5.3.orig/src/gcrypt.h.in libgcrypt-1.5.3/src/gcrypt.h.in
--- libgcrypt-1.5.3.orig/src/gcrypt.h.in	2013-07-25 11:10:04.000000000 +0200
+++ libgcrypt-1.5.3/src/gcrypt.h.in	2014-05-02 01:00:36.154606500 +0200
@@ -37,7 +37,7 @@
 # include <ws2tcpip.h>
 # include <time.h>
 # ifndef __GNUC__
-  typedef long ssize_t;
+  typedef SSIZE_T ssize_t;
   typedef int  pid_t;
 # endif /*!__GNUC__*/
 #else
diff -Nru -x '*~' libgcrypt-1.5.3.orig/src/secmem.c libgcrypt-1.5.3/src/secmem.c
--- libgcrypt-1.5.3.orig/src/secmem.c	2014-05-02 01:02:35.807424400 +0200
+++ libgcrypt-1.5.3/src/secmem.c	2014-05-02 01:00:36.156606700 +0200
@@ -30,7 +30,7 @@
 #include <stddef.h>
 #ifdef _WIN32
 #include <winsock2.h>
-#define ssize_t SSIZE_T
+typedef SSIZE_T ssize_t;
 typedef int pid_t;
 #endif
 
