diff -Nru libgpg-error-1.7.orig/lang/Makefile.in libgpg-error-1.7/lang/Makefile.in
--- libgpg-error-1.7.orig/lang/Makefile.in	2008-11-26 14:34:01.000000000 +0100
+++ libgpg-error-1.7/lang/Makefile.in	2009-10-17 03:43:09.187349000 +0200
@@ -212,7 +212,6 @@
 target_alias = @target_alias@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = cl
 EXTRA_DIST = README
 all: all-recursive
 
