diff -Nrub tiff-4.0.7/CMakeLists.txt.old tiff-4.0.7/CMakeLists.txt
--- tiff-4.0.7/CMakeLists.txt.old       Fri Dec  2 11:57:44 2016
+++ tiff-4.0.7/CMakeLists.txt   Fri Dec  2 11:58:09 2016
@@ -467,12 +467,12 @@
               HOST_BIG_ENDIAN HAVE_IEEEFP)

 # Large file support
-if (UNIX)
+#if (UNIX)
   # This might not catch every possibility catered for by
   # AC_SYS_LARGEFILE.
   add_definitions(-D_FILE_OFFSET_BITS=64)
   set(FILE_OFFSET_BITS 64)
-endif()
+#endif()

 # Documentation install directory (default to cmake project docdir)
 set(LIBTIFF_DOCDIR "${CMAKE_INSTALL_FULL_DOCDIR}")
