diff --git a/.qmake.cache b/.qmake.cache
index 1e2d334..571d13e 100755
--- a/.qmake.cache
+++ b/.qmake.cache
@@ -11,6 +11,8 @@ MOC_DIR = $$BUILD_DIR
 UI_DIR = $$BUILD_DIR
 RCC_DIR = $$BUILD_DIR
 INCLUDEPATH += $$ROOT_DIR/_include
+INCLUDEPATH += $$(PSDKDIR)/Include/ATL
+LIBS += -L$$(KDEROOT)/lib libfftw3f-3.lib libsamplerate-0.lib shell32.lib
 
 win32:DEFINES += _CRT_SECURE_NO_WARNINGS WIN32_LEAN_AND_MEAN
 
diff --git a/src/ws/win/ComSetup.h b/src/ws/win/ComSetup.h
index fc3816b..4e6eb08 100644
--- a/src/ws/win/ComSetup.h
+++ b/src/ws/win/ComSetup.h
@@ -24,6 +24,7 @@
 
 #include <objbase.h>
 #include <atlbase.h>
+extern CComModule _Module;
 #include <atlcom.h>
 
 
diff --git a/src/ws/win/Pac.cpp b/src/ws/win/Pac.cpp
index 3e3d72b..2aa2339 100644
--- a/src/ws/win/Pac.cpp
+++ b/src/ws/win/Pac.cpp
@@ -86,7 +86,7 @@ lastfm::Pac::resolve(const QNetworkRequest &request, const wchar_t* pacUrl)
     if (!m_hSession)
     {
         QByteArray user_agent = request.rawHeader("user-agent");
-        m_hSession = WinHttpOpen(CA2W(user_agent), WINHTTP_ACCESS_TYPE_NO_PROXY, 0, 0, WINHTTP_FLAG_ASYNC);
+        m_hSession = WinHttpOpen(QString(user_agent).utf16(), WINHTTP_ACCESS_TYPE_NO_PROXY, 0, 0, WINHTTP_FLAG_ASYNC);
     }
     if (m_hSession)
     {
