diff -Nru mpir-2.2.1.orig/configure mpir-2.2.1/configure
--- mpir-2.2.1.orig/configure	2010-12-18 02:16:35.000000000 +0200
+++ mpir-2.2.1/configure	2011-01-22 23:07:15.916293600 +0200
@@ -10055,7 +10055,7 @@
 #define __GMP_BITS_PER_MP_LIMB 123 /* dummy for GMP_NUMB_BITS etc */
 #define GMP_LIMB_BITS 123
 $DEFN_LONG_LONG_LIMB
-#include "$srcdir/gmp-h.in"
+#include "../mpir-2.2.1/gmp-h.in"
 
 #if ! __GMP_HAVE_PROTOTYPES
 die die die
@@ -24724,7 +24724,7 @@
 #define __GMP_BITS_PER_MP_LIMB 123 /* dummy for GMP_NUMB_BITS etc */
 #define GMP_LIMB_BITS 123
 $DEFN_LONG_LONG_LIMB
-#include "$srcdir/gmp-h.in"
+#include "../mpir-2.2.1/gmp-h.in"
 
 #ifndef __GMP_EXTERN_INLINE
 die die die
@@ -24935,9 +24935,9 @@
 #define __GMP_BITS_PER_MP_LIMB 123 /* dummy for GMP_NUMB_BITS etc */
 #define GMP_LIMB_BITS 123
 $DEFN_LONG_LONG_LIMB
-#include "$srcdir/gmp-h.in"
+#include "../mpir-2.2.1/gmp-h.in"
 
-#include "$srcdir/gmp-impl.h"
+#include "../mpir-2.2.1/gmp-impl.h"
 
 int
 main ()
@@ -25033,7 +25033,7 @@
 #define __GMP_BITS_PER_MP_LIMB 123 /* dummy for GMP_NUMB_BITS etc */
 #define GMP_LIMB_BITS 123
 $DEFN_LONG_LONG_LIMB
-#include "$srcdir/gmp-h.in"
+#include "../mpir-2.2.1/gmp-h.in"
 
 #if ! _GMP_H_HAVE_FILE
 die die die
@@ -27955,7 +27955,7 @@
 #define __GMP_BITS_PER_MP_LIMB 123 /* dummy for GMP_NUMB_BITS etc */
 #define GMP_LIMB_BITS 123
 $DEFN_LONG_LONG_LIMB
-#include \"$srcdir/gmp-h.in\"
+#include \"../mpir-2.2.1/gmp-h.in\"
 
 "; then :
 
