diff -Nru kmymoney-4.5.1.orig/kmymoney/reports/CMakeLists.txt kmymoney-4.5.1/kmymoney/reports/CMakeLists.txt
--- kmymoney-4.5.1.orig/kmymoney/reports/CMakeLists.txt	2010-11-14 15:28:29.000000000 +0100
+++ kmymoney-4.5.1/kmymoney/reports/CMakeLists.txt	2010-12-15 17:42:29.801474500 +0100
@@ -13,6 +13,7 @@
 )
 
 KDE4_ADD_LIBRARY(reports STATIC ${libreports_a_SOURCES})
+ADD_DEPENDENCIES(reports kmm_config)
 
 ############## tests ####################
 IF( KDE4_BUILD_TESTS )
diff -Nru kmymoney-4.5.1.orig/kmymoney/dialogs/settings/CMakeLists.txt kmymoney-4.5.1/kmymoney/dialogs/settings/CMakeLists.txt
--- kmymoney-4.5.1.orig/kmymoney/dialogs/settings/CMakeLists.txt	2010-11-14 16:28:43.000000000 +0200
+++ kmymoney-4.5.1/kmymoney/dialogs/settings/CMakeLists.txt	2010-12-16 09:02:49.181375500 +0200
@@ -31,3 +31,4 @@
 kde4_add_ui_files(libsettings_a_SOURCES ${libsettings_a_UI} )
 
 KDE4_ADD_LIBRARY(settings STATIC ${libsettings_a_SOURCES})
+ADD_DEPENDENCIES(settings kmm_config)
