diff -Nru kmymoney-3.98.1.orig/CMakeLists.txt kmymoney-3.98.1/CMakeLists.txt
--- kmymoney-3.98.1.orig/CMakeLists.txt	2010-05-21 08:41:07.000000000 +0200
+++ kmymoney-3.98.1/CMakeLists.txt	2010-06-07 23:47:44.487000000 +0200
@@ -127,8 +127,10 @@
 CHECK_FUNCTION_EXISTS(vsnprintf HAVE_VSNPRINTF)
 CHECK_FUNCTION_EXISTS(strtoll HAVE_STRTOLL)
 
-SET(CMAKE_REQUIRED_FLAGS "-std=c99")
-SET(CMAKE_REQUIRED_LIBRARIES m)
+IF(NOT WIN32)
+    SET(CMAKE_REQUIRED_FLAGS "-std=c99")
+    SET(CMAKE_REQUIRED_LIBRARIES m)
+ENDIF(NOT WIN32)
 CHECK_FUNCTION_EXISTS(round HAVE_ROUND)
 
 # check for function in header
@@ -142,7 +144,9 @@
 
 # special handling of crypt() check
 # ----------------------- crypt() ---------------------------------
-SET(CMAKE_REQUIRED_LIBRARIES crypt) 
+IF(NOT WIN32)
+    SET(CMAKE_REQUIRED_LIBRARIES crypt) 
+ENDIF(NOT WIN32)
 CHECK_FUNCTION_EXISTS(crypt HAVE_CRYPT)
 # -----------------------------------------------------------------
 
