diff -Nru mpir-2.3.0.orig/build.vc10/configure.bat mpir-2.3.0/build.vc10/configure.bat
--- mpir-2.3.0.orig/build.vc10/configure.bat	2010-12-16 04:30:49.000000000 +0200
+++ mpir-2.3.0/build.vc10/configure.bat	2011-03-10 09:45:50.016638000 +0200
@@ -69,6 +69,7 @@
 	set ARCH=amd64
 	set ARCHW=x64
 )
+set YASM="%YASMPATH%\vsyasm.exe"
 cl config.guess.c > nul 2>&1
 if errorlevel 1 goto :nocc1
 config.guess.exe print > config.guess.bat
@@ -77,7 +78,6 @@
 :nocc1
 if exist "%VS100COMNTOOLS%\..\..\VC\vcvarsall.bat" (
 	call "%VS100COMNTOOLS%\..\..\VC\vcvarsall.bat" %ARCH%
-        set YASM="%VS100COMNTOOLS%\..\..\VC\bin\vsyasm.exe"
 )
 cl config.guess.c > nul 2>&1
 if errorlevel 1 goto :nocc
diff -Nru mpir-2.3.0.orig/build.vc10/make.bat mpir-2.3.0/build.vc10/make.bat
--- mpir-2.3.0.orig/build.vc10/make.bat	2011-01-26 12:18:14.000000000 +0200
+++ mpir-2.3.0/build.vc10/make.bat	2011-03-12 22:41:59.996160800 +0200
@@ -39,7 +39,7 @@
 
 :make
 
-set MS_BUILD="C:\Windows\Microsoft.NET\Framework64\v4.0.30319\msbuild.exe"
+set MS_BUILD="C:\Windows\Microsoft.NET\Framework\v4.0.30319\msbuild.exe"
 
 set LIBBUILD=%LIBTYPE%_mpir_%BCPU%
 
diff -Nru mpir-2.3.0.orig/build.vc10/postbuild.bat mpir-2.3.0/build.vc10/postbuild.bat
--- mpir-2.3.0.orig/build.vc10/postbuild.bat	2011-01-26 12:18:14.000000000 +0200
+++ mpir-2.3.0/build.vc10/postbuild.bat	2011-03-10 00:14:18.414385800 +0200
@@ -9,11 +9,11 @@
 set str2=%str:~0,10%
 if "%str2%" NEQ "build.vc10" goto dele
 
-rem we now have: build.vc10\<win32|x64>\<debug|release>\mpir.<lib|dll>
+rem we now have: build.vc10\type\<win32|x64>\<debug|release>\mpir.<lib|dll>
 
 rem extract platform (plat=<win32|x64>), configuration (conf=<debug|release>) anbd file name     
 set file=
-for /f "tokens=1,2,3,4 delims=\" %%a in ("%str%") do set plat=%%b&set conf=%%c&set file=%%d
+for /f "tokens=1,2,3,4,5 delims=\" %%a in ("%str%") do set type=%%b&set plat=%%c&set conf=%%d&set file=%%e
 if /i "%file%" NEQ "" (goto next)
 call :seterr & echo ERROR: %1 is not supported & exit /b %errorlevel%
 
@@ -25,7 +25,7 @@
 set extn=%extn:~-4,3%
 
 rem set the target aand output directories
-set source="%plat%\%conf%"
+set source="%type%\%plat%\%conf%"
 set dest="%extn%\%plat%\%conf%"
 
 rem output parametrers for the MPIR tests
diff -Nru mpir-2.3.0.orig/mpirxx.h mpir-2.3.0/mpirxx.h
--- mpir-2.3.0.orig/mpirxx.h	2010-12-16 03:58:21.000000000 +0200
+++ mpir-2.3.0/mpirxx.h	2011-03-10 22:07:38.321159900 +0200
@@ -1355,6 +1355,7 @@
    since gmp-impl.h is not publicly available, I redefine it here
    I use a different name to avoid possible clashes */
 extern "C" {
+typedef void (*__gmp_freefunc_t) (void *, size_t);
 struct __gmp_alloc_cstring_c
 {
    void (*free_func) (void *, size_t);

diff -Nru mpir-2.3.0.orig/configure mpir-2.3.0/configure
--- mpir-2.3.0.orig/configure	2011-03-04 09:24:37.000000000 +0200
+++ mpir-2.3.0/configure	2011-03-10 23:25:10.016611300 +0200
@@ -10062,7 +10062,7 @@
 #define __GMP_BITS_PER_MP_LIMB 123 /* dummy for GMP_NUMB_BITS etc */
 #define GMP_LIMB_BITS 123
 $DEFN_LONG_LONG_LIMB
-#include "$srcdir/gmp-h.in"
+#include "../mpir-2.3.0/gmp-h.in"
 
 #if ! __GMP_HAVE_PROTOTYPES
 die die die
@@ -24722,7 +24722,7 @@
 #define __GMP_BITS_PER_MP_LIMB 123 /* dummy for GMP_NUMB_BITS etc */
 #define GMP_LIMB_BITS 123
 $DEFN_LONG_LONG_LIMB
-#include "$srcdir/gmp-h.in"
+#include "../mpir-2.3.0/gmp-h.in"
 
 #ifndef __GMP_EXTERN_INLINE
 die die die
@@ -24933,9 +24933,9 @@
 #define __GMP_BITS_PER_MP_LIMB 123 /* dummy for GMP_NUMB_BITS etc */
 #define GMP_LIMB_BITS 123
 $DEFN_LONG_LONG_LIMB
-#include "$srcdir/gmp-h.in"
+#include "../mpir-2.3.0/gmp-h.in"
 
-#include "$srcdir/gmp-impl.h"
+#include "../mpir-2.3.0/gmp-impl.h"
 
 int
 main ()
@@ -25031,7 +25031,7 @@
 #define __GMP_BITS_PER_MP_LIMB 123 /* dummy for GMP_NUMB_BITS etc */
 #define GMP_LIMB_BITS 123
 $DEFN_LONG_LONG_LIMB
-#include "$srcdir/gmp-h.in"
+#include "../mpir-2.3.0/gmp-h.in"
 
 #if ! _GMP_H_HAVE_FILE
 die die die
@@ -27934,7 +27934,7 @@
 #define __GMP_BITS_PER_MP_LIMB 123 /* dummy for GMP_NUMB_BITS etc */
 #define GMP_LIMB_BITS 123
 $DEFN_LONG_LONG_LIMB
-#include \"$srcdir/gmp-h.in\"
+#include \"../mpir-2.3.0/gmp-h.in\"
 
 "; then :
 
diff -Nru mpir-2.3.0.orig/Makefile.am mpir-2.3.0/Makefile.am
--- mpir-2.3.0.orig/Makefile.am	2011-03-04 09:24:25.000000000 +0200
+++ mpir-2.3.0/Makefile.am	2011-03-12 21:03:41.906923500 +0200
@@ -217,7 +217,7 @@
 if WANT_GMPCOMPAT
 	(rm -f $(DESTDIR)$(includedir)/gmp.h && cp mpir.h $(DESTDIR)$(includedir)/gmp.h)
 if WANT_CXX
-	(rm -f $(DESTDIR)$(includedir)/gmpxx.h && cp mpirxx.h $(DESTDIR)$(includedir)/gmpxx.h)
+	(rm -f $(DESTDIR)$(includedir)/gmpxx.h && cp $(srcdir)/mpirxx.h $(DESTDIR)$(includedir)/gmpxx.h)
 endif
 endif
 	@echo ''
diff -Nru mpir-2.3.0.orig/Makefile.in mpir-2.3.0/Makefile.in
--- mpir-2.3.0.orig/Makefile.in	2011-03-04 09:24:40.000000000 +0200
+++ mpir-2.3.0/Makefile.in	2011-03-12 21:04:56.430361000 +0200
@@ -1225,7 +1225,7 @@
 
 install-data-hook:
 @WANT_GMPCOMPAT_TRUE@	(rm -f $(DESTDIR)$(includedir)/gmp.h && cp mpir.h $(DESTDIR)$(includedir)/gmp.h)
-@WANT_CXX_TRUE@@WANT_GMPCOMPAT_TRUE@	(rm -f $(DESTDIR)$(includedir)/gmpxx.h && cp mpirxx.h $(DESTDIR)$(includedir)/gmpxx.h)
+@WANT_CXX_TRUE@@WANT_GMPCOMPAT_TRUE@	(rm -f $(DESTDIR)$(includedir)/gmpxx.h && cp $(srcdir)/mpirxx.h $(DESTDIR)$(includedir)/gmpxx.h)
 	@echo ''
 	@echo '+-------------------------------------------------------------+'
 	@echo '| CAUTION:                                                    |'
