diff --git a/tree.c b/tree.c
index 8baae3d..f958ad3 100644
--- a/tree.c
+++ b/tree.c
@@ -3320,6 +3320,29 @@ xmlAddChild(xmlNodePtr parent, xmlNodePtr cur) {
        xmlFreeNode(cur);
        return(parent->last);
    }
+
+    /*
+     * If cur is nbktext, merge with previous node
+     */
+   if ((parent->last != NULL) && (parent->last->type == XML_TEXT_NODE) &&
+       (xmlStrEqual(cur->name, "nbktext")) &&
+       (parent->last != cur)) {
+       xmlNodeAddContent(parent->last, cur->content);
+       xmlFreeNode(cur);
+       return(parent->last);
+   }
+
+    /*
+     * If parent->last is nbktext and we're not, merge with current node
+     */
+   if ((parent->last != NULL) && (parent->last->type == XML_TEXT_NODE) &&
+       (xmlStrEqual(parent->last->name, "nbktext")) &&
+       (parent->last != cur)) {
+       xmlNodeAddContent(parent->last, cur->content);
+        parent->last->name = xmlStringText;
+       xmlFreeNode(cur);
+       return(parent->last);
+   }
     }
 
     /*