diff -Nru -x '*~' binutils-2.22.52.orig/binutils/defparse.y binutils-2.22.52/binutils/defparse.y
--- binutils-2.22.52.orig/binutils/defparse.y	2012-02-24 15:20:16.000000000 +0100
+++ binutils-2.22.52/binutils/defparse.y	2012-07-26 00:19:25.253594500 +0200
@@ -52,7 +52,12 @@
 
 command: 
 		NAME opt_name opt_base { def_name ($2, $3); }
-	|	LIBRARY opt_name opt_base option_list { def_library ($2, $3); }
+	|	LIBRARY opt_name opt_base option_list
+		{
+		/* Ignore LIBRARY without argument, or empty name.  */
+		if ($2 && $2[0] != 0)
+		def_library ($2, $3);
+		}
 	|	EXPORTS explist 
 	|	DESCRIPTION ID { def_description ($2);}
 	|	STACKSIZE NUMBER opt_number { def_stacksize ($2, $3);}
diff -Nru -x '*~' binutils-2.22.52.orig/binutils/rcparse.y binutils-2.22.52/binutils/rcparse.y
--- binutils-2.22.52.orig/binutils/rcparse.y	2011-10-25 10:03:17.000000000 +0200
+++ binutils-2.22.52/binutils/rcparse.y	2012-07-26 00:19:25.260594500 +0200
@@ -1423,13 +1423,15 @@
 		  res_alloc (sizeof (rc_fixed_versioninfo)));
 	    memset ($$, 0, sizeof (rc_fixed_versioninfo));
 	  }
-	| fixedverinfo FILEVERSION numexpr cnumexpr cnumexpr cnumexpr
+		| fixedverinfo FILEVERSION numexpr optcnumexpr optcnumexpr
+		optcnumexpr
 	  {
 	    $1->file_version_ms = ($3 << 16) | $4;
 	    $1->file_version_ls = ($5 << 16) | $6;
 	    $$ = $1;
 	  }
-	| fixedverinfo PRODUCTVERSION numexpr cnumexpr cnumexpr cnumexpr
+		| fixedverinfo PRODUCTVERSION numexpr optcnumexpr optcnumexpr
+		optcnumexpr
 	  {
 	    $1->product_version_ms = ($3 << 16) | $4;
 	    $1->product_version_ls = ($5 << 16) | $6;
diff -Nru -x '*~' binutils-2.22.52.orig/ld/deffilep.y binutils-2.22.52/ld/deffilep.y
--- binutils-2.22.52.orig/ld/deffilep.y	2012-02-24 15:20:19.000000000 +0100
+++ binutils-2.22.52/ld/deffilep.y	2012-07-26 16:32:18.779568500 +0200
@@ -140,7 +140,11 @@
 
 command: 
 		NAME opt_name opt_base { def_image_name ($2, $3, 0); }
-	|	LIBRARY opt_name opt_base { def_image_name ($2, $3, 1); }
+	|	LIBRARY opt_name opt_base
+		{
+		if ($2 && $2[0] != 0)
+		def_image_name ($2, $3, 1);
+		}
 	|	DESCRIPTION ID { def_description ($2);}
 	|	STACKSIZE_K NUMBER opt_number { def_stacksize ($2, $3);}
 	|	HEAPSIZE NUMBER opt_number { def_heapsize ($2, $3);}
