diff -Nru kdesdk-4.7.0.orig/kompare/libdiff2/tests/CMakeLists.txt kdesdk-4.7.0/kompare/libdiff2/tests/CMakeLists.txt
--- kdesdk-4.7.0.orig/kompare/libdiff2/tests/CMakeLists.txt	2011-05-20 22:18:58.000000000 +0200
+++ kdesdk-4.7.0/kompare/libdiff2/tests/CMakeLists.txt	2011-08-22 23:41:42.467430300 +0200
@@ -19,6 +19,7 @@
 )
 
 kde4_add_unit_test(interactiveDiffTest ${interactiveDiffTest_SRCS})
+set_target_properties(interactiveDiffTest PROPERTIES COMPILE_FLAGS "-DDIFF2_EXPORT=")
 target_link_libraries(interactiveDiffTest
     ${QT_QTTEST_LIBRARY}
     ${KDE4_KDECORE_LIBS}
@@ -32,6 +33,7 @@
 )
 
 kde4_add_unit_test(levenshteintest ${levenshteinTest_SRCS})
+set_target_properties(levenshteintest PROPERTIES COMPILE_FLAGS "-DDIFF2_EXPORT=")
 target_link_libraries(levenshteintest
     ${QT_QTTEST_LIBRARY}
     ${QT_QTCORE_LIBRARY}
