--- configure	2007-11-05 21:52:57.000000000 +0100
+++ configure	2008-06-11 23:35:43.156250000 +0200
@@ -3328,6 +3328,10 @@
 	EXT="cygwin"
 	CFLAGS="$CFLAGS -DHAVE_POSIX_SIGNALS"
 	;;
+    MINGW*)
+        ARCH="cygwin"
+        EXT="mingw"
+ 	;;
     *)
 		echo "cfitsio: == Don't know what do do with $system"
 	;;
@@ -3387,6 +3391,10 @@
     SHLIB_LD="$CC -shared"
     SHLIB_SUFFIX=".dll"
     ;;
+  mingw)
+    SHLIB_LD="$CC -shared"
+    SHLIB_SUFFIX=".dll"
+    ;;
   darwin)
     SHLIB_LD="$CC -dynamiclib"
     SHLIB_SUFFIX=".dylib"
