diff -Nru pexports-0.44.orig/CMakeLists.txt pexports-0.44/CMakeLists.txt
--- pexports-0.44.orig/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ pexports-0.44/CMakeLists.txt	2010-04-21 23:49:37.396967600 +0200
@@ -0,0 +1,27 @@
+project(pexports)
+
+cmake_minimum_required(VERSION 2.8)
+
+find_package(BISON REQUIRED)
+find_package(FLEX REQUIRED)
+
+BISON_TARGET(parser hparse.y ${CMAKE_BINARY_DIR}/hparse.c)
+FLEX_TARGET(scanner hlex.l  ${CMAKE_BINARY_DIR}/hlex.c)
+ADD_FLEX_BISON_DEPENDENCY(scanner parser)
+
+include_directories(${CMAKE_BINARY_DIR} ${CMAKE_SOURCE_DIR})
+
+if(MSVC)
+    add_definitions(-D_CRT_SECURE_NO_WARNINGS -D_CRT_SECURE_NO_DEPRECATE -wd4996)
+endif(MSVC)
+
+add_executable(pexports
+   pexports.c
+   str_tree.c
+   ${CMAKE_BINARY_DIR}/hparse.c
+   ${CMAKE_BINARY_DIR}/hlex.c
+)
+
+install(TARGETS pexports RUNTIME DESTINATION bin
+                         LIBRARY DESTINATION lib
+                         ARCHIVE DESTINATION lib)
\ Kein Zeilenumbruch am Dateiende.
