diff -Nru OpenCV-2.3.0.orig/CMakeLists.txt OpenCV-2.3.0/CMakeLists.txt
--- OpenCV-2.3.0.orig/CMakeLists.txt	2011-08-17 20:37:56.720349800 +0200
+++ OpenCV-2.3.0/CMakeLists.txt	2011-08-17 20:36:27.094223500 +0200
@@ -1378,32 +1378,33 @@
         install(FILES ${root_files}
                 DESTINATION "."
                 COMPONENT main)
-        install(FILES ${root_src_files}
-                DESTINATION "."
-                COMPONENT src)
+#        install(FILES ${root_src_files}
+#                DESTINATION "."
+#                COMPONENT src)
         # Add the "win-install/OpenCVConfig.cmake" file to the "main" install component
         install(FILES
             "${CMAKE_BINARY_DIR}/win-install/OpenCVConfig.cmake"
             DESTINATION "."
             COMPONENT main
             )
-        install(FILES
-            "include/CMakeLists.txt"
-            DESTINATION "include/"
-            COMPONENT src
-            )
+#        install(FILES
+#            "include/CMakeLists.txt"
+#            DESTINATION "include/"
+#            COMPONENT src
+#            )
 
         if(CMAKE_INSTALL_DEBUG_LIBRARIES)
-            foreach(m calib3d core contrib features2d ffmpeg flann gpu highgui imgproc legacy ml objdetect video)
-                install(FILES  ${EXECUTABLE_OUTPUT_PATH}/Debug/opencv_${m}${OPENCV_DLLVERSION}d.dll
+            # ffmpeg 
+            foreach(m calib3d core contrib features2d flann gpu highgui imgproc legacy ml objdetect video)
+                install(FILES  ${EXECUTABLE_OUTPUT_PATH}/opencv_${m}${OPENCV_DLLVERSION}d.dll
             DESTINATION bin COMPONENT main)
-                install(FILES  ${LIBRARY_OUTPUT_PATH}/Debug/opencv_${m}${OPENCV_DLLVERSION}d.lib
+                install(FILES  ${LIBRARY_OUTPUT_PATH}/opencv_${m}${OPENCV_DLLVERSION}d.lib
             DESTINATION lib COMPONENT main)
             endforeach()
         endif()
 
         install(DIRECTORY data samples DESTINATION "." COMPONENT main PATTERN ".svn" EXCLUDE)
-        install(DIRECTORY 3rdparty android modules doc DESTINATION "." COMPONENT src PATTERN ".svn" EXCLUDE)
+#        install(DIRECTORY 3rdparty android modules doc DESTINATION "." COMPONENT src PATTERN ".svn" EXCLUDE)
         #install(DIRECTORY doc DESTINATION doc COMPONENT src FILES_MATCHING PATTERN "*.rst" "*.tex")
         #install(DIRECTORY doc/pics DESTINATION doc COMPONENT src PATTERN ".svn" EXCLUDE)
         #install(DIRECTORY doc/latex2sphinx DESTINATION doc COMPONENT src PATTERN ".svn" EXCLUDE)
