diff -Nru kdelibs-4.6.2.orig/cmake/modules/FindKDE4Internal.cmake kdelibs-4.6.2/cmake/modules/FindKDE4Internal.cmake
--- kdelibs-4.6.2.orig/cmake/modules/FindKDE4Internal.cmake	2011-04-01 15:56:18.000000000 +0200
+++ kdelibs-4.6.2/cmake/modules/FindKDE4Internal.cmake	2011-04-15 10:38:45.704400000 +0200
@@ -1224,11 +1224,6 @@
       set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--export-all-symbols -Wl,--disable-auto-import")
       set (CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,--export-all-symbols -Wl,--disable-auto-import")
 
-      # we always link against the release version of QT with mingw
-      # (even for debug builds). So we need to define QT_NO_DEBUG
-      # or else QPluginLoader rejects plugins because it thinks
-      # they're built against the wrong QT.
-      add_definitions(-DQT_NO_DEBUG)
    endif (MINGW)
 
    check_cxx_compiler_flag(-fPIE HAVE_FPIE_SUPPORT)
