Index: kopete/kopete/chatwindow/kopetechatwindow.cpp
===================================================================
--- kopete/kopete/chatwindow/kopetechatwindow.cpp	(revision 946324)
+++ kopete/kopete/chatwindow/kopetechatwindow.cpp	(working copy)
@@ -359,7 +359,7 @@
 	//Default to 'Return' and 'Enter' for sending messages
 	//'Return' is the key in the main part of the keyboard
 	//'Enter' is on the Numpad
-	KShortcut chatSendShortcut( QKeySequence(Qt::Key_Return), QKeySequence(Qt::Key_Enter) );
+	KShortcut chatSendShortcut( QKeySequence((int)Qt::Key_Return), QKeySequence((int)Qt::Key_Enter) );
 	chatSend->setShortcut( chatSendShortcut );
 	chatSend->setEnabled( false );
 
