diff -Nru kdegames-4.5.4.orig/ksirk/ksirk/iris/src/CMakeLists.txt kdegames-4.5.4/ksirk/ksirk/iris/src/CMakeLists.txt
--- kdegames-4.5.4.orig/ksirk/ksirk/iris/src/CMakeLists.txt	2009-12-14 11:44:48.000000000 +0100
+++ kdegames-4.5.4/ksirk/ksirk/iris/src/CMakeLists.txt	2010-12-14 00:57:24.312792200 +0100
@@ -93,34 +93,34 @@
   )
 
 kde4_moc_headers(iris_ksirk
-irisnet/corelib/jdnsshared.h
-irisnet/corelib/netavailability.h
-irisnet/corelib/netinterface.h
-irisnet/corelib/netnames.h
-irisnet/corelib/objectsession.h
-irisnet/noncore/processquit.h
-jdns/qjdns.h
-xmpp/base/randrandomnumbergenerator.h
-xmpp/base64/base64.h
-xmpp/cutestuff/bsocket.h
-xmpp/cutestuff/bytestream.h
-xmpp/cutestuff/httpconnect.h
-xmpp/cutestuff/socks.h
-xmpp/cutestuff/httppoll.h
-xmpp/sasl/digestmd5proplist.cpp
-xmpp/sasl/digestmd5response.cpp
-xmpp/sasl/plainmessage.cpp
-xmpp/xmpp-core/xmpp.h
-xmpp/xmpp-core/xmpp_clientstream.h
-xmpp/xmpp-core/xmpp_stream.h
-xmpp/xmpp-core/securestream.h
-xmpp/xmpp-core/simplesasl.h
-xmpp/xmpp-im/s5b.h
-xmpp/xmpp-im/xmpp_client.h
-xmpp/xmpp-im/xmpp_task.h
-xmpp/zlib/common.h
-xmpp/zlib/zlibcompressor.h
-xmpp/zlib/zlibdecompressor.h
+ irisnet/corelib/jdnsshared.h
+ irisnet/corelib/netavailability.h
+ irisnet/corelib/netinterface.h
+ irisnet/corelib/netnames.h
+ irisnet/corelib/objectsession.h
+ irisnet/noncore/processquit.h
+ jdns/qjdns.h
+# xmpp/base/randrandomnumbergenerator.h
+# xmpp/base64/base64.h
+ xmpp/cutestuff/bsocket.h
+# xmpp/cutestuff/bytestream.h
+# xmpp/cutestuff/httpconnect.h
+# xmpp/cutestuff/socks.h
+# xmpp/cutestuff/httppoll.h
+# xmpp/sasl/digestmd5proplist.cpp
+# xmpp/sasl/digestmd5response.cpp
+# xmpp/sasl/plainmessage.cpp
+ xmpp/xmpp-core/xmpp.h
+ xmpp/xmpp-core/xmpp_clientstream.h
+ xmpp/xmpp-core/xmpp_stream.h
+ xmpp/xmpp-core/securestream.h
+# xmpp/xmpp-core/simplesasl.h
+ xmpp/xmpp-im/s5b.h
+ xmpp/xmpp-im/xmpp_client.h
+# xmpp/xmpp-im/xmpp_task.h
+# xmpp/zlib/common.h
+# xmpp/zlib/zlibcompressor.h
+# xmpp/zlib/zlibdecompressor.h
 )
 
 
