diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' lldb-4.0.1.src.orig/cmake/modules/LLDBStandalone.cmake lldb-4.0.1.src/cmake/modules/LLDBStandalone.cmake
--- lldb-4.0.1.src.orig/cmake/modules/LLDBStandalone.cmake	2017-01-10 00:12:37.000000000 +0100
+++ lldb-4.0.1.src/cmake/modules/LLDBStandalone.cmake	2017-08-21 10:47:35.763933800 +0200
@@ -28,6 +28,11 @@
       RESULT_VARIABLE HAD_ERROR
       OUTPUT_VARIABLE CONFIG_OUTPUT
     )
+
+      # On Windows we might get trailing backslashes
+      string(REGEX REPLACE
+        "\\\\" "/"
+        CONFIG_OUTPUT ${CONFIG_OUTPUT})
     if(NOT HAD_ERROR)
       string(REGEX REPLACE
         "[ \t]*[\r\n]+[ \t]*" ";"
