commit b1ab5fb69e38bb1b5c388e23cf47723ee8dc37cc
Author: Montel Laurent <montel@kde.org>
Date:   Fri Jun 23 11:35:28 2017 +0200

    Fix build against windows

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 531ef2c3f..f40f75ac8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,6 +43,10 @@ find_package(KF5IdentityManagement ${IDENTITYMANAGER_LIB_VERSION} CONFIG REQUIRE
 
 add_definitions(-DTRANSLATION_DOMAIN=\"libkcalutils5\")
 add_definitions("-DQT_NO_CAST_FROM_ASCII -DQT_NO_CAST_TO_ASCII")
+if(BUILD_TESTING)
+   add_definitions(-DBUILD_TESTING)
+endif(BUILD_TESTING)
+
 
 ########### Targets ###########
 add_subdirectory(src)
diff --git a/autotests/CMakeLists.txt b/autotests/CMakeLists.txt
index f2fc3076b..17199d127 100644
--- a/autotests/CMakeLists.txt
+++ b/autotests/CMakeLists.txt
@@ -17,15 +17,9 @@ ecm_add_tests(testdndfactory.cpp teststringify.cpp
 
 
 ecm_add_test(testincidenceformatter.cpp
-        ${CMAKE_SOURCE_DIR}/src/incidenceformatter.cpp
-        ${CMAKE_SOURCE_DIR}/src/grantleetemplatemanager.cpp
-        ${CMAKE_SOURCE_DIR}/src/qtresourcetemplateloader.cpp
-        ${CMAKE_SOURCE_DIR}/src/grantleeki18nlocalizer.cpp
-        ${CMAKE_SOURCE_DIR}/src/stringify.cpp
-        ${CMAKE_BINARY_DIR}/src/kcalutils_debug.cpp
     TEST_NAME "testincidenceformatter"
     NAME_PREFIX "kcalutils-"
-    LINK_LIBRARIES Qt5::Core Qt5::Test KF5::CalendarCore KF5::I18n KF5::IdentityManagement Grantlee5::Templates
+    LINK_LIBRARIES KF5CalendarUtils Qt5::Core Qt5::Test KF5::CalendarCore KF5::I18n KF5::IdentityManagement Grantlee5::Templates
     
 )
 
diff --git a/src/grantleetemplatemanager_p.h b/src/grantleetemplatemanager_p.h
index 3228d1652..2e2902c6f 100644
--- a/src/grantleetemplatemanager_p.h
+++ b/src/grantleetemplatemanager_p.h
@@ -21,6 +21,7 @@
 #define GRANTLEETEMPLATEMANAGER_H_P
 
 #include <QSharedPointer>
+#include "kcalutils_private_export.h"
 
 namespace Grantlee
 {
@@ -34,7 +35,7 @@ typedef QSharedPointer<TemplateImpl> Template;
 class QString;
 class GrantleeKi18nLocalizer;
 
-class GrantleeTemplateManager
+class KCALUTILS_TESTS_EXPORT GrantleeTemplateManager
 {
 public:
     ~GrantleeTemplateManager();
diff --git a/src/kcalutils_private_export.h b/src/kcalutils_private_export.h
new file mode 100644
index 000000000..eeff56230
--- /dev/null
+++ b/src/kcalutils_private_export.h
@@ -0,0 +1,34 @@
+/*  This file is part of the KDE project
+    Copyright (C) 2007 David Faure <faure@kde.org>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Library General Public
+    License as published by the Free Software Foundation; either
+    version 2 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Library General Public License for more details.
+
+    You should have received a copy of the GNU Library General Public License
+    along with this library; see the file COPYING.LIB.  If not, write to
+    the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+    Boston, MA 02110-1301, USA.
+*/
+
+#ifndef KCALUTILSPRIVATE_EXPORT_H
+#define KCALUTILSPRIVATE_EXPORT_H
+
+#include "kcalutils_export.h"
+
+/* Classes which are exported only for unit tests */
+#ifdef BUILD_TESTING
+# ifndef KCALUTILS_TESTS_EXPORT
+#  define KCALUTILS_TESTS_EXPORT KCALUTILS_EXPORT
+# endif
+#else /* not compiling tests */
+# define KCALUTILS_TESTS_EXPORT
+#endif
+
+#endif
