--- tools\qdbus\qdbuscpp2xml\qdbuscpp2xml.pro.orig	2008-02-15 21:28:36.197500000 +0100
+++ tools\qdbus\qdbuscpp2xml\qdbuscpp2xml.pro	2008-02-15 21:29:09.478750000 +0100
@@ -2,8 +2,8 @@
 DESTDIR = ../../../bin
 TARGET = qdbuscpp2xml
 QT = core xml
-CONFIG += qdbus
-CONFIG -= app_bundle
+CONFIG += qdbus
+CONFIG -= app_bundle debug_and_release
 win32:CONFIG += console
 
 target.path=$$[QT_INSTALL_BINS]
--- tools\qdbus\qdbusxml2cpp\qdbusxml2cpp.pro.orig	2008-02-15 21:28:36.416250000 +0100
+++ tools\qdbus\qdbusxml2cpp\qdbusxml2cpp.pro	2008-02-15 21:29:09.510000000 +0100
@@ -2,8 +2,8 @@
 DESTDIR = ../../../bin
 TARGET = qdbusxml2cpp
 QT = core xml
-CONFIG += qdbus
-CONFIG -= app_bundle
+CONFIG += qdbus
+CONFIG -= app_bundle debug_and_release
 win32:CONFIG += console
 
 target.path=$$[QT_INSTALL_BINS]
--- tools\qdbus\qdbusviewer\qdbusviewer.pro.orig	2008-02-15 21:40:44.013500000 +0100
+++ tools\qdbus\qdbusviewer\qdbusviewer.pro	2008-02-15 21:40:44.794750000 +0100
@@ -14,7 +14,8 @@
 
 DESTDIR = ../../../bin
 
-CONFIG += qdbus
+CONFIG += qdbus
+CONFIG -= debug_and_release
 QT += xml
 
 target.path=$$[QT_INSTALL_BINS]
--- src\dbus\qdbus_symbols.cpp.orig	2008-04-18 16:34:22.000000000 +0200
+++ src\dbus\qdbus_symbols.cpp	2008-04-17 21:10:23.796875000 +0200
@@ -75,7 +75,11 @@
 
     static int majorversions[] = { 3, 2, -1 };
     lib->unload();
+#ifdef Q_CC_MINGW
+    lib->setFileName(QLatin1String("libdbus-1"));
+#else
     lib->setFileName(QLatin1String("dbus-1"));
+#endif
     for (uint i = 0; i < sizeof(majorversions) / sizeof(majorversions[0]); ++i) {
         lib->setFileNameAndVersion(lib->fileName(), majorversions[i]);
         if (lib->load() && lib->resolve("dbus_connection_open_private"))
