--- tools\qdbus\qdbuscpp2xml\qdbuscpp2xml.pro.orig	2008-02-15 21:28:36.197500000 +0100
+++ tools\qdbus\qdbuscpp2xml\qdbuscpp2xml.pro	2008-02-15 21:29:09.478750000 +0100
@@ -2,8 +2,8 @@
 DESTDIR = ../../../bin
 TARGET = qdbuscpp2xml
 QT = core xml
-CONFIG += qdbus
-CONFIG -= app_bundle
+CONFIG += qdbus release
+CONFIG -= app_bundle debug debug_and_release
 win32:CONFIG += console
 
 target.path=$$[QT_INSTALL_BINS]
--- tools\qdbus\qdbusxml2cpp\qdbusxml2cpp.pro.orig	2008-02-15 21:28:36.416250000 +0100
+++ tools\qdbus\qdbusxml2cpp\qdbusxml2cpp.pro	2008-02-15 21:29:09.510000000 +0100
@@ -2,8 +2,8 @@
 DESTDIR = ../../../bin
 TARGET = qdbusxml2cpp
 QT = core xml
-CONFIG += qdbus
-CONFIG -= app_bundle
+CONFIG += qdbus release
+CONFIG -= app_bundle debug debug_and_release
 win32:CONFIG += console
 
 target.path=$$[QT_INSTALL_BINS]
--- tools\qdbus\qdbusviewer\qdbusviewer.pro.orig	2008-02-15 21:40:44.013500000 +0100
+++ tools\qdbus\qdbusviewer\qdbusviewer.pro	2008-02-15 21:40:44.794750000 +0100
@@ -14,7 +14,8 @@
 
 DESTDIR = ../../../bin
 
-CONFIG += qdbus
+CONFIG += qdbus release
+CONFIG -= debug debug_and_release
 QT += xml
 
 target.path=$$[QT_INSTALL_BINS]
