diff --git a/src/dbus/dbus.pro b/src/dbus/dbus.pro
index 9ab3920..7c59808 100644
--- a/src/dbus/dbus.pro
+++ b/src/dbus/dbus.pro
@@ -20,7 +20,8 @@ unix {
         QtXml
 }
 win32 { 
-    LIBS_PRIVATE += -lws2_32 \
+    wince*:LIBS_PRIVATE += -lws2
+    else:LIBS_PRIVATE += -lws2_32 \
         -ladvapi32 \
         -lnetapi32 \
         -luser32
diff --git a/src/src.pro b/src/src.pro
index 8dec49b..26e8372 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -11,10 +11,10 @@ wince*:{
     SRC_SUBDIRS += src_corelib src_xml src_network src_gui src_sql src_testlib
     !vxworks:contains(QT_CONFIG, qt3support): SRC_SUBDIRS += src_qt3support
     include(tools/tools.pro)
-    contains(QT_CONFIG, dbus):SRC_SUBDIRS += src_dbus
 }
 win32:SRC_SUBDIRS += src_activeqt
 
+!symbian:contains(QT_CONFIG, dbus):SRC_SUBDIRS += src_dbus
 contains(QT_CONFIG, opengl)|contains(QT_CONFIG, opengles1)|contains(QT_CONFIG, opengles2): SRC_SUBDIRS += src_opengl
 contains(QT_CONFIG, openvg): SRC_SUBDIRS += src_openvg
 contains(QT_CONFIG, xmlpatterns): SRC_SUBDIRS += src_xmlpatterns
diff --git a/tools/qdbus/qdbusxml2cpp/qdbusxml2cpp.cpp b/tools/qdbus/qdbusxml2cpp/qdbusxml2cpp.cpp
index 7ac6894..cd52197 100644
--- a/tools/qdbus/qdbusxml2cpp/qdbusxml2cpp.cpp
+++ b/tools/qdbus/qdbusxml2cpp/qdbusxml2cpp.cpp
@@ -53,14 +53,9 @@
 #include "private/qdbusmetaobject_p.h"
 #include "private/qdbusintrospection_p.h"
 
-#include <sys/types.h>
 #include <stdio.h>
 #include <stdlib.h>
 
-#ifdef Q_WS_WIN
-#include <process.h>
-#endif
-
 #define PROGRAMNAME     "qdbusxml2cpp"
 #define PROGRAMVERSION  "0.7"
 #define PROGRAMCOPYRIGHT "Copyright (C) 2010 Nokia Corporation and/or its subsidiary(-ies)."
