diff -Nru libogg-1.2.0.orig/CMakeLists.txt libogg-1.2.0/CMakeLists.txt
--- libogg-1.2.0.orig/CMakeLists.txt	Thu Jan  1 02:00:00 1970
+++ libogg-1.2.0/CMakeLists.txt	Wed Jul  7 23:03:02 2010
@@ -0,0 +1,6 @@
+project (libogg)
+
+cmake_minimum_required (VERSION 2.6)
+
+add_subdirectory(include)
+add_subdirectory(src)
diff -Nru libogg-1.2.0.orig/include/CMakeLists.txt libogg-1.2.0/include/CMakeLists.txt
--- libogg-1.2.0.orig/include/CMakeLists.txt	Thu Jan  1 02:00:00 1970
+++ libogg-1.2.0/include/CMakeLists.txt	Wed Jul  7 23:02:41 2010
@@ -0,0 +1 @@
+install(FILES ogg/ogg.h ogg/os_types.h DESTINATION include/ogg)
\ No newline at end of file
diff -Nru libogg-1.2.0.orig/src/CMakeLists.txt libogg-1.2.0/src/CMakeLists.txt
--- libogg-1.2.0.orig/src/CMakeLists.txt	Thu Jan  1 02:00:00 1970
+++ libogg-1.2.0/src/CMakeLists.txt	Wed Jul  7 23:08:24 2010
@@ -0,0 +1,13 @@
+set(ogg_SRCS
+bitwise.c
+framing.c
+../win32/ogg.def
+)
+
+include_directories(${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR} ../include)
+
+add_library(ogg SHARED ${ogg_SRCS})
+
+install(TARGETS ogg RUNTIME DESTINATION bin
+                           LIBRARY DESTINATION lib
+                           ARCHIVE DESTINATION lib)
\ No newline at end of file
