diff -Nru dbus-1.4.1.orig/cmake/dbus/CMakeLists.txt dbus-1.4.1/cmake/dbus/CMakeLists.txt
--- dbus-1.4.1.orig/cmake/dbus/CMakeLists.txt	2010-12-13 16:16:12.000000000 +0100
+++ dbus-1.4.1/cmake/dbus/CMakeLists.txt	2011-01-10 16:41:48.668366900 +0100
@@ -6,6 +6,10 @@
 
 add_definitions(-DDBUS_COMPILATION)
 
+if(WIN32 AND DBUS_REPLACE_LOCAL_DIR)
+    add_definitions(-DDBUS_REPLACE_LOCAL_DIR)
+endif(WIN32 AND DBUS_REPLACE_LOCAL_DIR)
+
 set (dbusinclude_HEADERS
 	${DBUS_DIR}/dbus.h
 	${DBUS_DIR}/dbus-address.h
diff -Nru dbus-1.4.1.orig/dbus/dbus-sysdeps-win.c dbus-1.4.1/dbus/dbus-sysdeps-win.c
--- dbus-1.4.1.orig/dbus/dbus-sysdeps-win.c	2010-12-15 13:03:33.000000000 +0100
+++ dbus-1.4.1/dbus/dbus-sysdeps-win.c	2011-01-10 18:15:26.135667600 +0100
@@ -2069,7 +2069,33 @@
 _dbus_replace_install_prefix (const char *configure_time_path)
 {
 #ifndef DBUS_PREFIX
+# if defined(_WIN32) && defined(DBUS_REPLACE_LOCAL_DIR)
+  static char module_path[MAX_PATH];
+  char *configure_time_file;
+  int i;
+  
+  // if this is a path, try to get only the file name
+  if(NULL != strrchr(configure_time_path, '/')) configure_time_file = strrchr(configure_time_path, '/');
+  else return configure_time_path;
+
+  GetModuleFileName(NULL, module_path, MAX_PATH);
+  // the path is of the form C:\\Path\\to\\dbus\\bin\\dbus.exe
+  if(NULL != strrchr(module_path, '\\')) module_path[strrchr(module_path, '\\') - module_path] = '\0';
+  sprintf(module_path, "%s%s", module_path, configure_time_file);
+
+  /* Somehow, in some situations, backslashes get collapsed in the string.
+   * Since windows C library accepts both forward and backslashes as
+   * path separators, convert all backslashes to forward slashes.
+   */
+
+  for(i = 0; module_path[i] != '\0'; i++) {
+    if(module_path[i] == '\\')
+      module_path[i] = '/';
+  }
+  return module_path;
+# else
   return configure_time_path;
+# endif
 #else
   static char retval[1000];
   static char runtime_prefix[1000];
