diff -Nru -x '*~' kactivities-4.10.2.orig/src/lib/models/CMakeLists.txt kactivities-4.10.2/src/lib/models/CMakeLists.txt
--- kactivities-4.10.2.orig/src/lib/models/CMakeLists.txt	2013-03-01 07:26:31.971556000 +0100
+++ kactivities-4.10.2/src/lib/models/CMakeLists.txt	2013-07-19 11:26:11.377122500 +0200
@@ -112,7 +112,7 @@
     )
 
 kde4_add_library (
-    kactivities-models-component-plugin SHARED
+    kactivities-models-component-plugin MODULE
     ${kactivities-models-component_LIB_SRCS}
     )
 
