diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user zlib-1.2.12.orig/zconf.h.cmakein zlib-1.2.12/zconf.h.cmakein
--- zlib-1.2.12.orig/zconf.h.cmakein	2017-01-01 08:37:10.000000000 +0100
+++ zlib-1.2.12/zconf.h.cmakein	2022-05-03 11:29:38.470841500 +0200
@@ -433,10 +433,6 @@
    typedef unsigned long z_crc_t;
 #endif
 
-#ifdef HAVE_UNISTD_H    /* may be set to #if 1 by ./configure */
-#  define Z_HAVE_UNISTD_H
-#endif
-
 #ifdef HAVE_STDARG_H    /* may be set to #if 1 by ./configure */
 #  define Z_HAVE_STDARG_H
 #endif
