diff -Nrub -x '*~' -x '*\.orig' -x'*\.o' cfe-3.7.0.src.orig/tools/driver/clang_symlink.cmake cfe-3.7.0.src/tools/driver/clang_symlink.cmake
--- cfe-3.7.0.src.orig/tools/driver/clang_symlink.cmake	2013-08-28 03:58:44.000000000 +0200
+++ cfe-3.7.0.src/tools/driver/clang_symlink.cmake	2015-12-08 18:36:33.926457500 +0100
@@ -2,9 +2,9 @@
 # DESTDIR environment variable may be unset at configuration time.
 # See PR8397.
 
+set(CLANGXX_DESTDIR $ENV{DESTDIR})
 if(UNIX)
   set(CLANGXX_LINK_OR_COPY create_symlink)
-  set(CLANGXX_DESTDIR $ENV{DESTDIR})
 else()
   set(CLANGXX_LINK_OR_COPY copy)
 endif()
@@ -16,7 +16,7 @@
   set(EXECUTABLE_SUFFIX "")
 endif()
 
-set(bindir "${CLANGXX_DESTDIR}${CMAKE_INSTALL_PREFIX}/bin/")
+set(bindir "${CLANGXX_DESTDIR}/bin/")
 set(clang "clang${EXECUTABLE_SUFFIX}")
 set(clangxx "clang++${EXECUTABLE_SUFFIX}")
 set(clang_cl "clang-cl${EXECUTABLE_SUFFIX}")
@@ -34,7 +34,7 @@
   COMMAND "${CMAKE_COMMAND}" -E ${CLANGXX_LINK_OR_COPY} "${clang}" "${clang_cl}"
   WORKING_DIRECTORY "${bindir}")
 
-if (WIN32)
+if (MSVC)
   message("Creating cl executable based on ${clang}")
 
   execute_process(
