diff -Nru openssl-1.0.0.orig/include/openssl/dtls1.h openssl-1.0.0/include/openssl/dtls1.h
--- openssl-1.0.0.orig/include/openssl/dtls1.h	2009-09-09 19:05:42.000000000 +0200
+++ openssl-1.0.0/include/openssl/dtls1.h	2010-04-29 17:38:21.511143500 +0200
@@ -66,7 +66,7 @@
 #include <resource.h>
 #include <sys/timeb.h>
 #endif
-#ifdef OPENSSL_SYS_WIN32
+#if defined(OPENSSL_SYS_WIN32)  && !defined(_WINSOCK2API_)
 /* Needed for struct timeval */
 #include <winsock.h>
 #elif defined(OPENSSL_SYS_NETWARE) && !defined(_WINSOCK2API_)
diff -Nru openssl-1.0.0.orig/ssl/dtls1.h openssl-1.0.0/ssl/dtls1.h
--- openssl-1.0.0.orig/ssl/dtls1.h	2009-09-09 19:05:42.000000000 +0200
+++ openssl-1.0.0/ssl/dtls1.h	2010-04-29 17:37:41.150835000 +0200
@@ -66,7 +66,7 @@
 #include <resource.h>
 #include <sys/timeb.h>
 #endif
-#ifdef OPENSSL_SYS_WIN32
+#if defined(OPENSSL_SYS_WIN32) && !defined(_WINSOCK2API_)
 /* Needed for struct timeval */
 #include <winsock.h>
 #elif defined(OPENSSL_SYS_NETWARE) && !defined(_WINSOCK2API_)
diff -Nru openssl-1.0.0.orig/ssl/ssltest.c openssl-1.0.0/ssl/ssltest.c
--- openssl-1.0.0.orig/ssl/ssltest.c	2010-01-24 17:57:38.000000000 +0100
+++ openssl-1.0.0/ssl/ssltest.c	2010-04-29 17:37:00.586514900 +0200
@@ -189,7 +189,7 @@
 				     get screwed...
 				  */
 
-#ifdef OPENSSL_SYS_WINDOWS
+#if defined(OPENSSL_SYS_WINDOWS) && !defined(_WINSOCK2API_)
 #include <winsock.h>
 #else
 #include OPENSSL_UNISTD
diff -Nru openssl-1.0.0.orig/test/ssltest.c openssl-1.0.0/test/ssltest.c
--- openssl-1.0.0.orig/test/ssltest.c	2010-01-24 17:57:38.000000000 +0100
+++ openssl-1.0.0/test/ssltest.c	2010-04-29 17:37:17.670492000 +0200
@@ -189,7 +189,7 @@
 				     get screwed...
 				  */
 
-#ifdef OPENSSL_SYS_WINDOWS
+#if deffined(OPENSSL_SYS_WINDOWS)  && !defined(_WINSOCK2API_)
 #include <winsock.h>
 #else
 #include OPENSSL_UNISTD
