diff -Nru -x '*~' kde-runtime-4.10.2.orig/kioslave/trash/trashimpl.cpp kde-runtime-4.10.2/kioslave/trash/trashimpl.cpp
--- kde-runtime-4.10.2.orig/kioslave/trash/trashimpl.cpp	2013-03-01 06:34:47.415076000 +0000
+++ kde-runtime-4.10.2/kioslave/trash/trashimpl.cpp	2013-05-19 23:10:57.696329400 +0100
@@ -308,7 +308,11 @@
 {
     const QString realPath = KStandardDirs::realFilePath( path );
     // topdir ends with '/'
+#ifndef Q_OS_WIN
     if ( realPath.startsWith( topdir ) ) {
+#else
+    if ( realPath.startsWith( topdir, Qt::CaseInsensitive ) ) {
+#endif
         const QString rel = realPath.mid( topdir.length() );
         Q_ASSERT( rel[0] != QLatin1Char('/') );
         return rel;
