diff -Nru kde-runtime-4.7.0.orig/drkonqi/tests/backtraceparsertest/CMakeLists.txt kde-runtime-4.7.0/drkonqi/tests/backtraceparsertest/CMakeLists.txt
--- kde-runtime-4.7.0.orig/drkonqi/tests/backtraceparsertest/CMakeLists.txt	2011-05-20 22:32:25.000000000 +0200
+++ kde-runtime-4.7.0/drkonqi/tests/backtraceparsertest/CMakeLists.txt	2011-08-25 10:53:38.585941500 +0200
@@ -5,10 +5,12 @@
 target_link_libraries(backtraceparsertest_manual drkonqi_backtrace_parser)
 
 # create a link so that the data directory can be accessed from the build directory, where the test runs
-if ( NOT ${CMAKE_CURRENT_BINARY_DIR} EQUAL ${CMAKE_CURRENT_SOURCE_DIR} )
-    add_custom_command(TARGET backtraceparsertest POST_BUILD
-                       COMMAND ${CMAKE_COMMAND} -E create_symlink
-                            ${CMAKE_CURRENT_SOURCE_DIR}/backtraceparsertest_data backtraceparsertest_data
-                       WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-                      )
-endif ( NOT ${CMAKE_CURRENT_BINARY_DIR} EQUAL ${CMAKE_CURRENT_SOURCE_DIR} )
+if( NOT WIN32 )
+	if ( NOT ${CMAKE_CURRENT_BINARY_DIR} EQUAL ${CMAKE_CURRENT_SOURCE_DIR} )
+		add_custom_command(TARGET backtraceparsertest POST_BUILD
+						   COMMAND ${CMAKE_COMMAND} -E create_symlink
+								${CMAKE_CURRENT_SOURCE_DIR}/backtraceparsertest_data backtraceparsertest_data
+						   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+						  )
+	endif ( NOT ${CMAKE_CURRENT_BINARY_DIR} EQUAL ${CMAKE_CURRENT_SOURCE_DIR} )
+endif( NOT WIN32 )
\ No newline at end of file
