diff -Nru -x '*~' kde-runtime-4.12.0.orig/kioslave/thumbnail/icoutils_win.cpp kde-runtime-4.12.0/kioslave/thumbnail/icoutils_win.cpp
--- kde-runtime-4.12.0.orig/kioslave/thumbnail/icoutils_win.cpp	2013-12-11 19:40:13.000000000 +0100
+++ kde-runtime-4.12.0/kioslave/thumbnail/icoutils_win.cpp	2013-12-19 18:39:36.733528000 +0100
@@ -16,7 +16,6 @@
 #include "icoutils.h"
 
 #include <windows.h>
-#include <strsafe.h>
 
 #include <QBuffer>
 #include <QDir>
@@ -99,7 +98,7 @@
     LPCTSTR fileName;
     QList<LPTSTR> iconResources;
 
-    fileName = QDir::toNativeSeparators( inputFileName ).utf16();
+    fileName = (TCHAR*) QDir::toNativeSeparators( inputFileName ).utf16();
 
     hModule = LoadLibraryEx ( fileName, 0, LOAD_LIBRARY_AS_DATAFILE );
 
