diff -Nru kde-runtime-4.8.1.orig/kioslave/fish/generate_fishcode.py kde-runtime-4.8.1/kioslave/fish/generate_fishcode.py
--- kde-runtime-4.8.1.orig/kioslave/fish/generate_fishcode.py	2011-07-27 20:36:16.000000000 +0200
+++ kde-runtime-4.8.1/kioslave/fish/generate_fishcode.py	2012-03-19 16:55:15.342892700 +0100
@@ -18,7 +18,7 @@
 r = fp.read()
 fp.close()
 
-fp = open(sys.argv[3], "wb")
+fp = open(sys.argv[3], "wt")
 fp.write('#define CHECKSUM "%s"\n' % sm)
 fp.write('static const char *fishCode(\n')
 
