diff --git a/CMakeLists.txt b/CMakeLists.txt
index 84eda59..25ba403 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -80,9 +80,10 @@ if ( UNIX )
    add_subdirectory(solid-device-automounter)
    add_subdirectory(solid-networkstatus)
 endif ( UNIX )
-if(NOT WIN32 AND SHAREDDESKTOPONTOLOGIES_FOUND AND Nepomuk_FOUND AND STRIGI_STRIGIQTDBUSCLIENT_LIBRARY AND SOPRANO_PLUGIN_REDLANDBACKEND_FOUND AND SOPRANO_PLUGIN_VIRTUOSOBACKEND_FOUND AND SOPRANO_PLUGIN_RAPTORPARSER_FOUND)
+
+if(SHAREDDESKTOPONTOLOGIES_FOUND AND Nepomuk_FOUND AND STRIGI_STRIGIQTDBUSCLIENT_LIBRARY AND SOPRANO_PLUGIN_REDLANDBACKEND_FOUND AND SOPRANO_PLUGIN_VIRTUOSOBACKEND_FOUND AND SOPRANO_PLUGIN_RAPTORPARSER_FOUND)
   macro_optional_add_subdirectory(nepomuk)
-endif(NOT WIN32 AND SHAREDDESKTOPONTOLOGIES_FOUND AND Nepomuk_FOUND AND STRIGI_STRIGIQTDBUSCLIENT_LIBRARY AND SOPRANO_PLUGIN_REDLANDBACKEND_FOUND AND SOPRANO_PLUGIN_VIRTUOSOBACKEND_FOUND AND SOPRANO_PLUGIN_RAPTORPARSER_FOUND)
+endif(SHAREDDESKTOPONTOLOGIES_FOUND AND Nepomuk_FOUND AND STRIGI_STRIGIQTDBUSCLIENT_LIBRARY AND SOPRANO_PLUGIN_REDLANDBACKEND_FOUND AND SOPRANO_PLUGIN_VIRTUOSOBACKEND_FOUND AND SOPRANO_PLUGIN_RAPTORPARSER_FOUND)
 
 # Plugins/components required for basic utility
 if ( NOT WINCE )
diff --git a/nepomuk/common/removablemediacache.h b/nepomuk/common/removablemediacache.h
index 98d2c21..740628b 100644
--- a/nepomuk/common/removablemediacache.h
+++ b/nepomuk/common/removablemediacache.h
@@ -50,7 +50,7 @@ public:
     RemovableMediaCache(QObject *parent = 0);
     ~RemovableMediaCache();
 
-    class Entry {
+    class NEPOMUKCOMMON_EXPORT Entry {
     public:
         Entry();
         Entry(const Solid::Device& device);
diff --git a/nepomuk/services/backupsync/lib/syncresource.h b/nepomuk/services/backupsync/lib/syncresource.h
index a2a70cf..ac3d48c 100644
--- a/nepomuk/services/backupsync/lib/syncresource.h
+++ b/nepomuk/services/backupsync/lib/syncresource.h
@@ -51,13 +51,13 @@ namespace Nepomuk {
          *
          * \author Vishesh Handa <handa.vish@gmail.com>
          */
-        class NEPOMUKSYNC_EXPORT SyncResource : public QMultiHash<KUrl, Soprano::Node>
+        class SyncResource : public QMultiHash<KUrl, Soprano::Node>
         {
         public :
-            SyncResource();
-            SyncResource( const KUrl & uri );
-            SyncResource( const SyncResource & rhs );
-            virtual ~SyncResource();
+            NEPOMUKSYNC_EXPORT SyncResource();
+            NEPOMUKSYNC_EXPORT SyncResource( const KUrl & uri );
+            NEPOMUKSYNC_EXPORT SyncResource( const SyncResource & rhs );
+            NEPOMUKSYNC_EXPORT virtual ~SyncResource();
 
             /**
              * It uses the the first element's subject as the uri and ignores all further subjects.
@@ -65,32 +65,32 @@ namespace Nepomuk {
              */
             static SyncResource fromStatementList(const QList<Soprano::Statement> & list);
 
-            QList<Soprano::Statement> toStatementList() const;
+            NEPOMUKSYNC_EXPORT QList<Soprano::Statement> toStatementList() const;
 
-            bool isFileDataObject() const;
-            bool isFolder() const;
-            KUrl nieUrl() const;
+            NEPOMUKSYNC_EXPORT bool isFileDataObject() const;
+            NEPOMUKSYNC_EXPORT bool isFolder() const;
+            NEPOMUKSYNC_EXPORT KUrl nieUrl() const;
 
-            KUrl uri() const;
+            NEPOMUKSYNC_EXPORT KUrl uri() const;
 
             /**
              * If \p node is resource node the uri is set to the node's uri
              * Otherwise if \p node is a blank node then the uri
              * is set to its identifier
              */
-            void setUri( const Soprano::Node & node );
+            NEPOMUKSYNC_EXPORT void setUri( const Soprano::Node & node );
 
-            QList<Soprano::Node> property( const KUrl & url ) const;
+            NEPOMUKSYNC_EXPORT QList<Soprano::Node> property( const KUrl & url ) const;
 
             /**
              * Removes all the statements whose object is \p uri
              */
-            void removeObject( const KUrl & uri );
+            NEPOMUKSYNC_EXPORT void removeObject( const KUrl & uri );
 
-            SyncResource& operator=( const SyncResource & rhs );
-            bool operator==( const SyncResource & res ) const;
+            NEPOMUKSYNC_EXPORT SyncResource& operator=( const SyncResource & rhs );
+            NEPOMUKSYNC_EXPORT bool operator==( const SyncResource & res ) const;
 
-            bool isValid() const;
+            NEPOMUKSYNC_EXPORT bool isValid() const;
         private:
             class Private;
             QSharedDataPointer<Private> d;
diff --git a/nepomuk/services/storage/removablemediamodel.h b/nepomuk/services/storage/removablemediamodel.h
index 4926439..d0d92cc 100644
--- a/nepomuk/services/storage/removablemediamodel.h
+++ b/nepomuk/services/storage/removablemediamodel.h
@@ -114,6 +114,8 @@ private:
 
     class StatementIteratorBackend;
     class QueryResultIteratorBackend;
+
+    friend class RemovableMediaModelTest;
 };
 }
 
diff --git a/nepomuk/services/storage/test/removablemediamodeltest.cpp b/nepomuk/services/storage/test/removablemediamodeltest.cpp
index 2cac0aa..09be8b1 100644
--- a/nepomuk/services/storage/test/removablemediamodeltest.cpp
+++ b/nepomuk/services/storage/test/removablemediamodeltest.cpp
@@ -20,10 +20,13 @@
 */
 
 #include "removablemediamodeltest.h"
-
+#ifdef __GNUC__
 #define private public
+#endif
 #include "../removablemediamodel.h"
+#ifdef __GNUC__
 #undef private
+#endif
 
 #include <QtTest>
 #include "qtest_kde.h"
