diff -Nru libvorbis-1.3.1.orig/CMakeLists.txt libvorbis-1.3.1/CMakeLists.txt
--- libvorbis-1.3.1.orig/CMakeLists.txt	Thu Jan  1 02:00:00 1970
+++ libvorbis-1.3.1/CMakeLists.txt	Thu Jul  8 10:17:49 2010
@@ -0,0 +1,6 @@
+project (libvorbis)
+
+cmake_minimum_required (VERSION 2.6)
+
+add_subdirectory(include)
+add_subdirectory(lib)
diff -Nru libvorbis-1.3.1.orig/include/CMakeLists.txt libvorbis-1.3.1/include/CMakeLists.txt
--- libvorbis-1.3.1.orig/include/CMakeLists.txt	Thu Jan  1 02:00:00 1970
+++ libvorbis-1.3.1/include/CMakeLists.txt	Thu Jul  8 09:31:25 2010
@@ -0,0 +1 @@
+install(FILES vorbis/codec.h vorbis/vorbisenc.h vorbis/vorbisfile.h DESTINATION include/vorbis)
\ No newline at end of file
diff -Nru libvorbis-1.3.1.orig/lib/CMakeLists.txt libvorbis-1.3.1/lib/CMakeLists.txt
--- libvorbis-1.3.1.orig/lib/CMakeLists.txt	Thu Jan  1 02:00:00 1970
+++ libvorbis-1.3.1/lib/CMakeLists.txt	Thu Jul  8 10:24:39 2010
@@ -0,0 +1,51 @@
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
+
+find_package(Ogg REQUIRED)
+
+set(vorbis_SRCS
+analysis.c
+bitrate.c
+block.c
+codebook.c
+envelope.c
+floor0.c
+floor1.c
+info.c
+lookup.c
+lpc.c
+lsp.c
+mapping0.c
+mdct.c
+psy.c
+registry.c
+res0.c
+sharedbook.c
+smallft.c
+synthesis.c
+vorbisenc.c
+window.c
+../win32/vorbis.def
+)
+
+set(vorbisenc_SRCS
+vorbisenc.c
+../win32/vorbisenc.def
+)
+
+set(vorbisfile_SRCS
+vorbisfile.c
+../win32/vorbisfile.def
+)
+
+include_directories(${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR} ${OGG_INCLUDE_DIR} ../include)
+
+add_library(vorbis SHARED ${vorbis_SRCS})
+target_link_libraries(vorbis ${OGG_LIBRARY})
+add_library(vorbisenc SHARED ${vorbisenc_SRCS})
+target_link_libraries(vorbisenc ${OGG_LIBRARY} vorbis)
+add_library(vorbisfile SHARED ${vorbisfile_SRCS})
+target_link_libraries(vorbisfile ${OGG_LIBRARY} vorbis)
+
+install(TARGETS vorbis vorbisenc vorbisfile RUNTIME DESTINATION bin
+                           LIBRARY DESTINATION lib
+                           ARCHIVE DESTINATION lib)
\ No newline at end of file
diff -Nru libvorbis-1.3.1.orig/lib/cmake/modules/FindOgg.cmake libvorbis-1.3.1/lib/cmake/modules/FindOgg.cmake
--- libvorbis-1.3.1.orig/lib/cmake/modules/FindOgg.cmake	Thu Jan  1 02:00:00 1970
+++ libvorbis-1.3.1/lib/cmake/modules/FindOgg.cmake	Thu Jul  8 10:22:37 2010
@@ -0,0 +1,28 @@
+# - Try to find the Ogg library
+# This will define
+#
+#  OGG_FOUND
+#  OGG_INCLUDE_DIR
+#  OGG_LIBRARY
+
+
+IF (OGG_INCLUDE_DIR)
+    SET(OGG_FIND_QUIETLY TRUE)
+ENDIF (OGG_INCLUDE_DIR)
+
+find_path(OGG_INCLUDE_DIR ogg/ogg.h)
+
+find_library(OGG_LIBRARY NAMES ogg)
+
+mark_as_advanced(OGG_INCLUDE_DIR OGG_LIBRARY)
+
+
+if (OGG_LIBRARY)
+   set(OGG_FOUND TRUE)
+endif (OGG_LIBRARY)
+
+   set(_CMAKE_REQUIRED_LIBRARIES_TMP ${CMAKE_REQUIRED_LIBRARIES})
+   set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} ${OGG_LIBRARY})
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(OGG DEFAULT_MSG OGG_LIBRARY OGG_INCLUDE_DIR)
\ No newline at end of file
