diff -Nru libofx-0.9.1.orig/CMakeLists.txt libofx-0.9.1/CMakeLists.txt
--- libofx-0.9.1.orig/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ libofx-0.9.1/CMakeLists.txt	2011-01-11 03:57:56.436686900 +0100
@@ -0,0 +1,98 @@
+project(ofx)
+
+set (VERSION_MAJOR 0)
+set (VERSION_MINOR 9)
+set (VERSION_PATH 1)
+
+cmake_minimum_required(VERSION 2.6)
+
+list(APPEND CMAKE_MODULE_PATH  ${CMAKE_SOURCE_DIR})
+find_package(OpenSP REQUIRED)
+
+if(MSVC)
+    add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+endif(MSVC)
+
+include_directories(
+    inc
+    ${OPENSP_INCLUDES}
+    $ENV{KDEROOT}/include
+)
+
+add_definitions(
+    -DOS_WIN32
+    -DMAKEFILE_DTD_PATH="${CMAKE_PREFIX_PATH}/share/libofx/dtd/"
+)
+
+set (COEFF_HDRS 
+        inc/libofx.h
+        lib/messages.hh
+        lib/ofx_preproc.hh
+        lib/file_preproc.hh
+        lib/context.hh
+        lib/ofx_sgml.hh
+        lib/ofc_sgml.hh
+        lib/ofx_aggregate.hh
+        lib/ofx_error_msg.hh
+        lib/ofx_containers.hh
+        lib/ofx_request.hh
+        lib/ofx_request_accountinfo.hh
+        lib/ofx_request_statement.hh
+        lib/ofx_utilities.hh
+        lib/tree.hh
+        lib/win32.hh
+        )
+
+set (SRC_SOURCES
+         lib/messages.cpp
+         lib/ofx_utilities.cpp
+         lib/file_preproc.cpp
+         lib/context.cpp
+         lib/ofx_preproc.cpp
+         lib/ofx_container_generic.cpp
+         lib/ofx_container_main.cpp
+         lib/ofx_container_security.cpp
+         lib/ofx_container_statement.cpp
+         lib/ofx_container_account.cpp
+         lib/ofx_container_transaction.cpp
+         lib/ofx_containers_misc.cpp
+         lib/ofx_request.cpp
+         lib/ofx_request_accountinfo.cpp
+         lib/ofx_request_statement.cpp
+         lib/ofx_sgml.cpp
+         lib/ofc_sgml.cpp
+         lib/win32.cpp
+         ${COEFF_HDRS}
+)
+
+# We would like it to be shared, but libofx does not export symbols manually
+# thus it causes troubles on MSVC
+#add_library(ofx SHARED ${SRC_SOURCES})
+add_library(ofx ${SRC_SOURCES})
+
+FIND_LIBRARY( ICONV_LIBRARIES iconv )
+if(MSVC)
+    set_target_properties(ofx PROPERTIES OUTPUT_NAME "libofx")
+endif(MSVC)
+target_link_libraries(ofx ${OPENSP_LIBRARIES} ${ICONV_LIBRARIES})
+
+######### add a utility function so that we can test ofx files ##########
+set(ofxdump_SRCS
+    ofxdump/cmdline.c
+    ofxdump/ofxdump.cpp 
+)
+
+if(MSVC)
+    set(ofxdump_SRCS ${ofxdump_SRCS} lib/messages.cpp lib/getopt.c lib/getopt1.c)
+endif(MSVC)
+
+add_definitions(-DCMDLINE_PARSER_PACKAGE=\"ofxdump\" -DCMDLINE_PARSER_VERSION=\"0.9.1\")
+add_executable(ofxdump ${ofxdump_SRCS})
+target_link_libraries(ofxdump ofx)
+
+install(TARGETS ofx ofxdump RUNTIME DESTINATION bin ARCHIVE DESTINATION lib LIBRARY DESTINATION lib)
+install(FILES inc/libofx.h DESTINATION include/libofx)
+install(FILES dtd/opensp.dcl DESTINATION ${CMAKE_PREFIX_PATH}/share/libofx/dtd)
+install(FILES dtd/ofx160.dtd DESTINATION ${CMAKE_PREFIX_PATH}/share/libofx/dtd)
+install(FILES dtd/ofx201.dtd DESTINATION ${CMAKE_PREFIX_PATH}/share/libofx/dtd)
+install(FILES dtd/ofc.dtd DESTINATION ${CMAKE_PREFIX_PATH}/share/libofx/dtd)
diff -Nru libofx-0.9.1.orig/config.h libofx-0.9.1/config.h
--- libofx-0.9.1.orig/config.h	1970-01-01 01:00:00.000000000 +0100
+++ libofx-0.9.1/config.h	2011-01-11 03:57:56.438687000 +0100
@@ -0,0 +1,71 @@
+// Have these
+#define HAVE_EVENTGENERATOR_H 1
+#define HAVE_LIBCURL 1
+#define HAVE_MEMORY_H 1
+#define HAVE_PARSEREVENTGENERATORKIT_H 1
+#define HAVE_SGMLAPPLICATION_H 1
+#define HAVE_STDLIB_H 1
+#define HAVE_STRING_H 1
+#define HAVE_SYS_STAT_H 1
+#define HAVE_SYS_TYPES_H 1
+
+// Do not exist
+#undef HAVE_DLFCN_H
+#undef HAVE_GETOPT_H
+#undef HAVE_GETOPT_LONG
+#undef HAVE_INTTYPES_H
+#undef HAVE_STDINT_H
+#undef HAVE_STRINGS_H
+#undef HAVE_UNISTD_H
+
+
+// Unsure
+#define HAVE_ICONV 1
+#define HAVE_LIBXMLPP 1
+
+#undef LIBCURL_FEATURE_ASYNCHDNS
+
+/* Defined if libcurl supports IPv6 */
+#undef LIBCURL_FEATURE_IPV6
+
+/* Defined if libcurl supports KRB4 */
+#undef LIBCURL_FEATURE_KRB4
+
+/* Defined if libcurl supports libz */
+#undef LIBCURL_FEATURE_LIBZ
+
+/* Defined if libcurl supports SSL */
+#undef LIBCURL_FEATURE_SSL
+
+/* Defined if libcurl supports DICT */
+#undef LIBCURL_PROTOCOL_DICT
+
+/* Defined if libcurl supports FILE */
+#undef LIBCURL_PROTOCOL_FILE
+
+/* Defined if libcurl supports FTP */
+#undef LIBCURL_PROTOCOL_FTP
+
+/* Defined if libcurl supports FTPS */
+#undef LIBCURL_PROTOCOL_FTPS
+
+/* Defined if libcurl supports GOPHER */
+#undef LIBCURL_PROTOCOL_GOPHER
+
+/* Defined if libcurl supports HTTP */
+#undef LIBCURL_PROTOCOL_HTTP
+
+/* Defined if libcurl supports HTTPS */
+#undef LIBCURL_PROTOCOL_HTTPS
+
+/* Defined if libcurl supports LDAP */
+#undef LIBCURL_PROTOCOL_LDAP
+
+/* Defined if libcurl supports TELNET */
+#undef LIBCURL_PROTOCOL_TELNET
+
+
+// MSVC Defines we seem to need
+#ifdef _MSC_VER
+typedef int ssize_t;
+#endif
diff -Nru libofx-0.9.1.orig/FindOpenSP.cmake libofx-0.9.1/FindOpenSP.cmake
--- libofx-0.9.1.orig/FindOpenSP.cmake	1970-01-01 01:00:00.000000000 +0100
+++ libofx-0.9.1/FindOpenSP.cmake	2011-01-11 03:57:56.440687100 +0100
@@ -0,0 +1,36 @@
+# - Try to find the OpenSP library
+# 
+# Once done this will define
+#
+#  OPENSP_FOUND - system has OPENSP
+#  OPENSP_INCLUDES - the OPENSP include directories
+#  OPENSP_LIBRARIES - The libraries needed to use OPENSP
+
+if (WIN32)
+
+    set(OPENSP_FOUND FALSE)
+    find_path(OPENSP_INCLUDES ParserEventGeneratorKit.h
+        ${CMAKE_PREFIX_PATH}/include/opensp
+    )
+
+    find_library(OPENSP_LIBRARIES
+        NAMES sp133
+        PATHS
+            ${CMAKE_PREFIX_PATH}/lib
+    )
+
+    if (OPENSP_INCLUDES AND OPENSP_OPENSP_LIBRARIES)
+        set(OPENSP_FOUND TRUE)
+    endif (OPENSP_INCLUDES AND OPENSP_OPENSP_LIBRARIES)
+
+  if (OPENSP_FOUND)
+    if (NOT OPENSP_FIND_QUIETLY)
+      message(STATUS "Found OPENSP library: ${OPENSP_LIBRARIES}")
+    endif (NOT OPENSP_FIND_QUIETLY)
+
+  else (OPENSP_FOUND)
+    if (OPENSP_FIND_REQUIRED)
+      message(FATAL_ERROR "Could NOT find OPENSP library\nPlease install it first")
+    endif (OPENSP_FIND_REQUIRED)
+  endif (OPENSP_FOUND)
+endif (WIN32)
diff -Nru libofx-0.9.1.orig/lib/getopt.c libofx-0.9.1/lib/getopt.c
--- libofx-0.9.1.orig/lib/getopt.c	2004-04-24 01:30:34.000000000 +0200
+++ libofx-0.9.1/lib/getopt.c	2011-01-11 03:57:56.475689100 +0100
@@ -40,6 +40,7 @@
 #endif
 
 #include <stdio.h>
+#include "getopt.h"
 
 /* Comment out all this code if we are using the GNU C Library, and are not
    actually compiling the library itself.  This code is part of the GNU C
@@ -199,7 +200,7 @@
 # define my_index	strchr
 #else
 
-# if HAVE_STRING_H
+# if HAVE_STRING_H || defined(_MSC_VER)
 #  include <string.h>
 # else
 #  include <strings.h>
diff -Nru libofx-0.9.1.orig/lib/getopt.h libofx-0.9.1/lib/getopt.h
--- libofx-0.9.1.orig/lib/getopt.h	1970-01-01 01:00:00.000000000 +0100
+++ libofx-0.9.1/lib/getopt.h	2011-01-11 03:57:56.478689300 +0100
@@ -0,0 +1,20 @@
+#ifndef _getopt_h
+#define _getopt_h
+
+#ifndef _MSC_VER
+#include <getopt.h>
+#else
+struct option {
+    const char *name;
+    int has_arg;
+    int *flag;
+    int val;
+};
+
+extern char *optarg;
+extern int optind;
+
+
+int getopt_long(int argc, char *const *argv, const char *shortopts, const struct option *longopts, int *indexptr);
+#endif
+#endif
diff -Nru libofx-0.9.1.orig/lib/messages.cpp libofx-0.9.1/lib/messages.cpp
--- libofx-0.9.1.orig/lib/messages.cpp	2004-03-08 16:42:17.000000000 +0100
+++ libofx-0.9.1/lib/messages.cpp	2011-01-11 03:57:56.480689400 +0100
@@ -25,6 +25,10 @@
 SGMLApplication::OpenEntityPtr entity_ptr; /**< Global for determining the line number in OpenSP */
 SGMLApplication::Position position; /**< Global for determining the line number in OpenSP */
 
+#ifdef _MSC_VER
+#define volatile
+#endif
+
 volatile int ofx_PARSER_msg = false; /**< If set to true, parser events will be printed to the console */
 volatile int ofx_DEBUG_msg = false;/**< If set to true, general debug messages will be printed to the console */
 volatile int ofx_DEBUG1_msg = false;/**< If set to true, debug level 1 messages will be printed to the console */
diff -Nru libofx-0.9.1.orig/lib/ofx_container_transaction.cpp libofx-0.9.1/lib/ofx_container_transaction.cpp
--- libofx-0.9.1.orig/lib/ofx_container_transaction.cpp	2007-11-10 00:13:21.000000000 +0100
+++ libofx-0.9.1/lib/ofx_container_transaction.cpp	2011-01-11 03:57:56.483689600 +0100
@@ -22,6 +22,7 @@
 #endif
 
 #include <string>
+#include <cstdlib>
 #include "messages.hh"
 #include "libofx.h"
 #include "ofx_containers.hh"
diff -Nru libofx-0.9.1.orig/lib/ofx_preproc.cpp libofx-0.9.1/lib/ofx_preproc.cpp
--- libofx-0.9.1.orig/lib/ofx_preproc.cpp	2008-12-06 23:22:57.000000000 +0100
+++ libofx-0.9.1/lib/ofx_preproc.cpp	2011-01-11 03:57:56.486689700 +0100
@@ -23,6 +23,11 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string>
+#ifdef OS_WIN32
+# include <windows.h>
+# undef ERROR
+# undef DELETE
+#endif
 #include "ParserEventGeneratorKit.h"
 #include "libofx.h"
 #include "messages.hh"
@@ -530,6 +535,29 @@
     }
   }
 
+#ifdef _WIN32
+  if (!dtd_found) {
+    char module_path[MAX_PATH];
+    GetModuleFileName(NULL, module_path, MAX_PATH);
+    // the path is of the form C:\\Path\\to\\ofx\\bin\\ofxdump.exe
+    // cut of the string after C:\\Path\\to\\ofx
+    if(NULL != strrchr(module_path, '\\')) module_path[strrchr(module_path, '\\') - module_path] = '\0';
+    if(NULL != strrchr(module_path, '\\')) module_path[strrchr(module_path, '\\') - module_path] = '\0';
+    
+    dtd_path_filename=module_path;
+    dtd_path_filename.append("\\share\\libofx\\dtd\\"+dtd_filename);
+    dtd_file.clear();
+    dtd_file.open(dtd_path_filename.c_str());
+    if(!dtd_file){
+      message_out(DEBUG,"find_dtd():Unable to open the file "+dtd_path_filename);
+    }
+    else{
+      message_out(STATUS,"find_dtd():DTD found: "+dtd_path_filename);
+      dtd_file.close();
+      dtd_found=true;
+    }
+  }
+#endif
   if (!dtd_found) {
     for(i=0;i<DTD_SEARCH_PATH_NUM&&dtd_found==false;i++){
       dtd_path_filename=DTD_SEARCH_PATH[i];
diff -Nru libofx-0.9.1.orig/lib/ofx_request.cpp libofx-0.9.1/lib/ofx_request.cpp
--- libofx-0.9.1.orig/lib/ofx_request.cpp	2007-10-28 04:01:38.000000000 +0100
+++ libofx-0.9.1/lib/ofx_request.cpp	2011-01-11 03:57:56.489689900 +0100
@@ -22,6 +22,7 @@
 #endif
 
 #include <string>
+#include <cstring>
 #include "messages.hh"
 #include "libofx.h"
 #include "ofx_request.hh"
diff -Nru libofx-0.9.1.orig/lib/ofx_request_accountinfo.cpp libofx-0.9.1/lib/ofx_request_accountinfo.cpp
--- libofx-0.9.1.orig/lib/ofx_request_accountinfo.cpp	2007-10-28 04:01:38.000000000 +0100
+++ libofx-0.9.1/lib/ofx_request_accountinfo.cpp	2011-01-11 03:57:56.494690200 +0100
@@ -22,6 +22,7 @@
 #endif
 
 #include <string>
+#include <cstdlib>
 #include "libofx.h"
 #include "ofx_request_accountinfo.hh"
 
diff -Nru libofx-0.9.1.orig/lib/ofx_request_statement.cpp libofx-0.9.1/lib/ofx_request_statement.cpp
--- libofx-0.9.1.orig/lib/ofx_request_statement.cpp	2007-10-28 04:01:38.000000000 +0100
+++ libofx-0.9.1/lib/ofx_request_statement.cpp	2011-01-11 03:57:56.507690900 +0100
@@ -22,6 +22,7 @@
 #endif
 
 #include <string>
+#include <cstdlib>
 #include "libofx.h"
 #include "ofx_utilities.hh"
 #include "ofx_request_statement.hh"
diff -Nru libofx-0.9.1.orig/lib/ofx_utilities.cpp libofx-0.9.1/lib/ofx_utilities.cpp
--- libofx-0.9.1.orig/lib/ofx_utilities.cpp	2007-10-27 14:15:58.000000000 +0200
+++ libofx-0.9.1/lib/ofx_utilities.cpp	2011-01-11 04:16:11.518322000 +0100
@@ -22,6 +22,7 @@
 #include "ParserEventGeneratorKit.h"
 #include "SGMLApplication.h"
 #include <time.h>
+#include <stdlib.h>
 #include <string>
 #include <locale.h>
 #include "messages.hh"
@@ -113,7 +114,7 @@
   char time_zone_specified = false;
 
   time_t temptime;
-  std::time(&temptime);
+  ::time(&temptime);
   local_offset = difftime(mktime(localtime(&temptime)), mktime(gmtime(&temptime)));
   
   if(ofxdate.size()!=0){
diff -Nru libofx-0.9.1.orig/ofxdump/cmdline.c libofx-0.9.1/ofxdump/cmdline.c
--- libofx-0.9.1.orig/ofxdump/cmdline.c	2009-02-09 21:20:24.000000000 +0100
+++ libofx-0.9.1/ofxdump/cmdline.c	2011-01-11 03:57:56.523691800 +0100
@@ -330,8 +330,6 @@
 
   optarg = 0;
   optind = 0;
-  opterr = 1;
-  optopt = '?';
 
   while (1)
     {
diff -Nru libofx-0.9.1.orig/ofxdump/getopt.h libofx-0.9.1/ofxdump/getopt.h
--- libofx-0.9.1.orig/ofxdump/getopt.h	1970-01-01 01:00:00.000000000 +0100
+++ libofx-0.9.1/ofxdump/getopt.h	2011-01-11 03:57:56.525692000 +0100
@@ -0,0 +1 @@
+#include "../lib/getopt.h"
diff -Nru libofx-0.9.1.orig/ofxdump/ofxdump.cpp libofx-0.9.1/ofxdump/ofxdump.cpp
--- libofx-0.9.1.orig/ofxdump/ofxdump.cpp	2006-08-02 06:19:22.000000000 +0200
+++ libofx-0.9.1/ofxdump/ofxdump.cpp	2011-01-11 03:57:56.597696100 +0100
@@ -29,7 +29,8 @@
  ***************************************************************************/
 #include <iostream>
 #include <iomanip>
-#include <string>
+#include <cstring>
+#include <cstdlib>
 #include "libofx.h"
 #include <stdio.h>		/* for printf() */
 #include <config.h>		/* Include config constants, e.g., VERSION TF */
@@ -39,7 +40,6 @@
 
 using namespace std;
 
-
 int ofx_proc_security_cb(struct OfxSecurityData data, void * security_data)
 {
   char dest_string[255];
