diff -Nru kdelibs-4.5.4.orig/kdecore/CMakeLists.txt kdelibs-4.5.4/kdecore/CMakeLists.txt
--- kdelibs-4.5.4.orig/kdecore/CMakeLists.txt	2010-08-04 21:12:54.000000000 +0200
+++ kdelibs-4.5.4/kdecore/CMakeLists.txt	2011-01-04 16:54:10.108006500 +0100
@@ -493,6 +493,9 @@
 ########### install files ###############
 install( FILES ${CMAKE_CURRENT_BINARY_DIR}/all_languages DESTINATION ${LOCALE_INSTALL_DIR}  )
 install( FILES kdebug.areas kdebugrc DESTINATION ${CONFIG_INSTALL_DIR} )
+if( WIN32 )
+    install( FILES kdeglobals DESTINATION ${CONFIG_INSTALL_DIR} )
+endif( WIN32 )
 install( FILES
    kdecore_export.h
    compression/kfilterbase.h
diff -Nru kdelibs-4.5.4.orig/kdecore/io/kurl.cpp kdelibs-4.5.4/kdecore/io/kurl.cpp
--- kdelibs-4.5.4.orig/kdecore/io/kurl.cpp	2010-09-30 23:19:05.000000000 +0200
+++ kdelibs-4.5.4/kdecore/io/kurl.cpp	2010-12-28 18:24:19.724170900 +0100
@@ -852,7 +852,9 @@
 QString KUrl::path( AdjustPathOption trailing ) const
 {
 #ifdef Q_WS_WIN
+#ifdef DEBUG_KURL
   kWarning() << (isLocalFile() ? "converted to local file - the related call should be converted to toLocalFile()" : "") << QUrl::path();
+#endif
   return trailingSlash( trailing, isLocalFile() ? QUrl::toLocalFile() : QUrl::path() );
 #else
   return trailingSlash( trailing, QUrl::path() );
@@ -1081,6 +1083,10 @@
   }
 
   tmp = path();
+#ifdef Q_WS_WIN
+  if (isLocalFile())
+    tmp.prepend(QLatin1Char('/')); // KUrl::path() returns toLocalFile() on windows so we need to add the / back to create a proper url
+#endif
   result += toPrettyPercentEncoding(tmp, false);
 
   // adjust the trailing slash, if necessary
diff -Nru kdelibs-4.5.4.orig/kdecore/kdeglobals kdelibs-4.5.4/kdecore/kdeglobals
--- kdelibs-4.5.4.orig/kdecore/kdeglobals	1970-01-01 01:00:00.000000000 +0100
+++ kdelibs-4.5.4/kdecore/kdeglobals	2011-01-04 16:53:22.336274200 +0100
@@ -0,0 +1,2 @@
+[General]
+widgetStyle=oxygen
