diff -Nru -x '*~' ninja-1.5.3.orig/configure.py ninja-1.5.3/configure.py
--- ninja-1.5.3.orig/configure.py	2014-11-24 18:37:47.000000000 +0100
+++ ninja-1.5.3/configure.py	2014-12-03 11:11:26.032414700 +0100
@@ -164,7 +164,7 @@
         """Run a subcommand, quietly.  Prints the full command on error."""
         try:
             subprocess.check_call(cmdline, shell=True)
-        except subprocess.CalledProcessError, e:
+        except subprocess.CalledProcessError:
             print('when running: ', cmdline)
             raise
 
