diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' qtbase-opensource-src-5.6.2.orig/src/3rdparty/angle/src/libANGLE/renderer/d3d/d3d11/Blit11.cpp qtbase-opensource-src-5.6.2/src/3rdparty/angle/src/libANGLE/renderer/d3d/d3d11/Blit11.cpp
--- qtbase-opensource-src-5.6.2.orig/src/3rdparty/angle/src/libANGLE/renderer/d3d/d3d11/Blit11.cpp	2016-09-16 07:49:42.000000000 +0200
+++ qtbase-opensource-src-5.6.2/src/3rdparty/angle/src/libANGLE/renderer/d3d/d3d11/Blit11.cpp	2016-10-27 10:01:35.665229000 +0200
@@ -9,6 +9,7 @@
 #include "libANGLE/renderer/d3d/d3d11/Blit11.h"

 #include <float.h>
+#include <cmath>

 #include "libANGLE/renderer/d3d/d3d11/Renderer11.h"
 #include "libANGLE/renderer/d3d/d3d11/renderer11_utils.h"

