diff --git a/src/3rdparty/angle/src/libANGLE/renderer/d3d/d3d11/Query11.cpp b/src/3rdparty/angle/src/libANGLE/renderer/d3d/d3d11/Query11.cpp
index 972c289..cbea0b0 100644
--- a/src/3rdparty/angle/src/libANGLE/renderer/d3d/d3d11/Query11.cpp
+++ b/src/3rdparty/angle/src/libANGLE/renderer/d3d/d3d11/Query11.cpp
@@ -20,12 +20,12 @@ typedef struct D3D11_QUERY_DATA_SO_STATISTICS {
 } D3D11_QUERY_DATA_SO_STATISTICS;
 #endif // ANGLE_MINGW32_COMPAT
 
-#ifdef __MINGW32__
+#if defined(__MINGW32__) && !(defined(__MINGW64_VERSION_MAJOR) && __MINGW64_VERSION_MAJOR >= 5)
 typedef struct D3D11_QUERY_DATA_TIMESTAMP_DISJOINT {
   UINT64 Frequency;
   BOOL   Disjoint;
 } D3D11_QUERY_DATA_TIMESTAMP_DISJOINT;
-#endif // MINGW32
+#endif // defined(__MINGW32__) && !(defined(__MINGW64_VERSION_MAJOR) && __MINGW64_VERSION_MAJOR >= 5)
 
 namespace rx
 {
