diff --git a/mkspecs/win32-g++/qmake.conf b/mkspecs/win32-g++/qmake.conf
index 830dde9..0bd5e1a 100644
--- a/mkspecs/win32-g++/qmake.conf
+++ b/mkspecs/win32-g++/qmake.conf
@@ -30,6 +30,7 @@ QMAKE_CFLAGS_DEPS       = -M
 QMAKE_CFLAGS_WARN_ON    = -Wall -Wextra
 QMAKE_CFLAGS_WARN_OFF   = -w
 QMAKE_CFLAGS_RELEASE    = -O2
+QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO = -O2 -g
 QMAKE_CFLAGS_DEBUG      = -g
 QMAKE_CFLAGS_YACC       = -Wno-unused -Wno-parentheses
 QMAKE_CFLAGS_SSE2       = -msse2 -mstackrealign
diff --git a/src/corelib/io/qstandardpaths_win.cpp b/src/corelib/io/qstandardpaths_win.cpp
index 062ab49..f5af59b 100644
--- a/src/corelib/io/qstandardpaths_win.cpp
+++ b/src/corelib/io/qstandardpaths_win.cpp
@@ -212,6 +212,7 @@ QStringList QStandardPaths::standardLocations(StandardLocation type)
                     dirs.append(QCoreApplication::applicationDirPath());
                     dirs.append(QCoreApplication::applicationDirPath() + QLatin1String("/data"));
                 }
+                dirs.append(QCoreApplication::applicationDirPath() + QLatin1String("/../share"));
 #endif
             }
             break;
diff --git a/tools/configure/configureapp.cpp b/tools/configure/configureapp.cpp
index 06f78b1..8772ccf 100644
--- a/tools/configure/configureapp.cpp
+++ b/tools/configure/configureapp.cpp
@@ -2940,11 +2940,11 @@ void Configure::generateOutputVars()
     }
     if (dictionary[ "DBUS" ] != "no" && !dbusPath.isEmpty()) {
         qmakeVars += QString("QT_CFLAGS_DBUS = -I%1/include").arg(dbusPath);
-        qmakeVars += QString("QT_LIBS_DBUS = -L%1/lib").arg(dbusPath);
+        qmakeVars += QString("QT_LIBS_DBUS = -L%1/lib -ldbus-1").arg(dbusPath);
     }
     if (dictionary[ "SQL_MYSQL" ] != "no" && !mysqlPath.isEmpty()) {
         qmakeVars += QString("QT_CFLAGS_MYSQL = -I%1/include").arg(mysqlPath);
-        qmakeVars += QString("QT_LFLAGS_MYSQL = -L%1/lib").arg(mysqlPath);
+        qmakeVars += QString("QT_LFLAGS_MYSQL = -L%1/lib -llibmysql").arg(mysqlPath);
     }
     if (!psqlLibs.isEmpty())
         qmakeVars += QString("QT_LFLAGS_PSQL=") + psqlLibs.section("=", 1);
