diff --git a/qmake/generators/makefile.cpp b/qmake/generators/makefile.cpp
index d356ada..2058fa0 100644
--- a/qmake/generators/makefile.cpp
+++ b/qmake/generators/makefile.cpp
@@ -1182,7 +1182,7 @@ MakefileGenerator::filePrefixRoot(const QString &root, const QString &path)
 {
     QString ret(path);
     if(path.length() > 2 && path[1] == ':') //c:\foo
-        ret.insert(2, root);
+        ret.replace(0, 2, root);
     else
         ret.prepend(root);
     while (ret.endsWith('\\'))
diff --git a/src/3rdparty/angle/src/libANGLE/renderer/d3d/d3d11/Query11.cpp b/src/3rdparty/angle/src/libANGLE/renderer/d3d/d3d11/Query11.cpp
index 972c289..cbea0b0 100644
--- a/src/3rdparty/angle/src/libANGLE/renderer/d3d/d3d11/Query11.cpp
+++ b/src/3rdparty/angle/src/libANGLE/renderer/d3d/d3d11/Query11.cpp
@@ -20,12 +20,12 @@ typedef struct D3D11_QUERY_DATA_SO_STATISTICS {
 } D3D11_QUERY_DATA_SO_STATISTICS;
 #endif // ANGLE_MINGW32_COMPAT
 
-#ifdef __MINGW32__
+#if defined(__MINGW32__) && !(defined(__MINGW64_VERSION_MAJOR) && __MINGW64_VERSION_MAJOR >= 5)
 typedef struct D3D11_QUERY_DATA_TIMESTAMP_DISJOINT {
   UINT64 Frequency;
   BOOL   Disjoint;
 } D3D11_QUERY_DATA_TIMESTAMP_DISJOINT;
-#endif // MINGW32
+#endif // defined(__MINGW32__) && !(defined(__MINGW64_VERSION_MAJOR) && __MINGW64_VERSION_MAJOR >= 5)
 
 namespace rx
 {
diff --git a/src/corelib/io/qprocess_win.cpp b/src/corelib/io/qprocess_win.cpp
index 96b39ef..2f58427 100644
--- a/src/corelib/io/qprocess_win.cpp
+++ b/src/corelib/io/qprocess_win.cpp
@@ -867,16 +867,17 @@ bool QProcessPrivate::startDetached(const QString &program, const QStringList &a
     bool success = false;
     PROCESS_INFORMATION pinfo;
 
+    DWORD dwCreationFlags = (GetConsoleWindow() ? 0 : CREATE_NO_WINDOW);
+    dwCreationFlags |= CREATE_UNICODE_ENVIRONMENT;
     STARTUPINFOW startupInfo = { sizeof( STARTUPINFO ), 0, 0, 0,
                                  (ulong)CW_USEDEFAULT, (ulong)CW_USEDEFAULT,
                                  (ulong)CW_USEDEFAULT, (ulong)CW_USEDEFAULT,
                                  0, 0, 0, 0, 0, 0, 0, 0, 0, 0
                                };
     success = CreateProcess(0, (wchar_t*)args.utf16(),
-                            0, 0, FALSE, CREATE_UNICODE_ENVIRONMENT | CREATE_NEW_CONSOLE, 0,
+                            0, 0, FALSE, dwCreationFlags, 0,
                             workingDir.isEmpty() ? 0 : (wchar_t*)workingDir.utf16(),
                             &startupInfo, &pinfo);
-
     if (success) {
         CloseHandle(pinfo.hThread);
         CloseHandle(pinfo.hProcess);
