diff --git a/mkspecs/features/qt_build_extra.prf b/mkspecs/features/qt_build_extra.prf
index 378f5bb..4e53f8c 100644
--- a/mkspecs/features/qt_build_extra.prf
+++ b/mkspecs/features/qt_build_extra.prf
@@ -13,7 +13,7 @@ equals(TEMPLATE, subdirs): return()
 
 # It's likely that these extra flags will be wrong for host builds,
 # and the bootstrapped tools usually don't need them anyway.
-host_build:force_bootstrap: return()
+# host_build:force_bootstrap: return()
 
 # The headersclean check needs defines and includes even for
 # header-only modules.
