diff -Nru -x '*~' phonon-backend-vlc-0.7.0.orig/CMakeLists.txt phonon-backend-vlc-0.7.0/CMakeLists.txt
--- phonon-backend-vlc-0.7.0.orig/CMakeLists.txt	2013-11-04 17:49:24.000000000 +0100
+++ phonon-backend-vlc-0.7.0/CMakeLists.txt	2013-11-10 15:04:37.386381300 +0100
@@ -10,8 +10,9 @@
 
 # LibVLC
+set(LIBVLC_MIN_VERSION "2.0.1")
 find_package(LIBVLC REQUIRED)
-macro_ensure_version("2.0.1" ${LIBVLC_VERSION} LIBVLC_FOUND)
-macro_log_feature(LIBVLC_FOUND "LibVLC" "VLC C API Library" "git://git.videolan.org/vlc.git" TRUE "2.0.1")
+#macro_ensure_version("2.0.1" ${LIBVLC_VERSION} LIBVLC_FOUND)
+#macro_log_feature(LIBVLC_FOUND "LibVLC" "VLC C API Library" "git://git.videolan.org/vlc.git" TRUE "2.0.1")
 
 if(PHONON_FOUND_EXPERIMENTAL)
   add_definitions(-DPHONON_EXPERIMENTAL)
diff -Nru -x '*~' phonon-backend-vlc-0.7.0.orig/src/CMakeLists.txt phonon-backend-vlc-0.7.0/src/CMakeLists.txt
--- phonon-backend-vlc-0.7.0.orig/src/CMakeLists.txt	2013-11-04 17:49:24.000000000 +0100
+++ phonon-backend-vlc-0.7.0/src/CMakeLists.txt	2013-11-10 15:00:39.954801000 +0100
@@ -49,9 +49,9 @@
     utils/libvlc.cpp
 )
 
-if(${LIBVLC_VERSION} VERSION_GREATER "2.2.0" OR ${LIBVLC_VERSION} VERSION_EQUAL "2.2.0")
-    list(APPEND phonon_vlc_SRCS equalizereffect.cpp)
-endif()
+#if(${LIBVLC_VERSION} VERSION_GREATER "2.2.0" OR ${LIBVLC_VERSION} VERSION_EQUAL "2.2.0")
+#    list(APPEND phonon_vlc_SRCS equalizereffect.cpp)
+#endif()
 
 if(NOT PHONON_NO_GRAPHICSVIEW)
     list(APPEND phonon_vlc_SRCS video/videographicsobject.cpp)
