diff -Nru -x '*~' kdelibs-4.10.2.orig/kdecore/config/kconfig.cpp kdelibs-4.10.2/kdecore/config/kconfig.cpp
--- kdelibs-4.10.2.orig/kdecore/config/kconfig.cpp	2013-03-29 07:00:27.950067000 +0000
+++ kdelibs-4.10.2/kdecore/config/kconfig.cpp	2013-05-19 15:57:06.453007500 +0100
@@ -560,7 +560,11 @@
         return;
     }
 
+#ifndef Q_OS_WIN
     bSuppressGlobal = (file == sGlobalFileName);
+#else
+    bSuppressGlobal = (file.compare(sGlobalFileName, Qt::CaseInsensitive) == 0);
+#endif
 
     if (bDynamicBackend || !mBackend) // allow dynamic changing of backend
         mBackend = KConfigBackend::create(componentData, file);
@@ -616,7 +620,11 @@
     const QByteArray utf8Locale = locale.toUtf8();
     foreach(const QString& file, globalFiles) {
         KConfigBackend::ParseOptions parseOpts = KConfigBackend::ParseGlobal|KConfigBackend::ParseExpansions;
+#ifndef Q_OS_WIN
         if (file != sGlobalFileName)
+#else
+        if (file.compare(sGlobalFileName, Qt::CaseInsensitive) != 0)
+#endif
             parseOpts |= KConfigBackend::ParseDefaults;
 
         KSharedPtr<KConfigBackend> backend = KConfigBackend::create(componentData, file);
@@ -651,7 +659,11 @@
 
         const QByteArray utf8Locale = locale.toUtf8();
         foreach(const QString& file, files) {
+#ifndef Q_OS_WIN
             if (file == mBackend->filePath()) {
+#else
+            if (file.compare(mBackend->filePath(), Qt::CaseInsensitive) == 0) {
+#endif
                 switch (mBackend->parseConfig(utf8Locale, entryMap, KConfigBackend::ParseExpansions)) {
                 case KConfigBackend::ParseOk:
                     break;
diff -Nru -x '*~' kdelibs-4.10.2.orig/kdecore/config/kdesktopfile.cpp kdelibs-4.10.2/kdecore/config/kdesktopfile.cpp
--- kdelibs-4.10.2.orig/kdecore/config/kdesktopfile.cpp	2013-03-01 06:26:13.619785000 +0000
+++ kdelibs-4.10.2/kdecore/config/kdesktopfile.cpp	2013-05-19 23:16:54.188155600 +0100
@@ -153,7 +153,11 @@
 
   // Check if the .desktop file is installed as part of KDE or XDG.
   foreach (const QString &prefix, kdePrefixes) {
+#ifndef Q_OS_WIN
     if (realPath.startsWith(prefix))
+#else
+    if (realPath.startsWith(prefix, Qt::CaseInsensitive))
+#endif
       return true;
   }
 
diff -Nru -x '*~' kdelibs-4.10.2.orig/kdecore/io/kmountpoint.cpp kdelibs-4.10.2/kdecore/io/kmountpoint.cpp
--- kdelibs-4.10.2.orig/kdecore/io/kmountpoint.cpp	2013-03-01 06:26:13.639785000 +0000
+++ kdelibs-4.10.2/kdecore/io/kmountpoint.cpp	2013-05-19 15:43:45.329200400 +0100
@@ -489,7 +489,11 @@
     for (const_iterator it = begin(); it != end(); ++it) {
         const QString mountpoint = (*it)->d->mountPoint;
         const int length = mountpoint.length();
+#ifndef Q_OS_WIN
         if (realname.startsWith(mountpoint) && length > max) {
+#else
+        if (realname.startsWith(mountpoint, Qt::CaseInsensitive) && length > max) {
+#endif
             max = length;
             result = *it;
             // keep iterating to check for a better match (bigger max)
@@ -504,8 +508,13 @@
     if (realDevice.isEmpty()) // d->device can be empty in the loop below, don't match empty with it
         return Ptr();
     for (const_iterator it = begin(); it != end(); ++it) {
+#ifndef Q_OS_WIN
         if ((*it)->d->device == realDevice ||
             (*it)->d->mountedFrom == realDevice)
+#else
+        if (realDevice.compare((*it)->d->device, Qt::CaseInsensitive) == 0 ||
+            realDevice.compare((*it)->d->mountedFrom, Qt::CaseInsensitive) == 0)
+#endif
             return *it;
     }
     return Ptr();
diff -Nru -x '*~' kdelibs-4.10.2.orig/kdecore/kernel/kstandarddirs.cpp kdelibs-4.10.2/kdecore/kernel/kstandarddirs.cpp
--- kdelibs-4.10.2.orig/kdecore/kernel/kstandarddirs.cpp	2013-05-20 10:16:58.757169000 +0100
+++ kdelibs-4.10.2/kdecore/kernel/kstandarddirs.cpp	2013-05-19 23:23:40.740469600 +0100
@@ -325,7 +325,11 @@
     if (dir.at(dir.length() - 1) != QLatin1Char('/'))
         dir += QLatin1Char('/');
 
+#ifndef Q_OS_WIN
     if (!d->m_prefixes.contains(dir)) {
+#else
+    if (!d->m_prefixes.contains(dir, Qt::CaseInsensitive)) {
+#endif
         priorityAdd(d->m_prefixes, dir, priority);
         d->m_dircache.clear();
     }
@@ -345,7 +349,11 @@
     if (dir.at(dir.length() - 1) != QLatin1Char('/'))
         dir += QLatin1Char('/');
 
+#ifndef Q_OS_WIN
     if (!d->xdgconf_prefixes.contains(dir)) {
+#else
+    if (!d->xdgconf_prefixes.contains(dir, Qt::CaseInsensitive)) {
+#endif
         priorityAdd(d->xdgconf_prefixes, dir, priority);
         d->m_dircache.clear();
     }
@@ -365,7 +373,11 @@
     if (dir.at(dir.length() - 1) != QLatin1Char('/'))
         dir += QLatin1Char('/');
 
+#ifndef Q_OS_WIN
     if (!d->xdgdata_prefixes.contains(dir)) {
+#else
+    if (!d->xdgdata_prefixes.contains(dir, Qt::CaseInsensitive)) {
+#endif
         priorityAdd(d->xdgdata_prefixes, dir, priority);
         d->m_dircache.clear();
     }
@@ -413,7 +425,11 @@
     QByteArray typeBa = type;
     QStringList& rels = d->m_relatives[typeBa]; // find or insert
 
+#ifndef Q_OS_WIN
     if (!rels.contains(copy)) {
+#else
+    if (!rels.contains(copy, Qt::CaseInsensitive)) {
+#endif
         if (priority)
             rels.prepend(copy);
         else
@@ -439,7 +455,11 @@
 
     QByteArray typeBa = type;
     QStringList &paths = d->m_absolutes[typeBa];
+#ifndef Q_OS_WIN
     if (!paths.contains(copy)) {
+#else
+    if (!paths.contains(copy, Qt::CaseInsensitive)) {
+#endif
         if (priority)
             paths.prepend(copy);
         else
@@ -472,7 +492,7 @@
     QString filename(_filename);
 #ifdef Q_OS_WIN
     if(strcmp(type, "exe") == 0) {
-      if(!filename.endsWith(QLatin1String(".exe")))
+      if(!filename.endsWith(QLatin1String(".exe"), Qt::CaseInsensitive))
         filename += QLatin1String(".exe");
     }
 #endif
@@ -560,7 +580,7 @@
     QString filename(_filename);
 #ifdef Q_OS_WIN
     if(strcmp(type, "exe") == 0) {
-      if(!filename.endsWith(QLatin1String(".exe")))
+      if(!filename.endsWith(QLatin1String(".exe"), Qt::CaseInsensitive))
         filename += QLatin1String(".exe");
     }
 #endif
@@ -646,7 +666,11 @@
                 }
                 if ( !bIsDir )
                 {
+#ifndef Q_OS_WIN
                     if ( !unique || !relList.contains(relPart + fn) )
+#else
+                    if ( !unique || !relList.contains(relPart + fn, Qt::CaseInsensitive) )
+#endif
                     {
                         list.append( pathfn );
                         relList.append( relPart + fn );
@@ -703,7 +727,11 @@
             }
             if ( isReg )
             {
+#ifndef Q_OS_WIN
                 if (!unique || !relList.contains(relPart + fn))
+#else
+                if (!unique || !relList.contains(relPart + fn, Qt::CaseInsensitive))
+#endif
                 {
                     list.append( pathfn );
                     relList.append( relPart + fn );
@@ -723,7 +751,11 @@
             return; // File not found
         if ( S_ISREG( buff.st_mode))
         {
+#ifndef Q_OS_WIN
             if (!unique || !relList.contains(relPart + fn))
+#else
+            if (!unique || !relList.contains(relPart + fn, Qt::CaseInsensitive))
+#endif
             {
                 list.append( pathfn );
                 relList.append( relPart + fn );
@@ -979,7 +1011,7 @@
     }
     if (len == 0)
         return QString();
-    return QString::fromUtf16((const unsigned short*)buf.data()).replace(QLatin1Char('\\'),QLatin1Char('/')).toLower();
+    return QString::fromUtf16((const unsigned short*)buf.data()).replace(QLatin1Char('\\'),QLatin1Char('/'));
 #else
     char realpath_buffer[MAXPATHLEN + 1];
     memset(realpath_buffer, 0, MAXPATHLEN + 1);
@@ -1107,14 +1139,8 @@
 
         const QStringList dirs = m_relatives.value(type);
         const QString typeInstallPath = installPath(type); // could be empty
-// better #ifdef incasesensitive_filesystem
-#ifdef Q_WS_WIN
-        const QString installdir = typeInstallPath.isEmpty() ? QString() : realPath(typeInstallPath).toLower();
-        const QString installprefix = installPath("kdedir").toLower();
-#else
         const QString installdir = typeInstallPath.isEmpty() ? QString() : realPath(typeInstallPath);
         const QString installprefix = installPath("kdedir");
-#endif
         if (!dirs.isEmpty())
         {
             bool local = true;
@@ -1131,13 +1157,13 @@
                     for (QStringList::ConstIterator it2 = basedirs.begin();
                          it2 != basedirs.end(); ++it2)
                     {
-#ifdef Q_WS_WIN
-                        const QString path = realPath( *it2 + rest ).toLower();
-#else
                         const QString path = realPath( *it2 + rest );
-#endif
                         testdir.setPath(path);
+#ifndef Q_OS_WIN
                         if ((local || testdir.exists()) && !candidates.contains(path))
+#else
+                        if ((local || testdir.exists()) && !candidates.contains(path, Qt::CaseInsensitive))
+#endif
                             candidates.append(path);
                         local = false;
                     }
@@ -1156,22 +1182,26 @@
                  pit != prefixList->end();
                  ++pit)
             {
+#ifndef Q_OS_WIN
             if((*pit)!=installprefix||installdir.isEmpty())
+#else
+            if((*pit).compare(installprefix, Qt::CaseInsensitive)!=0||installdir.isEmpty())
+#endif
             {
                     for (QStringList::ConstIterator it = dirs.constBegin();
                          it != dirs.constEnd(); ++it)
                     {
                         if ((*it).startsWith(QLatin1Char('%')))
                             continue;
-#ifdef Q_WS_WIN
-                        const QString path = realPath( *pit + *it ).toLower();
-#else
                         const QString path = realPath( *pit + *it );
-#endif
                         testdir.setPath(path);
                         if (local && restrictionActive)
                             continue;
+#ifndef Q_OS_WIN
                         if ((local || testdir.exists()) && !candidates.contains(path))
+#else
+                        if ((local || testdir.exists()) && !candidates.contains(path, Qt::CaseInsensitive))
+#endif
                             candidates.append(path);
                     }
                     local = false;
@@ -1180,7 +1210,11 @@
             {
                     // we have a custom install path, so use this instead of <installprefix>/<relative dir>
                 testdir.setPath(installdir);
+#ifndef Q_OS_WIN
                     if(testdir.exists() && ! candidates.contains(installdir))
+#else
+                    if(testdir.exists() && ! candidates.contains(installdir, Qt::CaseInsensitive))
+#endif
                         candidates.append(installdir);
             }
         }
@@ -1190,7 +1224,11 @@
         if (!installdir.isEmpty()) {
             bool ok = true;
             foreach (const QString &s, candidates) {
+#ifndef Q_OS_WIN
                 if (installdir.startsWith(s)) {
+#else
+                if (installdir.startsWith(s, Qt::CaseInsensitive)) {
+#endif
                     ok = false;
                     break;
                 }
@@ -1205,12 +1243,12 @@
         {
             testdir.setPath(*it);
             if (testdir.exists()) {
-#ifdef Q_WS_WIN
-                const QString filename = realPath( *it ).toLower();
-#else
                 const QString filename = realPath( *it );
-#endif
+#ifndef Q_OS_WIN
                 if (!candidates.contains(filename)) {
+#else
+                if (!candidates.contains(filename, Qt::CaseInsensitive)) {
+#endif
                     candidates.append(filename);
                 }
             }
@@ -1482,7 +1520,7 @@
     Q_FOREACH(const QString& var, envVars.keys())
     {
         const QString varName = QLatin1Char('%') + var + QLatin1Char('%');
-        if(str.contains(varName))
+        if(str.contains(varName, Qt::CaseInsensitive))
         {
             ret.replace(varName, envVars[var], Qt::CaseInsensitive);
         }
@@ -1601,7 +1639,11 @@
 
     for (QStringList::ConstIterator it = candidates.begin();
          it != candidates.end(); ++it) {
+#ifndef Q_OS_WIN
         if (fullPath.startsWith(*it)) {
+#else
+        if (fullPath.startsWith(*it, Qt::CaseInsensitive)) {
+#endif
             return fullPath.mid((*it).length());
         }
     }
@@ -1735,7 +1777,11 @@
     kdedirList.append(installPath("kdedir"));
 
     QString execPrefix(QFile::decodeName(EXEC_INSTALL_PREFIX));
+#ifndef Q_OS_WIN
     if (!execPrefix.isEmpty() && !kdedirList.contains(execPrefix))
+#else
+    if (!execPrefix.isEmpty() && !kdedirList.contains(execPrefix, Qt::CaseInsensitive))
+#endif
         kdedirList.append(execPrefix);
 #ifdef __linux__
     const QString linuxExecPrefix = executablePrefix();
@@ -1876,7 +1922,11 @@
         // otherwise resourceDirs() will add kdedir/share/applications/kde4
         // as returned by installPath(), and that's incorrect.
         Q_FOREACH(const QString& dir, kdedirDataDirs) {
+#ifndef Q_OS_WIN
             if (!xdgdirList.contains(dir))
+#else
+            if (!xdgdirList.contains(dir, Qt::CaseInsensitive))
+#endif
                 xdgdirList.append(dir);
         }
     } else {
diff -Nru -x '*~' kdelibs-4.10.2.orig/kded/vfolder_menu.cpp kdelibs-4.10.2/kded/vfolder_menu.cpp
--- kdelibs-4.10.2.orig/kded/vfolder_menu.cpp	2013-03-01 06:26:13.743784000 +0000
+++ kdelibs-4.10.2/kded/vfolder_menu.cpp	2013-05-19 17:03:09.921769000 +0100
@@ -95,7 +95,11 @@
    QString previous = *it++;
    for(;it != m_allDirectories.end();)
    {
+#ifndef Q_OS_WIN
      if ((*it).startsWith(previous))
+#else
+     if ((*it).startsWith(previous, Qt::CaseInsensitive))
+#endif
      {
         it = m_allDirectories.erase(it);
      }
@@ -1268,7 +1272,11 @@
 
             QString prefix = e.attributes().namedItem("prefix").toAttr().value();
 
+#ifndef Q_OS_WIN
             if (m_defaultLegacyDirs.contains(dir))
+#else
+            if (m_defaultLegacyDirs.contains(dir, Qt::CaseInsensitive))
+#endif
             {
                if (!kdeLegacyDirsDone)
                {
diff -Nru -x '*~' kdelibs-4.10.2.orig/kdeui/actions/krecentfilesaction.cpp kdelibs-4.10.2/kdeui/actions/krecentfilesaction.cpp
--- kdelibs-4.10.2.orig/kdeui/actions/krecentfilesaction.cpp	2013-05-20 10:16:58.725969000 +0100
+++ kdelibs-4.10.2/kdeui/actions/krecentfilesaction.cpp	2013-05-17 23:49:16.331103600 +0100
@@ -171,7 +171,7 @@
     {
 #ifdef Q_OS_WIN
       const QString tmpFile = url.isLocalFile() ? QDir::toNativeSeparators( d->m_urls[action].pathOrUrl() ) : d->m_urls[action].pathOrUrl();
-      if ( tmpFile.endsWith(file) )
+      if ( tmpFile.endsWith(file, Qt::CaseInsensitive) )
 #else
       if ( d->m_urls[action].pathOrUrl().endsWith(file) )
 #endif
