diff --git a/plasma/paintutils.cpp b/plasma/paintutils.cpp
index b1d2ec5..d2ef4d6 100644
--- a/plasma/paintutils.cpp
+++ b/plasma/paintutils.cpp
@@ -147,6 +147,7 @@ QPixmap texturedText(const QString &text, const QFont &font, Plasma::Svg *textur
     buffPainter.setCompositionMode(QPainter::CompositionMode_SourceOver);
     buffPainter.drawImage(QPoint(0,0), image);
     buffPainter.drawPixmap(QPoint(1,1), pixmap);
+    buffPainter.end();
     return ret;
 }
 
