diff -Nru -x '*~' kdelibs-4.10.2.orig/kio/kio/krun.cpp kdelibs-4.10.2/kio/kio/krun.cpp
--- kdelibs-4.10.2.orig/kio/kio/krun.cpp	2013-03-01 06:26:14.263777000 +0000
+++ kdelibs-4.10.2/kio/kio/krun.cpp	2013-05-19 16:16:39.387867700 +0100
@@ -738,7 +738,7 @@
             const KUrl url = *it;
             bool supported = isProtocolInSupportedList(url, appSupportedProtocols);
             kDebug(7010) << "Looking at url=" << url << " supported=" << supported;
-            if (!supported && KProtocolInfo::protocolClass(url.protocol()) == ":local") {
+            if (!supported && KProtocolInfo::protocolClass(url.protocol().toLower()) == ":local") {
                 // Maybe we can resolve to a local URL?
                 KUrl localURL = KIO::NetAccess::mostLocalUrl(url, 0);
                 if (localURL != url) {
@@ -1149,7 +1149,7 @@
         d->m_bIsLocalFile = true;
     }
 
-    if (!d->m_externalBrowser.isEmpty() && d->m_strURL.protocol().startsWith(QLatin1String("http"))) {
+    if (!d->m_externalBrowser.isEmpty() && d->m_strURL.protocol().startsWith(QLatin1String("http"), Qt::CaseInsensitive)) {
         if (d->runExecutable(d->m_externalBrowser)) {
             return;
         }
