diff -Nru kdelibs-4.7.2.orig/cmake/modules/FindHUpnp.cmake kdelibs-4.7.2/cmake/modules/FindHUpnp.cmake
--- kdelibs-4.7.2.orig/cmake/modules/FindHUpnp.cmake	2011-05-20 22:24:53.000000000 +0200
+++ kdelibs-4.7.2/cmake/modules/FindHUpnp.cmake	2011-10-18 19:54:04.556010000 +0200
@@ -13,7 +13,7 @@
 
 find_path( HUPNP_INCLUDE_DIR HUpnpCore/HUpnp )
 
-find_library( HUPNP_LIBS HUpnp )
+find_library( HUPNP_LIBS NAMES HUpnp HUpnp1)
 
 if( HUPNP_INCLUDE_DIR AND EXISTS "${HUPNP_INCLUDE_DIR}/HUpnpCore/public/hupnpinfo.h" )
     file( STRINGS "${HUPNP_INCLUDE_DIR}/HUpnpCore/public/hupnpinfo.h" HUPNP_INFO_H REGEX "^#define HUPNP_CORE_.*_VERSION .*$" )
