diff -Nru kdelibs-4.8.0.orig/kjs/tests/testkjs.cpp kdelibs-4.8.0/kjs/tests/testkjs.cpp
--- kdelibs-4.8.0.orig/kjs/tests/testkjs.cpp	2011-07-27 20:34:37.000000000 +0200
+++ kdelibs-4.8.0/kjs/tests/testkjs.cpp	2012-01-22 22:36:12.646200000 +0100
@@ -175,7 +175,7 @@
 // be in a separate main function because the kjsmain function requires object
 // unwinding.
 
-#if defined(_DEBUG)
+#if defined(_DEBUG) || defined(__MINGW32__)
 #define TRY
 #define EXCEPT(x)
 #else
