diff -Nru -x '*~' kdelibs-4.10.0.orig/kdeui/windowmanagement/kwindowsystem_win.cpp kdelibs-4.10.0/kdeui/windowmanagement/kwindowsystem_win.cpp
--- kdelibs-4.10.0.orig/kdeui/windowmanagement/kwindowsystem_win.cpp	2013-01-23 22:44:20.000000000 +0100
+++ kdelibs-4.10.0/kdeui/windowmanagement/kwindowsystem_win.cpp	2013-02-19 12:36:12.374805300 +0100
@@ -126,6 +126,7 @@
 
 void KWindowSystemPrivate::activate ( )
 {
+#if 0
 	//prevent us from doing the same over and over again
 	if(activated)
 		return;
@@ -150,6 +151,7 @@
 
     //fetch window infos
     reloadStackList();
+#endif
 }
 
 KWindowSystemPrivate::~KWindowSystemPrivate()
@@ -317,7 +319,7 @@
 {
     KWindowSystem::s_d_func()->stackingOrder.clear();
     KWindowSystem::s_d_func()->winInfos.clear();
-    EnumWindows((WNDENUMPROC)EnumWindProc, 0 );
+//    EnumWindows((WNDENUMPROC)EnumWindProc, 0 );
 }
 
 
diff -Nru -x '*~' kdelibs-4.10.0.orig/kjs/CMakeLists.txt kdelibs-4.10.0/kjs/CMakeLists.txt
--- kdelibs-4.10.0.orig/kjs/CMakeLists.txt	2013-01-23 22:44:27.000000000 +0100
+++ kdelibs-4.10.0/kjs/CMakeLists.txt	2013-02-19 12:34:46.153745100 +0100
@@ -234,7 +234,7 @@
 # 'kjs_bin' because cmake doesn't like having a lib and app with the same name
 kde4_add_executable(kjs_bin NOGUI ${kjs_SRCS})
 
-set_target_properties(kjs_bin PROPERTIES OUTPUT_NAME ${kjs_bin_OUTPUT_NAME})
+set_target_properties(kjs_bin PROPERTIES RUNTIME_OUTPUT_NAME ${kjs_bin_OUTPUT_NAME})
 
 target_link_libraries(kjs_bin ${KJSLIBNAME})
 
