diff -Nru -x '*~' kdelibs-4.12.0.orig/kjs/wtf/Platform.h kdelibs-4.12.0/kjs/wtf/Platform.h
--- kdelibs-4.12.0.orig/kjs/wtf/Platform.h	2013-12-12 21:33:53.000000000 +0100
+++ kdelibs-4.12.0/kjs/wtf/Platform.h	2013-12-16 09:59:07.443088700 +0100
@@ -33,11 +33,11 @@
 #endif
 
 /* PLATFORM handles OS, operating environment, graphics API, and CPU */
-#define PLATFORM(WTF_FEATURE) (defined(WTF_PLATFORM_##WTF_FEATURE) && WTF_PLATFORM_##WTF_FEATURE)
-#define COMPILER(WTF_FEATURE) (defined(WTF_COMPILER_##WTF_FEATURE) && WTF_COMPILER_##WTF_FEATURE)
-#define HAVE(WTF_FEATURE) (defined(HAVE_##WTF_FEATURE) && HAVE_##WTF_FEATURE)
-#define USE(WTF_FEATURE) (defined(WTF_USE_##WTF_FEATURE) &&  WTF_USE_##WTF_FEATURE)
-#define ENABLE(WTF_FEATURE) (defined(ENABLE_##WTF_FEATURE) && ENABLE_##WTF_FEATURE)
+#define PLATFORM(WTF_FEATURE) (defined( WTF_PLATFORM_##WTF_FEATURE ) && WTF_PLATFORM_##WTF_FEATURE)
+#define COMPILER(WTF_FEATURE) (defined( WTF_COMPILER_##WTF_FEATURE ) && WTF_COMPILER_##WTF_FEATURE)
+#define HAVE(WTF_FEATURE) (defined( HAVE_##WTF_FEATURE ) && HAVE_##WTF_FEATURE)
+#define USE(WTF_FEATURE) (defined( WTF_USE_##WTF_FEATURE ) &&  WTF_USE_##WTF_FEATURE)
+#define ENABLE(WTF_FEATURE) (defined( ENABLE_##WTF_FEATURE ) && ENABLE_##WTF_FEATURE)
 
 /* Operating systems - low-level dependencies */
 
