diff -Nru kdelibs-4.7.0.orig/solid/solid/backends/wmi/wmimanager.cpp kdelibs-4.7.0/solid/solid/backends/wmi/wmimanager.cpp
--- kdelibs-4.7.0.orig/solid/solid/backends/wmi/wmimanager.cpp  2011-05-20 22:24:54.000000000 +0200
+++ kdelibs-4.7.0/solid/solid/backends/wmi/wmimanager.cpp   2011-09-15 23:57:34.869247600 +0200
@@ -109,7 +109,7 @@
 QStringList WmiManager::devicesFromQuery(const QString &parentUdi,
                                          Solid::DeviceInterface::Type type)
 {
-    qDebug() << parentUdi << type;
+//    qDebug() << parentUdi << type;
     if (!parentUdi.isEmpty())
     {
         QStringList result = findDeviceStringMatch("info.parent", parentUdi);
@@ -151,13 +151,13 @@
     qDebug() << "has to be implemented" << key << value;
     QStringList result;
 
-    qDebug() << result;
+//    qDebug() << result;
     return result;
 }
 
 QStringList WmiManager::findDeviceByDeviceInterface(Solid::DeviceInterface::Type type)
 {
-    qDebug() << type;
+//    qDebug() << type;
     QStringList result;
     WmiQuery::ItemList list;
 
@@ -208,7 +208,7 @@
         break;
     }
 
-    qDebug() << result;
+//    qDebug() << result;
     return result;
 }
 
diff -Nru kdelibs-4.7.0.orig/solid/solid/backends/wmi/wmiquery.cpp kdelibs-4.7.0/solid/solid/backends/wmi/wmiquery.cpp
--- kdelibs-4.7.0.orig/solid/solid/backends/wmi/wmiquery.cpp    2011-09-16 00:10:29.545556500 +0200
+++ kdelibs-4.7.0/solid/solid/backends/wmi/wmiquery.cpp 2011-09-15 23:58:42.689126600 +0200
@@ -158,8 +158,8 @@
               CoUninitialize();
             m_failed = true;
         }
-        else
-            qDebug() << "Connected to ROOT\\CIMV2 WMI namespace" << endl;
+//        else
+//            qDebug() << "Connected to ROOT\\CIMV2 WMI namespace" << endl;
     }
 
     if( !m_failed )
