diff -Nru kdepimlibs-4.8.0.orig/gpgme++/context.cpp kdepimlibs-4.8.0/gpgme++/context.cpp
--- kdepimlibs-4.8.0.orig/gpgme++/context.cpp	2011-07-27 20:34:55.000000000 +0200
+++ kdepimlibs-4.8.0/gpgme++/context.cpp	2012-02-06 20:28:09.968800000 +0100
@@ -116,9 +116,10 @@
   }
 
   const char * Error::asString() const {
-    if ( mMessage.empty() )
-      format_error( static_cast<gpgme_error_t>( mErr ), mMessage );
-    return mMessage.c_str();
+    static char buffer[ 1024 ];
+    gpgme_strerror_r( static_cast<gpgme_error_t>( mErr ), buffer, sizeof buffer );
+    buffer[ sizeof buffer - 1 ] = '\0';
+    return buffer;
   }
 
   int Error::code() const {
diff -Nru kdepimlibs-4.8.0.orig/gpgme++/error.h kdepimlibs-4.8.0/gpgme++/error.h
--- kdepimlibs-4.8.0.orig/gpgme++/error.h	2011-07-27 20:34:55.000000000 +0200
+++ kdepimlibs-4.8.0/gpgme++/error.h	2012-02-06 20:25:23.921400000 +0100
@@ -27,7 +27,6 @@
 #include <gpgme++/gpgmefw.h>
 #include <gpgme++/gpgme++_export.h>
 
-#include <string>
 #include <iosfwd>
 
 #include <gpg-error.h>
@@ -40,8 +39,8 @@
 
   class GPGMEPP_EXPORT Error {
   public:
-    Error() : mErr( 0 ), mMessage() {}
-    explicit Error( unsigned int e ) : mErr( e ), mMessage() {}
+    Error() : mErr( 0 ) {}
+    explicit Error( unsigned int e ) : mErr( e ) {}
 
     const char * source() const;
     const char * asString() const;
@@ -64,7 +63,6 @@
     GPGMEPP_MAKE_SAFE_BOOL_OPERATOR( mErr && !isCanceled() )
   private:
     unsigned int mErr;
-    mutable std::string mMessage;
   };
 
   GPGMEPP_EXPORT std::ostream & operator<<( std::ostream & os, const Error & err );
