diff -Nru dbus-1.4.0.orig/dbus/dbus-connection.c dbus-1.4.0/dbus/dbus-connection.c
--- dbus-1.4.0.orig/dbus/dbus-connection.c	2010-09-06 19:30:42.000000000 +0200
+++ dbus-1.4.0/dbus/dbus-connection.c	2010-12-22 11:08:06.118257800 +0100
@@ -44,6 +44,10 @@
 #include "dbus-bus.h"
 #include "dbus-marshal-basic.h"
 
+#ifdef _WIN32
+  #include <windows.h>
+#endif
+
 #ifdef DBUS_DISABLE_CHECKS
 #define TOOK_LOCK_CHECK(connection)
 #define RELEASING_LOCK_CHECK(connection)
@@ -1136,6 +1140,10 @@
 
   _dbus_verbose ("unlocking io_path_mutex\n");
   _dbus_mutex_unlock (connection->io_path_mutex);
+  //To avoid live lock, due to unfair scheduling
+#ifdef _WIN32
+  Sleep(0);
+#endif
 }
 
 /**
@@ -4146,6 +4154,10 @@
 
   _dbus_verbose ("unlocking dispatch_mutex\n");
   _dbus_mutex_unlock (connection->dispatch_mutex);
+  //To avoid live lock, due to unfair scheduling
+#ifdef _WIN32
+  Sleep(0);
+#endif
 }
 
 static void
