From 84273426d479f4f664c98f180bd2374d76e1bd23 Mon Sep 17 00:00:00 2001
From: Patrick von Reth <patrick.vonreth@gmail.com>
Date: Tue, 9 Aug 2011 11:03:29 +0200
Subject: [PATCH] a inline fix for msvc

---
 bus/bus.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/bus/bus.c b/bus/bus.c
index 04b1286..cd5a943 100644
--- a/bus/bus.c
+++ b/bus/bus.c
@@ -40,6 +40,9 @@
 #include <dbus/dbus-credentials.h>
 #include <dbus/dbus-internals.h>
 
+#ifdef _MSC_VER
+#define inline
+#endif
 #ifdef DBUS_CYGWIN
 #include <signal.h>
 #endif
-- 
1.7.4.msysgit.0

