Index: parley/src/export/exportdialog.cpp
===================================================================
--- parley/src/export/exportdialog.cpp	(Revision 1007397)
+++ parley/src/export/exportdialog.cpp	(Revision 1012792)
@@ -104,7 +104,7 @@
 KUrl ExportDialog::getFileName(const QString& filter)
 {
 
-    KFileDialog dlg( (m_doc->document()->url().fileName() == i18n("Untitled")) ? "": m_doc->document()->url().path(), filter, m_parent );
+    KFileDialog dlg( (m_doc->document()->url().fileName() == i18n("Untitled")) ? "": m_doc->document()->url().toLocalFile(), filter, m_parent );
     dlg.setOperationMode( KFileDialog::Saving );
     dlg.setMode( KFile::File );
     dlg.setWindowTitle(i18n("Export As"));
Index: parley/src/parleydocument.cpp
===================================================================
--- parley/src/parleydocument.cpp	(Revision 1007397)
+++ parley/src/parleydocument.cpp	(Revision 1012792)
@@ -215,8 +215,8 @@
     }
 
     // remove previous backup
-    QFile::remove(m_doc->url().path()+'~');
-    ::rename(QFile::encodeName(m_doc->url().path()), QFile::encodeName(m_doc->url().path()+'~'));
+    QFile::remove(m_doc->url().toLocalFile()+'~');
+    ::rename(QFile::encodeName(m_doc->url().toLocalFile()), QFile::encodeName(m_doc->url().toLocalFile()+'~'));
 
     m_doc->setCsvDelimiter(Prefs::separator());
 
@@ -248,19 +248,19 @@
         return;
     }
     
-    QFileInfo fileinfo(url.path());
+    QFileInfo fileinfo(url.toLocalFile());
     if (fileinfo.exists()) {
         if(KMessageBox::warningContinueCancel(0,
                 i18n("<qt>The file<p><b>%1</b></p>already exists. Do you want to overwrite it?</qt>",
-                url.path()),QString(),KStandardGuiItem::overwrite()) == KMessageBox::Cancel) {
+                url.toLocalFile()),QString(),KStandardGuiItem::overwrite()) == KMessageBox::Cancel) {
             return;
         }
     }
 
-    QString msg = i18nc("@info:status saving a file", "Saving %1", url.path());
+    QString msg = i18nc("@info:status saving a file", "Saving %1", url.toLocalFile());
 
-    QFile::remove(url.path()+'~'); // remove previous backup
-    QFile::rename(QFile::encodeName(url.path()), QFile::encodeName(QString(url.path()+'~')));
+    QFile::remove(url.toLocalFile()+'~'); // remove previous backup
+    QFile::rename(QFile::encodeName(url.toLocalFile()), QFile::encodeName(QString(url.toLocalFile()+'~')));
 
     m_doc->setCsvDelimiter(Prefs::separator());
 
