--- a/glib/glib/gslice.c	2022-01-14 17:22:12.518103600 +0100
+++ b/glib/glib/gslice.c	2022-01-14 17:23:21.437668900 +0100
@@ -1503,17 +1503,17 @@
   found_one = smc_tree_lookup (adress, &real_size);
   if (!found_one)
     {
-      fprintf (stderr, "GSlice: MemChecker: attempt to release non-allocated block: %p size=%" G_GSIZE_FORMAT "\n", pointer, size);
+      fprintf (stderr, "GSlice: MemChecker: attempt to release non-allocated block\n");
       return 0;
     }
   if (real_size != size && (real_size || size))
     {
-      fprintf (stderr, "GSlice: MemChecker: attempt to release block with invalid size: %p size=%" G_GSIZE_FORMAT " invalid-size=%" G_GSIZE_FORMAT "\n", pointer, real_size, size);
+      fprintf (stderr, "GSlice: MemChecker: attempt to release block with invalid size\n");
       return 0;
     }
   if (!smc_tree_remove (adress))
     {
-      fprintf (stderr, "GSlice: MemChecker: attempt to release non-allocated block: %p size=%" G_GSIZE_FORMAT "\n", pointer, size);
+      fprintf (stderr, "GSlice: MemChecker: attempt to release non-allocated block\n");
       return 0;
     }
   return 1; /* all fine */
