Index: kotext/KoTextEditingFactory.h
===================================================================
--- kotext/KoTextEditingFactory.h	(revision 974288)
+++ kotext/KoTextEditingFactory.h	(working copy)
@@ -55,6 +55,11 @@
      */
     QString id() const;
 
+    QString name() const
+    {
+        return QString();
+    }
+
     /**
      * return if the plugin this factory creates has to be shown in the context menu.
      * @see KoTextEditingPlugin::checkSection()
Index: kotext/opendocument/KoTextSharedLoadingData.cpp
===================================================================
--- kotext/opendocument/KoTextSharedLoadingData.cpp	(revision 974288)
+++ kotext/opendocument/KoTextSharedLoadingData.cpp	(working copy)
@@ -33,6 +33,7 @@
 #include <KoXmlNS.h>
 #include <KoOdfStylesReader.h>
 #include <KoOdfLoadingContext.h>
+#include <KoShapeLoadingContext.h>
 
 #include "styles/KoStyleManager.h"
 #include "styles/KoParagraphStyle.h"
