diff -Nru koffice-2.0.82.orig/filters/karbon/pdf/CMakeLists.txt koffice-2.0.82/filters/karbon/pdf/CMakeLists.txt
--- koffice-2.0.82.orig/filters/karbon/pdf/CMakeLists.txt	2009-09-10 18:35:37.000000000 +0200
+++ koffice-2.0.82/filters/karbon/pdf/CMakeLists.txt	2009-09-25 03:24:46.879151700 +0200
@@ -1,11 +1,10 @@
-
-include_directories( ${POPPLER_INCLUDE_DIR} )
+include_directories( ${POPPLER_INCLUDE_DIR} ${POPPLER_INCLUDE_DIR}/../poppler )
 
 set(karbonpdfimport_PART_SRCS PdfImport.cpp SvgOutputDev.cpp )
 
 kde4_add_plugin(karbonpdfimport WITH_PREFIX ${karbonpdfimport_PART_SRCS})
 
-target_link_libraries(karbonpdfimport komain poppler poppler-qt4)
+target_link_libraries(karbonpdfimport komain poppler poppler-qt4 jpeg openjpeg)
 
 install(TARGETS karbonpdfimport DESTINATION ${PLUGIN_INSTALL_DIR})
 
diff -Nru koffice-2.0.82.orig/krita/plugins/formats/pdf/CMakeLists.txt koffice-2.0.82/krita/plugins/formats/pdf/CMakeLists.txt
--- koffice-2.0.82.orig/krita/plugins/formats/pdf/CMakeLists.txt	2009-09-10 18:37:35.000000000 +0200
+++ koffice-2.0.82/krita/plugins/formats/pdf/CMakeLists.txt	2009-09-25 02:33:46.967151700 +0200
@@ -13,7 +13,7 @@
 
 
 
-target_link_libraries(kritapdfimport kritaui ${POPPLER_LIBRARY} m ${KDE4_KDE3SUPPORT_LIBS})
+target_link_libraries(kritapdfimport kritaui ${POPPLER_LIBRARY} ${KDE4_KDE3SUPPORT_LIBS})
 
 install(TARGETS kritapdfimport  DESTINATION ${PLUGIN_INSTALL_DIR})
 ########### install files ###############
diff -Nru koffice-2.0.82.orig/kspread/ui/CharSelectDia.h koffice-2.0.82/kspread/ui/CharSelectDia.h
--- koffice-2.0.82.orig/kspread/ui/CharSelectDia.h	2009-09-10 18:44:24.000000000 +0200
+++ koffice-2.0.82/kspread/ui/CharSelectDia.h	2009-09-25 01:05:37.289151700 +0200
@@ -22,14 +22,13 @@
 #define KOCHARSELECTDIA_H
 
 #include <kdialog.h>
-#include "kowidgets_export.h"
 
 class QWidget;
 
 /**
  * A dialog for selecting a character.
  */
-class KOWIDGETS_EXPORT CharSelectDia : public KDialog
+class CharSelectDia : public KDialog
 {
     Q_OBJECT
 
diff -Nru koffice-2.0.82.orig/libs/main/KoDockerManager.cpp koffice-2.0.82/libs/main/KoDockerManager.cpp
--- koffice-2.0.82.orig/libs/main/KoDockerManager.cpp	2009-09-10 18:43:28.000000000 +0200
+++ koffice-2.0.82/libs/main/KoDockerManager.cpp	2009-09-25 01:04:08.444151700 +0200
@@ -69,7 +69,7 @@
         bool isOnTop = true;
         int idx = view->children().indexOf(j.value());
         foreach(QDockWidget* dock, tabedDocks) {
-          if(view->shell()->children().indexOf(dock) > idx and dock->isVisible() and dock->isEnabled()) {
+          if(view->shell()->children().indexOf(dock) > idx && dock->isVisible() && dock->isEnabled()) {
             isOnTop = false;
             break;
           }
@@ -137,7 +137,7 @@
   QMapIterator<QString, KoToolDocker *> j(d->toolDockerMap);
   while (j.hasNext()) {
     j.next();
-    if (not d->activeToolDockerMap.contains(j.key())) {
+    if (! d->activeToolDockerMap.contains(j.key())) {
         kDebug(/*30004*/) << "removing" << j.key() << ((void*) j.value());
         j.value()->setVisible(false);
         j.value()->setEnabled(false);
diff -Nru koffice-2.0.82.orig/libs/widgets/kowidgets_export.h koffice-2.0.82/libs/widgets/kowidgets_export.h
--- koffice-2.0.82.orig/libs/widgets/kowidgets_export.h	2009-09-10 18:43:38.000000000 +0200
+++ koffice-2.0.82/libs/widgets/kowidgets_export.h	2009-09-25 00:54:28.997151700 +0200
@@ -27,17 +27,7 @@
 #if defined _WIN32 || defined _WIN64
 
 #ifndef KOWIDGETS_EXPORT
-# if defined(MAKE_KOPROGRESS_LIB) || \
-     defined(MAKE_KOASPECTBUTTON_LIB) || \
-     defined(MAKE_KOCHARACTERSELECTIONDIALOG_LIB) || \
-     defined(MAKE_KOLINESTYLESELECTOR_LIB) || \
-     defined(MAKE_KODOCUMENTSECTIONBOX_LIB) || \
-     defined(MAKE_KOOPENPANE_LIB) || \
-     defined(MAKE_KOPAGELAYOUT_LIB) || \
-     defined(MAKE_KOCOLORWIDGETS_LIB) || \
-     defined(MAKE_KOCVSIMPORT_LIB) || \
-     defined(MAKE_KORESOURCEWIDGET_LIB) || \
-     defined(MAKE_KOZOOMACTION_LIB)
+# if defined(MAKE_KOWIDGETS_LIB)
 /* We are building this library */
 #  define KOWIDGETS_EXPORT KDE_EXPORT
 # else
