diff -Nru kdebase-workspace-4.5.4.orig/kcontrol/CMakeLists.txt kdebase-workspace-4.5.4/kcontrol/CMakeLists.txt
--- kdebase-workspace-4.5.4.orig/kcontrol/CMakeLists.txt	2010-05-16 12:04:23.000000000 +0200
+++ kdebase-workspace-4.5.4/kcontrol/CMakeLists.txt	2011-01-10 00:21:11.299833600 +0100
@@ -47,7 +47,9 @@
 add_subdirectory( hardware )
 add_subdirectory( desktoppaths )
 
-if(FONTCONFIG_FOUND AND FREETYPE_FOUND AND NOT WIN32)
+if( FREETYPE_FOUND )
     add_subdirectory( fonts )
+endif( FREETYPE_FOUND )
+if(FONTCONFIG_FOUND AND FREETYPE_FOUND AND NOT WIN32)
     add_subdirectory( kfontinst )
 endif(FONTCONFIG_FOUND AND FREETYPE_FOUND AND NOT WIN32)
diff -Nru kdebase-workspace-4.5.4.orig/kcontrol/fonts/CMakeLists.txt kdebase-workspace-4.5.4/kcontrol/fonts/CMakeLists.txt
--- kdebase-workspace-4.5.4.orig/kcontrol/fonts/CMakeLists.txt	2008-07-03 07:05:23.000000000 +0200
+++ kdebase-workspace-4.5.4/kcontrol/fonts/CMakeLists.txt	2011-01-10 00:25:09.775473700 +0100
@@ -1,22 +1,33 @@
 
-KDE4_NO_ENABLE_FINAL(fonts)
-include_directories(${FONTCONFIG_INCLUDE_DIR})
-include_directories(${FREETYPE_INCLUDE_DIR})
+KDE4_NO_ENABLE_FINAL( fonts )
+if( FONTCONFIG_FOUND )
+    include_directories( ${FONTCONFIG_INCLUDE_DIR} )
+endif( FONTCONFIG_FOUND )
+include_directories( ${FREETYPE_INCLUDE_DIR} )
 
 ########### next target ###############
 
-set(kcm_fonts_PART_SRCS ../krdb/krdb.cpp fonts.cpp ${libkxftconfig_SRCS} )
+set( kcm_fonts_PART_SRCS ../krdb/krdb.cpp fonts.cpp )
+if( Q_WS_X11 )
+    set( kcm_fonts_PART_SRCS ${kcm_fonts_PART_SRCS} ${libkxftconfig_SRCS} )
+endif( Q_WS_X11 )
 
 
-kde4_add_plugin(kcm_fonts ${kcm_fonts_PART_SRCS})
+kde4_add_plugin( kcm_fonts ${kcm_fonts_PART_SRCS} )
 
 
-target_link_libraries(kcm_fonts  ${KDE4_KDEUI_LIBS}
-   ${FONTCONFIG_LIBRARIES}
-   ${FREETYPE_LIBRARIES} ${QT_QTXML_LIBRARY} ${X11_LIBRARIES}
+target_link_libraries( kcm_fonts ${KDE4_KDEUI_LIBS}
+   ${FREETYPE_LIBRARIES} ${QT_QTXML_LIBRARY}
 )
+if( FONTCONFIG_FOUND )
+    target_link_libraries( kcm_fonts ${FONTCONFIG_LIBRARIES} )
+endif( FONTCONFIG_FOUND )
+
+if( Q_WS_X11 )
+    target_link_libraries( kcm_fonts ${X11_LIBRARIES} )
+endif( Q_WS_X11 )
 
-install(TARGETS kcm_fonts  DESTINATION ${PLUGIN_INSTALL_DIR} )
+install( TARGETS kcm_fonts  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
 
 ########### install files ###############
diff -Nru kdebase-workspace-4.5.4.orig/kcontrol/fonts/fonts.cpp kdebase-workspace-4.5.4/kcontrol/fonts/fonts.cpp
--- kdebase-workspace-4.5.4.orig/kcontrol/fonts/fonts.cpp	2010-05-27 19:23:07.000000000 +0200
+++ kdebase-workspace-4.5.4/kcontrol/fonts/fonts.cpp	2011-01-10 00:34:42.315221100 +0100
@@ -50,14 +50,18 @@
 #endif
 #endif
 
+#ifdef Q_WS_X11
 #include <X11/Xlib.h>
+#endif
 
 #include <KPluginFactory>
 
+#ifdef Q_WS_X11
 // X11 headers
 #undef Bool
 #undef Unsorted
 #undef None
+#endif
 
 static const char * const aa_rgb_xpm[]={
 "12 12 3 1",
diff -Nru kdebase-workspace-4.5.4.orig/plasma/desktop/applets/windowlist/WindowList.cpp kdebase-workspace-4.5.4/plasma/desktop/applets/windowlist/WindowList.cpp
--- kdebase-workspace-4.5.4.orig/plasma/desktop/applets/windowlist/WindowList.cpp	2010-09-30 23:20:34.000000000 +0200
+++ kdebase-workspace-4.5.4/plasma/desktop/applets/windowlist/WindowList.cpp	2010-12-14 18:11:17.030884800 +0100
@@ -128,7 +128,7 @@
         if (menu && menu->activeAction() && menu->activeAction()->data().type() == QVariant::ULongLong) {
             QContextMenuEvent *cmEvent = static_cast<QContextMenuEvent *>(event);
             QList<QAction*> actionList;
-            TaskManager::TaskItem item(this, TaskManager::TaskManager::self()->findTask(menu->activeAction()->data().toULongLong()));
+            TaskManager::TaskItem item(this, TaskManager::TaskManager::self()->findTask((WId)menu->activeAction()->data().toULongLong()));
             TaskManager::GroupManager groupManager(this);
             TaskManager::BasicMenu taskMenu(NULL, &item, &groupManager, actionList);
             if (taskMenu.exec(cmEvent->globalPos())) {
@@ -161,7 +161,7 @@
             QDrag *drag = new QDrag(menu);
             QMimeData *mimeData = new QMimeData;
             QByteArray data;
-            WId window = menu->activeAction()->data().toULongLong();
+            WId window = (WId)menu->activeAction()->data().toULongLong();
 
             data.resize(sizeof(WId));
 
@@ -299,10 +299,10 @@
 void WindowList::triggered(QAction *action)
 {
     if (action->data().type() == QVariant::ULongLong) {
-        if (KWindowSystem::activeWindow() == action->data().toULongLong()) {
-            KWindowSystem::minimizeWindow(action->data().toULongLong());
+        if (KWindowSystem::activeWindow() == (WId)action->data().toULongLong()) {
+            KWindowSystem::minimizeWindow((WId)action->data().toULongLong());
         } else {
-            KWindowSystem::activateWindow(action->data().toULongLong());
+            KWindowSystem::activateWindow((WId)action->data().toULongLong());
         }
     }
 }
diff -Nru kdebase-workspace-4.5.4.orig/plasma/generic/shells/plasmoidviewer/fullview.h kdebase-workspace-4.5.4/plasma/generic/shells/plasmoidviewer/fullview.h
--- kdebase-workspace-4.5.4.orig/plasma/generic/shells/plasmoidviewer/fullview.h	2010-07-07 19:10:17.000000000 +0200
+++ kdebase-workspace-4.5.4/plasma/generic/shells/plasmoidviewer/fullview.h	2010-12-14 18:11:17.033884900 +0100
@@ -26,10 +26,7 @@
 #ifndef FULLVIEW_H
 #define FULLVIEW_H
 
-#define private public
 #include <Plasma/Applet>
-#undef private
-
 #include <Plasma/Corona>
 
 #include <QGraphicsView>
