diff -Nru -x '*~' strigi-0.7.5.orig/libstreams/ConfigureChecks.cmake strigi-0.7.5/libstreams/ConfigureChecks.cmake
--- strigi-0.7.5.orig/libstreams/ConfigureChecks.cmake	2011-06-02 06:28:57.000000000 -0300
+++ strigi-0.7.5/libstreams/ConfigureChecks.cmake	2013-05-25 11:08:55.091953400 -0300
@@ -26,6 +26,11 @@
 CHECK_FUNCTION_EXISTS(fchdir HAVE_FCHDIR)               # unused !
 CHECK_FUNCTION_EXISTS(gettimeofday HAVE_GETTIMEOFDAY)   # src/luceneindexer/cluceneindexmanager.cpp, src/luceneindexer/cluceneindexreader.cpp, src/streams/strigi/timeofday.h
 CHECK_FUNCTION_EXISTS(isblank HAVE_ISBLANK)             # src/streams/mailinputstream.cpp, src/streams/strigi/compat.cpp
+if(WIN32 AND "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
+    # On MSVC isblank is absent in both C and C++ mode, however intel compiler has a weird behavior where isblank is present only in C mode.
+    MESSAGE(STATUS "isblank - False positive for intel compiler")
+    SET(HAVE_ISBLANK "")
+endif(WIN32 AND "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
 CHECK_FUNCTION_EXISTS(mkstemp HAVE_MKSTEMP)             # src/streamanalyzer/helperendanalyzer.cpp
 CHECK_FUNCTION_EXISTS(nanosleep HAVE_NANOSLEEP)         # src/storage/sqlitestorage.cpp, src/daemon/indexscheduler.cpp, src/searchclient/cmdlinestrigi.cpp
 CHECK_FUNCTION_EXISTS(setenv HAVE_SETENV)               # src/xmlindexer/peranalyzerxml.cpp
