diff -Nru strigi-0.7.5.orig/CMakeLists.txt strigi-0.7.5/CMakeLists.txt
--- strigi-0.7.5.orig/CMakeLists.txt	2011-06-02 11:32:32.000000000 +0200
+++ strigi-0.7.5/CMakeLists.txt	2012-02-25 14:15:41.765250800 +0100
@@ -66,7 +66,7 @@
   add_subdirectory(strigiutils)
   if(NOT WIN32)
     add_subdirectory(strigidaemon)
-    add_subdirectory(strigiclient) # Must be AFTER strigidaemon
-  endif()
+  endif(NOT WIN32)
+  add_subdirectory(strigiclient) # Must be AFTER strigidaemon
 endif()
 
