diff -Nru strigi-0.7.2.orig/cmake/MacroCheckGccVisibility.cmake strigi-0.7.2/cmake/MacroCheckGccVisibility.cmake
--- strigi-0.7.2.orig/cmake/MacroCheckGccVisibility.cmake	2010-02-03 20:03:07.000000000 +0100
+++ strigi-0.7.2/cmake/MacroCheckGccVisibility.cmake	2010-12-23 14:14:44.357096800 +0100
@@ -13,7 +13,7 @@
    check_cxx_compiler_flag(-fvisibility=hidden ${GccVisibility})
 
    # get the gcc version
-   exec_program(${CMAKE_C_COMPILER} ARGS --version OUTPUT_VARIABLE _gcc_version_info)
+   exec_program(${CMAKE_C_COMPILER} ARGS  ${CMAKE_C_COMPILER_ARG1} --version OUTPUT_VARIABLE _gcc_version_info)
 
    string (REGEX MATCH "[345]\\.[0-9]\\.[0-9]" _gcc_version "${_gcc_version_info}")
    # gcc on mac just reports: "gcc (GCC) 3.3 20030304 ..." without the patch level, handle this here:
@@ -26,7 +26,7 @@
 
    set(_GCC_COMPILED_WITH_BAD_ALLOCATOR FALSE)
    if (GCC_IS_NEWER_THAN_4_1)
-      exec_program(${CMAKE_C_COMPILER} ARGS -v OUTPUT_VARIABLE _gcc_alloc_info)
+      exec_program(${CMAKE_C_COMPILER} ARGS ${CMAKE_C_COMPILER_ARG1} -v OUTPUT_VARIABLE _gcc_alloc_info)
       string(REGEX MATCH "(--enable-libstdcxx-allocator=mt)" _GCC_COMPILED_WITH_BAD_ALLOCATOR "${_gcc_alloc_info}")
    endif (GCC_IS_NEWER_THAN_4_1)
 
diff -Nru strigi-0.7.2.orig/src/combinedindexer/combinedindexmanager.cpp strigi-0.7.2/src/combinedindexer/combinedindexmanager.cpp
--- strigi-0.7.2.orig/src/combinedindexer/combinedindexmanager.cpp	2010-02-03 20:03:07.000000000 +0100
+++ strigi-0.7.2/src/combinedindexer/combinedindexmanager.cpp	2010-12-23 14:14:44.360096900 +0100
@@ -45,6 +45,8 @@
 #include <string>
 #include <set>
 #include <map>
+#include <iterator>
+
 using namespace std;
 using namespace Strigi;
 
diff -Nru strigi-0.7.2.orig/src/dummyindexer/grepindexreader.cpp strigi-0.7.2/src/dummyindexer/grepindexreader.cpp
--- strigi-0.7.2.orig/src/dummyindexer/grepindexreader.cpp	2010-02-03 20:03:07.000000000 +0100
+++ strigi-0.7.2/src/dummyindexer/grepindexreader.cpp	2010-12-23 14:14:44.362097100 +0100
@@ -24,6 +24,7 @@
 #include "indexwriter.h"
 #include "filelister.h"
 #include <set>
+#include <iterator>
 using namespace std;
 using namespace Strigi;
 
diff -Nru strigi-0.7.2.orig/src/streamanalyzer/analyzerloader.cpp strigi-0.7.2/src/streamanalyzer/analyzerloader.cpp
--- strigi-0.7.2.orig/src/streamanalyzer/analyzerloader.cpp	2010-02-03 20:03:07.000000000 +0100
+++ strigi-0.7.2/src/streamanalyzer/analyzerloader.cpp	2010-12-23 14:14:44.364097200 +0100
@@ -21,6 +21,7 @@
 #include "analyzerplugin.h"
 #include <string>
 #include <iostream>
+#include <iterator>
 #include <cstdio>
 #include <cstring>
 #include <assert.h>
diff -Nru strigi-0.7.2.orig/src/streamanalyzer/endplugins/jpegendanalyzer.cpp strigi-0.7.2/src/streamanalyzer/endplugins/jpegendanalyzer.cpp
--- strigi-0.7.2.orig/src/streamanalyzer/endplugins/jpegendanalyzer.cpp	2010-02-03 20:03:07.000000000 +0100
+++ strigi-0.7.2/src/streamanalyzer/endplugins/jpegendanalyzer.cpp	2010-12-23 14:14:44.367097300 +0100
@@ -29,12 +29,11 @@
 #include <exiv2/exif.hpp>
 #include <exiv2/image.hpp>
 #include <exiv2/error.hpp>
-#include <math.h>
 #ifdef _MSC_VER
 // at least MSVC2008 needs this define for defines like M_LN2
 #define _USE_MATH_DEFINES
-#include <cmath>
 #endif
+#include <math.h>
 
 #if defined(__SUNPRO_CC) || defined(_MSC_VER)
 #define HUGE_VALF HUGE_VAL
diff -Nru strigi-0.7.2.orig/src/streamanalyzer/fieldpropertiesdb.cpp strigi-0.7.2/src/streamanalyzer/fieldpropertiesdb.cpp
--- strigi-0.7.2.orig/src/streamanalyzer/fieldpropertiesdb.cpp	2010-02-03 20:03:07.000000000 +0100
+++ strigi-0.7.2/src/streamanalyzer/fieldpropertiesdb.cpp	2010-12-23 14:14:44.370097500 +0100
@@ -23,6 +23,7 @@
 #include "fieldtypes.h"
 #include <vector>
 #include <map>
+#include <iterator>
 #include <iostream>
 #include <set>
 #include <cstdlib>
diff -Nru strigi-0.7.2.orig/src/streams/strigi/strigiconfig.h.cmake strigi-0.7.2/src/streams/strigi/strigiconfig.h.cmake
--- strigi-0.7.2.orig/src/streams/strigi/strigiconfig.h.cmake	2010-02-03 20:03:07.000000000 +0100
+++ strigi-0.7.2/src/streams/strigi/strigiconfig.h.cmake	2010-12-23 14:17:50.436739900 +0100
@@ -84,43 +84,52 @@
 #endif
 
 // our needed types
-#if !@HAVE_INT8_T@
+#if !defined(HAVE_INT8_T)
+ #if !@HAVE_INT8_T@
  #define HAVE_INT8_T 1
  #if ${SIZEOF_CHAR}==1 //is char one byte?
   typedef signed char int8_t;
  #else
   #error Could not determine type for int8_t!
  #endif
+ #endif
 #endif
 
-#if !@HAVE_UINT8_T@
+#if !defined(HAVE_UINT8_T)
+ #if !@HAVE_UINT8_T@
  #define HAVE_UINT8_T 1
  #if ${SIZEOF_CHAR}==1 //is char one byte?
   typedef unsigned char uint8_t;
  #else
   #error Could not determine type for uint8_t!
  #endif
+ #endif
 #endif
 
-#if !@HAVE_INT16_T@
+#if !defined(HAVE_INT16_T)
+ #if !@HAVE_INT16_T@
  #define HAVE_INT16_T 1
  #if ${SIZEOF_SHORT}==2 //is short two bytes?
   typedef short int16_t;
  #else
   #error Could not determine type for int16_t!
  #endif
+ #endif
 #endif
 
-#if !@HAVE_UINT16_T@
+#if !defined(HAVE_UINT16_T)
+ #if !@HAVE_UINT16_T@
  #define HAVE_UINT16_T 1
  #if ${SIZEOF_SHORT}==2 //is short two bytes?
   typedef unsigned short uint16_t;
  #else
   #error Could not determine type for uint16_t!
  #endif
+ #endif
 #endif
 
-#if !@HAVE_INT32_T@
+#if !defined(HAVE_INT32_T)
+ #if !@HAVE_INT32_T@
  #define HAVE_INT32_T 1
  #if ${SIZEOF_INT}==4 //is int four bytes?
   typedef int int32_t;
@@ -129,9 +138,11 @@
  #else
   #error Could not determine type for int32_t!
  #endif
+ #endif
 #endif
 
-#if !@HAVE_UINT32_T@
+#if !defined(HAVE_UINT32_T) 
+ #if !@HAVE_UINT32_T@
  #define HAVE_UINT32_T 1
  #if ${SIZEOF_INT}==4 //is int four bytes?
   typedef unsigned int uint32_t;
@@ -140,9 +151,11 @@
  #else
   #error Could not determine type for uint32_t!
  #endif
+ #endif
 #endif
 
-#if !@HAVE_INT64_T@
+#if !defined(HAVE_INT64_T)
+ #if !@HAVE_INT64_T@
  #define HAVE_INT64_T 1
  #if ${SIZEOF_LONG}==8
   typedef long int64_t;
@@ -151,9 +164,11 @@
  #else
   #error Could not determine type for int64_t!
  #endif
+ #endif
 #endif
 
-#if !@HAVE_UINT64_T@
+#if !defined(HAVE_UINT64_T)
+ #if !@HAVE_UINT64_T@
  #define HAVE_UINT64_T 1
  #if ${SIZEOF_LONG}==8
   typedef unsigned long uint64_t;
@@ -164,11 +179,14 @@
  #else
   #error Could not determine type for uint64_t!
  #endif
+ #endif
 #endif
 
-#if !@HAVE_UINT@
+#if !defined(HAVE_UINT)
+ #if !@HAVE_UINT@
  typedef unsigned int uint;
  #define HAVE_UINT 1
+ #endif
 #endif
 
 #if !@HAVE_SIZE_T@
diff -Nru strigi-0.7.2.orig/src/xmlindexer/peranalyzerxml.cpp strigi-0.7.2/src/xmlindexer/peranalyzerxml.cpp
--- strigi-0.7.2.orig/src/xmlindexer/peranalyzerxml.cpp	2010-02-03 20:03:07.000000000 +0100
+++ strigi-0.7.2/src/xmlindexer/peranalyzerxml.cpp	2010-12-23 14:14:44.384098300 +0100
@@ -38,6 +38,7 @@
 #include <cstring>
 #include <cerrno>
 #include <algorithm>
+#include <iterator>
 #ifdef HAVE_UNISTD_H
  #include <unistd.h>
 #endif
diff -Nru strigi-0.7.2.orig/src/xmlindexer/perfieldxml.cpp strigi-0.7.2/src/xmlindexer/perfieldxml.cpp
--- strigi-0.7.2.orig/src/xmlindexer/perfieldxml.cpp	2010-02-03 20:03:07.000000000 +0100
+++ strigi-0.7.2/src/xmlindexer/perfieldxml.cpp	2010-12-23 14:14:44.386098400 +0100
@@ -37,6 +37,7 @@
 #include <cstring>
 #include <cerrno>
 #include <algorithm>
+#include <iterator>
 #ifdef HAVE_UNISTD_H
  #include <unistd.h>
 #endif
diff -Nru strigi-0.7.2.orig/StrigiConfig.cmake.in strigi-0.7.2/StrigiConfig.cmake.in
--- strigi-0.7.2.orig/StrigiConfig.cmake.in	2010-02-03 20:03:07.000000000 +0100
+++ strigi-0.7.2/StrigiConfig.cmake.in	2010-12-23 14:14:44.392098800 +0100
@@ -27,6 +27,7 @@
 GET_FILENAME_COMPONENT(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
 GET_FILENAME_COMPONENT(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
 GET_FILENAME_COMPONENT(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
+GET_FILENAME_COMPONENT(_IMPORT_PREFIX "${_IMPORT_PREFIX}" REALPATH)
 
 # install locations
 set(STRIGI_INCLUDE_DIR "${_IMPORT_PREFIX}/@INCLUDE_DESTINATION@")
