Index: C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/cmake/FindWcecompat.cmake
===================================================================
--- C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/cmake/FindWcecompat.cmake	(Revision 0)
+++ C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/cmake/FindWcecompat.cmake	(Revision 0)
@@ -0,0 +1,33 @@
+# Try to find Wcecompat functionality
+# Once done this will define
+#
+#  WCECOMPAT_FOUND - system has Wcecompat
+#  WCECOMPAT_INCLUDE_DIR - Wcecompat include directory
+#  WCECOMPAT_LIBRARIES - Libraries needed to use Wcecompat
+#
+# Copyright (c) 2010, Andreas Holzammer, <andy@kdab.com>
+#
+# Redistribution and use is allowed according to the terms of the BSD license.
+
+if(WCECOMPAT_INCLUDE_DIR AND WCECOMPAT_LIB_FOUND)
+  set(Wcecompat_FIND_QUIETLY TRUE)
+endif(WCECOMPAT_INCLUDE_DIR AND WCECOMPAT_LIB_FOUND)
+
+find_path(WCECOMPAT_INCLUDE_DIR errno.h PATH_SUFFIXES wcecompat)
+
+set(WCECOMPAT_LIB_FOUND FALSE)
+
+if(WCECOMPAT_INCLUDE_DIR)
+    find_library(WCECOMPAT_LIBRARIES NAMES wcecompat wcecompatex )
+    if(WCECOMPAT_LIBRARIES)
+      set(WCECOMPAT_LIB_FOUND TRUE)
+    endif(WCECOMPAT_LIBRARIES)
+endif(WCECOMPAT_INCLUDE_DIR)
+
+# I have no idea what this is about, but it seems to be used quite often, so I add this here
+set(WCECOMPAT_CONST const)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(Wcecompat  DEFAULT_MSG  WCECOMPAT_LIBRARIES  WCECOMPAT_LIB_FOUND)
+
+mark_as_advanced(WCECOMPAT_INCLUDE_DIR  WCECOMPAT_LIBRARIES  WCECOMPAT_CONST  WCECOMPAT_LIB_FOUND)
Index: C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/ConfigureChecks.cmake
===================================================================
--- C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/ConfigureChecks.cmake	(Revision 1132854)
+++ C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/ConfigureChecks.cmake	(Arbeitskopie)
@@ -29,7 +29,7 @@
 CHECK_FUNCTION_EXISTS(mkstemp HAVE_MKSTEMP)             # src/streamanalyzer/helperendanalyzer.cpp
 CHECK_FUNCTION_EXISTS(nanosleep HAVE_NANOSLEEP)         # src/storage/sqlitestorage.cpp, src/daemon/indexscheduler.cpp, src/searchclient/cmdlinestrigi.cpp
 CHECK_FUNCTION_EXISTS(setenv HAVE_SETENV)               # src/xmlindexer/peranalyzerxml.cpp
-CHECK_FUNCTION_EXISTS(strcasecmp HAVE_STRCASECMP)       # src/streamindexer/expatsaxendanalyzer.cpp, src/streamindexer/saxendanalyzer.cpp
+CHECK_SYMBOL_EXISTS(strcasecmp "string.h" HAVE_STRCASECMP)       # src/streamindexer/expatsaxendanalyzer.cpp, src/streamindexer/saxendanalyzer.cpp
 CHECK_FUNCTION_EXISTS(strcasestr HAVE_STRCASESTR)       # src/streams/mailinputstream.cpp
 CHECK_FUNCTION_EXISTS(strlwr HAVE_STRLWR)               # src/streamindexer/ifilterendanalyzer.cpp
 CHECK_FUNCTION_EXISTS(strncasecmp HAVE_STRNCASECMP)     # src/streams/mailinputstream.cpp
Index: C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/strigicmd/strigicmd.cpp
===================================================================
--- C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/strigicmd/strigicmd.cpp	(Revision 1132854)
+++ C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/strigicmd/strigicmd.cpp	(Arbeitskopie)
@@ -136,6 +136,9 @@
 }
 void
 printIndexedDocument(IndexedDocument indexedDoc) {
+//Windows ce does not have a console, so no matter of displaying something
+//and the lack of ctime implementation of the sdk
+#ifndef _WIN32_WCE
     printf ("\t- mimetype: %s\n", indexedDoc.mimetype.c_str());
     printf ("\t- sha1: %s\n", indexedDoc.sha1.c_str());
     cout << "\t- size: " << indexedDoc.size << endl;
@@ -166,6 +169,7 @@
                 printf ("\t\t%s\n", it->second.c_str());
         }
     }
+#endif
 }
 
 /*!
Index: C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/streamanalyzer/analyzerloader.cpp
===================================================================
--- C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/streamanalyzer/analyzerloader.cpp	(Revision 1132854)
+++ C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/streamanalyzer/analyzerloader.cpp	(Arbeitskopie)
@@ -44,6 +44,12 @@
 typedef HMODULE StgModuleType;
 #endif
 
+#if defined(UNICODE) || defined(_UNICODE)
+#define T(t) L##t
+#else
+#define T(t) t
+#endif
+
 using namespace std;
 using namespace Strigi;
 
@@ -83,7 +89,7 @@
 AnalyzerLoader::Private::Module::~Module() {
     void(*f)(const AnalyzerFactoryFactory*)
         = (void(*)(const AnalyzerFactoryFactory*))
-        DLSYM(mod, "deleteStrigiAnalyzerFactory");
+        DLSYM(mod, T("deleteStrigiAnalyzerFactory"));
     if (f) {
         f(factory);
     }
@@ -151,8 +157,14 @@
     // do not use RTLD_GLOBAL here
     handle = dlopen(lib, RTLD_LAZY); //note: If neither RTLD_GLOBAL nor RTLD_LOCAL are specified, the default is RTLD_LOCAL.
 #else
+#ifndef _WIN32_WCE
     handle = LoadLibrary(lib);
+#else
+	wchar_t* wlib = wce_mbtowc(lib);
+	handle = LoadLibraryW(wlib);
+	free(wlib);
 #endif
+#endif
     if (!handle) {
 #if defined(HAVE_DLFCN_H) && !defined(_WIN32)
         cerr << "Could not load '" << lib << "':" << dlerror() << endl;
@@ -162,7 +174,7 @@
         return;
     }
     const AnalyzerFactoryFactory* (*f)() = (const AnalyzerFactoryFactory* (*)())
-        DLSYM(handle, "strigiAnalyzerFactory");
+        DLSYM(handle, T("strigiAnalyzerFactory"));
     if (!f) {
 #ifndef WIN32
         fprintf(stderr, "%s\n", dlerror());
Index: C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/streamanalyzer/tests/indextests.cpp
===================================================================
--- C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/streamanalyzer/tests/indextests.cpp	(Revision 1132854)
+++ C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/streamanalyzer/tests/indextests.cpp	(Arbeitskopie)
@@ -37,6 +37,9 @@
 	#include <unistd.h>
 #else
 	#include <direct.h>
+	#ifdef _WIN32_WCE
+		#include <io.h>
+	#endif
 #endif
 
 /**
Index: C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/streamanalyzer/indexpluginloader.cpp
===================================================================
--- C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/streamanalyzer/indexpluginloader.cpp	(Revision 1132854)
+++ C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/streamanalyzer/indexpluginloader.cpp	(Arbeitskopie)
@@ -37,6 +37,12 @@
 #define DLCLOSE FreeLibrary
 #endif
 
+#if defined(UNICODE) || defined(_UNICODE)
+#define T(t) L##t
+#else
+#define T(t) t
+#endif
+
 #ifndef _WIN32
 typedef void* StgModuleType;
 #else
@@ -134,8 +140,14 @@
         // the default is RTLD_LOCAL.
         handle = dlopen(lib.c_str(), RTLD_LOCAL | RTLD_NOW);
 #else
-        handle = LoadLibrary(lib.c_str());
+#ifndef _WIN32_WCE
+		handle = LoadLibrary(lib.c_str());
+#else
+		wchar_t* wlib = wce_mbtowc(lib.c_str());
+		handle = LoadLibraryW(wlib);
+		free(wlib);
 #endif
+#endif
         if (!handle) {
 #if defined(HAVE_DLFCN_H) && !defined(_WIN32)
             cerr << "Could not load '" << lib << "':" << dlerror() << endl;
@@ -146,7 +158,7 @@
             return;
         }
         IndexManager*(*create)(const char*) = (IndexManager*(*)(const char*))
-            DLSYM(handle, "createIndexManager");
+            DLSYM(handle, T("createIndexManager"));
         if (!create) {
 #ifndef WIN32
             fprintf(stderr, "%s\n", dlerror());
@@ -157,7 +169,7 @@
             return;
         }
         void(*destroy)(IndexManager*) = (void(*)(IndexManager*))
-            DLSYM(handle, "deleteIndexManager");
+            DLSYM(handle, T("deleteIndexManager"));
         if (!destroy) {
 #ifndef WIN32
             fprintf(stderr, "%s\n", dlerror());
Index: C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/streams/substreamprovider.h
===================================================================
--- C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/streams/substreamprovider.h	(Revision 1132854)
+++ C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/streams/substreamprovider.h	(Arbeitskopie)
@@ -20,6 +20,7 @@
 #ifndef STRIGI_SUBSTREAMPROVIDER
 #define STRIGI_SUBSTREAMPROVIDER
 
+#include <time.h>
 #include <string>
 #include <map>
 #include "streambase.h"
Index: C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/streams/tests/inputstreamtests.h
===================================================================
--- C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/streams/tests/inputstreamtests.h	(Revision 1132854)
+++ C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/streams/tests/inputstreamtests.h	(Arbeitskopie)
@@ -21,6 +21,9 @@
 #define STRIGI_INPUTSTREAMTESTS
 #ifdef _WIN32
 #include <direct.h>
+#ifdef _WIN32_WCE
+#include <stddef.h>
+#endif
 #else
 #include <unistd.h>
 #endif
Index: C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/streams/CMakeLists.txt
===================================================================
--- C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/streams/CMakeLists.txt	(Revision 1132854)
+++ C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/src/streams/CMakeLists.txt	(Arbeitskopie)
@@ -1,4 +1,6 @@
+if(NOT WINCE)
 add_subdirectory(tests)
+endif(NOT WINCE)
 add_subdirectory(pdf)
 
 include_directories(
@@ -62,6 +64,9 @@
 )
 
 target_link_libraries(streams ${ZLIB_LIBRARIES} ${BZIP2_LIBRARIES} ${ICONV_LIBRARIES})
+if(WINCE)
+    target_link_libraries(streams ${WCECOMPAT_LIBRARIES})
+endif(WINCE)
 if(NOT WIN32)
   configure_file(${CMAKE_CURRENT_SOURCE_DIR}/libstreams.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/libstreams.pc)
 endif(NOT WIN32)
Index: C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/CMakeLists.txt
===================================================================
--- C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/CMakeLists.txt	(Revision 1132854)
+++ C:/kderootce/svn/tags/kdesupport-for-4.4/strigi/CMakeLists.txt	(Arbeitskopie)
@@ -220,6 +220,13 @@
   endif (NOT HAVE_LIBDL)
 endif(NOT WIN32)
 
+if(WINCE)
+    find_package(Wcecompat REQUIRED)
+    include_directories(${WCECOMPAT_INCLUDE_DIR})
+    set(CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES} ${WCECOMPAT_INCLUDE_DIR})
+    set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} ${WCECOMPAT_LIBRARIES})
+endif(WINCE)
+
 # check for visibility support
 macro_check_gcc_visibility(__STRIGI_HAVE_GCC_VISIBILITY)
 
