diff -ru libgpod-0.7.2-org/src/db-artwork-writer.c libgpod-0.7.2/src/db-artwork-writer.c
--- libgpod-0.7.2-org/src/db-artwork-writer.c	2009-04-15 23:18:35.000000000 +0200
+++ libgpod-0.7.2/src/db-artwork-writer.c	2010-02-03 09:21:15.955078100 +0100
@@ -44,7 +44,9 @@
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
+#ifndef WIN32
 #include <sys/mman.h>
+#endif
 #include <sys/stat.h>
 #include <stdio.h>
 #include <sys/types.h>
diff -ru libgpod-0.7.2-org/src/itdb_itunesdb.c libgpod-0.7.2/src/itdb_itunesdb.c
--- libgpod-0.7.2-org/src/itdb_itunesdb.c	2009-04-15 23:18:35.000000000 +0200
+++ libgpod-0.7.2/src/itdb_itunesdb.c	2010-02-03 09:21:16.001953100 +0100
@@ -5335,9 +5335,11 @@
 	itdb->filename = fn;
     }
 
+#ifndef WIN32
     /* make sure all buffers are flushed as some people tend to
        disconnect as soon as gtkpod returns */
     sync ();
+#endif
 
     return result;
 }
@@ -5397,9 +5399,11 @@
 	result = itdb_rename_files (itdb_get_mountpoint (itdb), error);
     }
 
+#ifndef WIN32
     /* make sure all buffers are flushed as some people tend to
        disconnect as soon as gtkpod returns */
     sync ();
+#endif
 
     return result;
 }
@@ -5556,9 +5560,11 @@
     if (result == TRUE)
 	result = itdb_rename_files (itdb_get_mountpoint (itdb), error);
 
+#ifndef WIN32
     /* make sure all buffers are flushed as some people tend to
        disconnect as soon as gtkpod returns */
     sync ();
+#endif
 
     return result;
 }
@@ -5685,9 +5691,11 @@
     wcontents_free (cts);
     g_free (fexp);
 
+#ifndef WIN32
     /* make sure all buffers are flushed as some people tend to
        disconnect as soon as gtkpod returns */
     sync ();
+#endif
 
     return result;
 }
@@ -6372,8 +6380,13 @@
 	goto err_out;
     }
 
+#ifdef WIN32
+    file_out =  g_open (to_file, O_CREAT|O_WRONLY|O_TRUNC|O_BINARY,
+	        0);
+#else
     file_out =  g_open (to_file, O_CREAT|O_WRONLY|O_TRUNC|O_BINARY,
                         S_IRWXU|S_IRWXG|S_IRWXO);
+#endif
     if (file_out < 0)
     {
 	g_set_error (error,
diff -ru libgpod-0.7.2-org/src/ithumb-writer.c libgpod-0.7.2/src/ithumb-writer.c
--- libgpod-0.7.2-org/src/ithumb-writer.c	2009-04-15 23:18:35.000000000 +0200
+++ libgpod-0.7.2/src/ithumb-writer.c	2010-02-03 09:21:16.033203100 +0100
@@ -673,7 +673,11 @@
 				return NULL;
 			}
 			dir = g_build_filename (photos_dir, "Thumbs", NULL);
-			mkdir (dir, 0777);
+#ifdef WIN32
+	        mkdir (dir);
+#else
+	        mkdir (dir, 0777);
+#endif
 			g_free (dir);
 			g_free (photos_dir);
 
@@ -697,7 +701,11 @@
 			return NULL;
 		}
 		dir = g_build_filename (control_dir, "Artwork", NULL);
+#ifdef WIN32
+	 mkdir (dir);
+#else
 		mkdir (dir, 0777);
+#endif
 		g_free (dir);
 		g_free (control_dir);
 
diff -ru libgpod-0.7.2-org/tests/test-checksum.c libgpod-0.7.2/tests/test-checksum.c
--- libgpod-0.7.2-org/tests/test-checksum.c	2009-04-15 23:18:36.000000000 +0200
+++ libgpod-0.7.2/tests/test-checksum.c	2010-02-03 09:21:16.056640600 +0100
@@ -1,3 +1,4 @@
+#ifndef WIN32
 /*
 |  Copyright (C) 2007 Christophe Fergeau <teuf@gnome.org>
 |
@@ -179,3 +180,10 @@
 
     return 0;
 }
+#else /* WIN32 */
+#warning __FILE__ " does not compile on win32"
+int main(int argc, char **argv)
+{
+    return 1;
+}
+#endif
Only in libgpod-0.7.2/tests: test-checksum.c.orig
