diff --git a/CMakeLists.txt b/CMakeLists.txt
index 251d5ad..af3b64b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,6 +47,11 @@ include_directories(
     ${QT_QTGUI_INCLUDE_DIR}
     )
 
+if(WINCE)
+    SET(QT_DBUSCPP2XML_EXECUTABLE ${HOST_BINDIR}/qdbuscpp2xml.exe)
+    SET(QT_DBUSXML2CPP_EXECUTABLE ${HOST_BINDIR}/qdbusxml2cpp.exe)
+endif(WINCE)
+
 configure_file(dbusmenu-qt.pc.in ${CMAKE_BINARY_DIR}/dbusmenu-qt.pc @ONLY)
 
 install(FILES ${CMAKE_BINARY_DIR}/dbusmenu-qt.pc
