diff -Nru kdegraphics-4.4.0.orig/CMakeLists.txt kdegraphics-4.4.0/CMakeLists.txt
--- kdegraphics-4.4.0.orig/CMakeLists.txt	2010-01-06 17:59:27.000000000 +0100
+++ kdegraphics-4.4.0/CMakeLists.txt	2010-01-13 13:24:06.933400000 +0100
@@ -58,7 +58,9 @@
 macro_optional_add_subdirectory(kruler)
 macro_optional_add_subdirectory(svgpart)
 if(QIMAGEBLITZ_FOUND)
-  macro_optional_add_subdirectory(kolourpaint)
+  if(NOT MINGW)
+    macro_optional_add_subdirectory(kolourpaint)
+  endif(NOT MINGW)
   macro_optional_add_subdirectory(okular)
 endif(QIMAGEBLITZ_FOUND)
 