diff -Nru -x '*~' ksirk-4.10.1.orig/ksirk/iris/src/CMakeLists.txt ksirk-4.10.1/ksirk/iris/src/CMakeLists.txt
--- ksirk-4.10.1.orig/ksirk/iris/src/CMakeLists.txt	2013-03-01 08:16:50.866751000 +0100
+++ ksirk-4.10.1/ksirk/iris/src/CMakeLists.txt	2013-03-09 23:03:52.115036300 +0100
@@ -75,6 +75,9 @@
   set(libiris_SRCS ${libiris_SRCS}
 		irisnet/corelib/netinterface_win.cpp
 	)
+    if(MINGW)
+        add_definitions(-DWIN32)
+    endif(MINGW)
 endif(WIN32)
 
 include_directories(
diff -Nru -x '*~' ksirk-4.10.2.orig/ksirkskineditor/CMakeLists.txt ksirk-4.10.2/ksirkskineditor/CMakeLists.txt
--- ksirk-4.10.2.orig/ksirkskineditor/CMakeLists.txt	2013-03-01 08:16:51.050749000 +0100
+++ ksirk-4.10.2/ksirkskineditor/CMakeLists.txt	2013-06-02 23:12:28.435400500 +0200
@@ -41,7 +41,7 @@
 
 kde4_add_kcfg_files( ksirkskineditor_SRCS ksirkskineditorsettings.kcfgc )
 
-kde4_add_app_icon(ksirkskineditor_SRCS "hi*-app-ksirkskineditor.png")
+kde4_add_app_icon(ksirkskineditor_SRCS "hi*-app-ksirk.png")
 
 kde4_add_executable(ksirkskineditor ${ksirkskineditor_SRCS})
 
