diff -Nrub -x '*~' -x '*\.orig' -x'*\.o' cfe-3.7.0.src.orig/CMakeLists.txt cfe-3.7.0.src/CMakeLists.txt
--- cfe-3.7.0.src.orig/CMakeLists.txt	2015-06-12 18:33:38.000000000 +0200
+++ cfe-3.7.0.src/CMakeLists.txt	2015-12-05 14:17:59.990917800 +0100
@@ -37,6 +37,10 @@
       OUTPUT_VARIABLE CONFIG_OUTPUT
     )
     if(NOT HAD_ERROR)
+      # On Windows we might get trailing backslashes
+      string(REGEX REPLACE
+        "\\\\" "/"
+        CONFIG_OUTPUT ${CONFIG_OUTPUT})
       string(REGEX REPLACE
         "[ \t]*[\r\n]+[ \t]*" ";"
         CONFIG_OUTPUT ${CONFIG_OUTPUT})
