diff --git a/src/contrib/CMakeLists.txt b/src/contrib/CMakeLists.txt
index 76f81e3..2579496 100644
--- a/src/contrib/CMakeLists.txt
+++ b/src/contrib/CMakeLists.txt
@@ -53,6 +53,8 @@ target_link_libraries(lucene++-contrib
 cotire(lucene++-contrib)
 
 install(TARGETS lucene++-contrib
-  DESTINATION "${LIB_DESTINATION}"
+  ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+  LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+  RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
   COMPONENT runtime
 )
diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 13ff0a6..9fd287d 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -52,6 +52,8 @@ install(FILES ${lucene_headers}
 )
 
 install(TARGETS lucene++
-  DESTINATION "${LIB_DESTINATION}"
+  ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+  LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+  RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
   COMPONENT runtime
 )
