diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' dbus-1.11.4.orig/dbus/dbus-transport.c dbus-1.11.4/dbus/dbus-transport.c
--- dbus-1.11.4.orig/dbus/dbus-transport.c	2016-08-12 12:11:25.000000000 +0200
+++ dbus-1.11.4/dbus/dbus-transport.c	2016-11-15 09:42:36.922246600 +0100
@@ -1074,9 +1074,11 @@
       if (!_dbus_string_copy (bytes, 0, buffer, _dbus_string_get_length (buffer)))
         succeeded = FALSE;
       
+#ifdef DBUS_ENABLE_VERBOSE_MODE
       _dbus_verbose (" %d unused bytes sent to message loader\n", 
                      _dbus_string_get_length (buffer) -
                      orig_len);
+#endif
       
       _dbus_message_loader_return_buffer (transport->loader,
                                           buffer);
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' dbus-1.11.4.orig/tools/dbus-launch-win.c dbus-1.11.4/tools/dbus-launch-win.c
--- dbus-1.11.4.orig/tools/dbus-launch-win.c	2015-05-14 15:48:53.000000000 +0200
+++ dbus-1.11.4/tools/dbus-launch-win.c	2016-11-15 09:42:46.781370300 +0100
@@ -29,6 +29,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <assert.h>
+#include <stdlib.h>
 
 /* Save string functions.  Instead of figuring out the exact _MSC_VER
    that work, override for everybody.  */
