--- dbus-1.4.24.orig/cmake/ConfigureChecks.cmake	2012-03-05 11:19:35.000000000 -0300
+++ dbus-1.4.24/cmake/ConfigureChecks.cmake	2013-05-22 16:40:44.646278300 -0300
@@ -92,9 +92,9 @@
 find_program(DOXYGEN doxygen)
 find_program(XMLTO xmlto)
 
-if(MSVC)
+if(MSVC OR ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel"))
    SET(DBUS_VA_COPY_FUNC "_DBUS_VA_COPY_ASSIGN";)
-else(MSVC)
+else(MSVC OR ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel"))
 write_file("${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/CMakeTmp/cmake_try_compile.c" "#include <stdarg.h>
 	#include <stdlib.h>
         static void f (int i, ...) {
@@ -141,7 +141,7 @@
     SET(DBUS_VA_COPY_AS_ARRAY "1" CACHE STRING "'va_lists' cannot be copies as values")
   endif(DBUS_HAVE___VA_COPY)
 endif(DBUS_HAVE_VA_COPY)
-endif(MSVC) # _not_ MSVC
+endif(MSVC OR ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")) # _not_ MSVC
 #### Abstract sockets
 
 if (DBUS_ENABLE_ABSTRACT_SOCKETS)
