Index: platforms/win/kwinstartmenu/misc.cpp
===================================================================
--- platforms/win/kwinstartmenu/misc.cpp	(revision 914666)
+++ platforms/win/kwinstartmenu/misc.cpp	(revision 914667)
@@ -1,6 +1,6 @@
 /* This file is part of the KDE project
 
-   Copyright (C) 2006-2008 Ralf Habacker <ralf.habacker@freenet.de>
+   Copyright (C) 2006-2009 Ralf Habacker <ralf.habacker@freenet.de>
    All rights reserved.
 
    This library is free software; you can redistribute it and/or
@@ -254,16 +254,18 @@
 
  This is done by the following rules: 
     1. If no entry in an installation points to an existing executable, this installation 
-       is removed and could be deleted     
+       has been removed. The relating start menu entries could be deleted.
     2. if start menu entries for non current kde installation have the same working 
        dir as the current installation, this should be removed too
 */
 void removeObsolateInstallations()
 {
-    kDebug() << getInstalledKDEVersions();
+    QStringList installedReleases = getInstalledKDEVersions();
     QString currentVersion = getKDEStartMenuRootEntry();
+    kDebug() << "installed releases" << installedReleases;
+    QFileInfo currentWorkDir(getWorkingDir());
 
-    foreach(QString release,getInstalledKDEVersions()) 
+    foreach(QString release,installedReleases) 
     {
         // skip current version 
         if (release == currentVersion)
@@ -276,15 +278,19 @@
         foreach(LinkFile lf, allReleasesFiles) 
         {
             lf.read(); // this in not done by the LinkFile class by default 
-            kDebug() << release << " : " << lf;
             QFileInfo fi(lf.execPath());
             if (fi.exists())
                 available = true;
-            if (lf.workingDir() == getWorkingDir())
+            QFileInfo f1(lf.workingDir());
+			// compare harmonized pathes
+            if (f1.absoluteFilePath() == currentWorkDir.absoluteFilePath())
                 sameWorkingDir = true;
         }
         if (!available || sameWorkingDir)
+        {
+            kDebug() << "removing obsolate installation" << getStartMenuPath() + "/" + release;
             removeDirectory(getStartMenuPath() + "/" + release);
+        }
     }
 }
 
@@ -295,14 +301,10 @@
     // generate list of installed linkfiles 
     QList<LinkFile> oldFiles;
     LinkFiles::scan(oldFiles, getKDEStartMenuPath());
-    foreach(const LinkFile& lf, oldFiles)
-      kDebug() << "oldFile: " << lf;
 
     // create list of currently available link files 
     QList<LinkFile> newFiles;
     generateMenuEntries(newFiles,KUrl("applications:/"));
-    foreach(const LinkFile& lf, newFiles)
-      kDebug() << "newFile: " << lf;
 
     // remove obsolate links 
     LinkFiles::cleanup(newFiles,oldFiles);
