commit c16692e69d6d815dfd19182db18534646a8787fa
Author: Montel Laurent <montel@kde.org>
Date:   Mon May 22 13:24:16 2017 +0200

    Add missing include

diff --git a/src/common/davprotocolbase.h b/src/common/davprotocolbase.h
index 8877eb4..3709c6b 100644
--- a/src/common/davprotocolbase.h
+++ b/src/common/davprotocolbase.h
@@ -27,6 +27,7 @@
 #include <QtCore/QMap>
 #include <QtXml/QDomDocument>
 #include <QSharedPointer>
+#include <QVariant>
 
 namespace KDAV
 {
