--- CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ CMakeLists.txt	2008-06-04 13:49:51.874407100 +0200
@@ -0,0 +1,66 @@
+# CMake build system for exiv2 library and executables
+# Copyright 2008 by Patrick Spendrin <ps_ml@gmx.de>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#    1. Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#    2. Redistributions in binary form must reproduce the above
+#       copyright notice, this list of conditions and the following
+#       disclaimer in the documentation and/or other materials provided
+#       with the distribution.
+#    3. The name of the author may not be used to endorse or promote
+#       products derived from this software without specific prior
+#       written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+# IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
+# GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+# IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
+# IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+project( exiv2 )
+
+cmake_minimum_required( VERSION 2.6 )
+
+set( PACKAGE_BUGREPORT  "ahuggel@gmx.net" )
+set( PACKAGE            "exiv2" )
+set( PACKAGE_NAME       "exiv2" )
+set( PACKAGE_STRING     "exiv2 0.16" )
+set( PACKAGE_TARNAME    "exiv2" )
+set( PACKAGE_VERSION    "0.16" )
+
+include( config/ConfigureChecks.cmake )
+
+set( INSTALL_TARGET_STANDARD_ARGS RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib )
+
+# add sources from XMPSDK directory
+set( XMPSRC ${CMAKE_SOURCE_DIR}/xmpsdk/src/XMPMeta.cpp
+            ${CMAKE_SOURCE_DIR}/xmpsdk/src/XMPMeta-GetSet.cpp
+            ${CMAKE_SOURCE_DIR}/xmpsdk/src/XMPMeta-Parse.cpp
+            ${CMAKE_SOURCE_DIR}/xmpsdk/src/XMPMeta-Serialize.cpp
+            ${CMAKE_SOURCE_DIR}/xmpsdk/src/XMPIterator.cpp
+            ${CMAKE_SOURCE_DIR}/xmpsdk/src/XMPUtils.cpp
+            ${CMAKE_SOURCE_DIR}/xmpsdk/src/XMPUtils-FileInfo.cpp
+            ${CMAKE_SOURCE_DIR}/xmpsdk/src/XMPCore_Impl.cpp
+            ${CMAKE_SOURCE_DIR}/xmpsdk/src/ExpatAdapter.cpp
+            ${CMAKE_SOURCE_DIR}/xmpsdk/src/ParseRDF.cpp
+            ${CMAKE_SOURCE_DIR}/xmpsdk/src/UnicodeConversions.cpp
+            ${CMAKE_SOURCE_DIR}/xmpsdk/src/MD5.cpp
+            ${CMAKE_SOURCE_DIR}/xmpsdk/src/WXMPMeta.cpp
+            ${CMAKE_SOURCE_DIR}/xmpsdk/src/WXMPIterator.cpp
+            ${CMAKE_SOURCE_DIR}/xmpsdk/src/WXMPUtils.cpp )
+
+add_subdirectory( src )
+if( BUILD_samples )
+    add_subdirectory( samples )
+endif( BUILD_samples )
--- config/ConfigureChecks.cmake	1970-01-01 01:00:00.000000000 +0100
+++ config/ConfigureChecks.cmake	2008-06-04 13:00:24.031250000 +0200
@@ -0,0 +1,153 @@
+# CMake build system for exiv2 library and executables
+# Copyright 2008 by Patrick Spendrin <ps_ml@gmx.de>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#    1. Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#    2. Redistributions in binary form must reproduce the above
+#       copyright notice, this list of conditions and the following
+#       disclaimer in the documentation and/or other materials provided
+#       with the distribution.
+#    3. The name of the author may not be used to endorse or promote
+#       products derived from this software without specific prior
+#       written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+# IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
+# GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+# IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
+# IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# ConfigureChecks for exiv2
+
+find_library( ICONV_LIBRARIES iconv )
+find_library( LIBINTL_LIBRARIES libintl )
+
+include( CheckIncludeFile )
+include( CheckFunctionExists )
+include( CheckSymbolExists )
+
+set( HAVE_XMP_TOOLKIT ON ) 
+set( STDC_HEADERS ON )
+
+#set( ENABLE_NLS ON )
+set( HAVE_PRINTUCS2 ON )
+set( HAVE_LENSDATA ON )
+set( HAVE_DECL_STRERROR_R 0 )
+
+include_directories( ${CMAKE_INCLUDE_PATH} ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_SOURCE_DIR}/xmpsdk/include )
+set( CMAKE_REQUIRED_INCLUDES ${CMAKE_INCLUDE_PATH} )
+link_directories( ${CMAKE_LIBRARY_PATH} )
+set( CMAKE_REQUIRED_LIBRARIES ${ICONV_LIBRARIES} )
+
+find_package( EXPAT )
+find_package( ZLIB )
+
+if( NOT EXPAT_FOUND )
+message(FATAL_ERROR "missing library expat required for xmp")
+endif( NOT EXPAT_FOUND )
+
+check_include_file( "inttypes.h" HAVE_INTTYPES_H )
+check_include_file( "libintl.h" HAVE_LIBINTL_H )
+check_include_file( "malloc.h" HAVE_MALLOC_H )
+check_include_file( "memory.h" HAVE_MEMORY_H )
+check_include_file( "iconv.h" HAVE_ICONV_H )
+check_include_file( "stdbool.h" HAVE_STDBOOL_H )
+check_include_file( "stdint.h" HAVE_STDINT_H )
+check_include_file( "stdlib.h" HAVE_STDLIB_H )
+check_include_file( "string.h" HAVE_STRING_H )
+check_include_file( "strings.h" HAVE_STRINGS_H )
+check_include_file( "unistd.h" HAVE_UNISTD_H )
+check_include_file( "wchar.h" HAVE_WCHAR_H )
+check_include_file( "sys/stat.h" HAVE_SYS_STAT_H )
+check_include_file( "sys/time.h" HAVE_SYS_TIME_H )
+check_include_file( "sys/types.h" HAVE_SYS_TYPES_H )
+check_include_file( "sys/mman.h" HAVE_SYS_MMAN_H )
+
+check_function_exists( alarm HAVE_ALARM )
+check_function_exists( gmtime_r HAVE_GMTIME_R )
+check_function_exists( iconv HAVE_ICONV )
+check_function_exists( memset HAVE_MEMSET )
+check_function_exists( mmap HAVE_MMAP )
+check_function_exists( munmap HAVE_MUNMAP )
+check_function_exists( realloc HAVE_REALLOC )
+check_function_exists( strchr HAVE_STRCHR )
+check_function_exists( strchr_r HAVE_STRCHR_R )
+check_function_exists( strerror HAVE_STRERROR )
+check_function_exists( strerror_r HAVE_STRERROR_R )
+check_function_exists( strtol HAVE_STRTOL )
+check_function_exists( timegm HAVE_TIMEGM )
+check_function_exists( vprintf HAVE_VPRINTF )
+
+if( ${HAVE_STDBOOL_H} )
+    check_symbol_exists( "_Bool" stdbool.h HAVE__BOOL )
+endif( ${HAVE_STDBOOL_H} )
+
+set( EXV_SYMBOLS
+HAVE_INTTYPES_H
+HAVE_LIBINTL_H
+HAVE_MALLOC_H
+HAVE_MEMORY_H
+HAVE_ICONV_H
+HAVE_STDBOOL_H
+HAVE_STDINT_H
+HAVE_STDLIB_H
+HAVE_STRING_H
+HAVE_STRINGS_H
+HAVE_UNISTD_H
+HAVE_WCHAR_H
+HAVE_SYS_STAT_H
+HAVE_SYS_TIME_H
+HAVE_SYS_TYPES_H
+HAVE_SYS_MMAN_H
+HAVE_ALARM
+HAVE_GMTIME_R
+HAVE_ICONV
+HAVE_MEMSET
+HAVE_MMAP
+HAVE_MUNMAP
+HAVE_REALLOC
+HAVE_STRCHR
+HAVE_STRCHR_R
+HAVE_STRERROR
+HAVE_STRERROR_R
+HAVE_STRTOL
+HAVE_TIMEGM
+HAVE_VPRINTF
+
+HAVE__BOOL
+HAVE_LENSDATA
+HAVE_DECL_STRERROR_R
+HAVE_PRINTUCS2
+HAVE_LIBZ
+HAVE_XMP_TOOLKIT
+PACKAGE_BUGREPORT
+PACKAGE
+PACKAGE_NAME
+PACKAGE_STRING
+PACKAGE_TARNAME
+PACKAGE_VERSION
+)
+foreach( entry ${EXV_SYMBOLS} )
+set( EXV_${entry} ${${entry}} )
+endforeach( entry ${EXV_SYMBOLS} )
+
+if( NOT MSVC )
+configure_file( config/exv_conf.h.cmake ${CMAKE_BINARY_DIR}/exv_conf.h )
+install( FILES ${CMAKE_BINARY_DIR}/exv_conf.h DESTINATION include/exiv2 )
+else( NOT MSVC )
+endif( NOT MSVC )
+
+if( NOT WIN32 )
+configure_file( config/exiv2.pc.cmake ${CMAKE_BINARY_DIR}/exiv2.pc )
+install( FILES ${CMAKE_BINARY_DIR}/exiv2.pc DESTINATION lib/pkgconfig )
+endif( NOT WIN32 )
\ Kein Zeilenumbruch am Dateiende.
--- config/exiv2.pc.cmake	1970-01-01 01:00:00.000000000 +0100
+++ config/exiv2.pc.cmake	2008-06-04 13:00:24.046875000 +0200
@@ -0,0 +1,10 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${prefix}/lib
+includedir=${prefix}/include
+
+Name: exiv2
+Description: Exif and IPTC metadata library and tools
+Version: @PACKAGE_VERSION@
+Libs: -L${libdir} -lexiv2 @EXV_LIB_STATIC@
+Cflags: -I${includedir}
--- config/exv_conf.h.cmake	1970-01-01 01:00:00.000000000 +0100
+++ config/exv_conf.h.cmake	2008-06-04 13:00:24.109375000 +0200
@@ -0,0 +1,195 @@
+/**/
+/* config.h.cmake.  Generated cmake.  */
+
+/* Define to 1 if you have the `alarm' function. */
+#cmakedefine EXV_HAVE_ALARM
+
+/* Define to 1 if you have the declaration of `strerror_r', and to 0 if you
+   don't. */
+#cmakedefine EXV_HAVE_DECL_STRERROR_R @HAVE_DECL_STRERROR_R@
+
+/* Define to 1 if you have the `gmtime_r' function. */
+#cmakedefine EXV_HAVE_GMTIME_R 1
+
+/* Define to 1 if you have the <inttypes.h> header file. */
+#cmakedefine EXV_HAVE_INTTYPES_H 1
+
+/* Define to 1 if you have the <libintl.h> header file. */
+#cmakedefine EXV_HAVE_LIBINTL_H 1
+
+#ifndef EXV_COMMERCIAL_VERSION
+
+/* Define to 1 to enable translation of Exif.Nikon3.LensData. */
+#cmakedefine EXV_HAVE_LENSDATA 1
+
+/* Define to 1 if translation of program messages to the user's
+   native language is requested. */
+#cmakedefine ENABLE_NLS 1
+
+/* Define to 1 if you have the `iconv' function. */
+#cmakedefine EXV_HAVE_ICONV 1
+
+/* Define to 1 to enable conversion of UCS2 encoded Windows tags to UTF-8. */
+#cmakedefine EXV_HAVE_PRINTUCS2 1
+
+#endif /* !EXV_COMMERCIAL_VERSION */
+
+/* Define to `const' or to empty, depending on the second argument of `iconv'. */
+#define ICONV_CONST const
+
+/* Define to 1 if you have the <libintl.h> header file. */
+#cmakedefine EXV_HAVE_LIBINTL_H 1
+
+/* Define to 1 if your system has a GNU libc compatible `malloc' function, and
+   to 0 otherwise. */
+#cmakedefine EXV_HAVE_MALLOC 1
+
+/* Define to 1 if you have the <malloc.h> header file. */
+#cmakedefine EXV_HAVE_MALLOC_H 1
+
+/* Define to 1 if you have the <memory.h> header file. */
+#cmakedefine EXV_HAVE_MEMORY_H 1
+
+/* Define to 1 if you have the `memset' function. */
+#cmakedefine EXV_HAVE_MEMSET 1
+
+/* Define to 1 if your system has a GNU libc compatible `realloc' function,
+   and to 0 otherwise. */
+#cmakedefine EXV_HAVE_REALLOC 1
+
+/* Define to 1 if stdbool.h conforms to C99. */
+#cmakedefine EXV_HAVE_STDBOOL_H 1
+
+/* Define to 1 if you have the <stdint.h> header file. */
+#cmakedefine EXV_HAVE_STDINT_H 1
+
+/* Define to 1 if you have the <stdlib.h> header file. */
+#cmakedefine EXV_HAVE_STDLIB_H 1
+
+/* Define to 1 if you have the `strchr' function. */
+#cmakedefine EXV_HAVE_STRCHR 1
+
+/* Define to 1 if you have the `strerror' function. */
+#cmakedefine EXV_HAVE_STRERROR 1
+
+/* Define to 1 if you have the `strerror_r' function. */
+#cmakedefine EXV_HAVE_STRERROR_R 1
+
+/* Define to 1 if you have the <strings.h> header file. */
+#cmakedefine EXV_HAVE_STRINGS_H 1
+
+/* Define to 1 if you have the <string.h> header file. */
+#cmakedefine EXV_HAVE_STRING_H 1
+
+/* Define to 1 if you have the `strtol' function. */
+#cmakedefine EXV_HAVE_STRTOL 1
+
+/* Define to 1 if you have the `mmap' function. */
+#cmakedefine EXV_HAVE_MMAP 1
+
+/* Define to 1 if you have the `munmap' function. */
+#cmakedefine EXV_HAVE_MUNMAP 1
+
+/* Define to 1 if you have the <sys/stat.h> header file. */
+#cmakedefine EXV_HAVE_SYS_STAT_H 1
+
+/* Define to 1 if you have the <sys/time.h> header file. */
+#cmakedefine EXV_HAVE_SYS_TIME_H 1
+
+/* Define to 1 if you have the <sys/types.h> header file. */
+#cmakedefine EXV_HAVE_SYS_TYPES_H 1
+
+/* Define to 1 if you have the `timegm' function. */
+#cmakedefine EXV_HAVE_TIMEGM 1
+
+/* Define to 1 if you have the <unistd.h> header file. */
+#cmakedefine EXV_HAVE_UNISTD_H 1
+
+/* Define to 1 if you have the `vprintf' function. */
+#cmakedefine EXV_HAVE_VPRINTF 1
+
+/* Define to 1 if you have the <wchar.h> header file. */
+#cmakedefine EXV_HAVE_WCHAR_H 1
+
+/* Define to 1 if you have the <sys/mman.h> header file. */
+#cmakedefine EXV_HAVE_SYS_MMAN_H 1
+
+/* Define to 1 if you have the `zlib' library. */
+#cmakedefine EXV_HAVE_LIBZ 1
+
+/* Define to 1 if you have the Adobe XMP Toolkit. */
+#cmakedefine EXV_HAVE_XMP_TOOLKIT 1
+
+/* Define to 1 if the system has the type `_Bool'. */
+#cmakedefine EXV_HAVE__BOOL 1
+
+/* Define to 1 if `lstat' dereferences a symlink specified with a trailing
+   slash. */
+#cmakedefine EXV_LSTAT_FOLLOWS_SLASHED_SYMLINK 1
+
+/* Define to the address where bug reports for this package should be sent. */
+#cmakedefine EXV_PACKAGE_BUGREPORT "@PACKAGE_BUGREPORT@"
+
+/* Define to the name of this package. */
+#cmakedefine EXV_PACKAGE "@PACKAGE@"
+
+/* Define to the full name of this package. */
+#cmakedefine EXV_PACKAGE_NAME "@PACKAGE_NAME@"
+
+/* Define to the full name and version of this package. */
+#cmakedefine EXV_PACKAGE_STRING "@PACKAGE_STRING@"
+
+/* Define to the one symbol short name of this package. */
+#cmakedefine EXV_PACKAGE_TARNAME "@PACKAGE_TARNAME@"
+
+/* Define to the version of this package. */
+#cmakedefine EXV_PACKAGE_VERSION "@PACKAGE_VERSION@"
+
+/* Define to 1 if you have the ANSI C header files. */
+#cmakedefine STDC_HEADERS 1
+
+/* Define to 1 if strerror_r returns char *. */
+#cmakedefine STRERROR_R_CHAR_P 1
+
+/* Define to 1 if you can safely include both <sys/time.h> and <time.h>. */
+#define TIME_WITH_SYS_TIME 1
+
+/* Define to 1 if your <sys/time.h> declares `struct tm'. */
+#cmakedefine TM_IN_SYS_TIME 1
+
+/* Define to empty if `const' does not conform to ANSI C. */
+#cmakedefine const
+
+/* Define to `__inline__' or `__inline' if that's what the C compiler
+   calls it, or to nothing if 'inline' is not supported under any name.  */
+#ifndef __cplusplus
+#cmakedefine inline
+#endif
+
+/* Define to rpl_malloc if the replacement function should be used. */
+#cmakedefine malloc
+
+/* Define to `int' if <sys/types.h> does not define. */
+#cmakedefine pid_t
+
+/* Define to rpl_realloc if the replacement function should be used. */
+#cmakedefine realloc
+
+/* Define to `unsigned' if <sys/types.h> does not define. */
+#cmakedefine size_t
+
+#if defined __CYGWIN32__ && !defined __CYGWIN__
+   /* For backwards compatibility with Cygwin b19 and
+      earlier, we define __CYGWIN__ here, so that
+      we can rely on checking just for that macro. */
+#define __CYGWIN__  __CYGWIN32__
+#endif
+     
+/* File path seperator */
+#if defined WIN32 && !defined __CYGWIN__
+#define EXV_SEPERATOR_STR "\\"
+#define EXV_SEPERATOR_CHR '\\'
+#else
+#define EXV_SEPERATOR_STR "/"
+#define EXV_SEPERATOR_CHR '/'
+#endif
--- samples/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ samples/CMakeLists.txt	2008-06-04 13:00:24.171875000 +0200
@@ -0,0 +1,57 @@
+# CMake build system for exiv2 library and executables
+# Copyright 2008 by Patrick Spendrin <ps_ml@gmx.de>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#    1. Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#    2. Redistributions in binary form must reproduce the above
+#       copyright notice, this list of conditions and the following
+#       disclaimer in the documentation and/or other materials provided
+#       with the distribution.
+#    3. The name of the author may not be used to endorse or promote
+#       products derived from this software without specific prior
+#       written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+# IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
+# GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+# IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
+# IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+set( SAMPLES    addmoddel.cpp
+                crwedit.cpp
+                crwparse.cpp
+                dataarea-test.cpp
+                exifcomment.cpp
+                exifdata-test.cpp
+                exifprint.cpp
+                ifd-test.cpp
+                iotest.cpp
+                iptceasy.cpp
+                iptcprint.cpp
+                iptctest.cpp
+                key-test.cpp
+                largeiptc-test.cpp
+                makernote-test.cpp
+                write-test.cpp
+                write2-test.cpp
+                tiffparse.cpp
+                xmpparse.cpp
+                xmpparser-test.cpp
+                xmpsample.cpp )
+
+foreach(entry ${SAMPLES})
+STRING( REPLACE ".cpp" "" target ${entry})
+add_executable( ${target} ${target}.cpp )
+target_link_libraries(${target} exiv2 )
+endforeach(entry ${SAMPLES})
--- src/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ src/CMakeLists.txt	2008-06-04 13:51:34.790825500 +0200
@@ -0,0 +1,169 @@
+# CMake build system for exiv2 library and executables
+# Copyright 2008 by Patrick Spendrin <ps_ml@gmx.de>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#    1. Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#    2. Redistributions in binary form must reproduce the above
+#       copyright notice, this list of conditions and the following
+#       disclaimer in the documentation and/or other materials provided
+#       with the distribution.
+#    3. The name of the author may not be used to endorse or promote
+#       products derived from this software without specific prior
+#       written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+# IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
+# GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+# IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
+# IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# Add standalone C++ header files to this list
+set( LIBEXIV2_HDR   basicio.hpp
+                    canonmn.hpp
+                    cr2image.hpp
+                    crwimage.hpp
+                    datasets.hpp
+                    error.hpp
+                    exif.hpp
+                    exv_msvc.h
+                    fujimn.hpp
+                    futils.hpp
+                    ifd.hpp
+                    image.hpp
+                    iptc.hpp
+                    jpgimage.hpp
+                    makernote.hpp
+                    makernote2.hpp
+                    metadatum.hpp
+                    minoltamn.hpp
+                    mn.hpp
+                    mrwimage.hpp
+                    nikonmn.hpp
+                    olympusmn.hpp
+                    orfimage.hpp
+                    panasonicmn.hpp
+                    pentaxmn.hpp
+                    pngchunk.hpp
+                    pngimage.hpp
+                    properties.hpp
+                    rafimage.hpp
+                    rcsid.hpp
+                    sigmamn.hpp
+                    sonymn.hpp
+                    tags.hpp
+                    tiffcomposite.hpp
+                    tifffwd.hpp
+                    tiffimage.hpp
+                    tiffparser.hpp
+                    tiffvisitor.hpp
+                    types.hpp
+                    value.hpp
+                    version.hpp
+                    xmp.hpp )
+
+# Add library C++ source files to this list
+set( LIBEXIV2_SRC   basicio.cpp
+                    canonmn.cpp
+                    cr2image.cpp
+                    crwimage.cpp
+                    datasets.cpp
+                    error.cpp
+                    exif.cpp
+                    futils.cpp
+                    fujimn.cpp
+                    ifd.cpp
+                    image.cpp
+                    iptc.cpp
+                    jpgimage.cpp
+                    makernote.cpp
+                    makernote2.cpp
+                    metadatum.cpp
+                    minoltamn.cpp
+                    mrwimage.cpp
+                    nikonmn.cpp
+                    olympusmn.cpp
+                    orfimage.cpp
+                    panasonicmn.cpp
+                    rafimage.cpp
+                    sigmamn.cpp
+                    pentaxmn.cpp
+                    sonymn.cpp
+                    tags.cpp
+                    tiffcomposite.cpp
+                    tiffimage.cpp
+                    tiffparser.cpp
+                    tiffvisitor.cpp
+                    types.cpp
+                    value.cpp
+                    version.cpp
+                    properties.cpp
+                    xmp.cpp )
+
+# Add source files of simple applications to this list
+set( TAGLIST_SRC    taglist.cpp )
+
+# Source files for the Exiv2 application
+set( EXIV2_SRC      exiv2.cpp
+                    actions.cpp
+                    utils.cpp )
+
+# Source files for the metacopy sample application
+set( MC_SRC         metacopy.cpp
+                    utils.cpp )
+
+if( HAVE_ZLIB )
+set( LIBEXIV2_SRC ${LIBEXIV2_SRC}   pngimage.cpp
+                                    pngchunk.cpp )
+endif( HAVE_ZLIB )
+
+if(NOT HAVE_TIMEGM )
+set( LIBEXIV2_SRC   ${LIBEXIV2_SRC} localtime.c )
+set( EXIV2_SRC      ${EXIV2_SRC}    localtime.c )
+endif( NOT HAVE_TIMEGM )
+
+if(MSVC)
+set( MC_SRC         ${MC_SRC}       getopt_win32.c )
+set( LIBEXIV2_SRC   ${LIBEXIV2_SRC} getopt_win32.c )
+endif(MSVC)
+
+# ******************************************************************************
+# Library
+include_directories( ${CMAKE_SOURCE_DIR}/xmpsdk/src )
+add_library( exiv2 STATIC ${XMPSRC} ${LIBEXIV2_SRC} )
+
+target_link_libraries( exiv2 ${EXPAT_LIBRARIES} )
+install( TARGETS exiv2 ${INSTALL_TARGET_STANDARD_ARGS} )
+
+# ******************************************************************************
+# Binary
+add_executable( exiv2bin ${EXIV2_SRC} )
+set_target_properties( exiv2bin PROPERTIES OUTPUT_NAME exiv2 )
+target_link_libraries( exiv2bin exiv2 )
+install( TARGETS exiv2bin ${INSTALL_TARGET_STANDARD_ARGS} )
+
+# ******************************************************************************
+# Binary
+add_executable( metacopy ${MC_SRC} )
+target_link_libraries( metacopy exiv2 )
+install( TARGETS metacopy ${INSTALL_TARGET_STANDARD_ARGS} )
+
+
+# ******************************************************************************
+# Binary
+add_executable( taglist ${TAGLIST_SRC} )
+target_link_libraries( taglist exiv2 )
+install( TARGETS taglist ${INSTALL_TARGET_STANDARD_ARGS} )
+
+install( FILES ${LIBEXIV2_HDR} DESTINATION include/exiv2 )
+
+install( FILES exiv2.1 DESTINATION share/man/man1 )
