Index: kipi-plugins-0.7.0/acquireimages/plugin_acquireimages.cpp
===================================================================
--- kipi-plugins-0.7.0/acquireimages/plugin_acquireimages.cpp
+++ kipi-plugins-0.7.0/acquireimages/plugin_acquireimages.cpp
@@ -40,7 +40,7 @@
 #include <klocale.h>
 #include <kmessagebox.h>
 
-#ifdef WIN32
+#ifdef Q_OS_WIN
 
 // Twain interface includes
 
@@ -105,7 +105,7 @@
 
 void Plugin_AcquireImages::slotActivate()
 {
-#ifdef WIN32
+#ifdef Q_OS_WIN
 
     TwainController *twainCtrl = new TwainController(m_interface);
 
