diff -Nru libarchive-2.8.4.orig/CMakeLists.txt libarchive-2.8.4/CMakeLists.txt
--- libarchive-2.8.4.orig/CMakeLists.txt	2010-06-27 20:48:32.000000000 +0200
+++ libarchive-2.8.4/CMakeLists.txt	2010-12-05 01:40:03.538263900 +0100
@@ -468,7 +468,9 @@
 #
 CHECK_SYMBOL_EXISTS(CreateHardLinkA "windows.h" HAVE_CREATEHARDLINKA)
 CHECK_SYMBOL_EXISTS(CreateHardLinkW "windows.h" HAVE_CREATEHARDLINKW)
-CHECK_SYMBOL_EXISTS(_CrtSetReportMode "crtdbg.h" HAVE__CrtSetReportMode)
+IF (NOT "CMAKE_C_COMPILER_ID" MATCHES "^GNU$")
+  CHECK_SYMBOL_EXISTS(_CrtSetReportMode "crtdbg.h" HAVE__CrtSetReportMode)
+ENDIF(NOT "CMAKE_C_COMPILER_ID" MATCHES "^GNU$")
 CHECK_FUNCTION_EXISTS_GLIBC(chflags HAVE_CHFLAGS)
 CHECK_FUNCTION_EXISTS_GLIBC(chown HAVE_CHOWN)
 CHECK_FUNCTION_EXISTS_GLIBC(chroot HAVE_CHROOT)
@@ -843,7 +845,7 @@
 # aggressive about diagnosing build problems; this can get
 # relaxed somewhat in final shipping versions.
 IF ("CMAKE_C_COMPILER_ID" MATCHES "^GNU$")
-  ADD_DEFINITIONS(-Wall -Werror)
+  ADD_DEFINITIONS(-Wall)
 ENDIF ("CMAKE_C_COMPILER_ID" MATCHES "^GNU$")
 
 IF(ENABLE_TEST)
