diff -Nru -x '*~' nmealib.orig/CMakeLists.txt nmealib/CMakeLists.txt
--- nmealib.orig/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ nmealib/CMakeLists.txt	2012-04-24 09:48:35.291101700 +0200
@@ -0,0 +1,39 @@
+PROJECT(nmealib)
+
+cmake_minimum_required(VERSION 2.8)
+
+set (NMEALIB_SOURCES
+    src/context.c
+    src/generate.c
+    src/generator.c
+    src/gmath.c
+    src/info.c
+    src/parse.c
+    src/parser.c
+    src/sentence.c
+    src/time.c
+    src/tok.c
+)
+
+set(NMEALIB_HEADERS
+    include/nmea/config.h
+    include/nmea/context.h
+    include/nmea/generate.h
+    include/nmea/generator.h
+    include/nmea/gmath.h
+    include/nmea/info.h
+    include/nmea/nmea.h
+    include/nmea/parse.h
+    include/nmea/parser.h
+    include/nmea/sentence.h
+    include/nmea/time.h
+    include/nmea/tok.h
+    include/nmea/units.h
+)
+
+include_directories(src include)
+
+add_library(nmealib ${NMEALIB_SOURCES})
+target_link_libraries(nmealib)
+install(TARGETS nmealib RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+install(FILES ${NMEALIB_HEADERS} DESTINATION include/nmea)
