diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..2bc4093
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,46 @@
+cmake_minimum_required(VERSION 3.12)
+project(peparser)
+
+include(GNUInstallDirs)
+
+find_package(Boost REQUIRED
+    COMPONENTS
+        filesystem
+        program_options
+)
+
+add_executable(peparser
+    actions.cpp
+    activationcontext.cpp
+    block.cpp
+    dependencycheck.cpp
+    etoken.cpp
+    json/json.cpp
+    main.cpp
+    peparser.cpp
+    peparser.rc
+    resourcepath.cpp
+    resourcetable.cpp
+    signer.cpp
+    versionstring.cpp
+    widestring.cpp
+)
+
+target_compile_definitions(peparser
+    PRIVATE
+        UNICODE
+
+        # https://svn.boost.org/trac10/ticket/13326
+        BOOST_PROGRAM_OPTIONS_DYN_LINK=1
+)
+
+target_link_libraries(peparser
+    Boost::filesystem
+    Boost::program_options
+)
+
+install(TARGETS peparser
+    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+    LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+)
diff --git a/peparser.rc b/peparser.rc
index 84beb0d..8833259 100644
--- a/peparser.rc
+++ b/peparser.rc
@@ -7,7 +7,7 @@
 //
 // Generated from the TEXTINCLUDE 2 resource.
 //
-#include "afxres.h"
+#include "windows.h"
 
 /////////////////////////////////////////////////////////////////////////////
 #undef APSTUDIO_READONLY_SYMBOLS
