diff -Nru boost_1_47_0.orig/tools/build/v2/engine/build.bat boost_1_47_0/tools/build/v2/engine/build.bat
--- boost_1_47_0.orig/tools/build/v2/engine/build.bat	2011-06-06 22:36:21.000000000 +0200
+++ boost_1_47_0/tools/build/v2/engine/build.bat	2011-08-14 23:52:49.942039600 +0200
@@ -1,4 +1,5 @@
 @ECHO OFF
+SETLOCAL ENABLEDELAYEDEXPANSION
 
 REM ~ Copyright 2002-2007 Rene Rivera.
 REM ~ Distributed under the Boost Software License, Version 1.0.
@@ -290,7 +291,7 @@
 if NOT "_%CWFolder%_" == "__" (
     set "BOOST_JAM_TOOLSET_ROOT=%CWFolder%\"
     )
-set "PATH=%BOOST_JAM_TOOLSET_ROOT%Other Metrowerks Tools\Command Line Tools;%PATH%"
+set "PATH=%BOOST_JAM_TOOLSET_ROOT%Other Metrowerks Tools\Command Line Tools;!PATH!"
 set "BOOST_JAM_CC=mwcc -runtime ss -cwd include -DNT -lkernel32.lib -ladvapi32.lib -luser32.lib"
 set "BOOST_JAM_OPT_JAM=-o bootstrap\jam0.exe"
 set "BOOST_JAM_OPT_MKJAMBASE=-o bootstrap\mkjambase0.exe"
@@ -303,7 +304,7 @@
     )
 call :Call_If_Exists "%BOOST_JAM_TOOLSET_ROOT%bin\VCVARS32.BAT"
 if not "_%BOOST_JAM_TOOLSET_ROOT%_" == "__" (
-    set "PATH=%BOOST_JAM_TOOLSET_ROOT%bin;%PATH%"
+    set "PATH=%BOOST_JAM_TOOLSET_ROOT%bin;!PATH!"
     )
 set "BOOST_JAM_CC=cl /nologo /GZ /Zi /MLd /Fobootstrap/ /Fdbootstrap/ -DNT -DYYDEBUG kernel32.lib advapi32.lib user32.lib"
 set "BOOST_JAM_OPT_JAM=/Febootstrap\jam0"
@@ -318,7 +319,7 @@
 if "_%VCINSTALLDIR%_" == "__" call :Call_If_Exists "%BOOST_JAM_TOOLSET_ROOT%bin\VCVARS32.BAT"
 if NOT "_%BOOST_JAM_TOOLSET_ROOT%_" == "__" (
     if "_%VCINSTALLDIR%_" == "__" (
-        set "PATH=%BOOST_JAM_TOOLSET_ROOT%bin;%PATH%"
+        set "PATH=%BOOST_JAM_TOOLSET_ROOT%bin;!PATH!"
         ) )
 set "BOOST_JAM_CC=cl /nologo /GZ /Zi /MLd /Fobootstrap/ /Fdbootstrap/ -DNT -DYYDEBUG kernel32.lib advapi32.lib user32.lib"
 set "BOOST_JAM_OPT_JAM=/Febootstrap\jam0"
@@ -333,7 +334,7 @@
 if "_%VCINSTALLDIR%_" == "__" call :Call_If_Exists "%BOOST_JAM_TOOLSET_ROOT%VCVARSALL.BAT" %BOOST_JAM_ARGS%
 if NOT "_%BOOST_JAM_TOOLSET_ROOT%_" == "__" (
     if "_%VCINSTALLDIR%_" == "__" (
-        set "PATH=%BOOST_JAM_TOOLSET_ROOT%bin;%PATH%"
+        set "PATH=%BOOST_JAM_TOOLSET_ROOT%bin;!PATH!"
         ) )
 set "BOOST_JAM_CC=cl /nologo /RTC1 /Zi /MTd /Fobootstrap/ /Fdbootstrap/ -DNT -DYYDEBUG -wd4996 kernel32.lib advapi32.lib user32.lib"
 set "BOOST_JAM_OPT_JAM=/Febootstrap\jam0"
@@ -348,7 +349,7 @@
 if "_%VCINSTALLDIR%_" == "__" call :Call_If_Exists "%BOOST_JAM_TOOLSET_ROOT%VCVARSALL.BAT" %BOOST_JAM_ARGS%
 if NOT "_%BOOST_JAM_TOOLSET_ROOT%_" == "__" (
     if "_%VCINSTALLDIR%_" == "__" (
-        set "PATH=%BOOST_JAM_TOOLSET_ROOT%bin;%PATH%"
+        set "PATH=%BOOST_JAM_TOOLSET_ROOT%bin;!PATH!"
         ) )
 set "BOOST_JAM_CC=cl /nologo /RTC1 /Zi /MTd /Fobootstrap/ /Fdbootstrap/ -DNT -DYYDEBUG -wd4996 kernel32.lib advapi32.lib user32.lib"
 set "BOOST_JAM_OPT_JAM=/Febootstrap\jam0"
@@ -363,7 +364,7 @@
 if "_%VCINSTALLDIR%_" == "__" call :Call_If_Exists "%BOOST_JAM_TOOLSET_ROOT%VCVARSALL.BAT" %BOOST_JAM_ARGS%
 if NOT "_%BOOST_JAM_TOOLSET_ROOT%_" == "__" (
     if "_%VCINSTALLDIR%_" == "__" (
-        set "PATH=%BOOST_JAM_TOOLSET_ROOT%bin;%PATH%"
+        set "PATH=%BOOST_JAM_TOOLSET_ROOT%bin;!PATH!"
         ) )
 set "BOOST_JAM_CC=cl /nologo /RTC1 /Zi /MTd /Fobootstrap/ /Fdbootstrap/ -DNT -DYYDEBUG -wd4996 kernel32.lib advapi32.lib user32.lib"
 set "BOOST_JAM_OPT_JAM=/Febootstrap\jam0"
@@ -379,7 +380,7 @@
         set "BOOST_JAM_TOOLSET_ROOT=%FOUND_PATH%..\"
         ) )
 if not "_%BOOST_JAM_TOOLSET_ROOT%_" == "__" (
-    set "PATH=%BOOST_JAM_TOOLSET_ROOT%Bin;%PATH%"
+    set "PATH=%BOOST_JAM_TOOLSET_ROOT%Bin;!PATH!"
     )
 set "BOOST_JAM_CC=bcc32 -WC -w- -q -I%BOOST_JAM_TOOLSET_ROOT%Include -L%BOOST_JAM_TOOLSET_ROOT%Lib /DNT -nbootstrap"
 set "BOOST_JAM_OPT_JAM=-ejam0"
@@ -417,7 +418,7 @@
 :Skip_INTEL_WIN32
 if NOT "_%BOOST_JAM_TOOLSET%_" == "_mingw_" goto Skip_MINGW
 if not "_%BOOST_JAM_TOOLSET_ROOT%_" == "__" (
-    set "PATH=%BOOST_JAM_TOOLSET_ROOT%bin;%PATH%"
+    set "PATH=%BOOST_JAM_TOOLSET_ROOT%bin;!PATH!"
     )
 set "BOOST_JAM_CC=gcc -DNT"
 set "BOOST_JAM_OPT_JAM=-o bootstrap\jam0.exe"
