diff --git a/CMakeLists.txt b/CMakeLists.txt
index d2bc809..b33be8e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -334,7 +334,7 @@ add_subdirectory(libs)
 add_subdirectory(tools)
 ##########################################################################
 
-if(NOT BUILD_PROJECTS STREQUAL "NONE") 
+if(NOT BUILD_PROJECTS) 
   install(EXPORT Boost DESTINATION ${BOOST_INSTALL_EXPORTS_FILE})
 endif()
 
