diff -N -u -r poppler-data-0.2.02/CMakeLists.txt poppler-data-0.2.0/CMakeLists.txt
--- CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ CMakeLists.txt	2007-12-20 01:58:55.593750000 +0100
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 2.6)
+
+set(datadir ${CMAKE_INSTALL_PREFIX}/share)
+set(pkgdatadir ${datadir}/poppler)
+
+install(DIRECTORY ${CMAKE_SOURCE_DIR}/cidToUnicode  DESTINATION ${pkgdatadir})
+install(DIRECTORY ${CMAKE_SOURCE_DIR}/cMap          DESTINATION ${pkgdatadir})
+install(DIRECTORY ${CMAKE_SOURCE_DIR}/nameToUnicode DESTINATION ${pkgdatadir})
+install(DIRECTORY ${CMAKE_SOURCE_DIR}/unicodeMap    DESTINATION ${pkgdatadir})
+install(FILES ${CMAKE_SOURCE_DIR}/COPYING README    DESTINATION ${pkgdatadir})
