diff -Nru -x '*~' kde-workspace-4.10.2.orig/kstyles/oxygen/oxygenstyle.cpp kde-workspace-4.10.2/kstyles/oxygen/oxygenstyle.cpp
--- kde-workspace-4.10.2.orig/kstyles/oxygen/oxygenstyle.cpp	2013-03-01 06:32:25.052846000 +0000
+++ kde-workspace-4.10.2/kstyles/oxygen/oxygenstyle.cpp	2013-05-24 17:21:05.361266800 +0100
@@ -3511,8 +3511,8 @@
         const bool hasAlpha( helper().hasAlphaChannel( widget ) );
         if(  hasAlpha && StyleConfigData::toolTipTransparent() )
         {
-
-            blurHelper().registerWidget( widget->window() );
+			if( widget && widget->window() )
+			{ blurHelper().registerWidget( widget->window() ); }
             topColor.setAlpha( 220 );
             bottomColor.setAlpha( 220 );
         }
