diff -Nru -x '*~' kde-workspace-4.10.0.orig/kcontrol/style/CMakeLists.txt kde-workspace-4.10.0/kcontrol/style/CMakeLists.txt
--- kde-workspace-4.10.0.orig/kcontrol/style/CMakeLists.txt	2013-01-23 22:46:32.000000000 +0100
+++ kde-workspace-4.10.0/kcontrol/style/CMakeLists.txt	2013-03-01 23:44:32.942564800 +0100
@@ -11,8 +11,11 @@
 
 kde4_add_plugin(kcm_style ${kcm_style_PART_SRCS})
 
-target_link_libraries(kcm_style kdecorations ${KDE4_KIO_LIBS} ${QIMAGEBLITZ_LIBRARIES}
+target_link_libraries(kcm_style ${KDE4_KIO_LIBS} ${QIMAGEBLITZ_LIBRARIES}
     ${X11_LIBRARIES} ${KDE4_PLASMA_LIBS} ${KDE4_KNEWSTUFF3_LIBS})
+if(NOT WIN32)
+    target_link_libraries(kcm_style kdecorations)
+endif(NOT WIN32)
 
 install(TARGETS kcm_style  DESTINATION ${PLUGIN_INSTALL_DIR})
 
diff -Nru -x '*~' kde-workspace-4.10.0.orig/kcontrol/style/kcmstyle.cpp kde-workspace-4.10.0/kcontrol/style/kcmstyle.cpp
--- kde-workspace-4.10.0.orig/kcontrol/style/kcmstyle.cpp	2013-01-23 22:46:32.000000000 +0100
+++ kde-workspace-4.10.0/kcontrol/style/kcmstyle.cpp	2013-03-02 00:07:22.723396500 +0100
@@ -47,7 +47,10 @@
 #include <KColorScheme>
 #include <KStandardDirs>
 #include <knewstuff3/downloaddialog.h>
+#ifdef Q_WS_X11
 #include <../../kwin/libkdecorations/kdecoration.h>
+#endif
+
 
 #include <QtCore/QFile>
 #include <QtCore/QSettings>
@@ -435,6 +438,7 @@
         notification->sendEvent();
     }
 
+#ifdef Q_WS_X11
     // If user select ButtonVertical, we add (if needed) a button to titlebar
     if (style == "ButtonVertical") {
         KConfig _kwinConfig("kwinrc", KConfig::NoGlobals);
@@ -448,6 +452,7 @@
         kwinConfig.writeEntry("ButtonsOnLeft", buttonsOnLeft);
         kwinConfig.writeEntry("CustomButtonPositions", "true");
     }
+#endif
 
     args = QList<QVariant>() << "appmenu" << (style != "InApplication");
     method = QDBusMessage::createMethodCall("org.kde.kded",
diff -Nru -x '*~' kde-workspace-4.10.0.orig/libs/plasmagenericshell/tests/CMakeLists.txt kde-workspace-4.10.0/libs/plasmagenericshell/tests/CMakeLists.txt
--- kde-workspace-4.10.0.orig/libs/plasmagenericshell/tests/CMakeLists.txt	2013-01-23 22:46:50.000000000 +0100
+++ kde-workspace-4.10.0/libs/plasmagenericshell/tests/CMakeLists.txt	2013-03-02 00:01:57.278782200 +0100
@@ -27,7 +27,7 @@
 kde4_add_kcfg_files(backgrounddialog_SRCS ../../../plasma/desktop/shell/data/plasma-shell-desktop.kcfgc)
 
 kde4_add_executable(plasmabackgrounddialog ${backgrounddialog_SRCS})
-target_link_libraries(plasmabackgrounddialog ${KDE4_PLASMA_LIBS} ${KDE4_KFILE_LIBRARY} kworkspace)
+target_link_libraries(plasmabackgrounddialog ${KDE4_PLASMA_LIBS} ${QT_QTUITOOLS_LIBRARY} ${KDE4_KFILE_LIBRARY} kworkspace)
 
 endif (KDE4_BUILD_TESTS)
 
