--- cmake-2.6.2\Source\CMakeLists.txt.old 2008-08-01 11:34:00.000000000 +0200
+++ cmake-2.6.2\Source\CMakeLists.txt 2008-08-06 02:19:33.265625000 +0200
@@ -430,6 +430,13 @@
 ADD_EXECUTABLE(cpack CPack/cpack.cxx)
 TARGET_LINK_LIBRARIES(cpack CPackLib)
 
+IF(WIN32)
+  TARGET_LINK_LIBRARIES(cmake comsuppw wbemuuid)
+  TARGET_LINK_LIBRARIES(cpack comsuppw wbemuuid)
+  TARGET_LINK_LIBRARIES(ctest comsuppw wbemuuid)
+  TARGET_LINK_LIBRARIES(DumpDocumentation comsuppw wbemuuid)
+ENDIF(WIN32)
+
 # Curses GUI
 IF(BUILD_CursesDialog)
   INCLUDE(${CMake_SOURCE_DIR}/Source/CursesDialog/CMakeLists.txt)
