diff -Nru libmsn-4.1.orig/msn/CMakeLists.txt libmsn-4.1/msn/CMakeLists.txt
--- libmsn-4.1.orig/msn/CMakeLists.txt	2010-01-05 13:07:08.000000000 +0100
+++ libmsn-4.1/msn/CMakeLists.txt	2010-10-12 18:22:09.659966400 +0200
@@ -18,11 +18,6 @@
     passport.cpp
 )
 
-if(WIN32)
-set(msn_STAT_SRCS   ${msn_STAT_SRCS}
-                    ${OPENSSL_INCLUDE_DIR}/openssl/applink.c)
-endif(WIN32)
-
 set(msn_HEADERS
     authdata.h
     message.h
