diff -udrN libmsn-4.0-beta4-orig/msn/message.h libmsn-4.0-beta4/msn/message.h
--- libmsn-4.0-beta4-orig/msn/message.h	2008-12-16 23:41:43.000000000 +0100
+++ libmsn-4.0-beta4/msn/message.h	2009-01-25 00:15:13.018464900 +0100
@@ -25,6 +25,9 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
+#if defined(WIN32) && !defined(_MSC_VER)
+#include <windows.h>
+#endif
 #include <string>
 #include <map>
 #include <vector>
