diff -Nru a/sdk/include/vlc/plugins/vlc_common.h b/sdk/include/vlc/plugins/vlc_common.h
--- a/sdk/include/vlc/plugins/vlc_common.h
+++ b/sdk/include/vlc/plugins/vlc_common.h
@@ -685,14 +685,14 @@
          | ((x & 0x00FF000000000000) >> 40)
          | ((x & 0xFF00000000000000) >> 56);
 #else
-    return ((x & 0x00000000000000FFLLU) << 56)
-         | ((x & 0x000000000000FF00LLU) << 40)
-         | ((x & 0x0000000000FF0000LLU) << 24)
-         | ((x & 0x00000000FF000000LLU) <<  8)
-         | ((x & 0x000000FF00000000LLU) >>  8)
-         | ((x & 0x0000FF0000000000LLU) >> 24)
-         | ((x & 0x00FF000000000000LLU) >> 40)
-         | ((x & 0xFF00000000000000LLU) >> 56);
+    return ((x & 0x00000000000000FFULL) << 56)
+         | ((x & 0x000000000000FF00ULL) << 40)
+         | ((x & 0x0000000000FF0000ULL) << 24)
+         | ((x & 0x00000000FF000000ULL) <<  8)
+         | ((x & 0x000000FF00000000ULL) >>  8)
+         | ((x & 0x0000FF0000000000ULL) >> 24)
+         | ((x & 0x00FF000000000000ULL) >> 40)
+         | ((x & 0xFF00000000000000ULL) >> 56);
 #endif
 }
 
