--- src\Makefile.am.orig	2007-09-06 16:27:01.687500000 +0200
+++ src\Makefile.am	2007-09-06 16:25:00.562500000 +0200
@@ -58,7 +58,7 @@
 	$(LTRCCOMPILE) -i $< -o $@
 
 gpg_error_res = versioninfo.o
-gpg_error_res_ldflag = -Wl,$(gpg_error_res)
+gpg_error_res_ldflag = -Wl,.libs/$(gpg_error_res)
 no_undefined = -no-undefined
 export_symbols = -export-symbols $(srcdir)/gpg-error.def
 
--- src\Makefile.in.orig	2007-09-06 16:26:55.156250000 +0200
+++ src\Makefile.in	2007-09-06 16:26:06.031250000 +0200
@@ -298,7 +298,7 @@
 @HAVE_W32_SYSTEM_FALSE@gpg_error_res = 
 @HAVE_W32_SYSTEM_TRUE@gpg_error_res = versioninfo.o
 @HAVE_W32_SYSTEM_FALSE@gpg_error_res_ldflag = 
-@HAVE_W32_SYSTEM_TRUE@gpg_error_res_ldflag = -Wl,$(gpg_error_res)
+@HAVE_W32_SYSTEM_TRUE@gpg_error_res_ldflag = -Wl,.libs/$(gpg_error_res)
 @HAVE_W32_SYSTEM_FALSE@no_undefined = 
 @HAVE_W32_SYSTEM_TRUE@no_undefined = -no-undefined
 @HAVE_W32_SYSTEM_FALSE@export_symbols = 
--- src\w32-gettext.c.orig	2005-11-02 09:33:55.000000000 +0100
+++ src\w32-gettext.c	2007-09-06 16:17:39.046875000 +0200
@@ -522,10 +522,10 @@
 # define SUBLANG_AZERI_CYRILLIC 0x02
 # endif
 # ifndef SUBLANG_BENGALI_INDIA
-# define SUBLANG_BENGALI_INDIA 0x00
+# define SUBLANG_BENGALI_INDIA 0x01
 # endif
 # ifndef SUBLANG_BENGALI_BANGLADESH
-# define SUBLANG_BENGALI_BANGLADESH 0x01
+# define SUBLANG_BENGALI_BANGLADESH 0x02
 # endif
 # ifndef SUBLANG_CHINESE_MACAU
 # define SUBLANG_CHINESE_MACAU 0x05
@@ -618,16 +618,16 @@
 # define SUBLANG_NEPALI_INDIA 0x02
 # endif
 # ifndef SUBLANG_PUNJABI_INDIA
-# define SUBLANG_PUNJABI_INDIA 0x00
+# define SUBLANG_PUNJABI_INDIA 0x01
 # endif
 # ifndef SUBLANG_PUNJABI_PAKISTAN
-# define SUBLANG_PUNJABI_PAKISTAN 0x01
+# define SUBLANG_PUNJABI_PAKISTAN 0x02
 # endif
 # ifndef SUBLANG_ROMANIAN_ROMANIA
-# define SUBLANG_ROMANIAN_ROMANIA 0x00
+# define SUBLANG_ROMANIAN_ROMANIA 0x01
 # endif
 # ifndef SUBLANG_ROMANIAN_MOLDOVA
-# define SUBLANG_ROMANIAN_MOLDOVA 0x01
+# define SUBLANG_ROMANIAN_MOLDOVA 0x02
 # endif
 # ifndef SUBLANG_SERBIAN_LATIN
 # define SUBLANG_SERBIAN_LATIN 0x02
@@ -635,12 +635,12 @@
 # ifndef SUBLANG_SERBIAN_CYRILLIC
 # define SUBLANG_SERBIAN_CYRILLIC 0x03
 # endif
-# ifndef SUBLANG_SINDHI_INDIA
-# define SUBLANG_SINDHI_INDIA 0x00
-# endif
 # ifndef SUBLANG_SINDHI_PAKISTAN
 # define SUBLANG_SINDHI_PAKISTAN 0x01
 # endif
+# ifndef SUBLANG_SINDHI_AFGHANISTAN
+# define SUBLANG_SINDHI_AFGHANISTAN 0x02
+# endif
 # ifndef SUBLANG_SPANISH_GUATEMALA
 # define SUBLANG_SPANISH_GUATEMALA 0x04
 # endif
@@ -698,14 +698,14 @@
 # ifndef SUBLANG_TAMAZIGHT_ARABIC
 # define SUBLANG_TAMAZIGHT_ARABIC 0x01
 # endif
-# ifndef SUBLANG_TAMAZIGHT_LATIN
-# define SUBLANG_TAMAZIGHT_LATIN 0x02
+# ifndef SUBLANG_TAMAZIGHT_ALGERIA_LATIN
+# define SUBLANG_TAMAZIGHT_ALGERIA_LATIN 0x02
 # endif
 # ifndef SUBLANG_TIGRINYA_ETHIOPIA
-# define SUBLANG_TIGRINYA_ETHIOPIA 0x00
+# define SUBLANG_TIGRINYA_ETHIOPIA 0x01
 # endif
 # ifndef SUBLANG_TIGRINYA_ERITREA
-# define SUBLANG_TIGRINYA_ERITREA 0x01
+# define SUBLANG_TIGRINYA_ERITREA 0x02
 # endif
 # ifndef SUBLANG_URDU_PAKISTAN
 # define SUBLANG_URDU_PAKISTAN 0x01
@@ -1074,8 +1074,8 @@
     case LANG_SINDHI:
       switch (sub)
 	{
-	case SUBLANG_SINDHI_INDIA: return "sd_IN";
 	case SUBLANG_SINDHI_PAKISTAN: return "sd_PK";
+	case SUBLANG_SINDHI_AFGHANISTAN: return "sd_AF";
 	}
       return "sd";
     case LANG_SINHALESE: return "si_LK";
@@ -1128,7 +1128,7 @@
 	{
 	/* FIXME: Adjust this when Tamazight locales appear on Unix.  */
 	case SUBLANG_TAMAZIGHT_ARABIC: return "ber_MA@arabic";
-	case SUBLANG_TAMAZIGHT_LATIN: return "ber_MA@latin";
+	case SUBLANG_TAMAZIGHT_ALGERIA_LATIN: return "ber_MA@latin";
 	}
       return "ber_MA";
     case LANG_TAMIL:
