diff -Nru -x '*~' poppler-0.28.1.orig/config.h.cmake poppler-0.28.1/config.h.cmake
--- poppler-0.28.1.orig/config.h.cmake	2014-09-28 17:19:59.000000000 +0200
+++ poppler-0.28.1/config.h.cmake	2014-12-04 17:11:22.407854900 +0100
@@ -229,3 +229,7 @@
 
 /* Define for large files, on AIX-style hosts. */
 /* TODO This is wrong, port if needed #undef _LARGE_FILES */
+
+#cmakedefine HAVE_STRCPY_S 1
+
+#cmakedefine HAVE_STRCAT_S 1
\ Kein Zeilenumbruch am Dateiende.
