diff -Nrub -x '*~' -x '*\.orig' -x'*\.o' poppler-0.38.0.orig/config.h.cmake poppler-0.38.0/config.h.cmake
--- poppler-0.38.0.orig/config.h.cmake	2015-10-28 23:11:49.000000000 +0100
+++ poppler-0.38.0/config.h.cmake	2015-12-08 23:42:24.499813200 +0100
@@ -222,6 +222,9 @@
 /* MS has defined snprintf as deprecated */
 #ifdef _MSC_VER
+#if _MSC_VER < 1900
 #define snprintf _snprintf
+#endif
+#include <algorithm>
 #endif
 
 /* Number of bits in a file offset, on hosts where this is settable. */
@@ -232,3 +235,6 @@
 
 /* Define for large files, on AIX-style hosts. */
 /* TODO This is wrong, port if needed #undef _LARGE_FILES */
+
+#cmakedefine HAVE_STRCAT_S 1
+#cmakedefine HAVE_STRCPY_S 1
\ No newline at end of file
diff -Nrub -x '*~' -x '*\.orig' -x'*\.o' poppler-0.38.0.orig/poppler/poppler-config.h.cmake poppler-0.38.0/poppler/poppler-config.h.cmake
--- poppler-0.38.0.orig/poppler/poppler-config.h.cmake	2015-10-05 16:04:39.000000000 +0200
+++ poppler-0.38.0/poppler/poppler-config.h.cmake	2015-12-08 23:40:24.572953800 +0100
@@ -160,6 +160,7 @@
 #include <windef.h>
 #else
 #define CDECL
+#define WIN32_LEAN_AND_MEAN 1
 #endif
 
 #if defined(_WIN32)
@@ -186,7 +187,7 @@
 #define GCC_PRINTF_FORMAT(fmt_index, va_index)
 #endif
 
-#if defined(_MSC_VER)
+#if defined(_MSC_VER) && _MSC_VER <= 1600
 #define fmax(a, b) std::max(a, b)
 #define fmin(a, b) std::min(a, b)
 #endif
