diff -Nru libofx-0.9.2.orig/CMakeLists.txt libofx-0.9.2/CMakeLists.txt
--- libofx-0.9.2.orig/CMakeLists.txt	1970-01-01 02:00:00.000000000 +0200
+++ libofx-0.9.2/CMakeLists.txt	2011-02-15 21:01:22.922812600 +0200
@@ -0,0 +1,98 @@
+project(ofx)
+
+set (VERSION_MAJOR 0)
+set (VERSION_MINOR 9)
+set (VERSION_PATH 1)
+
+cmake_minimum_required(VERSION 2.6)
+
+list(APPEND CMAKE_MODULE_PATH  ${CMAKE_SOURCE_DIR})
+find_package(OpenSP REQUIRED)
+
+if(MSVC)
+    add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+endif(MSVC)
+
+include_directories(
+    inc
+    ${OPENSP_INCLUDES}
+    $ENV{KDEROOT}/include
+)
+
+add_definitions(
+    -DOS_WIN32
+    -DMAKEFILE_DTD_PATH="${CMAKE_PREFIX_PATH}/share/libofx/dtd/"
+)
+
+set (COEFF_HDRS 
+        inc/libofx.h
+        lib/messages.hh
+        lib/ofx_preproc.hh
+        lib/file_preproc.hh
+        lib/context.hh
+        lib/ofx_sgml.hh
+        lib/ofc_sgml.hh
+        lib/ofx_aggregate.hh
+        lib/ofx_error_msg.hh
+        lib/ofx_containers.hh
+        lib/ofx_request.hh
+        lib/ofx_request_accountinfo.hh
+        lib/ofx_request_statement.hh
+        lib/ofx_utilities.hh
+        lib/tree.hh
+        lib/win32.hh
+        )
+
+set (SRC_SOURCES
+         lib/messages.cpp
+         lib/ofx_utilities.cpp
+         lib/file_preproc.cpp
+         lib/context.cpp
+         lib/ofx_preproc.cpp
+         lib/ofx_container_generic.cpp
+         lib/ofx_container_main.cpp
+         lib/ofx_container_security.cpp
+         lib/ofx_container_statement.cpp
+         lib/ofx_container_account.cpp
+         lib/ofx_container_transaction.cpp
+         lib/ofx_containers_misc.cpp
+         lib/ofx_request.cpp
+         lib/ofx_request_accountinfo.cpp
+         lib/ofx_request_statement.cpp
+         lib/ofx_sgml.cpp
+         lib/ofc_sgml.cpp
+         lib/win32.cpp
+         ${COEFF_HDRS}
+)
+
+# We would like it to be shared, but libofx does not export symbols manually
+# thus it causes troubles on MSVC
+#add_library(ofx SHARED ${SRC_SOURCES})
+add_library(ofx ${SRC_SOURCES})
+
+FIND_LIBRARY( ICONV_LIBRARIES iconv )
+if(MSVC)
+    set_target_properties(ofx PROPERTIES OUTPUT_NAME "libofx")
+endif(MSVC)
+target_link_libraries(ofx ${OPENSP_LIBRARIES} ${ICONV_LIBRARIES})
+
+######### add a utility function so that we can test ofx files ##########
+set(ofxdump_SRCS
+    ofxdump/cmdline.c
+    ofxdump/ofxdump.cpp 
+)
+
+if(MSVC)
+    set(ofxdump_SRCS ${ofxdump_SRCS} lib/messages.cpp lib/getopt.c lib/getopt1.c)
+endif(MSVC)
+
+add_definitions(-DCMDLINE_PARSER_PACKAGE=\"ofxdump\" -DCMDLINE_PARSER_PACKAGE_NAME=\"ofxdump\" -DCMDLINE_PARSER_VERSION=\"0.9.1\")
+add_executable(ofxdump ${ofxdump_SRCS})
+target_link_libraries(ofxdump ofx)
+
+install(TARGETS ofx ofxdump RUNTIME DESTINATION bin ARCHIVE DESTINATION lib LIBRARY DESTINATION lib)
+install(FILES inc/libofx.h DESTINATION include/libofx)
+install(FILES dtd/opensp.dcl DESTINATION ${CMAKE_PREFIX_PATH}/share/libofx/dtd)
+install(FILES dtd/ofx160.dtd DESTINATION ${CMAKE_PREFIX_PATH}/share/libofx/dtd)
+install(FILES dtd/ofx201.dtd DESTINATION ${CMAKE_PREFIX_PATH}/share/libofx/dtd)
+install(FILES dtd/ofc.dtd DESTINATION ${CMAKE_PREFIX_PATH}/share/libofx/dtd)
diff -Nru libofx-0.9.2.orig/config.h libofx-0.9.2/config.h
--- libofx-0.9.2.orig/config.h	1970-01-01 02:00:00.000000000 +0200
+++ libofx-0.9.2/config.h	2011-02-15 21:01:22.922812600 +0200
@@ -0,0 +1,71 @@
+// Have these
+#define HAVE_EVENTGENERATOR_H 1
+#define HAVE_LIBCURL 1
+#define HAVE_MEMORY_H 1
+#define HAVE_PARSEREVENTGENERATORKIT_H 1
+#define HAVE_SGMLAPPLICATION_H 1
+#define HAVE_STDLIB_H 1
+#define HAVE_STRING_H 1
+#define HAVE_SYS_STAT_H 1
+#define HAVE_SYS_TYPES_H 1
+
+// Do not exist
+#undef HAVE_DLFCN_H
+#undef HAVE_GETOPT_H
+#undef HAVE_GETOPT_LONG
+#undef HAVE_INTTYPES_H
+#undef HAVE_STDINT_H
+#undef HAVE_STRINGS_H
+#undef HAVE_UNISTD_H
+
+
+// Unsure
+#define HAVE_ICONV 1
+#define HAVE_LIBXMLPP 1
+
+#undef LIBCURL_FEATURE_ASYNCHDNS
+
+/* Defined if libcurl supports IPv6 */
+#undef LIBCURL_FEATURE_IPV6
+
+/* Defined if libcurl supports KRB4 */
+#undef LIBCURL_FEATURE_KRB4
+
+/* Defined if libcurl supports libz */
+#undef LIBCURL_FEATURE_LIBZ
+
+/* Defined if libcurl supports SSL */
+#undef LIBCURL_FEATURE_SSL
+
+/* Defined if libcurl supports DICT */
+#undef LIBCURL_PROTOCOL_DICT
+
+/* Defined if libcurl supports FILE */
+#undef LIBCURL_PROTOCOL_FILE
+
+/* Defined if libcurl supports FTP */
+#undef LIBCURL_PROTOCOL_FTP
+
+/* Defined if libcurl supports FTPS */
+#undef LIBCURL_PROTOCOL_FTPS
+
+/* Defined if libcurl supports GOPHER */
+#undef LIBCURL_PROTOCOL_GOPHER
+
+/* Defined if libcurl supports HTTP */
+#undef LIBCURL_PROTOCOL_HTTP
+
+/* Defined if libcurl supports HTTPS */
+#undef LIBCURL_PROTOCOL_HTTPS
+
+/* Defined if libcurl supports LDAP */
+#undef LIBCURL_PROTOCOL_LDAP
+
+/* Defined if libcurl supports TELNET */
+#undef LIBCURL_PROTOCOL_TELNET
+
+
+// MSVC Defines we seem to need
+#ifdef _MSC_VER
+typedef int ssize_t;
+#endif
diff -Nru libofx-0.9.2.orig/FindOpenSP.cmake libofx-0.9.2/FindOpenSP.cmake
--- libofx-0.9.2.orig/FindOpenSP.cmake	1970-01-01 02:00:00.000000000 +0200
+++ libofx-0.9.2/FindOpenSP.cmake	2011-02-15 21:01:22.922812600 +0200
@@ -0,0 +1,36 @@
+# - Try to find the OpenSP library
+# 
+# Once done this will define
+#
+#  OPENSP_FOUND - system has OPENSP
+#  OPENSP_INCLUDES - the OPENSP include directories
+#  OPENSP_LIBRARIES - The libraries needed to use OPENSP
+
+if (WIN32)
+
+    set(OPENSP_FOUND FALSE)
+    find_path(OPENSP_INCLUDES ParserEventGeneratorKit.h
+        ${CMAKE_PREFIX_PATH}/include/opensp
+    )
+
+    find_library(OPENSP_LIBRARIES
+        NAMES sp133
+        PATHS
+            ${CMAKE_PREFIX_PATH}/lib
+    )
+
+    if (OPENSP_INCLUDES AND OPENSP_OPENSP_LIBRARIES)
+        set(OPENSP_FOUND TRUE)
+    endif (OPENSP_INCLUDES AND OPENSP_OPENSP_LIBRARIES)
+
+  if (OPENSP_FOUND)
+    if (NOT OPENSP_FIND_QUIETLY)
+      message(STATUS "Found OPENSP library: ${OPENSP_LIBRARIES}")
+    endif (NOT OPENSP_FIND_QUIETLY)
+
+  else (OPENSP_FOUND)
+    if (OPENSP_FIND_REQUIRED)
+      message(FATAL_ERROR "Could NOT find OPENSP library\nPlease install it first")
+    endif (OPENSP_FIND_REQUIRED)
+  endif (OPENSP_FOUND)
+endif (WIN32)
diff -Nru libofx-0.9.2.orig/lib/getopt.c libofx-0.9.2/lib/getopt.c
--- libofx-0.9.2.orig/lib/getopt.c	2004-04-24 02:30:34.000000000 +0300
+++ libofx-0.9.2/lib/getopt.c	2011-02-15 21:01:22.922812600 +0200
@@ -40,6 +40,7 @@
 #endif
 
 #include <stdio.h>
+#include "getopt.h"
 
 /* Comment out all this code if we are using the GNU C Library, and are not
    actually compiling the library itself.  This code is part of the GNU C
@@ -199,7 +200,7 @@
 # define my_index	strchr
 #else
 
-# if HAVE_STRING_H
+# if HAVE_STRING_H || defined(_MSC_VER)
 #  include <string.h>
 # else
 #  include <strings.h>
diff -Nru libofx-0.9.2.orig/lib/getopt.h libofx-0.9.2/lib/getopt.h
--- libofx-0.9.2.orig/lib/getopt.h	1970-01-01 02:00:00.000000000 +0200
+++ libofx-0.9.2/lib/getopt.h	2011-02-15 21:01:22.938437600 +0200
@@ -0,0 +1,20 @@
+#ifndef _getopt_h
+#define _getopt_h
+
+#ifndef _MSC_VER
+#include <getopt.h>
+#else
+struct option {
+    const char *name;
+    int has_arg;
+    int *flag;
+    int val;
+};
+
+extern char *optarg;
+extern int optind;
+
+
+int getopt_long(int argc, char *const *argv, const char *shortopts, const struct option *longopts, int *indexptr);
+#endif
+#endif
diff -Nru libofx-0.9.2.orig/lib/messages.cpp libofx-0.9.2/lib/messages.cpp
--- libofx-0.9.2.orig/lib/messages.cpp	2011-02-12 18:51:02.000000000 +0200
+++ libofx-0.9.2/lib/messages.cpp	2011-02-15 21:01:22.938437600 +0200
@@ -25,6 +25,10 @@
 SGMLApplication::OpenEntityPtr entity_ptr; /**< Global for determining the line number in OpenSP */
 SGMLApplication::Position position; /**< Global for determining the line number in OpenSP */
 
+#ifdef _MSC_VER
+#define volatile
+#endif
+
 volatile int ofx_PARSER_msg = false; /**< If set to true, parser events will be printed to the console */
 volatile int ofx_DEBUG_msg = false;/**< If set to true, general debug messages will be printed to the console */
 volatile int ofx_DEBUG1_msg = false;/**< If set to true, debug level 1 messages will be printed to the console */
diff -Nru libofx-0.9.2.orig/ofxdump/cmdline.c libofx-0.9.2/ofxdump/cmdline.c
--- libofx-0.9.2.orig/ofxdump/cmdline.c	2011-02-12 19:18:41.000000000 +0200
+++ libofx-0.9.2/ofxdump/cmdline.c	2011-02-15 21:06:04.773972100 +0200
@@ -380,7 +380,6 @@
                const char *long_opt, char short_opt,
                const char *additional_error)
 {
-  FIX_UNUSED (field);
   char *stop_char = 0;
   const char *val = value;
   int found;
@@ -480,8 +479,6 @@
 
   optarg = 0;
   optind = 0;
-  opterr = params->print_errors;
-  optopt = '?';
 
   while (1)
     {
diff -Nru libofx-0.9.2.orig/ofxdump/getopt.h libofx-0.9.2/ofxdump/getopt.h
--- libofx-0.9.2.orig/ofxdump/getopt.h	1970-01-01 02:00:00.000000000 +0200
+++ libofx-0.9.2/ofxdump/getopt.h	2011-02-15 21:01:22.938437600 +0200
@@ -0,0 +1 @@
+#include "../lib/getopt.h"
