Index: libnova/src/CMakeLists.txt
===================================================================
--- libnova/src/CMakeLists.txt	(revision: 270)
+++ libnova/src/CMakeLists.txt	(working copy)
@@ -35,6 +35,7 @@
 	${HEADER_PATH}/hyperbolic_motion.h
 	${HEADER_PATH}/parallax.h
 	${HEADER_PATH}/airmass.h
+	${HEADER_PATH}/heliocentric_time.h
 )
 
 add_library(${LIBRARY_NAME} 
@@ -71,16 +72,20 @@
 	hyperbolic_motion.c
 	parallax.c
 	airmass.c
+	heliocentric_time.c
 )
 
-if(MSVC)
+if(MSVC_IDE)
     if(BUILD_SHARED_LIBRARY)
-        set_target_properties(${LIBRARY_NAME} PROPERTIES PREFIX "../../bin/" IMPORT_PREFIX "../")
+        set_target_properties(${LIBRARY_NAME} PROPERTIES PREFIX "../" IMPORT_PREFIX "../")
     else(BUILD_SHARED_LIBRARY)
         set_target_properties(${LIBRARY_NAME} PROPERTIES PREFIX "../")
     endif(BUILD_SHARED_LIBRARY)
-endif(MSVC)
+endif(MSVC_IDE)
 
+if(CMAKE_COMPILER_IS_GNUCC)
+    target_link_libraries(${LIBRARY_NAME} m)
+endif(CMAKE_COMPILER_IS_GNUCC)
 
 
 
Index: libnova/src/heliocentric_time.c
===================================================================
--- libnova/src/heliocentric_time.c	(revision: 270)
+++ libnova/src/heliocentric_time.c	(working copy)
@@ -34,17 +34,19 @@
 */
 double ln_get_heliocentric_time_diff (double JD, struct ln_equ_posn *object)
 {
+	double theta, ra, dec, c_dec, obliq;
+	
 	struct ln_nutation nutation;
 	struct ln_helio_posn earth;
 
 	ln_get_nutation (JD, &nutation);
 	ln_get_earth_helio_coords (JD, &earth);
 
-	double theta = ln_deg_to_rad (ln_range_degrees (earth.L + 180));
-	double ra = ln_deg_to_rad (object->ra);
-	double dec = ln_deg_to_rad (object->dec);
-	double c_dec = cos (dec);
-	double obliq = ln_deg_to_rad (nutation.ecliptic);
+	theta = ln_deg_to_rad (ln_range_degrees (earth.L + 180));
+	ra = ln_deg_to_rad (object->ra);
+	dec = ln_deg_to_rad (object->dec);
+	c_dec = cos (dec);
+	obliq = ln_deg_to_rad (nutation.ecliptic);
 
 	/* L.Binnendijk Properties of Double Stars, Philadelphia, University of Pennselvania Press, pp. 228-232, 1960 */
 	return -0.0057755 * earth.R * (
Index: libnova/CMakeLists.txt
===================================================================
--- libnova/CMakeLists.txt	(revision: 270)
+++ libnova/CMakeLists.txt	(working copy)
@@ -45,22 +45,23 @@
 
 
 # set and create the EXECUTABLE_OUTPUT_PATH
-set(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/bin)
-file(MAKE_DIRECTORY ${EXECUTABLE_OUTPUT_PATH})
+#set(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/bin)
+#file(MAKE_DIRECTORY ${EXECUTABLE_OUTPUT_PATH})
 
 # set and create the LIBRARY_OUTPUT_PATH
-set(LIBRARY_OUTPUT_PATH ${PROJECT_BINARY_DIR}/lib)
-file(MAKE_DIRECTORY ${LIBRARY_OUTPUT_PATH})
+#set(LIBRARY_OUTPUT_PATH ${PROJECT_BINARY_DIR}/lib)
+#file(MAKE_DIRECTORY ${LIBRARY_OUTPUT_PATH})
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
 
-
 # define a macro to setup executable program which are linked with libnova
 macro(SETUP_EXECUTABLE EXECUTABLE_NAME)
     add_executable(${EXECUTABLE_NAME} ${EXECUTABLE_NAME}.c)
     target_link_libraries(${EXECUTABLE_NAME} libnova)
     set_target_properties(${EXECUTABLE_NAME} PROPERTIES DEBUG_POSTFIX "${CMAKE_DEBUG_POSTFIX}")
-    if(MSVC)
+    if(MSVC_IDE)
         set_target_properties(${EXECUTABLE_NAME} PROPERTIES PREFIX "../")
-    endif(MSVC)
+    endif(MSVC_IDE)
     INSTALL(TARGETS ${EXECUTABLE_NAME} RUNTIME DESTINATION bin)
 endmacro(SETUP_EXECUTABLE EXECUTABLE_NAME)
 