diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user -x .idea -x .clang-format gettext-1.0.orig/gettext-runtime/gnulib-lib/progreloc.c gettext-1.0/gettext-runtime/gnulib-lib/progreloc.c
--- gettext-1.0.orig/gettext-runtime/gnulib-lib/progreloc.c	2026-01-02 10:16:14.000000000 +0000
+++ gettext-1.0/gettext-runtime/gnulib-lib/progreloc.c	2026-06-15 21:01:59.474995026 +0000
@@ -257,7 +257,7 @@
     char buf[6+10+5];
     sprintf (buf, "/proc/%d/exe", getpid ());
     char *link = xreadlink (buf);
-    if (link != NULL && link[0] != '[')
+    if (link != NULL && link[0] != '[') {
       return link;
     }
     if (executable_fd < 0)
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user -x .idea -x .clang-format gettext-1.0.orig/gettext-tools/gnulib-lib/progreloc.c gettext-1.0/gettext-tools/gnulib-lib/progreloc.c
--- gettext-1.0.orig/gettext-tools/gnulib-lib/progreloc.c	2026-01-02 10:16:14.000000000 +0000
+++ gettext-1.0/gettext-tools/gnulib-lib/progreloc.c	2026-06-15 21:01:59.474995026 +0000
@@ -257,7 +257,7 @@
     char buf[6+10+5];
     sprintf (buf, "/proc/%d/exe", getpid ());
     char *link = xreadlink (buf);
-    if (link != NULL && link[0] != '[')
+    if (link != NULL && link[0] != '[') {
       return link;
     }
     if (executable_fd < 0)
