diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user gettext-0.22.3.orig/gettext-tools/src/msgfilter.c gettext-0.22.3/gettext-tools/src/msgfilter.c
--- gettext-0.22.3.orig/gettext-tools/src/msgfilter.c	2023-09-18 22:09:26.000000000 +0200
+++ gettext-0.22.3/gettext-tools/src/msgfilter.c	2023-10-28 16:25:54.504649000 +0200
@@ -63,6 +63,23 @@
 
 #define _(str) gettext (str)
 
+#ifdef _MSC_VER
+// https://invent.kde.org/packaging/kdewin/-/blob/076ae9319e8bfc89b457ecf8797d01bd2600cf7a/src/stdlib.c#L112
+inline int unsetenv (const char *name)
+{
+    if (name == NULL || *name == '\0' || strchr (name, '=') != NULL)
+    {
+      errno = EINVAL;
+      return -1;
+    }
+
+    _putenv_s(name, "");
+    return 0;
+}
+
+
+#endif
+
 
 /* We use a child process, and communicate through a bidirectional pipe.  */
 
