diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user gettext-0.22.3.orig/gettext-tools/gnulib-m4/libxml.m4 gettext-0.22.3/gettext-tools/gnulib-m4/libxml.m4
--- gettext-0.22.3.orig/gettext-tools/gnulib-m4/libxml.m4	2022-09-11 12:22:30.000000000 +0200
+++ gettext-0.22.3/gettext-tools/gnulib-m4/libxml.m4	2023-10-26 14:03:30.597819300 +0200
@@ -48,7 +48,7 @@
         dnl libxml2 is built in such a way that uses of xmlFree work fine with
         dnl -Wl,--enable-auto-import but lead to a link error with
         dnl -Wl,--disable-auto-import.
-        AC_LIB_LINKFLAGS_BODY([xml2])
+        AC_LIB_LINKFLAGS_BODY([libxml2])
         LIBS="$gl_save_LIBS $LIBXML2 $LIBICONV"
         AC_LINK_IFELSE(
           [AC_LANG_PROGRAM(
