diff -Nru openssl-0.9.8k.orig/e_os2.h openssl-0.9.8k/e_os2.h
--- openssl-0.9.8k.orig/e_os2.h
+++ openssl-0.9.8k/e_os2.h
@@ -264,7 +264,7 @@
 # define OPENSSL_IMPLEMENT_GLOBAL(type,name)			     \
 	extern type _hide_##name;				     \
 	type *_shadow_##name(void) { return &_hide_##name; }	     \
-	static type _hide_##name
+	type _hide_##name
 # define OPENSSL_DECLARE_GLOBAL(type,name) type *_shadow_##name(void)
 # define OPENSSL_GLOBAL_REF(name) (*(_shadow_##name()))
 #else
diff -Nru openssl-0.9.8k.orig/include/openssl/e_os2.h openssl-0.9.8k/include/openssl/e_os2.h
--- openssl-0.9.8k.orig/include/openssl/e_os2.h
+++ openssl-0.9.8k/include/openssl/e_os2.h
@@ -264,7 +264,7 @@
 # define OPENSSL_IMPLEMENT_GLOBAL(type,name)			     \
 	extern type _hide_##name;				     \
 	type *_shadow_##name(void) { return &_hide_##name; }	     \
-	static type _hide_##name
+	type _hide_##name
 # define OPENSSL_DECLARE_GLOBAL(type,name) type *_shadow_##name(void)
 # define OPENSSL_GLOBAL_REF(name) (*(_shadow_##name()))
 #else
