diff -Nru -x '*~' kde-baseapps-4.10.0.orig/dolphin/src/tests/CMakeLists.txt kde-baseapps-4.10.0/dolphin/src/tests/CMakeLists.txt
--- kde-baseapps-4.10.0.orig/dolphin/src/tests/CMakeLists.txt	2013-01-23 22:47:58.000000000 +0100
+++ kde-baseapps-4.10.0/dolphin/src/tests/CMakeLists.txt	2013-03-02 00:38:41.079214700 +0100
@@ -20,7 +20,11 @@
     ../kitemviews/kfileitemlistview.cpp
     ../kitemviews/kitemmodelbase.cpp
     ../kitemviews/kitemlistview.cpp
+    ../kitemviews/kitemlistcontainer.cpp
+    ../kitemviews/kitemlistwidget.cpp
     ../kitemviews/kitemlistviewaccessible.cpp
+    ../kitemviews/kstandarditemlistview.cpp
+    ../kitemviews/kstandarditemlistwidget.cpp
 )
 kde4_add_unit_test(kitemlistcontrollertest TEST ${kitemlistcontrollertest_SRCS})
 target_link_libraries(kitemlistcontrollertest dolphinprivate ${KDE4_KIO_LIBS} ${QT_QTTEST_LIBRARY})
@@ -34,6 +38,10 @@
     ../kitemviews/kitemmodelbase.cpp
     ../kitemviews/kitemlistview.cpp
     ../kitemviews/kitemlistviewaccessible.cpp
+    ../kitemviews/kitemlistcontainer.cpp
+    ../kitemviews/kitemlistwidget.cpp
+    ../kitemviews/kstandarditemlistview.cpp
+    ../kitemviews/kstandarditemlistwidget.cpp
 )
 kde4_add_unit_test(kfileitemlistviewtest TEST ${kfileitemlistviewtest_SRCS})
 target_link_libraries(kfileitemlistviewtest dolphinprivate ${KDE4_KIO_LIBS} ${QT_QTTEST_LIBRARY})
diff -Nru -x '*~' kde-baseapps-4.10.0.orig/plasma/applets/folderview/dialogshadows.cpp kde-baseapps-4.10.0/plasma/applets/folderview/dialogshadows.cpp
--- kde-baseapps-4.10.0.orig/plasma/applets/folderview/dialogshadows.cpp	2013-01-23 22:48:01.000000000 +0100
+++ kde-baseapps-4.10.0/plasma/applets/folderview/dialogshadows.cpp	2013-03-01 18:14:54.658424100 +0100
@@ -161,8 +161,12 @@
 
 QPixmap DialogShadows::Private::initEmptyPixmap(const QSize &size)
 {
+#ifdef Q_WS_X11
     Pixmap emptyXPix = XCreatePixmap(QX11Info::display(), QX11Info::appRootWindow(), size.width(), size.height(), 32);
     QPixmap tempEmptyPix = QPixmap::fromX11Pixmap(emptyXPix, QPixmap::ExplicitlyShared);
+#else
+    QPixmap tempEmptyPix(size.width(), size.height());
+#endif
     tempEmptyPix.fill(Qt::transparent);
     return tempEmptyPix;
 }
