Index: nepomuk/core/ontology/entity.cpp
===================================================================
--- nepomuk/core/ontology/entity.cpp	(revision 915828)
+++ nepomuk/core/ontology/entity.cpp	(revision 915829)
@@ -32,19 +32,6 @@
 
 #include <kicon.h>
 
-#ifndef _MSC_VER
-uint qHash( const QUrl& url )
-{
-    return qHash( url.toString() );
-}
-#endif
-
-// uint qHash( const Nepomuk::Types::Entity& c )
-// {
-//     return (uint)(ulong)c.d.data();
-// }
-
-
 Nepomuk::Types::EntityPrivate::EntityPrivate( const QUrl& uri_ )
     : uri( uri_ ),
       available( uri_.isValid() ? -1 : 0 ),
Index: nepomuk/core/ontology/entitymanager.h
===================================================================
--- nepomuk/core/ontology/entitymanager.h	(revision 915828)
+++ nepomuk/core/ontology/entitymanager.h	(revision 915829)
@@ -25,10 +25,6 @@
 #include <QtCore/QSharedData>
 #include <QtCore/QMutex>
 
-inline uint qHash( const QUrl& url ) {
-    return qHash( url.toString() );
-}
-
 namespace Soprano {
     class Statement;
 }
Index: kdeui/windowmanagement/kwindowinfo_win.cpp
===================================================================
--- kdeui/windowmanagement/kwindowinfo_win.cpp	(revision 915711)
+++ kdeui/windowmanagement/kwindowinfo_win.cpp	(revision 915712)
@@ -147,7 +147,7 @@
         return NET::PopupMenu;
     else if(windowStyleEx & WS_EX_TOOLWINDOW && supported_types & NET::TooltipMask)
         return NET::Tooltip;
-    else if(!windowStyle & WS_CHILD  && supported_types & NET::NormalMask)
+    else if(!(windowStyle & WS_CHILD) && supported_types & NET::NormalMask)
         return NET::Normal;
         
     return wt;
Index: interfaces/ktexteditor/codecompletionmodelcontrollerinterface.cpp
===================================================================
--- interfaces/ktexteditor/codecompletionmodelcontrollerinterface.cpp	(revision 907328)
+++ interfaces/ktexteditor/codecompletionmodelcontrollerinterface.cpp	(revision 907329)
@@ -19,6 +19,8 @@
 
 #include "codecompletionmodelcontrollerinterface.h"
 
+#include <QtCore/QModelIndex>
+
 #include <ktexteditor/view.h>
 #include <ktexteditor/document.h>
 
