Index: hupnp_av/hupnp_av.pro
===================================================================
--- hupnp_av/hupnp_av.pro	(Revision 129)
+++ hupnp_av/hupnp_av.pro	(Arbeitskopie)
@@ -12,15 +12,11 @@
 
 INCLUDEPATH += ./include/
 
-isEmpty(PREFIX) {
-    INCLUDEPATH += ../hupnp/include
-    LIBS += -L"./../hupnp/bin/" -lHUpnp
-}
-else {
-    INCLUDEPATH += $$PREFIX/include/HUpnpCore/
-    LIBS += -L"$$PREFIX/lib" -lHUpnp
-}
 
+INCLUDEPATH += ../hupnp/include
+LIBS += -L"./../hupnp/bin/" -lHUpnp1
+
+
 OBJECTS_DIR = obj
 DESTDIR     = ./bin
 MOC_DIR     = obj
