diff -N -r -u fontconfig-2.4.2-orig/cmake/modules/FindFreetype.cmake fontconfig-2.4.2/cmake/modules/FindFreetype.cmake
--- fontconfig-2.4.2-orig/cmake/modules/FindFreetype.cmake	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/cmake/modules/FindFreetype.cmake	2008-02-10 16:34:13.625000000 +0100
@@ -0,0 +1,73 @@
+# - Try to find the freetype library
+# Once done this will define
+#
+#  FREETYPE_FOUND - system has Freetype
+#  FREETYPE_INCLUDE_DIR - the FREETYPE include directory
+#  FREETYPE_LIBRARIES - Link these to use FREETYPE
+#
+# Copyright (c) 2006, Laurent Montel, <montel@kde.org>
+#
+# Redistribution and use is allowed according to the terms of the BSD license.
+# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+
+
+if (FREETYPE_LIBRARIES AND FREETYPE_INCLUDE_DIR)
+
+  # in cache already
+  set(FREETYPE_FOUND TRUE)
+
+else (FREETYPE_LIBRARIES AND FREETYPE_INCLUDE_DIR)
+
+  FIND_PROGRAM(FREETYPECONFIG_EXECUTABLE NAMES freetype-config PATHS
+     /opt/local/bin
+  )
+
+  #reset vars
+  set(FREETYPE_LIBRARIES)
+  set(FREETYPE_INCLUDE_DIR)
+
+  # if freetype-config has been found
+  if(FREETYPECONFIG_EXECUTABLE)
+
+    EXEC_PROGRAM(${FREETYPECONFIG_EXECUTABLE} ARGS --libs RETURN_VALUE _return_VALUE OUTPUT_VARIABLE FREETYPE_LIBRARIES)
+
+    EXEC_PROGRAM(${FREETYPECONFIG_EXECUTABLE} ARGS --cflags RETURN_VALUE _return_VALUE OUTPUT_VARIABLE _freetype_pkgconfig_output)
+    if(FREETYPE_LIBRARIES AND _freetype_pkgconfig_output)
+      set(FREETYPE_FOUND TRUE)
+
+      # freetype-config can print out more than one -I, so we need to chop it up
+      # into a list and process each entry separately
+      SEPARATE_ARGUMENTS(_freetype_pkgconfig_output)
+      FOREACH(value ${_freetype_pkgconfig_output})
+        STRING(REGEX REPLACE "-I(.+)" "\\1" value "${value}")
+        set(FREETYPE_INCLUDE_DIR ${FREETYPE_INCLUDE_DIR} ${value})
+      ENDFOREACH(value)
+    endif(FREETYPE_LIBRARIES AND _freetype_pkgconfig_output)
+
+    MARK_AS_ADVANCED(FREETYPE_LIBRARIES FREETYPE_INCLUDE_DIR)
+
+    set( FREETYPE_LIBRARIES ${FREETYPE_LIBRARIES} CACHE INTERNAL "The libraries for freetype" )
+
+  else(FREETYPECONFIG_EXECUTABLE)
+	find_path (FREETYPE_INCLUDE_DIR freetype/freetype.h
+        PATH ${CMAKE_INCLUDE_PATH}
+        PATH_SUFFIXES freetype2
+	)
+    find_library(FREETYPE_LIBRARIES freetype)
+    if(FREETYPE_INCLUDE_DIR AND FREETYPE_LIBRARIES)
+        set(FREETYPE_FOUND TRUE)
+    endif(FREETYPE_INCLUDE_DIR AND FREETYPE_LIBRARIES)
+  endif(FREETYPECONFIG_EXECUTABLE)
+
+
+  IF (FREETYPE_FOUND)
+    IF (NOT Freetype_FIND_QUIETLY)
+       MESSAGE(STATUS "Found Freetype: ${FREETYPE_LIBRARIES}")
+    ENDIF (NOT Freetype_FIND_QUIETLY)
+  ELSE (FREETYPE_FOUND)
+    IF (Freetype_FIND_REQUIRED)
+       MESSAGE(FATAL_ERROR "Could not find FreeType library")
+    ENDIF (Freetype_FIND_REQUIRED)
+  ENDIF (FREETYPE_FOUND)
+
+endif (FREETYPE_LIBRARIES AND FREETYPE_INCLUDE_DIR)
diff -N -r -u fontconfig-2.4.2-orig/cmake/modules/FindKDEWIN.cmake fontconfig-2.4.2/cmake/modules/FindKDEWIN.cmake
--- fontconfig-2.4.2-orig/cmake/modules/FindKDEWIN.cmake	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/cmake/modules/FindKDEWIN.cmake	2008-02-10 16:34:13.640625000 +0100
@@ -0,0 +1,64 @@
+# - Try to find the directory in which the kdewin32 library and other win32 related libraries lives
+# 
+# used environment vars 
+#  KDEWIN_DIR  - kdewin root dir 
+#
+# this will define
+#  KDEWIN_FOUND - system has KDEWIN
+#  KDEWIN_DIR - the KDEWIN root installation dir
+#
+# Copyright (c) 2007, Ralf Habacker, <ralf.habacker@freenet.de>
+#
+# Redistribution and use is allowed according to the terms of the BSD license.
+# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+
+if (WIN32)
+  IF (NOT KDEWIN_DIR)
+    # check for enviroment variable
+    file(TO_CMAKE_PATH "$ENV{KDEWIN_DIR}" KDEWIN_DIR)
+    if(NOT KDEWIN_DIR)
+      file(TO_CMAKE_PATH "$ENV{PROGRAMFILES}" _progFiles)
+      if (MINGW)
+        set (DIR "kdewin-mingw")
+      else (MINGW)
+        set (DIR "kdewin-msvc")
+      endif (MINGW)
+      
+      # search in the default program install folder
+      find_file(KDEWIN_DIR_tmp ${DIR} kdewin kdewin32 win32libs gnuwin32
+      PATHS
+        "${_progFiles}"
+      )
+      set (KDEWIN_DIR ${KDEWIN_DIR_tmp})
+    endif (NOT KDEWIN_DIR)
+    if (KDEWIN_DIR)
+      message(STATUS "Found windows supplementary package location: ${KDEWIN_DIR}")
+    endif (KDEWIN_DIR)
+  endif (NOT KDEWIN_DIR)
+
+  # this must be set every time 
+  if (KDEWIN_DIR)
+    # add include path and library to all targets, this is required because 
+    # cmake's 2.4.6 FindZLib.cmake does not use CMAKE_REQUIRED... vars
+    set(CMAKE_INCLUDE_PATH ${CMAKE_INCLUDE_PATH} ${KDEWIN_DIR}/include)
+    set(CMAKE_LIBRARY_PATH ${CMAKE_LIBRARY_PATH} ${KDEWIN_DIR}/lib)
+     
+    set (KDEWIN_FOUND 1)
+    if (NOT KDEWIN_FIND_QUIETLY)
+      message(STATUS "Found kdewin32 library: ${KDEWIN32_LIBRARY}")
+    endif (NOT KDEWIN_FIND_QUIETLY)
+
+  else(KDEWIN_DIR)
+    if( KDEWIN_FIND_REQUIRED )
+      message(FATAL_ERROR "Could not find the location of the windows supplementary packages which is \n"
+                      "\t\tenvironment variable KDEWIN_DIR\n"
+                      "\t\t<ProgramFiles>/${DIR}\n"
+                      "\t\t<ProgramFiles>/kdewin\n" 
+                      "\t\t<ProgramFiles>/kdewin32\n" 
+                      "\t\t<ProgramFiles>/win32libs\n"
+                      "\t\t<ProgramFiles>/gnuwin32\n"
+      )
+    endif( KDEWIN_FIND_REQUIRED )
+   endif(KDEWIN_DIR)
+  
+endif (WIN32)
diff -N -r -u fontconfig-2.4.2-orig/cmake/modules/FindKDEWIN32.cmake fontconfig-2.4.2/cmake/modules/FindKDEWIN32.cmake
--- fontconfig-2.4.2-orig/cmake/modules/FindKDEWIN32.cmake	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/cmake/modules/FindKDEWIN32.cmake	2008-02-10 16:34:13.687500000 +0100
@@ -0,0 +1,74 @@
+# - Try to find the KDEWIN32 library
+# 
+# Once done this will define
+#
+#  KDEWIN32_FOUND - system has KDEWIN32
+#  KDEWIN32_INCLUDES - the KDEWIN32 include directories
+#  KDEWIN32_LIBRARIES - The libraries needed to use KDEWIN32
+#
+# Copyright (c) 2006, Alexander Neundorf, <neundorf@kde.org>
+# Copyright (c) 2007, Ralf Habacker, <ralf.habacker@freenet.de>
+#
+# Redistribution and use is allowed according to the terms of the BSD license.
+# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+
+
+if (WIN32)
+  include(FindLibraryWithDebug)
+
+  if (NOT KDEWIN32_DIR)
+    if(NOT KDEWIN_FOUND)
+      find_package(KDEWIN)
+    endif(NOT KDEWIN_FOUND)
+
+    find_path(KDEWIN32_INCLUDE_DIR winposix_export.h
+      ${CMAKE_INCLUDE_PATH}
+      ${CMAKE_INSTALL_PREFIX}/include
+    )
+ 
+    # search for kdewin32 in the default install directory for applications (default of (n)make install)
+    FILE(TO_CMAKE_PATH "${CMAKE_LIBRARY_PATH}" _cmakeLibraryPathCmakeStyle)
+    find_library_with_debug(KDEWIN32_LIBRARY
+      WIN32_DEBUG_POSTFIX d
+      NAMES kdewin32
+      PATHS 
+        ${_cmakeLibraryPathCmakeStyle}
+        ${CMAKE_INSTALL_PREFIX}/lib
+      NO_SYSTEM_ENVIRONMENT_PATH
+    )
+
+    # kdelibs/win/ has to be built before the rest of kdelibs/
+    # eventually it will be moved out from kdelibs/
+    if (KDEWIN32_LIBRARY AND KDEWIN32_INCLUDE_DIR)
+      set(KDEWIN32_FOUND TRUE)
+      # add needed system libs
+      set(KDEWIN32_LIBRARIES ${KDEWIN32_LIBRARY} user32 shell32 ws2_32 netapi32 userenv)
+  
+      if (MINGW)
+        #mingw compiler
+        set(KDEWIN32_INCLUDES ${KDEWIN32_INCLUDE_DIR} ${KDEWIN32_INCLUDE_DIR}/mingw ${QT_INCLUDES})
+      else (MINGW)
+        # msvc compiler
+        # add the MS SDK include directory if available
+        file(TO_CMAKE_PATH "$ENV{MSSDK}" MSSDK_DIR)
+        set(KDEWIN32_INCLUDES ${KDEWIN32_INCLUDE_DIR} ${KDEWIN32_INCLUDE_DIR}/msvc  ${QT_INCLUDES} ${MSSDK_DIR})
+      endif (MINGW)
+  
+    endif (KDEWIN32_LIBRARY AND KDEWIN32_INCLUDE_DIR)
+    # required for configure
+    set(CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES} ${KDEWIN32_INCLUDES})
+    set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} ${KDEWIN32_LIBRARIES})      
+
+  endif (NOT KDEWIN32_DIR)
+
+  if (KDEWIN32_FOUND)
+    if (NOT KDEWIN32_FIND_QUIETLY)
+      message(STATUS "Found kdewin32 library: ${KDEWIN32_LIBRARY}")
+    endif (NOT KDEWIN32_FIND_QUIETLY)
+
+  else (KDEWIN32_FOUND)
+    if (KDEWIN32_FIND_REQUIRED)
+      message(FATAL_ERROR "Could NOT find KDEWIN32 library\nPlease install it first")
+    endif (KDEWIN32_FIND_REQUIRED)
+  endif (KDEWIN32_FOUND)
+endif (WIN32)
diff -N -r -u fontconfig-2.4.2-orig/cmake/modules/FindLibraryWithDebug.cmake fontconfig-2.4.2/cmake/modules/FindLibraryWithDebug.cmake
--- fontconfig-2.4.2-orig/cmake/modules/FindLibraryWithDebug.cmake	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/cmake/modules/FindLibraryWithDebug.cmake	2008-02-10 16:34:14.000000000 +0100
@@ -0,0 +1,113 @@
+#
+#  FIND_LIBRARY_WITH_DEBUG
+#  -> enhanced FIND_LIBRARY to allow the search for an
+#     optional debug library with a WIN32_DEBUG_POSTFIX similar
+#     to CMAKE_DEBUG_POSTFIX when creating a shared lib
+#     it has to be the second and third argument
+
+# Copyright (c) 2007, Christian Ehrlicher, <ch.ehrlicher@gmx.de>
+# Redistribution and use is allowed according to the terms of the BSD license.
+# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+
+MACRO(FIND_LIBRARY_WITH_DEBUG var_name win32_dbg_postfix_name dgb_postfix libname)
+
+  IF(NOT "${win32_dbg_postfix_name}" STREQUAL "WIN32_DEBUG_POSTFIX")
+
+     # no WIN32_DEBUG_POSTFIX -> simply pass all arguments to FIND_LIBRARY
+     FIND_LIBRARY(${var_name}
+                  ${win32_dbg_postfix_name}
+                  ${dgb_postfix}
+                  ${libname}
+                  ${ARGN}
+     )
+
+  ELSE(NOT "${win32_dbg_postfix_name}" STREQUAL "WIN32_DEBUG_POSTFIX")
+
+    IF(NOT WIN32)
+      # on non-win32 we don't need to take care about WIN32_DEBUG_POSTFIX
+
+      FIND_LIBRARY(${var_name} ${libname} ${ARGN})
+
+    ELSE(NOT WIN32)
+
+      # 1. get all possible libnames
+      SET(args ${ARGN})
+      SET(newargs "")
+      SET(libnames_release "")
+      SET(libnames_debug "")
+
+      LIST(LENGTH args listCount)
+
+      IF("${libname}" STREQUAL "NAMES")
+        SET(append_rest 0)
+        LIST(APPEND args " ")
+
+        FOREACH(i RANGE ${listCount})
+          LIST(GET args ${i} val)
+
+          IF(append_rest)
+            LIST(APPEND newargs ${val})
+          ELSE(append_rest)
+            IF("${val}" STREQUAL "PATHS")
+              LIST(APPEND newargs ${val})
+              SET(append_rest 1)
+            ELSE("${val}" STREQUAL "PATHS")
+              LIST(APPEND libnames_release "${val}")
+              LIST(APPEND libnames_debug   "${val}${dgb_postfix}")
+            ENDIF("${val}" STREQUAL "PATHS")
+          ENDIF(append_rest)
+
+        ENDFOREACH(i)
+
+      ELSE("${libname}" STREQUAL "NAMES")
+
+        # just one name
+        LIST(APPEND libnames_release "${libname}")
+        LIST(APPEND libnames_debug   "${libname}${dgb_postfix}")
+
+        SET(newargs ${args})
+
+      ENDIF("${libname}" STREQUAL "NAMES")
+
+      # search the release lib
+      FIND_LIBRARY(${var_name}_RELEASE
+                   NAMES ${libnames_release}
+                   ${newargs}
+      )
+
+      # search the debug lib
+      FIND_LIBRARY(${var_name}_DEBUG
+                   NAMES ${libnames_debug}
+                   ${newargs}
+      )
+
+      IF(${var_name}_RELEASE AND ${var_name}_DEBUG)
+
+        # both libs found
+        SET(${var_name} optimized ${${var_name}_RELEASE}
+                        debug     ${${var_name}_DEBUG})
+
+      ELSE(${var_name}_RELEASE AND ${var_name}_DEBUG)
+
+        IF(${var_name}_RELEASE)
+
+          # only release found
+          SET(${var_name} ${${var_name}_RELEASE})
+
+        ELSE(${var_name}_RELEASE)
+
+          # only debug (or nothing) found
+          SET(${var_name} ${${var_name}_DEBUG})
+
+        ENDIF(${var_name}_RELEASE)
+       
+      ENDIF(${var_name}_RELEASE AND ${var_name}_DEBUG)
+
+      MARK_AS_ADVANCED(${var_name}_RELEASE)
+      MARK_AS_ADVANCED(${var_name}_DEBUG)
+
+    ENDIF(NOT WIN32)
+
+  ENDIF(NOT "${win32_dbg_postfix_name}" STREQUAL "WIN32_DEBUG_POSTFIX")
+
+ENDMACRO(FIND_LIBRARY_WITH_DEBUG)
diff -N -r -u fontconfig-2.4.2-orig/cmake/modules/FindLibXml2.cmake fontconfig-2.4.2/cmake/modules/FindLibXml2.cmake
--- fontconfig-2.4.2-orig/cmake/modules/FindLibXml2.cmake	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/cmake/modules/FindLibXml2.cmake	2008-02-10 16:34:14.015625000 +0100
@@ -0,0 +1,62 @@
+# - Try to find LibXml2
+# Once done this will define
+#
+#  LIBXML2_FOUND - system has LibXml2
+#  LIBXML2_INCLUDE_DIR - the LibXml2 include directory
+#  LIBXML2_LIBRARIES - the libraries needed to use LibXml2
+#  LIBXML2_DEFINITIONS - Compiler switches required for using LibXml2
+
+# Copyright (c) 2006, Alexander Neundorf, <neundorf@kde.org>
+#
+# Redistribution and use is allowed according to the terms of the BSD license.
+# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+
+
+IF (LIBXML2_INCLUDE_DIR AND LIBXML2_LIBRARIES)
+   # in cache already
+   SET(LibXml2_FIND_QUIETLY TRUE)
+ENDIF (LIBXML2_INCLUDE_DIR AND LIBXML2_LIBRARIES)
+
+IF (NOT WIN32)
+   # use pkg-config to get the directories and then use these values
+   # in the FIND_PATH() and FIND_LIBRARY() calls
+   INCLUDE(UsePkgConfig)
+   PKGCONFIG(libxml-2.0 _LibXml2IncDir _LibXml2LinkDir _LibXml2LinkFlags _LibXml2Cflags)
+   SET(LIBXML2_DEFINITIONS ${_LibXml2Cflags})
+ENDIF (NOT WIN32)
+
+FIND_PATH(LIBXML2_INCLUDE_DIR libxml/xpath.h
+   PATHS
+   ${_LibXml2IncDir}
+   PATH_SUFFIXES libxml2
+   )
+
+FIND_LIBRARY(LIBXML2_LIBRARIES NAMES xml2 libxml2
+   PATHS
+   ${_LibXml2LinkDir}
+   )
+
+IF (LIBXML2_INCLUDE_DIR AND LIBXML2_LIBRARIES)
+   SET(LIBXML2_FOUND TRUE)
+ELSE (LIBXML2_INCLUDE_DIR AND LIBXML2_LIBRARIES)
+   SET(LIBXML2_FOUND FALSE)
+ENDIF (LIBXML2_INCLUDE_DIR AND LIBXML2_LIBRARIES)
+
+FIND_PROGRAM(XMLLINT_EXECUTABLE xmllint)
+
+if( NOT XMLLINT_EXECUTABLE )
+   MESSAGE(STATUS "xmllint program not found. Install it if you want validate generated doc file.")
+endif(NOT XMLLINT_EXECUTABLE )
+
+IF (LIBXML2_FOUND)
+   IF (NOT LibXml2_FIND_QUIETLY)
+      MESSAGE(STATUS "Found LibXml2: ${LIBXML2_LIBRARIES}")
+   ENDIF (NOT LibXml2_FIND_QUIETLY)
+ELSE (LIBXML2_FOUND)
+   IF (LibXml2_FIND_REQUIRED)
+      MESSAGE(SEND_ERROR "Could NOT find LibXml2")
+   ENDIF (LibXml2_FIND_REQUIRED)
+ENDIF (LIBXML2_FOUND)
+
+MARK_AS_ADVANCED(LIBXML2_INCLUDE_DIR LIBXML2_LIBRARIES)
+
diff -N -r -u fontconfig-2.4.2-orig/CMakeLists.txt fontconfig-2.4.2/CMakeLists.txt
--- fontconfig-2.4.2-orig/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/CMakeLists.txt	2008-02-10 16:34:14.015625000 +0100
@@ -0,0 +1,47 @@
+project( fontconfig )
+set( CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules )
+
+find_package( EXPAT )
+find_package( Freetype REQUIRED )
+find_package( LibXML2  REQUIRED )
+
+if( MSVC )
+    find_package( kdewin32 REQUIRED )
+endif( MSVC )
+
+include_directories(
+    ${FREETYPE_INCLUDE_DIR}
+    ${LIBXML2_INCLUDE_DIR}
+    ${CMAKE_INCLUDE_PATH}
+    ${CMAKE_CURRENT_BINARY_DIR}
+    ${CMAKE_CURRENT_SOURCE_DIR}
+    ${CMAKE_CURRENT_SOURCE_DIR}/src
+)
+
+if( EXPAT_FOUND )
+include_directories( ${EXPAT_INCLUDE_DIR} )
+endif( EXPAT_FOUND )
+
+if( MSVC )
+    include_directories( ${CMAKE_INCLUDE_PATH}/msvc )
+    add_definitions( -D_CRT_SECURE_NO_DEPRECATE -D_CRT_NONSTDC_NO_DEPRECATE -wd4018)
+endif(MSVC)
+
+link_directories( ${CMAKE_LIBRARY_PATH} )
+
+include( ConfigureChecks.cmake )
+
+add_subdirectory( fontconfig )
+add_subdirectory( fc-case )
+add_subdirectory( fc-lang )
+add_subdirectory( fc-glyphname )
+add_subdirectory( fc-arch )
+add_subdirectory( src )
+add_subdirectory( fc-cache )
+add_subdirectory( fc-cat )
+add_subdirectory( fc-list )
+add_subdirectory( fc-match )
+add_subdirectory( conf.d )
+add_subdirectory( doc )
+add_subdirectory( test )
+
diff -N -r -u fontconfig-2.4.2-orig/conf.d/CMakeLists.txt fontconfig-2.4.2/conf.d/CMakeLists.txt
--- fontconfig-2.4.2-orig/conf.d/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/conf.d/CMakeLists.txt	2008-02-10 16:34:14.031250000 +0100
@@ -0,0 +1,3 @@
+file( GLOB conf_files RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} "*.conf" )
+install( FILES README ${conf_files} DESTINATION etc/fonts/conf.d )
+install( FILES README ${conf_files} DESTINATION etc/fonts/conf.avail )
\ No newline at end of file
diff -N -r -u fontconfig-2.4.2-orig/fonts.conf.cmake fontconfig-2.4.2/fonts.conf.cmake
--- fontconfig-2.4.2-orig/fonts.conf.cmake	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/fonts.conf.cmake	2008-02-10 16:34:14.078125000 +0100
@@ -0,0 +1,150 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!-- /etc/fonts/fonts.conf file to configure system font access -->
+<fontconfig>
+
+<!--
+	DO NOT EDIT THIS FILE.
+	IT WILL BE REPLACED WHEN FONTCONFIG IS UPDATED.
+	LOCAL CHANGES BELONG IN 'local.conf'.
+
+	The intent of this standard configuration file is to be adequate for
+	most environments.  If you have a reasonably normal environment and
+	have found problems with this configuration, they are probably
+	things that others will also want fixed.  Please submit any
+	problems to the fontconfig bugzilla system located at fontconfig.org
+
+	Note that the normal 'make install' procedure for fontconfig is to
+	replace any existing fonts.conf file with the new version.  Place
+	any local customizations in local.conf which this file references.
+
+	Keith Packard
+-->
+
+<!-- Font directory list -->
+
+	<dir>@FC_DEFAULT_FONTS@</dir>
+	@FC_FONTPATH@
+	<dir>~/.fonts</dir>
+
+<!-- Font cache directory list -->
+
+	<cachedir>@FC_CACHEDIR@</cachedir>
+	<cachedir>~/.fontconfig</cachedir>
+
+<!--
+  Accept deprecated 'mono' alias, replacing it with 'monospace'
+-->
+	<match target="pattern">
+		<test qual="any" name="family">
+			<string>mono</string>
+		</test>
+		<edit name="family" mode="assign">
+			<string>monospace</string>
+		</edit>
+	</match>
+
+<!--
+  Accept alternate 'sans serif' spelling, replacing it with 'sans-serif'
+-->
+	<match target="pattern">
+		<test qual="any" name="family">
+			<string>sans serif</string>
+		</test>
+		<edit name="family" mode="assign">
+			<string>sans-serif</string>
+		</edit>
+	</match>
+
+<!--
+  Accept deprecated 'sans' alias, replacing it with 'sans-serif'
+-->
+	<match target="pattern">
+		<test qual="any" name="family">
+			<string>sans</string>
+		</test>
+		<edit name="family" mode="assign">
+			<string>sans-serif</string>
+		</edit>
+	</match>
+
+<!--
+  Load local system customization file
+-->
+	<include ignore_missing="yes">conf.d</include>
+
+	<config>
+<!--
+  These are the default Unicode chars that are expected to be blank
+  in fonts.  All other blank chars are assumed to be broken and
+  won't appear in the resulting charsets
+ -->
+		<blank>
+			<int>0x0020</int>	<!-- SPACE -->
+			<int>0x00A0</int>	<!-- NO-BREAK SPACE -->
+			<int>0x00AD</int>	<!-- SOFT HYPHEN -->
+			<int>0x034F</int>	<!-- COMBINING GRAPHEME JOINER -->
+			<int>0x0600</int>	<!-- ARABIC NUMBER SIGN -->
+			<int>0x0601</int>	<!-- ARABIC SIGN SANAH -->
+			<int>0x0602</int>	<!-- ARABIC FOOTNOTE MARKER -->
+			<int>0x0603</int>	<!-- ARABIC SIGN SAFHA -->
+			<int>0x06DD</int>	<!-- ARABIC END OF AYAH -->
+			<int>0x070F</int>	<!-- SYRIAC ABBREVIATION MARK -->
+			<int>0x115F</int>	<!-- HANGUL CHOSEONG FILLER -->
+			<int>0x1160</int>	<!-- HANGUL JUNGSEONG FILLER -->
+			<int>0x1680</int>	<!-- OGHAM SPACE MARK -->
+			<int>0x17B4</int>	<!-- KHMER VOWEL INHERENT AQ -->
+			<int>0x17B5</int>	<!-- KHMER VOWEL INHERENT AA -->
+			<int>0x180E</int>	<!-- MONGOLIAN VOWEL SEPARATOR -->
+			<int>0x2000</int>	<!-- EN QUAD -->
+			<int>0x2001</int>	<!-- EM QUAD -->
+			<int>0x2002</int>	<!-- EN SPACE -->
+			<int>0x2003</int>	<!-- EM SPACE -->
+			<int>0x2004</int>	<!-- THREE-PER-EM SPACE -->
+			<int>0x2005</int>	<!-- FOUR-PER-EM SPACE -->
+			<int>0x2006</int>	<!-- SIX-PER-EM SPACE -->
+			<int>0x2007</int>	<!-- FIGURE SPACE -->
+			<int>0x2008</int>	<!-- PUNCTUATION SPACE -->
+			<int>0x2009</int>	<!-- THIN SPACE -->
+			<int>0x200A</int>	<!-- HAIR SPACE -->
+			<int>0x200B</int>	<!-- ZERO WIDTH SPACE -->
+			<int>0x200C</int>	<!-- ZERO WIDTH NON-JOINER -->
+			<int>0x200D</int>	<!-- ZERO WIDTH JOINER -->
+			<int>0x200E</int>	<!-- LEFT-TO-RIGHT MARK -->
+			<int>0x200F</int>	<!-- RIGHT-TO-LEFT MARK -->
+			<int>0x2028</int>	<!-- LINE SEPARATOR -->
+			<int>0x2029</int>	<!-- PARAGRAPH SEPARATOR -->
+			<int>0x202A</int>	<!-- LEFT-TO-RIGHT EMBEDDING -->
+			<int>0x202B</int>	<!-- RIGHT-TO-LEFT EMBEDDING -->
+			<int>0x202C</int>	<!-- POP DIRECTIONAL FORMATTING -->
+			<int>0x202D</int>	<!-- LEFT-TO-RIGHT OVERRIDE -->
+			<int>0x202E</int>	<!-- RIGHT-TO-LEFT OVERRIDE -->
+			<int>0x202F</int>	<!-- NARROW NO-BREAK SPACE -->
+			<int>0x205F</int>	<!-- MEDIUM MATHEMATICAL SPACE -->
+			<int>0x2060</int>	<!-- WORD JOINER -->
+			<int>0x2061</int>	<!-- FUNCTION APPLICATION -->
+			<int>0x2062</int>	<!-- INVISIBLE TIMES -->
+			<int>0x2063</int>	<!-- INVISIBLE SEPARATOR -->
+			<int>0x206A</int>	<!-- INHIBIT SYMMETRIC SWAPPING -->
+			<int>0x206B</int>	<!-- ACTIVATE SYMMETRIC SWAPPING -->
+			<int>0x206C</int>	<!-- INHIBIT ARABIC FORM SHAPING -->
+			<int>0x206D</int>	<!-- ACTIVATE ARABIC FORM SHAPING -->
+			<int>0x206E</int>	<!-- NATIONAL DIGIT SHAPES -->
+			<int>0x206F</int>	<!-- NOMINAL DIGIT SHAPES -->
+			<int>0x3000</int>	<!-- IDEOGRAPHIC SPACE -->
+			<int>0x3164</int>	<!-- HANGUL FILLER -->
+			<int>0xFEFF</int>	<!-- ZERO WIDTH NO-BREAK SPACE -->
+			<int>0xFFA0</int>	<!-- HALFWIDTH HANGUL FILLER -->
+			<int>0xFFF9</int>	<!-- INTERLINEAR ANNOTATION ANCHOR -->
+			<int>0xFFFA</int>	<!-- INTERLINEAR ANNOTATION SEPARATOR -->
+			<int>0xFFFB</int>	<!-- INTERLINEAR ANNOTATION TERMINATOR -->
+		</blank>
+ <!--
+   Rescan configuration every 30 seconds when FcFontSetList is called
+ -->
+		<rescan>
+			<int>30</int>
+		</rescan>
+	</config>
+
+</fontconfig>
diff -N -r -u fontconfig-2.4.2-orig/config.h.cmake fontconfig-2.4.2/config.h.cmake
--- fontconfig-2.4.2-orig/config.h.cmake	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/config.h.cmake	2008-02-10 16:34:14.078125000 +0100
@@ -0,0 +1,179 @@
+/* config.h.in.  Generated from configure.in by autoheader.  */
+
+/* Font configuration directory */
+#cmakedefine CONFDIR
+
+/* Use libxml2 instead of Expat */
+#define ENABLE_LIBXML2 1
+
+/* Additional font directories */
+#define FC_ADD_FONTS
+
+/* System font directory */
+#define FC_DEFAULT_FONTS "WINDOWSFONTDIR"
+
+/* temporary cachedir */
+#define FC_CACHEDIR ${FC_CACHEDIR}
+
+/* Define to 1 if you have the `chsize' function. */
+#cmakedefine HAVE_CHSIZE 1
+
+/* Define to 1 if you have the <dirent.h> header file, and it defines `DIR'.
+   */
+#cmakedefine HAVE_DIRENT_H 1
+
+/* Define to 1 if you have the <dlfcn.h> header file. */
+#cmakedefine HAVE_DLFCN_H 1
+
+/* Define to 1 if you don't have `vprintf' but do have `_doprnt.' */
+#cmakedefine HAVE_DOPRNT 1
+
+/* Found a useable expat library */
+#cmakedefine HAVE_EXPAT 1
+
+/* Define to 1 if you have the <fcntl.h> header file. */
+#cmakedefine HAVE_FCNTL_H 1
+
+/* Define to 1 if you have the `ftruncate' function. */
+#cmakedefine HAVE_FTRUNCATE 1
+
+/* FT_Bitmap_Size structure includes y_ppem field */
+#cmakedefine HAVE_FT_BITMAP_SIZE_Y_PPEM 1
+
+/* Define to 1 if you have the `FT_Get_BDF_Property' function. */
+#cmakedefine HAVE_FT_GET_BDF_PROPERTY 1
+
+/* Define to 1 if you have the `FT_Get_Next_Char' function. */
+#cmakedefine HAVE_FT_GET_NEXT_CHAR 1
+
+/* Define to 1 if you have the `FT_Get_PS_Font_Info' function. */
+#cmakedefine HAVE_FT_GET_PS_FONT_INFO 1
+
+/* Define to 1 if you have the `FT_Get_X11_Font_Format' function. */
+#cmakedefine HAVE_FT_GET_X11_FONT_FORMAT 1
+
+/* Define to 1 if you have the `FT_Has_PS_Glyph_Names' function. */
+#cmakedefine HAVE_FT_HAS_PS_GLYPH_NAMES 1
+
+/* Define to 1 if you have the `geteuid' function. */
+#cmakedefine HAVE_GETEUID 1
+
+/* Define to 1 if you have the `getopt' function. */
+#cmakedefine HAVE_GETOPT 1
+
+/* Define to 1 if you have the `getopt_long' function. */
+#cmakedefine HAVE_GETOPT_LONG 1
+
+/* Define to 1 if you have the `getpagesize' function. */
+#cmakedefine HAVE_GETPAGESIZE 1
+
+/* Define to 1 if you have the `getuid' function. */
+#cmakedefine HAVE_GETUID 1
+
+/* Define to 1 if you have the <inttypes.h> header file. */
+#cmakedefine HAVE_INTTYPES_H 1
+
+/* Define to 1 if you have the `link' function. */
+#cmakedefine HAVE_LINK 1
+
+/* Define to 1 if you have the `lrand48' function. */
+#cmakedefine HAVE_LRAND48 1
+
+/* Define to 1 if you have the `memmove' function. */
+#cmakedefine HAVE_MEMMOVE 1
+
+/* Define to 1 if you have the <memory.h> header file. */
+#cmakedefine HAVE_MEMORY_H 1
+
+/* Define to 1 if you have the `memset' function. */
+#cmakedefine HAVE_MEMSET 1
+
+/* Define to 1 if you have the `mkstemp' function. */
+#cmakedefine HAVE_MKSTEMP 1
+
+/* Define to 1 if you have a working `mmap' system call. */
+#cmakedefine HAVE_MMAP 1
+
+/* Define to 1 if you have the <ndir.h> header file, and it defines `DIR'. */
+#cmakedefine HAVE_NDIR_H 1
+
+/* Define to 1 if you have the `rand' function. */
+#cmakedefine HAVE_RAND 1
+
+/* Define to 1 if you have the `random' function. */
+#cmakedefine HAVE_RANDOM 1
+
+/* Define to 1 if you have the <stdint.h> header file. */
+#cmakedefine HAVE_STDINT_H 1
+
+/* Define to 1 if you have the <stdlib.h> header file. */
+#cmakedefine HAVE_STDLIB_H 1
+
+/* Define to 1 if you have the `strchr' function. */
+#cmakedefine HAVE_STRCHR 1
+
+/* Define to 1 if you have the <strings.h> header file. */
+#cmakedefine HAVE_STRINGS_H 1
+
+/* Define to 1 if you have the <string.h> header file. */
+#cmakedefine HAVE_STRING_H 1
+
+/* Define to 1 if you have the `strrchr' function. */
+#cmakedefine HAVE_STRRCHR 1
+
+/* Define to 1 if you have the `strtol' function. */
+#cmakedefine HAVE_STRTOL 1
+
+/* Define to 1 if you have the `sysconf' function. */
+#cmakedefine HAVE_SYSCONF 1
+
+/* Define to 1 if you have the <sys/dir.h> header file, and it defines `DIR'.    */
+#cmakedefine HAVE_SYS_DIR_H 1
+
+/* Define to 1 if you have the <sys/ndir.h> header file, and it defines `DIR'.    */
+#cmakedefine HAVE_SYS_NDIR_H 1
+
+/* Define to 1 if you have the <sys/stat.h> header file. */
+#cmakedefine HAVE_SYS_STAT_H 1
+
+/* Define to 1 if you have the <sys/types.h> header file. */
+#cmakedefine HAVE_SYS_TYPES_H 1
+
+/* Define to 1 if you have the <unistd.h> header file. */
+#cmakedefine HAVE_UNISTD_H 1
+
+/* Define to 1 if you have the `vprintf' function. */
+#cmakedefine HAVE_VPRINTF 1
+
+/* Can use #warning in C files */
+#cmakedefine HAVE_WARNING_CPP_DIRECTIVE 1
+
+/* Use xmlparse.h instead of expat.h */
+#cmakedefine HAVE_XMLPARSE_H 1
+
+/* Define to 1 if you have the `XML_SetDoctypeDeclHandler' function. */
+#cmakedefine HAVE_XML_SETDOCTYPEDECLHANDLER 1
+
+/* Name of package */
+#define PACKAGE "fontconfig"
+
+/* Define to the address where bug reports for this package should be sent. */
+#define PACKAGE_BUGREPORT 
+
+/* Define to the full name of this package. */
+#define PACKAGE_NAME "fontconfig"
+
+/* Define to the full name and version of this package. */
+#define PACKAGE_STRING "fontconfig"
+
+/* Define to the one symbol short name of this package. */
+#define PACKAGE_TARNAME ""
+
+/* Define to the version of this package. */
+#cmakedefine PACKAGE_VERSION 1
+
+/* Define to 1 if you have the ANSI C header files. */
+#define STDC_HEADERS 1
+
+/* Version number of package */
+#define VERSION "2.4.2"
diff -N -r -u fontconfig-2.4.2-orig/ConfigureChecks.cmake fontconfig-2.4.2/ConfigureChecks.cmake
--- fontconfig-2.4.2-orig/ConfigureChecks.cmake	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/ConfigureChecks.cmake	2008-02-10 16:34:14.203125000 +0100
@@ -0,0 +1,70 @@
+# Checking Headers and Functions for fontconfig
+
+include( CheckIncludeFile )
+include( CheckFunctionExists )
+
+if(WIN32)
+    if(MSVC)
+        set( CMAKE_REQUIRED_INCLUDES ${CMAKE_INCLUDE_PATH} ${CMAKE_INCLUDE_PATH}/msvc )
+    else(MSVC)
+        set( CMAKE_REQUIRED_INCLUDES ${CMAKE_INCLUDE_PATH} ${CMAKE_INCLUDE_PATH}/mingw )
+    endif(MSVC)
+endif(WIN32)
+
+set( CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} ${FREETYPE_LIBRARIES} ${LIBXML2_LIBRARIES} )
+set( CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES} ${FREETYPE_INCLUDE_DIR} ${LIBXML2_INCLUDE_DIR} )
+if( EXPAT_FOUND )
+    set( CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} ${EXPAT_LIBRARY} )
+    set( CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES} ${EXPAT_INCLUDE_DIR} )
+endif( EXPAT_FOUND )
+check_include_file( "dirent.h" HAVE_DIRENT_H )
+check_include_file( "dlfcn.h" HAVE_DLFCN_H )
+check_include_file( "fcntl.h" HAVE_FCNTL_H )
+check_include_file( "inttypes.h" HAVE_INTTYPES_H )
+check_include_file( "memory.h" HAVE_MEMORY_H )
+check_include_file( "ndir.h" HAVE_NDIR_H )
+check_include_file( "stdint.h" HAVE_STDINT_H )
+check_include_file( "stdlib.h" HAVE_STDLIB_H )
+check_include_file( "strings.h" HAVE_STRINGS_H )
+check_include_file( "string.h" HAVE_STRING_H )
+check_include_file( "sys/dir.h" HAVE_SYS_DIR_H )
+check_include_file( "sys/ndir.h" HAVE_SYS_NDIR_H )
+check_include_file( "sys/stat.h" HAVE_SYS_STAT_H )
+check_include_file( "sys/types.h" HAVE_SYS_TYPES_H )
+check_include_file( "unistd.h" HAVE_UNISTD_H )
+check_include_file( "xmlparse.h" HAVE_XMLPARSE_H )
+
+check_function_exists( chsize HAVE_CHSIZE )
+check_function_exists( _doprnt HAVE_DOPRNT )
+check_function_exists( ftruncate HAVE_FTRUNCATE )
+check_function_exists( FT_Get_BDF_Property HAVE_FT_GET_BDF_PROPERTY )
+check_function_exists( FT_Get_Next_Char HAVE_FT_GET_NEXT_CHAR )
+check_function_exists( FT_Get_PS_Font_Info HAVE_FT_GET_PS_FONT_INFO )
+check_function_exists( FT_Get_X11_Font_Format HAVE_FT_GET_X11_FONT_FORMAT )
+check_function_exists( FT_Has_PS_Glyph_Names HAVE_FT_HAS_PS_GLYPH_NAMES )
+check_function_exists( geteuid HAVE_GETEUID )
+check_function_exists( getopt HAVE_GETOPT )
+check_function_exists( getopt_long HAVE_GETOPT_LONG )
+check_function_exists( getpagesize HAVE_GETPAGESIZE )
+check_function_exists( getuid HAVE_GETUID )
+check_function_exists( link HAVE_LINK )
+check_function_exists( lrand48 HAVE_LRAND48 )
+check_function_exists( memmove HAVE_MEMMOVE )
+check_function_exists( memset HAVE_MEMSET )
+check_function_exists( mkstemp HAVE_MKSTEMP )
+check_function_exists( mmap HAVE_MMAP )
+check_function_exists( rand HAVE_RAND )
+check_function_exists( random HAVE_RANDOM )
+check_function_exists( strchr HAVE_STRCHR )
+check_function_exists( strrchr HAVE_STRRCHR )
+check_function_exists( strtol HAVE_STRTOL )
+check_function_exists( sysconf HAVE_SYSCONF )
+check_function_exists( vprintf HAVE_VPRINTF )
+check_function_exists( XML_SetDoctypeDeclHandler HAVE_XML_SETDOCTYPEDECLHANDLER )
+set( FC_DEFAULT_FONTS "%WINDIR%\\fonts" )
+add_definitions( -DHAVE_CONFIG_H )
+add_definitions( -DFONTCONFIG_PATH='"/etc/fonts"' )
+set( FC_CACHEDIR "\"%TEMP%\\fc_cache\"" )
+configure_file( ${CMAKE_CURRENT_SOURCE_DIR}/config.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config.h )
+configure_file( ${CMAKE_CURRENT_SOURCE_DIR}/fonts.conf.cmake ${CMAKE_CURRENT_BINARY_DIR}/fonts.conf )
+install( FILES ${CMAKE_CURRENT_BINARY_DIR}/fonts.conf DESTINATION etc/fonts )
diff -N -r -u fontconfig-2.4.2-orig/doc/CMakeLists.txt fontconfig-2.4.2/doc/CMakeLists.txt
--- fontconfig-2.4.2-orig/doc/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/doc/CMakeLists.txt	2008-02-10 16:34:14.359375000 +0100
@@ -0,0 +1,16 @@
+set( sub_dir_name edit-sgml )
+
+include_directories( ${CMAKE_CURRENT_BINARY_DIR} )
+
+add_executable( ${sub_dir_name} ${sub_dir_name}.c )
+
+target_link_libraries( ${sub_dir_name} fontconfig )
+
+file( GLOB man3_srcs RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} "*.3" )
+file( GLOB man5_srcs RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} "*.5" )
+file( GLOB html_srcs RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} "fontconfig-devel/*.html" )
+
+install( FILES ${man3_srcs} DESTINATION share/man/man3 )
+install( FILES ${man5_srcs} DESTINATION share/man/man5 )
+install( FILES ${html_srcs} DESTINATION share/doc/fontconfig )
+install( FILES fontconfig-devel.pdf fontconfig-user.pdf fontconfig-devel.txt fontconfig-user.txt  fontconfig-user.html DESTINATION share/doc/fontconfig )
diff -N -r -u fontconfig-2.4.2-orig/fc-arch/CMakeLists.txt fontconfig-2.4.2/fc-arch/CMakeLists.txt
--- fontconfig-2.4.2-orig/fc-arch/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/fc-arch/CMakeLists.txt	2008-02-10 16:34:14.390625000 +0100
@@ -0,0 +1,13 @@
+set( sub_dir_name fc-arch )
+
+set( DATA_FILE auto )
+set( TEMPLATE_FILE ${CMAKE_CURRENT_SOURCE_DIR}/fcarch.tmpl.h )
+set( OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/fcarch.h )
+
+include_directories( ${CMAKE_CURRENT_BINARY_DIR} )
+
+add_executable( ${sub_dir_name} ${sub_dir_name}.c )
+
+add_custom_command( TARGET ${sub_dir_name} POST_BUILD
+                    COMMAND ${CMAKE_CURRENT_BINARY_DIR}/${sub_dir_name} ARGS ${DATA_FILE} < ${TEMPLATE_FILE} > ${OUTPUT_FILE} 
+                    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR} VERBATIM )
diff -N -r -u fontconfig-2.4.2-orig/fc-cache/CMakeLists.txt fontconfig-2.4.2/fc-cache/CMakeLists.txt
--- fontconfig-2.4.2-orig/fc-cache/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/fc-cache/CMakeLists.txt	2008-02-10 16:34:14.437500000 +0100
@@ -0,0 +1,15 @@
+set( sub_dir_name fc-cache )
+
+include_directories( ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_BINARY_DIR}/.. ${CMAKE_CURRENT_BINARY_DIR}/../src/ )
+include_directories( ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/../src/ )
+
+if(MSVC)
+add_definitions( -DWINPOSIX_EXPORT= )
+add_executable( ${sub_dir_name} ${sub_dir_name}.c ../src/dirent.c ../src/unistd.c )
+else(MSVC)
+add_executable( ${sub_dir_name} ${sub_dir_name}.c )
+endif(MSVC)
+
+target_link_libraries( ${sub_dir_name} fontconfig )
+
+install( TARGETS ${sub_dir_name} DESTINATION bin )
\ No newline at end of file
diff -N -r -u fontconfig-2.4.2-orig/fc-case/CMakeLists.txt fontconfig-2.4.2/fc-case/CMakeLists.txt
--- fontconfig-2.4.2-orig/fc-case/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/fc-case/CMakeLists.txt	2008-02-10 16:34:14.500000000 +0100
@@ -0,0 +1,13 @@
+set( sub_dir_name fc-case )
+
+set( DATA_FILE ${CMAKE_CURRENT_SOURCE_DIR}/CaseFolding.txt )
+set( TEMPLATE_FILE ${CMAKE_CURRENT_SOURCE_DIR}/fccase.tmpl.h )
+set( OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/fccase.h )
+
+include_directories( ${CMAKE_CURRENT_BINARY_DIR} )
+
+add_executable( ${sub_dir_name} ${sub_dir_name}.c )
+
+add_custom_command( TARGET ${sub_dir_name} POST_BUILD
+                    COMMAND ${CMAKE_CURRENT_BINARY_DIR}/${sub_dir_name} ARGS ${DATA_FILE} < ${TEMPLATE_FILE} > ${OUTPUT_FILE} 
+                    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR} VERBATIM )
diff -N -r -u fontconfig-2.4.2-orig/fc-cat/CMakeLists.txt fontconfig-2.4.2/fc-cat/CMakeLists.txt
--- fontconfig-2.4.2-orig/fc-cat/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/fc-cat/CMakeLists.txt	2008-02-10 16:34:14.500000000 +0100
@@ -0,0 +1,14 @@
+set( sub_dir_name fc-cat )
+
+include_directories( ${CMAKE_CURRENT_BINARY_DIR} )
+
+if(MSVC)
+add_definitions( -DWINPOSIX_EXPORT= )
+add_executable( ${sub_dir_name} ${sub_dir_name}.c ../src/dirent.c ../src/unistd.c )
+else(MSVC)
+add_executable( ${sub_dir_name} ${sub_dir_name}.c )
+endif(MSVC)
+
+target_link_libraries( ${sub_dir_name} fontconfig )
+
+install( TARGETS ${sub_dir_name} DESTINATION bin )
\ No newline at end of file
diff -N -r -u fontconfig-2.4.2-orig/fc-glyphname/CMakeLists.txt fontconfig-2.4.2/fc-glyphname/CMakeLists.txt
--- fontconfig-2.4.2-orig/fc-glyphname/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/fc-glyphname/CMakeLists.txt	2008-02-10 16:34:14.531250000 +0100
@@ -0,0 +1,13 @@
+set( sub_dir_name fc-glyphname )
+
+set( DATA_FILE ${CMAKE_CURRENT_SOURCE_DIR}/zapfdingbats.txt )
+set( TEMPLATE_FILE ${CMAKE_CURRENT_SOURCE_DIR}/fcglyphname.tmpl.h )
+set( OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/fcglyphname.h )
+
+include_directories( ${CMAKE_CURRENT_BINARY_DIR} )
+
+add_executable( ${sub_dir_name} ${sub_dir_name}.c )
+
+add_custom_command( TARGET ${sub_dir_name} POST_BUILD
+                    COMMAND ${CMAKE_CURRENT_BINARY_DIR}/${sub_dir_name} ARGS ${DATA_FILE} < ${TEMPLATE_FILE} > ${OUTPUT_FILE} 
+                    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR} VERBATIM )
diff -N -r -u fontconfig-2.4.2-orig/fc-lang/CMakeLists.txt fontconfig-2.4.2/fc-lang/CMakeLists.txt
--- fontconfig-2.4.2-orig/fc-lang/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/fc-lang/CMakeLists.txt	2008-02-10 16:34:14.546875000 +0100
@@ -0,0 +1,15 @@
+set( sub_dir_name fc-lang )
+
+file( GLOB DATA_FILE RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} "*.orth" )
+set( TEMPLATE_FILE fclang.tmpl.h )
+set( OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/fclang.h )
+
+file( WRITE ${CMAKE_CURRENT_BINARY_DIR}/fcaliastail.h )
+
+include_directories( ${CMAKE_CURRENT_BINARY_DIR} )
+add_definitions(-Dfontconfig_EXPORTS)
+add_executable( ${sub_dir_name} ${sub_dir_name}.c )
+
+add_custom_command( TARGET ${sub_dir_name} POST_BUILD
+                    COMMAND ${CMAKE_CURRENT_BINARY_DIR}/${sub_dir_name} -d . ${DATA_FILE} < ${TEMPLATE_FILE} > ${OUTPUT_FILE}
+                    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} VERBATIM )
diff -N -r -u fontconfig-2.4.2-orig/fc-list/CMakeLists.txt fontconfig-2.4.2/fc-list/CMakeLists.txt
--- fontconfig-2.4.2-orig/fc-list/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/fc-list/CMakeLists.txt	2008-02-10 16:34:14.546875000 +0100
@@ -0,0 +1,14 @@
+set( sub_dir_name fc-list )
+
+include_directories( ${CMAKE_CURRENT_BINARY_DIR} )
+
+if(MSVC)
+add_definitions( -DWINPOSIX_EXPORT= )
+add_executable( ${sub_dir_name} ${sub_dir_name}.c ../src/dirent.c ../src/unistd.c )
+else(MSVC)
+add_executable( ${sub_dir_name} ${sub_dir_name}.c )
+endif(MSVC)
+
+target_link_libraries( ${sub_dir_name} fontconfig )
+
+install( TARGETS ${sub_dir_name} DESTINATION bin )
\ No newline at end of file
diff -N -r -u fontconfig-2.4.2-orig/fc-match/CMakeLists.txt fontconfig-2.4.2/fc-match/CMakeLists.txt
--- fontconfig-2.4.2-orig/fc-match/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/fc-match/CMakeLists.txt	2008-02-10 16:34:14.593750000 +0100
@@ -0,0 +1,14 @@
+set( sub_dir_name fc-match )
+
+include_directories( ${CMAKE_CURRENT_BINARY_DIR} )
+
+if(MSVC)
+add_definitions( -DWINPOSIX_EXPORT= )
+add_executable( ${sub_dir_name} ${sub_dir_name}.c ../src/dirent.c ../src/unistd.c )
+else(MSVC)
+add_executable( ${sub_dir_name} ${sub_dir_name}.c )
+endif(MSVC)
+
+target_link_libraries( ${sub_dir_name} fontconfig )
+
+install( TARGETS ${sub_dir_name} DESTINATION bin )
\ No newline at end of file
diff -N -r -u fontconfig-2.4.2-orig/fontconfig/CMakeLists.txt fontconfig-2.4.2/fontconfig/CMakeLists.txt
--- fontconfig-2.4.2-orig/fontconfig/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/fontconfig/CMakeLists.txt	2008-02-10 16:34:14.625000000 +0100
@@ -0,0 +1,3 @@
+# fontconfig
+
+install( FILES fcfreetype.h fcprivate.h fontconfig.h DESTINATION include/fontconfig )
\ No newline at end of file
diff -N -r -u fontconfig-2.4.2-orig/fontconfig/fcfreetype.h fontconfig-2.4.2/fontconfig/fcfreetype.h
--- fontconfig-2.4.2-orig/fontconfig/fcfreetype.h	2006-12-02 21:31:21.000000000 +0100
+++ fontconfig-2.4.2/fontconfig/fcfreetype.h	2008-02-10 16:45:34.609375000 +0100
@@ -27,8 +27,16 @@
 #include <ft2build.h>
 #include FT_FREETYPE_H
 
-#ifndef FcPublic
-#define FcPublic
+#if _WIN32
+# if defined(fontconfig_EXPORTS)
+#  define FcPublic __declspec(dllexport)
+# else
+#  define FcPublic __declspec(dllimport)
+# endif
+#else
+# ifndef FcPublic
+#  define FcPublic
+# endif
 #endif
 
 _FCFUNCPROTOBEGIN
diff -N -r -u fontconfig-2.4.2-orig/fontconfig/fontconfig.h fontconfig-2.4.2/fontconfig/fontconfig.h
--- fontconfig-2.4.2-orig/fontconfig/fontconfig.h	2006-12-03 00:24:50.000000000 +0100
+++ fontconfig-2.4.2/fontconfig/fontconfig.h	2008-02-10 16:34:15.265625000 +0100
@@ -27,17 +27,25 @@
 
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <unistd.h>
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif
 #include <stdarg.h>
 
 #if defined(__GNUC__) && (__GNUC__ >= 4)
 #define FC_ATTRIBUTE_SENTINEL(x) __attribute__((__sentinel__(0)))
 #else
 #define FC_ATTRIBUTE_SENTINEL(x)
 #endif
 
-#ifndef FcPublic
-#define FcPublic
+#if _WIN32
+# if defined(fontconfig_EXPORTS)
+#  define FcPublic __declspec(dllexport)
+# else
+#  define FcPublic __declspec(dllimport)
+# endif
+#else
+# define FcPublic
 #endif
 
 typedef unsigned char	FcChar8;
@@ -881,25 +890,25 @@
 FcStrSetCreate (void);
 
 FcPublic FcBool
-FcStrSetMember (FcPublic FcStrSet *set, const FcChar8 *s);
+FcStrSetMember (FcStrSet *set, const FcChar8 *s);
 
 FcPublic FcBool
-FcStrSetEqual (FcPublic FcStrSet *sa, FcPublic FcStrSet *sb);
+FcStrSetEqual (FcStrSet *sa, FcStrSet *sb);
 
 FcPublic FcBool
-FcStrSetAdd (FcPublic FcStrSet *set, const FcChar8 *s);
+FcStrSetAdd (FcStrSet *set, const FcChar8 *s);
 
 FcPublic FcBool
-FcStrSetAddFilename (FcPublic FcStrSet *set, const FcChar8 *s);
+FcStrSetAddFilename (FcStrSet *set, const FcChar8 *s);
 
 FcPublic FcBool
-FcStrSetDel (FcPublic FcStrSet *set, const FcChar8 *s);
+FcStrSetDel (FcStrSet *set, const FcChar8 *s);
 
 FcPublic void
-FcStrSetDestroy (FcPublic FcStrSet *set);
+FcStrSetDestroy (FcStrSet *set);
 
 FcPublic FcStrList *
-FcStrListCreate (FcPublic FcStrSet *set);
+FcStrListCreate (FcStrSet *set);
 
 FcPublic FcChar8 *
 FcStrListNext (FcStrList *list);
diff -N -r -u fontconfig-2.4.2-orig/src/CMakeLists.txt fontconfig-2.4.2/src/CMakeLists.txt
--- fontconfig-2.4.2-orig/src/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/src/CMakeLists.txt	2008-02-10 16:34:15.562500000 +0100
@@ -0,0 +1,47 @@
+set(sub_dir_name src)
+
+set( fontconfig_SRCS
+fcatomic.c
+fcblanks.c
+fccache.c
+fccfg.c
+fccharset.c
+fcdbg.c
+fcdefault.c
+fcdir.c
+fcfreetype.c
+fcfs.c
+fcinit.c
+fclang.c
+fclist.c
+fcmatch.c
+fcmatrix.c
+fcname.c
+fcpat.c
+fcserialize.c
+fcstr.c
+fcxml.c
+ftglue.c
+)
+
+include_directories(
+${CMAKE_CURRENT_BINARY_DIR}
+${CMAKE_CURRENT_BINARY_DIR}/../fc-lang
+)
+
+if(MSVC)
+add_definitions( -DWINPOSIX_EXPORT= -DPIC )
+set( fontconfig_SRCS ${fontconfig_SRCS} dirent.c unistd.c mmap.c )
+endif(MSVC)
+
+add_library( fontconfig SHARED ${fontconfig_SRCS} )
+set_target_properties(fontconfig PROPERTIES OUTPUT_NAME fontconfig )
+
+target_link_libraries( fontconfig ${FREETYPE_LIBRARIES} )
+target_link_libraries( fontconfig ${LIBXML2_LIBRARIES} )
+if( EXPAT_FOUND )
+    target_link_libraries( fontconfig ${EXPAT_LIBRARY} )
+endif( EXPAT_FOUND )
+
+install( TARGETS fontconfig RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib )
+
diff -N -r -u fontconfig-2.4.2-orig/src/dirent.c fontconfig-2.4.2/src/dirent.c
--- fontconfig-2.4.2-orig/src/dirent.c	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/src/dirent.c	2008-02-10 16:34:15.828125000 +0100
@@ -0,0 +1,158 @@
+/* ====================================================================
+ * The Apache Software License, Version 1.1
+ *
+ * Copyright (c) 2000-2002 The Apache Software Foundation.  All rights
+ * reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *
+ * 3. The end-user documentation included with the redistribution,
+ *    if any, must include the following acknowledgment:
+ *       "This product includes software developed by the
+ *        Apache Software Foundation (http://www.apache.org/)."
+ *    Alternately, this acknowledgment may appear in the software itself,
+ *    if and wherever such third-party acknowledgments normally appear.
+ *
+ * 4. The names "Apache" and "Apache Software Foundation" must
+ *    not be used to endorse or promote products derived from this
+ *    software without prior written permission. For written
+ *    permission, please contact apache@apache.org.
+ *
+ * 5. Products derived from this software may not be called "Apache",
+ *    nor may "Apache" appear in their name, without prior written
+ *    permission of the Apache Software Foundation.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESSED OR IMPLIED
+ * WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED.  IN NO EVENT SHALL THE APACHE SOFTWARE FOUNDATION OR
+ * ITS CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+ * USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+ * OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ * ====================================================================
+ *
+ * This software consists of voluntary contributions made by many
+ * individuals on behalf of the Apache Software Foundation.  For more
+ * information on the Apache Software Foundation, please see
+ * <http://www.apache.org/>.
+ *
+ * Portions of this software are based upon public domain software
+ * originally written at the National Center for Supercomputing Applications,
+ * University of Illinois, Urbana-Champaign.
+ */
+
+#include <winposix_export.h>
+#include <windows.h>
+
+#include <malloc.h>
+#include <string.h>
+#include <errno.h>
+
+#include "dirent.h"
+
+/**********************************************************************
+ * Implement dirent-style opendir/readdir/closedir on Window 95/NT
+ *
+ * Functions defined are opendir(), readdir() and closedir() with the
+ * same prototypes as the normal dirent.h implementation.
+ *
+ * Does not implement telldir(), seekdir(), rewinddir() or scandir(). 
+ * The dirent struct is compatible with Unix, except that d_ino is 
+ * always 1 and d_off is made up as we go along.
+ *
+ * The DIR typedef is not compatible with Unix.
+ **********************************************************************/
+
+#ifndef __MINGW32__
+
+KDEWIN32_EXPORT DIR * opendir(const char *dir)
+{
+    DIR *dp;
+    char *filespec;
+    long handle;
+    int index;
+
+    filespec = malloc(strlen(dir) + 2 + 1);
+    strcpy(filespec, dir);
+    index = strlen(filespec) - 1;
+    if (index >= 0 && (filespec[index] == '/' || filespec[index] == '\\'))
+        filespec[index] = '\0';
+    strcat(filespec, "\\*");
+
+    dp = (DIR *)malloc(sizeof(DIR));
+    dp->offset = 0;
+    dp->finished = 0;
+    dp->dir = strdup(dir);
+
+    if ((handle = _findfirst(filespec, &(dp->fileinfo))) < 0) {
+        if (errno == ENOENT)
+            dp->finished = 1;
+        else
+        return NULL;
+    }
+
+    dp->handle = handle;
+    free(filespec);
+
+    return dp;
+}
+
+KDEWIN32_EXPORT struct dirent * readdir(DIR *dp)
+{
+    if (!dp || dp->finished) return NULL;
+
+    if (dp->offset != 0) {
+        if (_findnext(dp->handle, &(dp->fileinfo)) < 0) {
+            dp->finished = 1;
+            /* posix does not set errno in this case */
+            errno = 0;
+            return NULL;
+        }
+    }
+    dp->offset++;
+
+    strncpy(dp->dent.d_name, dp->fileinfo.name, _MAX_FNAME);
+#ifdef KDEWIN32_HAVE_DIRENT_D_TYPE
+    dp->dent.d_type = DT_UNKNOWN;
+#endif    
+    dp->dent.d_ino = 1;
+    dp->dent.d_reclen = strlen(dp->dent.d_name);
+    dp->dent.d_off = dp->offset;
+
+    return &(dp->dent);
+}
+
+KDEWIN32_EXPORT int closedir(DIR *dp)
+{
+    if (!dp) return 0;
+    if ((HANDLE)dp->handle != INVALID_HANDLE_VALUE) _findclose(dp->handle);
+    if (dp->dir) free(dp->dir);
+    if (dp) free(dp);
+
+    return 0;
+}
+
+#endif // #ifndef __MINGW32__
+
+
+KDEWIN32_EXPORT struct dirent* readdir_r(DIR *dirp, struct dirent *entry, struct dirent **result)
+{
+	//todo: remove this when readdir_r.c will be ported
+	result = 0;
+	return 0;
+}
diff -N -r -u fontconfig-2.4.2-orig/src/fccache.c fontconfig-2.4.2/src/fccache.c
--- fontconfig-2.4.2-orig/src/fccache.c	2006-12-02 22:33:41.000000000 +0100
+++ fontconfig-2.4.2/src/fccache.c	2008-02-10 16:34:15.875000000 +0100
@@ -224,6 +224,10 @@
 #  endif
 # endif
 #endif
+
+#ifdef _WIN32
+#  define FcRandom() rand()
+#endif
 /*
  * Generate a random level number, distributed
  * so that each level is 1/4 as likely as the one before
@@ -707,7 +711,7 @@
     parent = FcStrDirname (dir);
     if (!parent)
 	return FcFalse;
-    if (access ((char *) parent, W_OK|X_OK) == 0)
+    if (access ((char *) parent, W_OK) == 0)
 	ret = mkdir ((char *) dir, 0777) == 0;
     else if (access ((char *) parent, F_OK) == -1)
 	ret = FcMakeDirectory (parent) && (mkdir ((char *) dir, 0777) == 0);
@@ -740,7 +744,7 @@
     if (!list)
 	return FcFalse;
     while ((test_dir = FcStrListNext (list))) {
-	if (access ((char *) test_dir, W_OK|X_OK) == 0)
+	if (access ((char *) test_dir, W_OK) == 0)
 	{
 	    cache_dir = test_dir;
 	    break;
diff -N -r -u fontconfig-2.4.2-orig/src/fcfreetype.c fontconfig-2.4.2/src/fcfreetype.c
--- fontconfig-2.4.2-orig/src/fcfreetype.c	2006-12-02 21:32:15.000000000 +0100
+++ fontconfig-2.4.2/src/fcfreetype.c	2008-02-10 16:34:15.890625000 +0100
@@ -1818,7 +1818,7 @@
 #define FT_Get_Next_Char(face, ucs4, gi) ((ucs4) >= 0xffffff ? \
 					  (*(gi) = 0), 0 : \
 					  (*(gi) = 1), (ucs4) + 1)
-#warning "No FT_Get_Next_Char: Please install freetype version 2.1.0 or newer"
+//#warning "No FT_Get_Next_Char: Please install freetype version 2.1.0 or newer"
 #endif
 
 typedef struct _FcCharEnt {
diff -N -r -u fontconfig-2.4.2-orig/src/mmap.c fontconfig-2.4.2/src/mmap.c
--- fontconfig-2.4.2-orig/src/mmap.c	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/src/mmap.c	2008-02-10 16:34:15.906250000 +0100
@@ -0,0 +1,209 @@
+/*
+   This file is part of the KDE libraries
+   Copyright (c) 2006 Christian Ehrlicher <ch.ehrlicher@gmx.de>
+
+   These sources are based on ftp://g.oswego.edu/pub/misc/malloc.c
+   file by Doug Lea, released to the public domain.
+
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Library General Public
+   License version 2 as published by the Free Software Foundation.
+
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+
+   You should have received a copy of the GNU Library General Public License
+   along with this library; see the file COPYING.LIB.  If not, write to
+   the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110-1301, USA.
+*/
+
+#include <winsock2.h>
+#include <winposix_export.h>
+#include <windows.h>
+
+#include <assert.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <io.h>
+#include <sys/mman.h>
+#include <sys/socket.h>
+
+#ifndef SECTION_MAP_EXECUTE_EXPLICIT
+//not defined in the February 2003 version of the Platform  SDK 
+#define SECTION_MAP_EXECUTE_EXPLICIT 0x0020
+#endif
+
+#ifndef FILE_MAP_EXECUTE
+//not defined in the February 2003 version of the Platform  SDK 
+#define FILE_MAP_EXECUTE SECTION_MAP_EXECUTE_EXPLICIT
+#endif
+
+#define MUNMAP_FAILURE  (-1)
+
+#define USE_MALLOC_LOCK 1
+
+struct mmapInfos {
+    HANDLE hFile;   // the duplicated fd
+    HANDLE hMap;    // handle returned by CreateFileMapping
+    void* start;    // ptr returned by MapViewOfFile
+};
+
+CRITICAL_SECTION cs;
+
+static g_curMMapInfos = 0;
+static g_maxMMapInfos = -1;
+static struct mmapInfos *g_mmapInfos = NULL;
+#define NEW_MMAP_STRUCT_CNT 10
+
+static int mapProtFlags(int flags, DWORD *dwAccess)
+{
+    if ( ( flags & PROT_READ ) == PROT_READ ) {
+        if ( ( flags & PROT_WRITE ) == PROT_WRITE ) {
+            *dwAccess = FILE_MAP_WRITE;
+            if ( ( flags & PROT_EXEC ) == PROT_EXEC ) {
+                return PAGE_EXECUTE_READWRITE;
+            }
+            return PAGE_READWRITE;
+        }
+        if ( ( flags & PROT_EXEC ) == PROT_EXEC ) {
+            *dwAccess = FILE_MAP_EXECUTE;
+            return PAGE_EXECUTE_READ;
+        }
+        *dwAccess = FILE_MAP_READ;
+        return PAGE_READONLY;
+    }
+    if ( ( flags & PROT_WRITE ) == PROT_WRITE ) {
+        *dwAccess = FILE_MAP_COPY;
+        return PAGE_WRITECOPY;
+    }   
+    if ( ( flags & PROT_EXEC ) == PROT_EXEC ) {
+        *dwAccess = FILE_MAP_EXECUTE;
+        return PAGE_EXECUTE_READ;
+    }
+    *dwAccess = 0;
+    return 0;   
+}
+
+void *mmap(void *start, size_t length, int prot , int flags, int fd, off_t offset)
+{
+    struct mmapInfos mmi;
+    DWORD dwAccess;
+    DWORD flProtect;
+    DWORD dwFlags;
+    HANDLE hfd;
+
+    if ( g_maxMMapInfos == -1 ) {
+        g_maxMMapInfos = 0;
+        InitializeCriticalSection( &cs );
+    }
+
+    flProtect = mapProtFlags( flags, &dwAccess );
+    if ( flProtect == 0 ) {
+        _set_errno( EINVAL );
+        return MAP_FAILED;
+    }
+    // we don't support this atm
+    if ( prot == MAP_FIXED ) {
+        _set_errno( ENOTSUP );
+        return MAP_FAILED;
+    }
+
+    if ( fd == -1 ) {
+        _set_errno( EBADF );
+        return MAP_FAILED;
+    }
+
+    // fd can be a crt or a win32 handle -> convert to win32 handle
+    if(!GetHandleInformation( (HANDLE)fd, &dwFlags )) {
+        if(GetLastError() == ERROR_INVALID_HANDLE) {
+            hfd = (HANDLE)_get_osfhandle( fd );
+            if ( hfd == INVALID_HANDLE_VALUE )
+                return MAP_FAILED;
+        } else {
+            return MAP_FAILED;
+        }
+    } else {
+        hfd = (HANDLE)fd;
+    }
+
+    if ( !DuplicateHandle( GetCurrentProcess(), hfd, GetCurrentProcess(),
+                           &mmi.hFile, 0, FALSE, DUPLICATE_SAME_ACCESS ) ) {
+#ifdef _DEBUG
+        DWORD dwLastErr = GetLastError();
+#endif
+        return MAP_FAILED;
+    }
+    mmi.hMap = CreateFileMapping( mmi.hFile, NULL, flProtect,
+                                  0, length, NULL );
+    if ( mmi.hMap == 0 ) {
+        CloseHandle( mmi.hFile );
+        _set_errno( EACCES );
+        return MAP_FAILED;
+    }
+
+    mmi.start = MapViewOfFile( mmi.hMap, dwAccess, 0, offset, 0 );
+    if ( mmi.start == 0 ) {
+        DWORD dwLastErr = GetLastError();
+        CloseHandle( mmi.hMap );
+        CloseHandle( mmi.hFile );
+        if ( dwLastErr == ERROR_MAPPED_ALIGNMENT )
+            _set_errno( EINVAL );
+        else
+            _set_errno( EACCES );
+        return MAP_FAILED;
+    }
+    EnterCriticalSection( &cs );
+    if ( g_mmapInfos == NULL ) {
+        g_maxMMapInfos = NEW_MMAP_STRUCT_CNT;
+        g_mmapInfos = ( struct mmapInfos* )calloc( g_maxMMapInfos,
+                      sizeof( struct mmapInfos ) );
+    }
+    if( g_curMMapInfos == g_maxMMapInfos) {
+        g_maxMMapInfos += NEW_MMAP_STRUCT_CNT;
+        g_mmapInfos = ( struct mmapInfos* )realloc( g_mmapInfos,
+                      g_maxMMapInfos * sizeof( struct mmapInfos ) );
+    }
+    memcpy( &g_mmapInfos[g_curMMapInfos], &mmi, sizeof( struct mmapInfos) );
+    g_curMMapInfos++;
+    
+    LeaveCriticalSection( &cs );
+    
+    return mmi.start;
+}
+
+int munmap(void *start, size_t length)
+{
+    int i, j;
+
+    for( i = 0; i < g_curMMapInfos; i++ ) {
+        if( g_mmapInfos[i].start == start )
+            break;
+    }
+    if( i == g_curMMapInfos ) {
+        _set_errno( EINVAL );
+        return -1;
+    }
+
+    UnmapViewOfFile( g_mmapInfos[ i ].start );
+    CloseHandle( g_mmapInfos[ i ].hMap );
+    CloseHandle( g_mmapInfos[ i ].hFile );
+
+    EnterCriticalSection( &cs );
+    for( j = i + 1; j < g_curMMapInfos; j++ ) {
+        memcpy( &g_mmapInfos[ j - 1 ], &g_mmapInfos[ j ],
+                sizeof( struct mmapInfos ) );
+    }
+    g_curMMapInfos--;
+    
+    if( g_curMMapInfos == 0 ) {
+        free( g_mmapInfos );
+        g_mmapInfos = NULL;
+        g_maxMMapInfos = 0;
+    }
+    LeaveCriticalSection( &cs );
+    
+    return 0;
+}
diff -N -r -u fontconfig-2.4.2-orig/src/unistd.c fontconfig-2.4.2/src/unistd.c
--- fontconfig-2.4.2-orig/src/unistd.c	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.4.2/src/unistd.c	2008-02-10 16:34:15.953125000 +0100
@@ -0,0 +1,507 @@
+/* This file is part of the KDE project
+   Copyright (C) 2003-2004 Jaroslaw Staniek <js@iidea.pl>
+
+   This program is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Library General Public
+   License as published by the Free Software Foundation; either
+   version 2 of the License, or (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+
+   You should have received a copy of the GNU Library General Public License
+   along with this program; see the file COPYING.  If not, write to
+   the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110-1301, USA.
+*/
+
+#define _WIN32_WINNT 0x0500
+#include <winposix_export.h>
+#include <winsock2.h>
+#include <windows.h>
+
+#include <direct.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+// BEGIN stat.c
+KDEWIN32_EXPORT int lstat(const char *path, struct stat *sb)
+{
+  return _stat(path,(struct _stat*)sb);
+}
+
+KDEWIN32_EXPORT int fchmod(int __fd, mode_t __mode)
+{
+  return 0;
+}
+// END stat.c
+
+KDEWIN32_EXPORT int getgroups(int size, gid_t list[])
+{
+	/* TODO */
+	return 0;
+}
+
+KDEWIN32_EXPORT int readlink(const char *__path, char *__buf, int __buflen)
+{
+    if (!__path) {
+      errno = EINVAL;
+      return -1;
+    }
+    if ( (__buflen < 0) || ((int)strlen(__path)>(__buflen-1)) )
+    {
+      errno = ENAMETOOLONG;
+      return -1;
+    }
+    if (access(__path, R_OK) == 0) {
+      /* ok, copy to buf */
+      strncpy(__buf,__path,__buflen);
+      errno = 0;
+      return 0;
+    }
+    errno = ENOENT;
+    return -1;
+}
+
+KDEWIN32_EXPORT int symlink(const char *__name1, const char *__name2)
+{
+	if(!CopyFileA(__name1, __name2, FALSE)) {
+	    switch(GetLastError()) {
+	        case ERROR_FILE_EXISTS:
+    	        errno = EEXIST;
+    	        break;
+    	    case ERROR_CANNOT_MAKE:
+    	        errno = EACCES;
+    	        break;
+    	    default:
+    	        errno = EFAULT;
+    	        break;
+        }
+	    return -1; 
+	}
+	return 0;
+}
+
+KDEWIN32_EXPORT int link(const char *__name1, const char *__name2)
+{
+	return symlink(__name1, __name2);
+}
+
+KDEWIN32_EXPORT int chown(const char *__path, uid_t __owner, gid_t __group)
+{ 
+  return 0; 
+}
+
+KDEWIN32_EXPORT int fchown(int __fd, uid_t __owner, gid_t __group )
+{
+  return 0; 
+}
+
+/* Get the real user ID of the calling process.  */
+KDEWIN32_EXPORT uid_t getuid()
+{
+  return -2;    // keep in sync with Qt!
+}
+
+/* Get the effective user ID of the calling process.  */
+KDEWIN32_EXPORT uid_t geteuid (void)
+{
+  return -2;    // keep in sync with Qt!
+}
+
+/* Get the real group ID of the calling process.  */
+KDEWIN32_EXPORT gid_t getgid (void)
+{
+  return -2;    // keep in sync with Qt!
+}
+
+/* Get the effective group ID of the calling process.  */
+KDEWIN32_EXPORT gid_t getegid (void)
+{
+  return -2;    // keep in sync with Qt!
+}
+
+KDEWIN32_EXPORT int pipe(int *fd)
+{
+  /** @todo */
+  return _pipe( fd, 256, O_BINARY ); /* OK? */
+}
+
+KDEWIN32_EXPORT pid_t fork(void)
+{
+  /** @todo */
+  return -1;
+}
+
+KDEWIN32_EXPORT pid_t setsid(void)
+{
+  /** @todo */
+  return -1;
+}
+
+KDEWIN32_EXPORT unsigned alarm(unsigned __secs )
+{
+  /** @todo */
+  return 1;
+}
+
+typedef unsigned int size_t;
+
+/*#define INCL_WINSOCK_API_PROTOTYPES 0
+#include <winsock2.h>*/
+
+// using winsock gethostname(), which would be taken normally requires WSAStartup called before 
+// which will not be done in every case
+//http://msdn.microsoft.com/library/default.asp?url=/library/en-us/winsock/winsock/gethostname_2.asp
+
+KDEWIN32_EXPORT int kde_gethostname(char *__name, size_t __len)
+{
+  DWORD len = __len;
+  if (0==GetComputerNameA(__name, &len))
+    return -1;
+  return 0;
+}
+
+#define getlogin_buf_size 255
+char getlogin_buf[getlogin_buf_size+1];
+
+KDEWIN32_EXPORT char* getlogin()
+{
+/*! @todo make this reentrant!*/
+	size_t size = sizeof(getlogin_buf);
+	*getlogin_buf = 0;
+	if (!GetUserNameA(getlogin_buf, (LPDWORD)&size))
+		return 0;
+	return getlogin_buf;
+}
+
+KDEWIN32_EXPORT void usleep(unsigned int usec)
+{
+	Sleep(usec/1000);
+}
+
+KDEWIN32_EXPORT void sleep(unsigned int sec)
+{
+	Sleep(sec*1000);
+}
+
+
+KDEWIN32_EXPORT int setreuid(uid_t ruid, uid_t euid)
+{
+	/*! @todo */
+	return 0;
+}
+
+
+/* this is based on code taken from the GNU libc, distributed under the LGPL license */
+
+/* Generate a unique temporary file name from TEMPLATE.
+
+   TEMPLATE has the form:
+
+   <path>/ccXXXXXX<suffix>
+
+   SUFFIX_LEN tells us how long <suffix> is (it can be zero length).
+
+   The last six characters of TEMPLATE before <suffix> must be "XXXXXX";
+   they are replaced with a string that makes the filename unique.
+
+   Returns a file descriptor open on the file for reading and writing.  */
+
+KDEWIN32_EXPORT int mkstemps (char* _template, int suffix_len)
+{
+  static const char letters[] = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+  char *XXXXXX;
+  int len;
+  int count;
+  int value;
+
+  len = strlen (_template);
+
+  if ((int) len < 6 + suffix_len || strncmp (&_template[len - 6 - suffix_len], "XXXXXX", 6))
+      return -1;
+
+  XXXXXX = &_template[len - 6 - suffix_len];
+
+  value = rand();
+  for (count = 0; count < 256; ++count)
+  {
+      int v = value;
+      int fd;
+
+      /* Fill in the random bits.  */
+      XXXXXX[0] = letters[v % 62];
+      v /= 62;
+      XXXXXX[1] = letters[v % 62];
+      v /= 62;
+      XXXXXX[2] = letters[v % 62];
+      v /= 62;
+      XXXXXX[3] = letters[v % 62];
+      v /= 62;
+      XXXXXX[4] = letters[v % 62];
+      v /= 62;
+      XXXXXX[5] = letters[v % 62];
+
+      fd = open (_template, O_RDWR|O_CREAT|O_EXCL, 0600);
+      if (fd >= 0)
+		/* The file does not exist.  */
+		return fd;
+
+      /* This is a random value.  It is only necessary that the next
+	 TMP_MAX values generated by adding 7777 to VALUE are different
+	 with (module 2^32).  */
+      value += 7777;
+    }
+  /* We return the null string if we can't find a unique file name.  */
+  _template[0] = '\0';
+  return -1;
+}
+
+// from kdecore/fakes.c
+KDEWIN32_EXPORT int mkstemp (char* _template)
+{
+  return mkstemps( _template, 0 );
+}
+
+// from kdecore/fakes.c
+int seteuid(uid_t euid)
+{
+    return setreuid(-1, euid); /* Well, if you have neither you are in trouble :) */
+}
+
+// from kdecore/fakes.c
+/* Generate a unique temporary directory name from TEMPLATE.
+
+   TEMPLATE has the form:
+
+   <path>/ccXXXXXX
+
+
+   The last six characters of TEMPLATE must be "XXXXXX";
+   they are replaced with a string that makes the filename unique.
+
+   Returns a file descriptor open on the file for reading and writing.  */
+
+KDEWIN32_EXPORT char* mkdtemp (char* _template)
+{
+  static const char letters[] = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+  char *XXXXXX;
+  int len;
+  int count;
+  int value;
+
+  len = strlen (_template);
+
+  if ((int) len < 6 || strncmp (&_template[len - 6], "XXXXXX", 6))
+      return 0;
+
+  XXXXXX = &_template[len - 6];
+
+  value = rand();
+  for (count = 0; count < 256; ++count)
+  {
+      int v = value;
+
+      /* Fill in the random bits.  */
+      XXXXXX[0] = letters[v % 62];
+      v /= 62;
+      XXXXXX[1] = letters[v % 62];
+      v /= 62;
+      XXXXXX[2] = letters[v % 62];
+      v /= 62;
+      XXXXXX[3] = letters[v % 62];
+      v /= 62;
+      XXXXXX[4] = letters[v % 62];
+      v /= 62;
+      XXXXXX[5] = letters[v % 62];
+
+      /* This is a random value.  It is only necessary that the next
+	 TMP_MAX values generated by adding 7777 to VALUE are different
+	 with (module 2^32).  */
+      value += 7777;
+
+      if (!mkdir(_template))
+		return _template;	
+    }
+    return 0;
+}
+
+KDEWIN32_EXPORT int revoke(const char *tty)
+{
+        errno = ENOTSUP;
+        return -1;
+}
+
+/* getpagesize for windows */
+//static 
+KDEWIN32_EXPORT long getpagesize (void)
+{
+    static long g_pagesize = 0;
+    if (! g_pagesize) {
+        SYSTEM_INFO system_info;
+        GetSystemInfo (&system_info);
+        g_pagesize = system_info.dwPageSize;
+    }
+    return g_pagesize;
+}
+
+#ifndef __MINGW32__
+// XGetopt.cpp  Version 1.2
+//
+// Author:  Hans Dietrich
+//          hdietrich2@hotmail.com
+//
+// Description:
+//     XGetopt.cpp implements getopt(), a function to parse command lines.
+//
+// History
+//     Version 1.2 - 2003 May 17
+//     - Added Unicode support
+//
+//     Version 1.1 - 2002 March 10
+//     - Added example to XGetopt.cpp module header 
+//
+// This software is released into the public domain.
+// You are free to use it in any way you like.
+//
+// This software is provided "as is" with no expressed
+// or implied warranty.  I accept no liability for any
+// damage or loss of business that this software may cause.
+//
+// CE: from http://www.codeproject.com/cpp/xgetopt.asp
+//     removed unicode support to compile with mingw
+char *optarg = NULL;	// global argument pointer
+int   optind = 0; 	// global argv index
+int   opterr = 0; 	
+
+KDEWIN32_EXPORT int getopt(int argc, char **argv, const char *optstring)
+{
+	static char *next = NULL;
+	char c = '?';
+	char *cp = NULL;
+	if (optind == 0)
+		next = NULL;
+
+	optarg = NULL;
+
+	if (next == NULL || *next == '\0')
+	{
+		if (optind == 0)
+			optind++;
+
+		if (optind >= argc || argv[optind][0] != '-' || argv[optind][1] == '\0')
+		{
+			optarg = NULL;
+			if (optind < argc)
+				optarg = argv[optind];
+			return -1;
+		}
+
+		if (strcmp(argv[optind], "--") == 0)
+		{
+			optind++;
+			optarg = NULL;
+			if (optind < argc)
+				optarg = argv[optind];
+			return -1;
+		}
+
+		next = argv[optind];
+		next++;		// skip past -
+		optind++;
+	}
+
+	c = *next++;
+	cp = strchr(optstring, c);
+
+	if (cp == NULL || c == ':')
+		return '?';
+
+	cp++;
+	if (*cp == ':')
+	{
+		if (*next != '\0')
+		{
+			optarg = next;
+			next = NULL;
+		}
+		else if (optind < argc)
+		{
+			optarg = argv[optind];
+			optind++;
+		}
+		else
+		{
+			return '?';
+		}
+	}
+
+	return c;
+}
+#endif  // __MINGW32__
+
+
+int truncate(const char *path, off_t length)
+{
+    HANDLE hFile;
+    LARGE_INTEGER fileSize;
+
+    if( length < 0 )
+    {
+        errno = EINVAL;
+        return -1;
+    }
+
+    hFile = CreateFileA( path, GENERIC_WRITE, 0, 0, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, 0 );
+
+    if( hFile == INVALID_HANDLE_VALUE )
+    {
+        DWORD dwLastErr = GetLastError();
+        errno = EACCES;
+        return -1;
+    }
+
+    if( GetFileSizeEx( hFile, &fileSize ) == 0 )
+    {
+        DWORD dwLastErr = GetLastError();
+        CloseHandle( hFile );
+        errno = EACCES;
+        return -1;
+    }
+
+    if( fileSize.QuadPart < length)
+    {
+        CloseHandle( hFile );
+        errno = EINVAL;
+        return -1;
+    }
+
+    fileSize.QuadPart = length;
+
+    if( SetFilePointerEx( hFile, fileSize, 0, FILE_BEGIN ) == 0 )
+    {
+        DWORD dwLastErr = GetLastError();
+        CloseHandle( hFile );
+        errno = EACCES;
+        return -1;
+    }
+
+    CloseHandle( hFile );
+    return 0;
+}
+
+#ifndef __MINGW32__
+int ftruncate(int fd, off_t length)
+{
+  return _chsize (fd, length);
+}
+#endif  // __MINGW32__
+
