--- include\libxslt\xsltexports.h.orig	2008-05-16 16:39:25.453125000 +0200
+++ include\libxslt\xsltexports.h	2008-05-17 13:55:26.546875000 +0200
@@ -93,10 +93,7 @@
   #undef XSLTPUBFUN
   #undef XSLTPUBVAR
   #undef XSLTCALL
-/*
   #if defined(IN_LIBXSLT) && !defined(LIBXSLT_STATIC)
-*/
-  #if !defined(LIBXSLT_STATIC)
     #define XSLTPUBFUN __declspec(dllexport)
     #define XSLTPUBVAR __declspec(dllexport) extern
   #else
--- include\libexslt\exsltexports.h.orig	2008-07-20 14:54:57.937500000 +0200
+++ include\libexslt\exsltexports.h	2008-07-20 14:57:09.171875000 +0200
@@ -91,10 +91,7 @@
   #undef EXSLTPUBFUN
   #undef EXSLTPUBVAR
   #undef EXSLTCALL
-/*
   #if defined(IN_LIBEXSLT) && !defined(LIBEXSLT_STATIC)
-*/
-  #if !defined(LIBEXSLT_STATIC)
     #define EXSLTPUBFUN __declspec(dllexport)
     #define EXSLTPUBVAR __declspec(dllexport) extern
   #else
