diff -Nru poppler-0.12.0.orig/poppler/Gfx.cc poppler-0.12.0/poppler/Gfx.cc
--- poppler-0.12.0.orig/poppler/Gfx.cc	2009-09-02 23:10:28.000000000 +0200
+++ poppler-0.12.0/poppler/Gfx.cc	2009-09-17 01:15:43.262000000 +0200
@@ -293,8 +293,8 @@
 
 #define numOps (sizeof(opTab) / sizeof(Operator))
 
-static inline GBool isSameGfxColor(const GfxColor &colorA, const GfxColor &colorB, uint nComps, double delta) {
-  for (uint k = 0; k < nComps; ++k) {
+static inline GBool isSameGfxColor(const GfxColor &colorA, const GfxColor &colorB, Guint nComps, double delta) {
+  for (Guint k = 0; k < nComps; ++k) {
     if (abs(colorA.c[k] - colorB.c[k]) > delta) {
       return false;
     }
