From 726abf9a6b177f80b627f46ff453c272a499d783 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michael=20Dr=C3=BCing?= <michael@drueing.de>
Date: Tue, 7 Feb 2017 18:31:26 +0100
Subject: [PATCH] Fix double export in already-exported class

---
 src/widgets/krun.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/widgets/krun.h b/src/widgets/krun.h
index 9de9e69..9885a83 100644
--- a/src/widgets/krun.h
+++ b/src/widgets/krun.h
@@ -297,9 +297,9 @@ public:
      * @deprecated since 5.31, use runUrl() with RunFlags instead.
      */
 #ifndef KIOWIDGETS_NO_DEPRECATED
-    static bool KIOWIDGETS_DEPRECATED_EXPORT runUrl(const QUrl &url, const QString &mimetype, QWidget *window,
-                                                    bool tempFile = false, bool runExecutables = true,
-                                                    const QString &suggestedFileName = QString(), const QByteArray &asn = QByteArray());
+    static bool KIOWIDGETS_DEPRECATED runUrl(const QUrl &url, const QString &mimetype, QWidget *window,
+                                             bool tempFile = false, bool runExecutables = true,
+                                             const QString &suggestedFileName = QString(), const QByteArray &asn = QByteArray());
 #endif
 
     /**
-- 
2.9.3

