diff -Nru -x '*~' kapptemplate-4.10.2.orig/cmake/modules/KAppTemplateMacro.cmake kapptemplate-4.10.2/cmake/modules/KAppTemplateMacro.cmake
--- kapptemplate-4.10.2.orig/cmake/modules/KAppTemplateMacro.cmake	2013-03-15 15:05:00.000000000 +0100
+++ kapptemplate-4.10.2/cmake/modules/KAppTemplateMacro.cmake	2013-05-02 00:01:03.318678500 +0200
@@ -30,9 +30,7 @@
 
         if(WIN32)
             add_custom_command(OUTPUT ${_template}
-                COMMAND zip ARGS -r
-                    ${_template} ${CMAKE_CURRENT_SOURCE_DIR}/${_templateName}
-                    -x .svn _svn .kdev_ignore
+                COMMAND 7za ARGS a -r -tzip ${_template} ${CMAKE_CURRENT_SOURCE_DIR}/${_templateName}/*
                 DEPENDS ${_deps}
             )
         else(WIN32)
Files kapptemplate-4.10.2.orig/templates/C++/akonadiresource.7z and kapptemplate-4.10.2/templates/C++/akonadiresource.7z differ
Files kapptemplate-4.10.2.orig/templates/C++/akonadiresource.zip and kapptemplate-4.10.2/templates/C++/akonadiresource.zip differ
