Index: kdenetwork/krdc/CMakeLists.txt
===================================================================
--- kdenetwork/krdc/CMakeLists.txt	(Revision 1356485)
+++ kdenetwork/krdc/CMakeLists.txt	(Arbeitskopie)
@@ -87,6 +87,8 @@
     config/general.ui
 )
 
+kde4_add_app_icon(krdc_SRCS "${KDE4_INSTALL_DIR}/share/icons/oxygen/*/apps/krdc.png")
+
 kde4_add_executable(krdc ${krdc_SRCS})
 
 target_link_libraries(krdc
