diff -Nru -x '*~' kdenetwork-4.10.2.orig/kopete/protocols/jabber/libiris/CMakeLists.txt kdenetwork-4.10.2/kopete/protocols/jabber/libiris/CMakeLists.txt
--- kdenetwork-4.10.2.orig/kopete/protocols/jabber/libiris/CMakeLists.txt	2013-03-01 06:22:13.593039000 +0100
+++ kdenetwork-4.10.2/kopete/protocols/jabber/libiris/CMakeLists.txt	2013-04-26 13:39:41.794889400 +0200
@@ -5,9 +5,12 @@
 FIND_PACKAGE(ZLIB REQUIRED)
 
 remove_definitions(-DQT_NO_HTTP)
-if(NOT MSVC)
+if(NOT WIN32)
   add_definitions(-fPIC)
-endif(NOT MSVC)
+endif(NOT WIN32)
+if(MINGW)
+  add_definitions(-DWIN32)
+endif(MINGW)
 
 include_directories(
 ${CMAKE_CURRENT_SOURCE_DIR}/iris
