diff -Nrub -x '*~' -x '*\.orig' -x'*\.o' mysql-5.6.26.orig/cmake/install_macros.cmake mysql-5.6.26/cmake/install_macros.cmake
--- mysql-5.6.26.orig/cmake/install_macros.cmake	2015-07-14 23:34:24.000000000 +0200
+++ mysql-5.6.26/cmake/install_macros.cmake	2015-09-23 02:24:14.070955900 +0200
@@ -16,42 +16,42 @@
 GET_FILENAME_COMPONENT(MYSQL_CMAKE_SCRIPT_DIR ${CMAKE_CURRENT_LIST_FILE} PATH)
 INCLUDE(${MYSQL_CMAKE_SCRIPT_DIR}/cmake_parse_arguments.cmake)
 MACRO (INSTALL_DEBUG_SYMBOLS targets)
-  IF(MSVC)
-  FOREACH(target ${targets})
-    GET_TARGET_PROPERTY(location ${target} LOCATION)
-    GET_TARGET_PROPERTY(type ${target} TYPE)
-    IF(NOT INSTALL_LOCATION)
-      IF(type MATCHES "STATIC_LIBRARY"
-          OR type MATCHES "MODULE_LIBRARY"
-          OR type MATCHES "SHARED_LIBRARY")
-        SET(INSTALL_LOCATION "lib")
-      ELSEIF(type MATCHES "EXECUTABLE")
-        SET(INSTALL_LOCATION "bin")
-      ELSE()
-        MESSAGE(FATAL_ERROR
-          "cannot determine type of ${target}. Don't now where to install")
-     ENDIF()
-    ENDIF()
-    STRING(REPLACE ".exe" ".pdb" pdb_location ${location})
-    STRING(REPLACE ".dll" ".pdb" pdb_location ${pdb_location})
-    STRING(REPLACE ".lib" ".pdb" pdb_location ${pdb_location})
-    IF(CMAKE_GENERATOR MATCHES "Visual Studio")
-      STRING(REPLACE
-        "${CMAKE_CFG_INTDIR}" "\${CMAKE_INSTALL_CONFIG_NAME}"
-        pdb_location ${pdb_location})
-    ENDIF()
-    IF(target STREQUAL "mysqld")
-	  SET(comp Server)
-    ELSE()
-      SET(comp Debuginfo)
-    ENDIF()	  
-    # No .pdb file for static libraries.
-    IF(NOT type MATCHES "STATIC_LIBRARY")
-      INSTALL(FILES ${pdb_location}
-        DESTINATION ${INSTALL_LOCATION} COMPONENT ${comp})
-    ENDIF()
-  ENDFOREACH()
-  ENDIF()
+#   IF(MSVC)
+#   FOREACH(target ${targets})
+#     GET_TARGET_PROPERTY(location ${target} LOCATION)
+#     GET_TARGET_PROPERTY(type ${target} TYPE)
+#     IF(NOT INSTALL_LOCATION)
+#       IF(type MATCHES "STATIC_LIBRARY"
+#           OR type MATCHES "MODULE_LIBRARY"
+#           OR type MATCHES "SHARED_LIBRARY")
+#         SET(INSTALL_LOCATION "lib")
+#       ELSEIF(type MATCHES "EXECUTABLE")
+#         SET(INSTALL_LOCATION "bin")
+#       ELSE()
+#         MESSAGE(FATAL_ERROR
+#           "cannot determine type of ${target}. Don't now where to install")
+#      ENDIF()
+#     ENDIF()
+#     STRING(REPLACE ".exe" ".pdb" pdb_location ${location})
+#     STRING(REPLACE ".dll" ".pdb" pdb_location ${pdb_location})
+#     STRING(REPLACE ".lib" ".pdb" pdb_location ${pdb_location})
+#     IF(CMAKE_GENERATOR MATCHES "Visual Studio")
+#       STRING(REPLACE
+#         "${CMAKE_CFG_INTDIR}" "\${CMAKE_INSTALL_CONFIG_NAME}"
+#         pdb_location ${pdb_location})
+#     ENDIF()
+#     IF(target STREQUAL "mysqld")
+# 	  SET(comp Server)
+#     ELSE()
+#       SET(comp Debuginfo)
+#     ENDIF()	  
+#     # No .pdb file for static libraries.
+#     IF(NOT type MATCHES "STATIC_LIBRARY")
+#       INSTALL(FILES ${pdb_location}
+#         DESTINATION ${INSTALL_LOCATION} COMPONENT ${comp})
+#     ENDIF()
+#   ENDFOREACH()
+#   ENDIF()
 ENDMACRO()
 
 # Installs manpage for given file (either script or executable)
diff -Nrub -x '*~' -x '*\.orig' -x'*\.o' mysql-5.6.26.orig/CMakeLists.txt mysql-5.6.26/CMakeLists.txt
--- mysql-5.6.26.orig/CMakeLists.txt	2015-07-14 23:34:24.000000000 +0200
+++ mysql-5.6.26/CMakeLists.txt	2015-09-23 10:11:38.778982600 +0200
@@ -204,7 +204,7 @@
   OPTION(WITH_VALGRIND "Valgrind instrumentation" OFF)
 ENDIF()
 IF(NOT WITHOUT_SERVER)
-  OPTION (WITH_UNIT_TESTS "Compile MySQL with unit tests" ON)
+  OPTION (WITH_UNIT_TESTS "Compile MySQL with unit tests" OFF)
 ENDIF()
 MARK_AS_ADVANCED(CYBOZU BACKUP_TEST WITHOUT_SERVER DISABLE_SHARED)
 
@@ -485,10 +485,10 @@
 ADD_SUBDIRECTORY(scripts)
 
 IF(NOT WITHOUT_SERVER)
-  ADD_SUBDIRECTORY(mysql-test)
-  ADD_SUBDIRECTORY(mysql-test/lib/My/SafeProcess)
+#   ADD_SUBDIRECTORY(mysql-test)
+#   ADD_SUBDIRECTORY(mysql-test/lib/My/SafeProcess)
   ADD_SUBDIRECTORY(support-files)
-  ADD_SUBDIRECTORY(sql-bench)
+#   ADD_SUBDIRECTORY(sql-bench)
   IF(EXISTS ${CMAKE_SOURCE_DIR}/internal/CMakeLists.txt)
     ADD_SUBDIRECTORY(internal)
   ENDIF()
