diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user libxml2-2.10.3.orig/CMakeLists.txt libxml2-2.10.3/CMakeLists.txt
--- libxml2-2.10.3.orig/CMakeLists.txt	2022-10-14 14:23:53.000000000 +0200
+++ libxml2-2.10.3/CMakeLists.txt	2022-11-05 14:23:05.711802900 +0100
@@ -655,7 +655,11 @@
 
 set(XML_INCLUDEDIR "-I\${includedir}/libxml2")
 set(XML_LIBDIR "-L\${libdir}")
-set(XML_LIBS "-lxml2")
+if(MSVC)
+	set(XML_LIBS "-llibxml2")
+else()
+	set(XML_LIBS "-lxml2")
+endif()
 set(XML_PRIVATE_LIBS "${Z_LIBS} ${LZMA_LIBS} ${THREAD_LIBS} ${ICONV_LIBS} ${ICU_LIBS} ${LIBM} ${WIN32_EXTRA_LIBADD}")
 
 file(RELATIVE_PATH PACKAGE_RELATIVE_PATH "${CMAKE_INSTALL_FULL_LIBDIR}/pkgconfig" "${CMAKE_INSTALL_PREFIX}")
