diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user libxml2-2.10.3.orig/CMakeLists.txt libxml2-2.10.3/CMakeLists.txt
--- libxml2-2.10.3.orig/CMakeLists.txt	2022-11-14 14:03:48.090986000 +0100
+++ libxml2-2.10.3/CMakeLists.txt	2022-11-14 14:03:24.437656800 +0100
@@ -653,7 +653,7 @@
 	configure_file(python/setup.py.in setup.py @ONLY)
 endif()
 
-set(XML_INCLUDEDIR "-I\${includedir}/libxml2")
+set(XML_INCLUDEDIR "-I\${includedir} -I\${includedir}/libxml2")
 set(XML_LIBDIR "-L\${libdir}")
 if(MSVC)
 	set(XML_LIBS "-llibxml2")
