diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/cmake/modules/FindLibraryWithDebug.cmake libmsn-4.0-beta1/cmake/modules/FindLibraryWithDebug.cmake
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/cmake/modules/FindLibraryWithDebug.cmake	1970-01-01 01:00:00.000000000 +0100
+++ libmsn-4.0-beta1/cmake/modules/FindLibraryWithDebug.cmake	2008-11-16 03:02:36.265625000 +0100
@@ -0,0 +1,113 @@
+#
+#  FIND_LIBRARY_WITH_DEBUG
+#  -> enhanced FIND_LIBRARY to allow the search for an
+#     optional debug library with a WIN32_DEBUG_POSTFIX similar
+#     to CMAKE_DEBUG_POSTFIX when creating a shared lib
+#     it has to be the second and third argument
+
+# Copyright (c) 2007, Christian Ehrlicher, <ch.ehrlicher@gmx.de>
+# Redistribution and use is allowed according to the terms of the BSD license.
+# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+
+MACRO(FIND_LIBRARY_WITH_DEBUG var_name win32_dbg_postfix_name dgb_postfix libname)
+
+  IF(NOT "${win32_dbg_postfix_name}" STREQUAL "WIN32_DEBUG_POSTFIX")
+
+     # no WIN32_DEBUG_POSTFIX -> simply pass all arguments to FIND_LIBRARY
+     FIND_LIBRARY(${var_name}
+                  ${win32_dbg_postfix_name}
+                  ${dgb_postfix}
+                  ${libname}
+                  ${ARGN}
+     )
+
+  ELSE(NOT "${win32_dbg_postfix_name}" STREQUAL "WIN32_DEBUG_POSTFIX")
+
+    IF(NOT WIN32)
+      # on non-win32 we don't need to take care about WIN32_DEBUG_POSTFIX
+
+      FIND_LIBRARY(${var_name} ${libname} ${ARGN})
+
+    ELSE(NOT WIN32)
+
+      # 1. get all possible libnames
+      SET(args ${ARGN})
+      SET(newargs "")
+      SET(libnames_release "")
+      SET(libnames_debug "")
+
+      LIST(LENGTH args listCount)
+
+      IF("${libname}" STREQUAL "NAMES")
+        SET(append_rest 0)
+        LIST(APPEND args " ")
+
+        FOREACH(i RANGE ${listCount})
+          LIST(GET args ${i} val)
+
+          IF(append_rest)
+            LIST(APPEND newargs ${val})
+          ELSE(append_rest)
+            IF("${val}" STREQUAL "PATHS")
+              LIST(APPEND newargs ${val})
+              SET(append_rest 1)
+            ELSE("${val}" STREQUAL "PATHS")
+              LIST(APPEND libnames_release "${val}")
+              LIST(APPEND libnames_debug   "${val}${dgb_postfix}")
+            ENDIF("${val}" STREQUAL "PATHS")
+          ENDIF(append_rest)
+
+        ENDFOREACH(i)
+
+      ELSE("${libname}" STREQUAL "NAMES")
+
+        # just one name
+        LIST(APPEND libnames_release "${libname}")
+        LIST(APPEND libnames_debug   "${libname}${dgb_postfix}")
+
+        SET(newargs ${args})
+
+      ENDIF("${libname}" STREQUAL "NAMES")
+
+      # search the release lib
+      FIND_LIBRARY(${var_name}_RELEASE
+                   NAMES ${libnames_release}
+                   ${newargs}
+      )
+
+      # search the debug lib
+      FIND_LIBRARY(${var_name}_DEBUG
+                   NAMES ${libnames_debug}
+                   ${newargs}
+      )
+
+      IF(${var_name}_RELEASE AND ${var_name}_DEBUG)
+
+        # both libs found
+        SET(${var_name} optimized ${${var_name}_RELEASE}
+                        debug     ${${var_name}_DEBUG})
+
+      ELSE(${var_name}_RELEASE AND ${var_name}_DEBUG)
+
+        IF(${var_name}_RELEASE)
+
+          # only release found
+          SET(${var_name} ${${var_name}_RELEASE})
+
+        ELSE(${var_name}_RELEASE)
+
+          # only debug (or nothing) found
+          SET(${var_name} ${${var_name}_DEBUG})
+
+        ENDIF(${var_name}_RELEASE)
+       
+      ENDIF(${var_name}_RELEASE AND ${var_name}_DEBUG)
+
+      MARK_AS_ADVANCED(${var_name}_RELEASE)
+      MARK_AS_ADVANCED(${var_name}_DEBUG)
+
+    ENDIF(NOT WIN32)
+
+  ENDIF(NOT "${win32_dbg_postfix_name}" STREQUAL "WIN32_DEBUG_POSTFIX")
+
+ENDMACRO(FIND_LIBRARY_WITH_DEBUG)
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/cmake/modules/FindOpenSSL.cmake libmsn-4.0-beta1/cmake/modules/FindOpenSSL.cmake
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/cmake/modules/FindOpenSSL.cmake	1970-01-01 01:00:00.000000000 +0100
+++ libmsn-4.0-beta1/cmake/modules/FindOpenSSL.cmake	2008-11-16 03:02:36.265625000 +0100
@@ -0,0 +1,63 @@
+# - Try to find the OpenSSL encryption library
+# Once done this will define
+#
+#  OPENSSL_FOUND - system has the OpenSSL library
+#  OPENSSL_INCLUDE_DIR - the OpenSSL include directory
+#  OPENSSL_LIBRARIES - The libraries needed to use OpenSSL
+#  OPENSSL_EAY_LIBRARIES - The additional libraries needed to use OpenSSL on windows
+
+# Copyright (c) 2006, Alexander Neundorf, <neundorf@kde.org>
+#
+# Redistribution and use is allowed according to the terms of the BSD license.
+# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+
+INCLUDE(FindLibraryWithDebug)
+
+# on win32 we additional need to link to libeay32.lib
+MACRO(OPENSSL_ADD_LIB_EAY_LIBS)
+   FIND_LIBRARY_WITH_DEBUG(OPENSSL_EAY_LIBRARIES
+                   WIN32_DEBUG_POSTFIX d
+                   NAMES eay libeay libeay32 libeay32MD)
+ENDMACRO(OPENSSL_ADD_LIB_EAY_LIBS)
+
+IF(OPENSSL_LIBRARIES)
+   SET(OpenSSL_FIND_QUIETLY TRUE)
+ENDIF(OPENSSL_LIBRARIES)
+
+IF(SSL_EAY_DEBUG AND SSL_EAY_RELEASE)
+   SET(LIB_FOUND 1)
+ENDIF(SSL_EAY_DEBUG AND SSL_EAY_RELEASE)
+
+FIND_PATH(OPENSSL_INCLUDE_DIR openssl/ssl.h )
+
+FIND_LIBRARY_WITH_DEBUG(OPENSSL_LIBRARIES
+                WIN32_DEBUG_POSTFIX d
+                NAMES ssl ssleay ssleay32 ssleay32MD)
+
+IF(WIN32)
+   OPENSSL_ADD_LIB_EAY_LIBS()
+   IF(OPENSSL_INCLUDE_DIR AND OPENSSL_LIBRARIES AND OPENSSL_EAY_LIBRARIES)
+      SET(OPENSSL_FOUND TRUE)
+   ELSE(OPENSSL_INCLUDE_DIR AND OPENSSL_LIBRARIES AND OPENSSL_EAY_LIBRARIES)
+      SET(OPENSSL_FOUND FALSE)
+   ENDIF (OPENSSL_INCLUDE_DIR AND OPENSSL_LIBRARIES AND OPENSSL_EAY_LIBRARIES)
+ELSE(WIN32)
+   IF(OPENSSL_INCLUDE_DIR AND OPENSSL_LIBRARIES)
+      SET(OPENSSL_FOUND TRUE)
+   ELSE(OPENSSL_INCLUDE_DIR AND OPENSSL_LIBRARIES)
+      SET(OPENSSL_FOUND FALSE)
+   ENDIF (OPENSSL_INCLUDE_DIR AND OPENSSL_LIBRARIES)
+ENDIF(WIN32)
+
+IF (OPENSSL_FOUND)
+   IF (NOT OpenSSL_FIND_QUIETLY)
+      MESSAGE(STATUS "Found OpenSSL: ${OPENSSL_LIBRARIES}")
+   ENDIF (NOT OpenSSL_FIND_QUIETLY)
+ELSE (OPENSSL_FOUND)
+   IF (OpenSSL_FIND_REQUIRED)
+      MESSAGE(FATAL_ERROR "Could NOT find OpenSSL")
+   ENDIF (OpenSSL_FIND_REQUIRED)
+ENDIF (OPENSSL_FOUND)
+
+MARK_AS_ADVANCED(OPENSSL_INCLUDE_DIR OPENSSL_LIBRARIES)
+
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/CMakeLists.txt libmsn-4.0-beta1/CMakeLists.txt
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/CMakeLists.txt	2008-10-27 16:08:03.000000000 +0100
+++ libmsn-4.0-beta1/CMakeLists.txt	2008-11-16 03:02:36.265625000 +0100
@@ -1,6 +1,10 @@
 CMAKE_MINIMUM_REQUIRED(VERSION 2.6)
 project(libmsn)
 
-include_directories(${CMAKE_CURRENT_SOURCE_DIR})
+set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules ${CMAKE_MODULE_PATH})
+# find directories for openssl
+find_package(OpenSSL REQUIRED)
+
+include_directories(${CMAKE_CURRENT_SOURCE_DIR} ${OPENSSL_INCLUDE_DIR})
 add_subdirectory(msn)
 
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/authdata.h libmsn-4.0-beta1/msn/authdata.h
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/authdata.h	2007-09-03 02:18:57.000000000 +0200
+++ libmsn-4.0-beta1/msn/authdata.h	2008-11-16 03:11:49.312500000 +0100
@@ -28,9 +28,11 @@
 #include <string>
 #include <msn/passport.h>
 
+#include "libmsn_export.h"
+
 namespace MSN
 {
-    class AuthData
+    class LIBMSN_EXPORT AuthData
     {
 public:
         Passport username;
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/buddy.h libmsn-4.0-beta1/msn/buddy.h
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/buddy.h	2008-09-21 21:01:04.000000000 +0200
+++ libmsn-4.0-beta1/msn/buddy.h	2008-11-16 03:11:54.843750000 +0100
@@ -31,6 +31,8 @@
 #include <map>
 #include <msn/passport.h>
 
+#include "libmsn_export.h"
+
 namespace MSN
 {
     /** The online state of a buddy.
@@ -63,7 +65,7 @@
      *  store contact information about a buddy that was retrieved during
      *  the buddy list synchronisation process.
      */
-    class Buddy
+    class LIBMSN_EXPORT Buddy
     {
 public:
         /** The PhoneNumbers class contains information about one or more phone numbers
@@ -116,7 +118,7 @@
      *  Each group is represented by a @a groupID and has a user-visible
      *  @a name.
      */
-    class Group
+    class LIBMSN_EXPORT Group
     {
 public:
         std::string groupID;
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/CMakeLists.txt libmsn-4.0-beta1/msn/CMakeLists.txt
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/CMakeLists.txt	2008-11-12 16:45:40.000000000 +0100
+++ libmsn-4.0-beta1/msn/CMakeLists.txt	2008-11-16 03:24:07.156250000 +0100
@@ -17,6 +17,10 @@
     buddy.cpp
     passport.cpp
 )
+if(WIN32)
+set(msn_STAT_SRCS   ${msn_STAT_SRCS}
+                    ${OPENSSL_INCLUDE_DIR}/openssl/applink.c)
+endif(WIN32)
 
 set(msn_HEADERS
     authdata.h
@@ -39,9 +43,15 @@
 set_target_properties(msn PROPERTIES VERSION 0.1.0 
                                      SOVERSION 0.1
 )
+if(NOT WIN32)
 target_link_libraries(msn crypto)
+else(NOT WIN32)
+target_link_libraries(msn ${OPENSSL_EAY_LIBRARIES})
+endif(NOT WIN32)
 
 ########### install files ###############
-install(TARGETS msn DESTINATION  ${CMAKE_INSTALL_PREFIX}/lib)
+install(TARGETS msn RUNTIME DESTINATION bin
+                    LIBRARY DESTINATION lib
+                    ARCHIVE DESTINATION lib)
 install(FILES ${msn_HEADERS} DESTINATION ${CMAKE_INSTALL_PREFIX}/include/msn)
 
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/connection.h libmsn-4.0-beta1/msn/connection.h
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/connection.h	2008-09-21 21:01:04.000000000 +0200
+++ libmsn-4.0-beta1/msn/connection.h	2008-11-16 03:12:01.406250000 +0100
@@ -30,6 +30,12 @@
 #include <map>
 #include <stdexcept>
 
+#ifdef _MSC_VER
+#pragma warning( disable : 4290 )
+#endif
+
+#include "libmsn_export.h"
+
 namespace MSN
 {
     class callback;
@@ -46,7 +52,7 @@
      *       that is waiting to be processed.  Member functions should use that
      *       for necessary data, and <em>not</em> call Connection::readExactly.
      */
-    class Connection
+    class LIBMSN_EXPORT Connection
     {      
 public:
         /** The socket which connects this Connection to another computer
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/errorcodes.h libmsn-4.0-beta1/msn/errorcodes.h
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/errorcodes.h	2007-09-03 02:18:57.000000000 +0200
+++ libmsn-4.0-beta1/msn/errorcodes.h	2008-11-16 03:12:07.546875000 +0100
@@ -29,6 +29,8 @@
 #include <list>
 #include <msn/switchboardserver.h>
 
+#include "libmsn_export.h"
+
 /** \mainpage libmsn 3.2 Reference
  *
  * <code>libmsn</code> is a C++ library for Microsoft's MSN Messenger service. It
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/externals.h libmsn-4.0-beta1/msn/externals.h
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/externals.h	2008-10-07 20:46:08.000000000 +0200
+++ libmsn-4.0-beta1/msn/externals.h	2008-11-16 03:12:16.156250000 +0100
@@ -28,13 +28,15 @@
 #include <msn/buddy.h>
 #include <msn/util.h>
 
+#include "libmsn_export.h"
+
 namespace MSN
 {
     class ListSyncInfo;
     /** The application should implement these callback functions to 
      * be able to receive notifications from the library.
      */
-    class Callbacks
+    class LIBMSN_EXPORT Callbacks
     {
     public:
         virtual void registerSocket(void *sock, int read, int write, bool isSSL) = 0;
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/libmsn_export.h libmsn-4.0-beta1/msn/libmsn_export.h
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/libmsn_export.h	1970-01-01 01:00:00.000000000 +0100
+++ libmsn-4.0-beta1/msn/libmsn_export.h	2008-11-16 03:23:47.921875000 +0100
@@ -0,0 +1,21 @@
+#ifndef _LIBMSN_EXPORT_H_
+#define _LIBMSN_EXPORT_H_
+
+/* export classes under windows
+ * visibility can be added later under all other systems
+*/
+#ifdef _MSC_VER
+#pragma warning( disable : 4251, 4996 )
+#endif
+
+#ifdef _WIN32
+# ifdef msn_EXPORTS
+#  define LIBMSN_EXPORT __declspec(dllexport)
+# else
+#  define LIBMSN_EXPORT __declspec(dllimport)
+#endif
+#else
+# define LIBMSN_EXPORT
+#endif
+
+#endif // _LIBMSN_EXPORT_H_
\ No newline at end of file
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/message.h libmsn-4.0-beta1/msn/message.h
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/message.h	2008-09-10 21:34:08.000000000 +0200
+++ libmsn-4.0-beta1/msn/message.h	2008-11-16 03:10:32.687500000 +0100
@@ -30,6 +30,12 @@
 #include <vector>
 #include <stdexcept>
 
+#include "libmsn_export.h"
+
+#ifdef _MSC_VER
+#pragma warning( disable : 4290 )
+#endif
+
 namespace MSN 
 {
     
@@ -39,7 +45,7 @@
      *
      *  @todo  Complete read/write support for formatting messages.
      */
-    class Message
+    class LIBMSN_EXPORT Message
     {
 public:
         enum FontEffects
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/msnobject.cpp libmsn-4.0-beta1/msn/msnobject.cpp
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/msnobject.cpp	2008-09-21 21:01:04.000000000 +0200
+++ libmsn-4.0-beta1/msn/msnobject.cpp	2008-11-16 03:02:36.375000000 +0100
@@ -108,7 +108,7 @@
     {
         if(msnObjects.empty()) return false;
 
-        __gnu_cxx::slist<MSNObjectUnit>::iterator i = msnObjects.begin();
+        std::list<MSNObjectUnit>::iterator i = msnObjects.begin();
         for(; i!=msnObjects.end();i++)
         {
             if((*i).realLocation == filename && 
@@ -126,8 +126,8 @@
         bool deleted=false;
         if(msnObjects.empty()) return false;
 
-        __gnu_cxx::slist<MSNObjectUnit>::iterator i = msnObjects.begin();
-        __gnu_cxx::slist<MSNObjectUnit>::iterator d;
+        std::list<MSNObjectUnit>::iterator i = msnObjects.begin();
+        std::list<MSNObjectUnit>::iterator d;
         for(;i!=msnObjects.end();i++)
         {
             if((*i).Type == Type)
@@ -144,7 +144,7 @@
     bool MSNObject::getMSNObjectXMLByType(int Type, std::string & xml)
     {
         if(msnObjects.empty()) return false;
-        __gnu_cxx::slist<MSNObjectUnit>::iterator i = msnObjects.begin();
+        std::list<MSNObjectUnit>::iterator i = msnObjects.begin();
         
         for( ; i!=msnObjects.end();i++)
         {
@@ -163,7 +163,7 @@
         XMLNode msnObject = XMLNode::parseString(xml.c_str());
         std::string SHA1D = msnObject.getAttribute("SHA1D",0);
 
-        __gnu_cxx::slist<MSNObjectUnit>::iterator i = msnObjects.begin();
+        std::list<MSNObjectUnit>::iterator i = msnObjects.begin();
         
         for( ; i!=msnObjects.end();i++)
         {
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/msnobject.h libmsn-4.0-beta1/msn/msnobject.h
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/msnobject.h	2008-09-21 21:01:04.000000000 +0200
+++ libmsn-4.0-beta1/msn/msnobject.h	2008-11-16 03:10:49.250000000 +0100
@@ -25,11 +25,12 @@
 
 #include <string>
 #include <list>
-#include <ext/slist>
+
+#include "libmsn_export.h"
 
 namespace MSN
 {
-    class MSNObject
+    class LIBMSN_EXPORT MSNObject
     {
 private:
         unsigned int current_id;
@@ -47,7 +48,7 @@
             std::string XMLString;
         } MSNObjectUnit;
 
-        __gnu_cxx::slist<MSNObjectUnit> msnObjects;
+        std::list<MSNObjectUnit> msnObjects;
 public:
         MSNObject() : current_id(0) {};
         ~MSNObject() {};
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/notificationserver.h libmsn-4.0-beta1/msn/notificationserver.h
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/notificationserver.h	2008-10-18 02:41:53.000000000 +0200
+++ libmsn-4.0-beta1/msn/notificationserver.h	2008-11-16 03:11:07.718750000 +0100
@@ -37,13 +37,19 @@
 #include <cassert>
 #include <sys/types.h>
 
+#include "libmsn_export.h"
+
+#ifdef _WIN32
+typedef unsigned uint;
+#endif
+
 namespace MSN
 {    
     class SwitchboardServerConnection;
     
     /** Contains information about synchronising the contact list with the server.
      */
-    class ListSyncInfo
+    class LIBMSN_EXPORT ListSyncInfo
     {
 public:
         /** Constants specifying the status of the list synchronisation.
@@ -115,7 +121,7 @@
     };
     
     // Intermediate steps in connection:
-    class connectinfo // : public callback_data
+    class LIBMSN_EXPORT connectinfo // : public callback_data
     {
 public:
         Passport username;
@@ -131,7 +137,7 @@
      *  list, online status, and dispatching message requests or invitations to
      *  or from switchboard servers.
      */
-    class NotificationServerConnection : public Connection
+    class LIBMSN_EXPORT NotificationServerConnection : public Connection
     {
 private:
         typedef void (NotificationServerConnection::*NotificationServerCallback)(std::vector<std::string> & args, int trid, void *);
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/p2p.h libmsn-4.0-beta1/msn/p2p.h
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/p2p.h	2008-10-02 03:39:45.000000000 +0200
+++ libmsn-4.0-beta1/msn/p2p.h	2008-11-16 03:11:15.359375000 +0100
@@ -35,6 +35,8 @@
 #include <fstream>
 #include <iostream>
 
+#include "libmsn_export.h"
+
 namespace MSN
 {
     class SwitchboardServerConnection;
@@ -42,7 +44,7 @@
 
     /** Manages all p2p communication. 
      */
-    class P2P
+    class LIBMSN_EXPORT P2P
     {
         public:
             P2P();
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/passport.h libmsn-4.0-beta1/msn/passport.h
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/passport.h	2008-09-10 21:34:08.000000000 +0200
+++ libmsn-4.0-beta1/msn/passport.h	2008-11-16 03:22:36.375000000 +0100
@@ -29,6 +29,8 @@
 #include <stdexcept>
 #include <iostream>
 
+#include "libmsn_export.h"
+
 namespace MSN
 {
     /** An InvalidPassport exception will be thrown whenever
@@ -47,7 +49,7 @@
      *  @todo Document validation rules.
      *  @todo Investigate subclassing std::string to reduce code duplication.
      */
-    class Passport
+    class LIBMSN_EXPORT Passport
     {
 public:
         Passport(std::string email_) : email(email_) { validate(); };
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/soap.cpp libmsn-4.0-beta1/msn/soap.cpp
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/soap.cpp	2008-09-21 21:01:04.000000000 +0200
+++ libmsn-4.0-beta1/msn/soap.cpp	2008-11-16 03:02:36.406250000 +0100
@@ -30,6 +30,7 @@
 
 #include <cctype>
 #include <iostream>
+#include <algorithm>
 #include <string.h>
 #include <soap.h>
 
@@ -1527,7 +1528,7 @@
                     if(!Member.nChildNode("PassportName"))
                             continue;
                     std::string a = Member.getChildNode("PassportName").getText();
-                    std::transform (a.begin(),a.end(), a.begin(), tolower); 
+                    transform (a.begin(),a.end(), a.begin(), tolower); 
 
                     if(!listInfo->contactList[a])
                         listInfo->contactList[a]=new Buddy(a);
@@ -1686,7 +1687,7 @@
             std::string isMobileIMEnabled(thisContact.getChildNode("isMobileIMEnabled").getText());
             std::string isMessengerUser(thisContact.getChildNode("isMessengerUser").getText());
             std::string contactType1(thisContact.getChildNode("contactType").getText());
-            std::transform (passportName.begin(), passportName.end(), passportName.begin(), tolower); 
+            transform (passportName.begin(), passportName.end(), passportName.begin(), tolower); 
 
             if(!listInfo->contactList[passportName])
                 listInfo->contactList[passportName]=new Buddy(passportName,displayName);
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/soap.h libmsn-4.0-beta1/msn/soap.h
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/soap.h	2008-09-21 21:01:04.000000000 +0200
+++ libmsn-4.0-beta1/msn/soap.h	2008-11-16 03:11:27.640625000 +0100
@@ -35,13 +35,15 @@
 #include <vector>
 #include <map>
 
+#include "libmsn_export.h"
+
 namespace MSN
 {
     class NotificationServerConnection;
 
     /** Represents a Soap Connection made by NotificationServerConnection
      */
-    class Soap : public Connection
+    class LIBMSN_EXPORT Soap : public Connection
     {
 private:
         NotificationServerConnection & notificationServer;
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/sstream_fix.h libmsn-4.0-beta1/msn/sstream_fix.h
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/sstream_fix.h	2007-09-03 02:18:57.000000000 +0200
+++ libmsn-4.0-beta1/msn/sstream_fix.h	2008-11-16 03:11:33.250000000 +0100
@@ -40,6 +40,8 @@
 #include <string>
 #include <strstream>
 
+#include "libmsn_export.h"
+
 namespace std
 {
 
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/switchboardserver.h libmsn-4.0-beta1/msn/switchboardserver.h
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/switchboardserver.h	2008-10-01 04:25:32.000000000 +0200
+++ libmsn-4.0-beta1/msn/switchboardserver.h	2008-11-16 03:11:41.703125000 +0100
@@ -33,6 +33,8 @@
 #include <string>
 #include <cassert>
 
+#include "libmsn_export.h"
+
 namespace MSN
 {
     class NotificationServerConnection;
@@ -40,7 +42,7 @@
 
     /** Represents a connection to a MSN switchboard.
      */
-    class SwitchboardServerConnection : public Connection
+    class LIBMSN_EXPORT SwitchboardServerConnection : public Connection
     {
 private:
         typedef void (SwitchboardServerConnection::*SwitchboardServerCallback)(std::vector<std::string> & args, int trid, void *);
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/util.cpp libmsn-4.0-beta1/msn/util.cpp
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/util.cpp	2008-11-12 22:07:09.000000000 +0100
+++ libmsn-4.0-beta1/msn/util.cpp	2008-11-16 03:02:36.453125000 +0100
@@ -32,6 +32,10 @@
 #include <sys/timeb.h>
 #include "md5.h"
 
+#ifdef _WIN32
+#define random rand
+#endif
+
 namespace MSN 
 {    
     std::pair<std::string, int> splitServerAddress(const std::string & address, int default_port)
diff -Nru D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/util.h libmsn-4.0-beta1/msn/util.h
--- D:\cygopt\tempdownloads\libmsn-4.0-beta1/msn/util.h	2008-11-12 22:07:09.000000000 +0100
+++ libmsn-4.0-beta1/msn/util.h	2008-11-16 03:02:36.453125000 +0100
@@ -37,6 +37,10 @@
 #include <stdexcept>
 #include <string>
 
+#ifdef _MSC_VER
+#pragma warning( disable : 4290 )
+#endif
+
 // this is for CHL command
 #define szClientID "PROD0114ES4Z%Q5W"
 #define szClientCode "PK}_A_0N_K%O?A9S"
