diff -Nru libgpg-error-1.10.orig/src/init.h libgpg-error-1.10/src/init.h
--- libgpg-error-1.10.orig/src/init.h	2010-09-16 20:32:44.000000000 +0200
+++ libgpg-error-1.10/src/init.h	2010-11-01 23:27:29.773288400 +0100
@@ -61,9 +61,7 @@
 
 
 /* Explicit constructor for w32-gettext.c  */
-#ifndef DLL_EXPORT
 void _gpg_w32__init_gettext_module (void);
-#endif
 
 #endif /*HAVE_W32_SYSTEM*/
 
diff -Nru libgpg-error-1.10.orig/src/w32-gettext.c libgpg-error-1.10/src/w32-gettext.c
--- libgpg-error-1.10.orig/src/w32-gettext.c	2010-11-01 23:27:43.412068500 +0100
+++ libgpg-error-1.10/src/w32-gettext.c	2010-11-01 23:19:57.285407600 +0100
@@ -33,7 +33,9 @@
 #ifdef HAVE_SYS_TYPES_H
 #include <sys/types.h>
 #endif
+#ifdef HAVE_STDINT_H
 #include <stdint.h>
+#endif
 #ifndef HAVE_W32CE_SYSTEM
 # include <locale.h>
 #endif /*HAVE_W32CE_SYSTEM*/
