diff -Nru digikam-2.0.0.orig/extra/kipi-plugins/gpssync/tests/CMakeLists.txt digikam-2.0.0/extra/kipi-plugins/gpssync/tests/CMakeLists.txt
--- digikam-2.0.0.orig/extra/kipi-plugins/gpssync/tests/CMakeLists.txt	2011-07-28 01:27:36.000000000 +0200
+++ digikam-2.0.0/extra/kipi-plugins/gpssync/tests/CMakeLists.txt	2011-08-25 10:02:10.019844200 +0200
@@ -7,6 +7,7 @@
                         ${KDE4_KDECORE_LIBS}
                         ${QT_QTTEST_LIBRARY}
                         ${QT_QTGUI_LIBRARY}
+                        ${QT_QTXML_LIBRARY}
                         ${KGEOMAP_LIBRARIES}
                      )
 
@@ -41,7 +42,7 @@
                    )
 
 TARGET_LINK_LIBRARIES(gpssync_test_kipiimageitem
-                      ${KDE4_KDECORE_LIBS}
+                      ${KDE4_KDEUI_LIBS}
                       ${QT_QTTEST_LIBRARY}
                       ${QT_QTGUI_LIBRARY}
                       ${KEXIV2_LIBRARIES}
diff -Nru digikam-2.0.0.orig/extra/libkgeomap/test/CMakeLists.txt digikam-2.0.0/extra/libkgeomap/test/CMakeLists.txt
--- digikam-2.0.0.orig/extra/libkgeomap/test/CMakeLists.txt	2011-07-28 01:27:44.000000000 +0200
+++ digikam-2.0.0/extra/libkgeomap/test/CMakeLists.txt	2011-08-25 10:02:10.019844200 +0200
@@ -35,7 +35,7 @@
         test_itemmarkertiler.cpp
        )
     KDE4_ADD_EXECUTABLE(kgeomap_test_itemmarkertiler ${test_itemmarkertiler_sources})
-    TARGET_LINK_LIBRARIES(kgeomap_test_itemmarkertiler kgeomap ${QT_QTTEST_LIBRARY})
+    TARGET_LINK_LIBRARIES(kgeomap_test_itemmarkertiler kgeomap ${KDE4_KDECORE_LIBS} ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${QT_QTTEST_LIBRARY})
     ADD_TEST(kgeomap_test_itemmarkertiler ${EXECUTABLE_OUTPUT_PATH}/kgeomap_test_itemmarkertiler)
 
     # test the primitive datatypes and helper functions
@@ -43,7 +43,7 @@
         test_primitives.cpp
        )
     KDE4_ADD_EXECUTABLE(kgeomap_test_primitives ${test_primitives_sources})
-    TARGET_LINK_LIBRARIES(kgeomap_test_primitives kgeomap ${QT_QTTEST_LIBRARY})
+    TARGET_LINK_LIBRARIES(kgeomap_test_primitives kgeomap ${QT_QTCORE_LIBRARY} ${QT_QTTEST_LIBRARY})
     ADD_TEST(kgeomap_test_primitives ${EXECUTABLE_OUTPUT_PATH}/kgeomap_test_primitives)
 
     # test the GeoCoordinates class
@@ -51,7 +51,7 @@
         test_geocoordinates.cpp
        )
     KDE4_ADD_EXECUTABLE(kgeomap_test_geocoordinates ${test_geocoordinates_sources})
-    TARGET_LINK_LIBRARIES(kgeomap_test_geocoordinates kgeomap ${QT_QTTEST_LIBRARY})
+    TARGET_LINK_LIBRARIES(kgeomap_test_geocoordinates kgeomap ${QT_QTCORE_LIBRARY} ${QT_QTTEST_LIBRARY})
     ADD_TEST(kgeomap_test_geocoordinates ${EXECUTABLE_OUTPUT_PATH}/kgeomap_test_geocoordinates)
 
     # test the TileIndex class
@@ -59,7 +59,7 @@
         test_tileindex.cpp
        )
     KDE4_ADD_EXECUTABLE(kgeomap_test_tileindex ${test_tileindex_sources})
-    TARGET_LINK_LIBRARIES(kgeomap_test_tileindex kgeomap ${QT_QTTEST_LIBRARY})
+    TARGET_LINK_LIBRARIES(kgeomap_test_tileindex kgeomap ${KDE4_KDECORE_LIBS} ${QT_QTCORE_LIBRARY} ${QT_QTTEST_LIBRARY})
     ADD_TEST(kgeomap_test_tileindex ${EXECUTABLE_OUTPUT_PATH}/kgeomap_test_tileindex)
 
     # test the LookupAltitudeGeonames class
@@ -67,7 +67,7 @@
         test_lookup_altitude_geonames.cpp
        )
     KDE4_ADD_EXECUTABLE(kgeomap_test_lookup_altitude_geonames ${test_lookup_altitude_geonames_sources})
-    TARGET_LINK_LIBRARIES(kgeomap_test_lookup_altitude_geonames kgeomap ${QT_QTTEST_LIBRARY})
+    TARGET_LINK_LIBRARIES(kgeomap_test_lookup_altitude_geonames kgeomap ${QT_QTCORE_LIBRARY} ${QT_QTTEST_LIBRARY})
     # do not add this as a test because it only works if there is an internet connection
 #     ADD_TEST(kgeomap_test_lookup_altitude_geonames ${EXECUTABLE_OUTPUT_PATH}/kgeomap_test_lookup_altitude_geonames)
 ENDIF(KDE4_BUILD_TESTS)
diff -Nru digikam-2.0.0.orig/extra/libmediawiki/libmediawiki/CMakeLists.txt digikam-2.0.0/extra/libmediawiki/libmediawiki/CMakeLists.txt
--- digikam-2.0.0.orig/extra/libmediawiki/libmediawiki/CMakeLists.txt	2011-07-28 01:27:42.000000000 +0200
+++ digikam-2.0.0/extra/libmediawiki/libmediawiki/CMakeLists.txt	2011-08-25 10:02:10.019844200 +0200
@@ -23,9 +23,9 @@
 # ============================================================
 
 CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/version.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/version.h)
-IF( KDE4_BUILD_TESTS )
+IF( KDE4_BUILD_TESTS AND NOT WIN32 )
 	set( CMAKE_CXX_FLAGS "-fprofile-arcs -ftest-coverage")
-ENDIF( KDE4_BUILD_TESTS )
+ENDIF( KDE4_BUILD_TESTS AND NOT WIN32 )
 SET(mediawiki_LIB_SRCS image.cpp
                        imageinfo.cpp
                        mediawiki.cpp
