diff -Nru openbabel-2.2.3.orig/CMakeLists.txt openbabel-2.2.3/CMakeLists.txt
--- openbabel-2.2.3.orig/CMakeLists.txt	2009-07-29 03:05:04.000000000 +0200
+++ openbabel-2.2.3/CMakeLists.txt	2010-12-08 01:42:13.828272100 +0100
@@ -70,6 +70,10 @@
   set(MODULE_EXTENSION ".so")
 endif(WIN32)
 
+if(MSVC10)
+    add_definitions(/Zc:auto-)
+endif(MSVC10)
+
 # configure checks
 find_package(LibXml2)
 find_package(Boost)
diff -Nru openbabel-2.2.3.orig/include/openbabel/generic.h openbabel-2.2.3/include/openbabel/generic.h
--- openbabel-2.2.3.orig/include/openbabel/generic.h	2008-12-01 18:42:19.000000000 +0100
+++ openbabel-2.2.3/include/openbabel/generic.h	2010-12-08 00:40:25.429163800 +0100
@@ -486,7 +486,7 @@
     //! double is angle in radians
     std::vector<triple<OBAtom*,OBAtom*,double> > _ads;
 
-    OBTorsion(): _bc(NULL, NULL)      {      }
+    OBTorsion(): _bc(reinterpret_cast<OBAtom*>(NULL), reinterpret_cast<OBAtom*>(NULL))      {      }
     //protected for use only by friend classes
     OBTorsion(OBAtom *, OBAtom *, OBAtom *, OBAtom *);
 
diff -Nru openbabel-2.2.3.orig/src/formats/inchiformat.cpp openbabel-2.2.3/src/formats/inchiformat.cpp
--- openbabel-2.2.3.orig/src/formats/inchiformat.cpp	2009-06-15 01:05:02.000000000 +0200
+++ openbabel-2.2.3/src/formats/inchiformat.cpp	2010-12-08 02:25:25.419502600 +0100
@@ -27,6 +27,7 @@
 #include <strstream>
 #endif
 #include <set>
+#include <iterator>
 #include <vector>
 #include "openbabel/chiral.h"
 
diff -Nru openbabel-2.2.3.orig/src/formats/pngformat.cpp openbabel-2.2.3/src/formats/pngformat.cpp
--- openbabel-2.2.3.orig/src/formats/pngformat.cpp	2008-12-01 18:42:02.000000000 +0100
+++ openbabel-2.2.3/src/formats/pngformat.cpp	2010-12-08 02:27:18.855990800 +0100
@@ -14,6 +14,7 @@
 #include <sstream>
 #include <string>
 #include <algorithm>
+#include <iterator>
 #include <openbabel/babelconfig.h>
 #include <openbabel/format.h>
 #include <openbabel/obconversion.h>
diff -Nru openbabel-2.2.3.orig/src/generic.cpp openbabel-2.2.3/src/generic.cpp
--- openbabel-2.2.3.orig/src/generic.cpp	2009-07-02 01:05:02.000000000 +0200
+++ openbabel-2.2.3/src/generic.cpp	2010-12-08 00:43:47.786738000 +0100
@@ -851,7 +851,7 @@
   **\brief Angle default constructor
   */
   OBAngle::OBAngle():
-    _vertex(NULL), _termini(NULL, NULL), _radians(0.0)
+    _vertex(NULL), _termini(reinterpret_cast<OBAtom*>(NULL), reinterpret_cast<OBAtom*>(NULL)), _radians(0.0)
   {  }
 
   /*!
