diff -Nrub -x '*~' -x '*\.orig' -x'*\.o' llvm-3.7.0.src.orig/tools/llvm-ar/install_symlink.cmake llvm-3.7.0.src/tools/llvm-ar/install_symlink.cmake
--- llvm-3.7.0.src.orig/tools/llvm-ar/install_symlink.cmake	2015-06-09 23:50:22.000000000 +0200
+++ llvm-3.7.0.src/tools/llvm-ar/install_symlink.cmake	2015-12-08 18:41:51.337058900 +0100
@@ -2,9 +2,9 @@
 # DESTDIR environment variable may be unset at configuration time.
 # See PR8397.
 
+set(DESTDIR $ENV{DESTDIR})
 if(UNIX)
   set(LINK_OR_COPY create_symlink)
-  set(DESTDIR $ENV{DESTDIR})
 else()
   set(LINK_OR_COPY copy)
 endif()
@@ -16,7 +16,7 @@
   set(EXECUTABLE_SUFFIX "")
 endif()
 
-set(bindir "${DESTDIR}${CMAKE_INSTALL_PREFIX}/bin/")
+set(bindir "${DESTDIR}/bin/")
 
 message("Creating llvm-ranlib")
 
Dateien llvm-3.7.0.src.orig/utils/llvm-build/llvmbuild/__pycache__/__init__.cpython-34.pyc und llvm-3.7.0.src/utils/llvm-build/llvmbuild/__pycache__/__init__.cpython-34.pyc sind verschieden.
Dateien llvm-3.7.0.src.orig/utils/llvm-build/llvmbuild/__pycache__/componentinfo.cpython-34.pyc und llvm-3.7.0.src/utils/llvm-build/llvmbuild/__pycache__/componentinfo.cpython-34.pyc sind verschieden.
Dateien llvm-3.7.0.src.orig/utils/llvm-build/llvmbuild/__pycache__/configutil.cpython-34.pyc und llvm-3.7.0.src/utils/llvm-build/llvmbuild/__pycache__/configutil.cpython-34.pyc sind verschieden.
Dateien llvm-3.7.0.src.orig/utils/llvm-build/llvmbuild/__pycache__/main.cpython-34.pyc und llvm-3.7.0.src/utils/llvm-build/llvmbuild/__pycache__/main.cpython-34.pyc sind verschieden.
Dateien llvm-3.7.0.src.orig/utils/llvm-build/llvmbuild/__pycache__/util.cpython-34.pyc und llvm-3.7.0.src/utils/llvm-build/llvmbuild/__pycache__/util.cpython-34.pyc sind verschieden.
