diff --git a/cmake/modules/LLVMInstallSymlink.cmake b/cmake/modules/LLVMInstallSymlink.cmake
index 482697b..a4bb0a5 100644
--- a/cmake/modules/LLVMInstallSymlink.cmake
+++ b/cmake/modules/LLVMInstallSymlink.cmake
@@ -3,14 +3,14 @@
 # See PR8397.
 
 function(install_symlink name target outdir)
+  set(DESTDIR $ENV{DESTDIR})
   if(UNIX)
     set(LINK_OR_COPY create_symlink)
-    set(DESTDIR $ENV{DESTDIR})
   else()
     set(LINK_OR_COPY copy)
   endif()
 
-  set(bindir "${DESTDIR}${CMAKE_INSTALL_PREFIX}/${outdir}/")
+  set(bindir "${DESTDIR}/${outdir}/")
 
   message("Creating ${name}")
 
