diff --git a/Tools/qmake/config.tests/icu/icu.cpp b/Tools/qmake/config.tests/icu/icu.cpp
index c1e6c6a..002e062 100644
--- a/Tools/qmake/config.tests/icu/icu.cpp
+++ b/Tools/qmake/config.tests/icu/icu.cpp
@@ -31,16 +31,7 @@
 **
 ****************************************************************************/
 
-#include <unicode/utypes.h>
-#include <unicode/ucol.h>
-#include <unicode/ustring.h>
-
 int main(int, char **)
 {
-    UErrorCode status = U_ZERO_ERROR;
-    UCollator *collator = ucol_open("ru_RU", &status);
-    if (U_FAILURE(status))
-        return 0;
-    ucol_close(collator);
     return 0;
 }
