diff --git a/Tools/qmake/mkspecs/features/default_post.prf b/Tools/qmake/mkspecs/features/default_post.prf
index 2dbb50c..813525c 100644
--- a/Tools/qmake/mkspecs/features/default_post.prf
+++ b/Tools/qmake/mkspecs/features/default_post.prf
@@ -91,12 +91,6 @@ contains(TEMPLATE, derived) {
     # on Linux and Mac OS X. On Windows we do have a convenience copy in
     # Qt5's top-level repository, so let's add that to the PATH if we can
     # find it.
-    win32 {
-        GNUTOOLS_DIR=$$[QT_HOST_DATA]/../gnuwin32/bin
-        exists($$GNUTOOLS_DIR/gperf.exe) {
-            GNUTOOLS = "(set $$escape_expand(\\\")PATH=$$toSystemPath($$GNUTOOLS_DIR);%PATH%$$escape_expand(\\\"))"
-        }
-    }
 
     for(generator, GENERATORS) {
         eval($${generator}.CONFIG = target_predeps no_link)
diff --git a/Tools/qmake/mkspecs/features/default_pre.prf b/Tools/qmake/mkspecs/features/default_pre.prf
index 38443a4..1ab3fb7 100644
--- a/Tools/qmake/mkspecs/features/default_pre.prf
+++ b/Tools/qmake/mkspecs/features/default_pre.prf
@@ -106,7 +106,9 @@ if(win32|mac):!macx-xcode {
 # A newer version of flex is required on Windows. At the moment the only
 # one that appears to provide binaries and is not cygwin is winflex.
 FLEX = flex
+BISON = bison
 win32: FLEX = win_flex
+win32: BISON = win_bison
 
 BIN_EXTENSION =
 win32: BIN_EXTENSION = .exe
diff --git a/Tools/qmake/mkspecs/features/functions.prf b/Tools/qmake/mkspecs/features/functions.prf
index 46293fe..dd7be68 100644
--- a/Tools/qmake/mkspecs/features/functions.prf
+++ b/Tools/qmake/mkspecs/features/functions.prf
@@ -202,9 +202,6 @@ defineTest(programExistsInPath) {
     PATH = "$$(PATH)"
     paths=$$split(PATH, $$QMAKE_DIRLIST_SEP)
 
-    GNUTOOLS_DIR=$$[QT_HOST_DATA]/../gnuwin32/bin
-    exists($$GNUTOOLS_DIR): paths += $$GNUTOOLS_DIR
-
     for(p, paths): exists($$p/$$program):return(true)
     return(false)
 }
