diff -Nru kstars-4.7.0.orig/kstars/kstarsactions.cpp kstars-4.7.0/kstars/kstarsactions.cpp
--- kstars-4.7.0.orig/kstars/kstarsactions.cpp	2011-05-20 22:34:16.000000000 +0200
+++ kstars-4.7.0/kstars/kstarsactions.cpp	2011-08-22 18:27:18.677686300 +0200
@@ -16,6 +16,11 @@
  ***************************************************************************/
 
 //needed in slotRunScript() for chmod() syscall (remote script downloaded to temp file)
+
+#ifdef _WIN32
+#include <windows.h>
+#undef interface
+#endif
 #include <sys/stat.h>
 
 #include <QCheckBox>
diff -Nru kstars-4.7.0.orig/kstars/skycomponents/deepstarcomponent.cpp kstars-4.7.0/kstars/skycomponents/deepstarcomponent.cpp
--- kstars-4.7.0.orig/kstars/skycomponents/deepstarcomponent.cpp	2011-05-20 22:34:17.000000000 +0200
+++ kstars-4.7.0/kstars/skycomponents/deepstarcomponent.cpp	2011-08-22 18:09:38.002019100 +0200
@@ -15,6 +15,9 @@
  *                                                                         *
  ***************************************************************************/
 
+#ifdef _WIN32
+#include <windows.h>
+#endif
 #include "deepstarcomponent.h"
 
 #include <QPixmap>
diff -Nru kstars-4.7.0.orig/kstars/skycomponents/horizoncomponent.cpp kstars-4.7.0/kstars/skycomponents/horizoncomponent.cpp
--- kstars-4.7.0.orig/kstars/skycomponents/horizoncomponent.cpp	2011-05-20 22:34:17.000000000 +0200
+++ kstars-4.7.0/kstars/skycomponents/horizoncomponent.cpp	2011-08-22 18:18:16.042649300 +0200
@@ -97,8 +97,10 @@
         labelPoint2.setAz( labelPoint.az().Degrees() + 1.0 );
         labelPoint2.HorizontalToEquatorial( data->lst(), data->geo()->lat() );
     }
-        
+
+#ifdef __GNUC__
     #warning Still have to port HorizonComponent::draw()
+#endif
     #if 0
     if ( ! selected() ) return;
 
diff -Nru kstars-4.7.0.orig/kstars/skycomponents/skymesh.cpp kstars-4.7.0/kstars/skycomponents/skymesh.cpp
--- kstars-4.7.0.orig/kstars/skycomponents/skymesh.cpp	2011-05-20 22:34:17.000000000 +0200
+++ kstars-4.7.0/kstars/skycomponents/skymesh.cpp	2011-08-22 18:13:43.973087800 +0200
@@ -340,7 +340,9 @@
 
 void SkyMesh::draw(QPainter& psky, MeshBufNum_t bufNum)
 {
+#ifdef __GNUC__
     #warning PORT SKYMESH DRAW
+#endif
     #if 0
     SkyMap*     map  = SkyMap::Instance();
     KStarsData* data = KStarsData::Instance();
diff -Nru kstars-4.7.0.orig/kstars/skycomponents/starcomponent.cpp kstars-4.7.0/kstars/skycomponents/starcomponent.cpp
--- kstars-4.7.0.orig/kstars/skycomponents/starcomponent.cpp	2011-05-20 22:34:17.000000000 +0200
+++ kstars-4.7.0/kstars/skycomponents/starcomponent.cpp	2011-08-22 18:09:38.004019200 +0200
@@ -15,6 +15,9 @@
  *                                                                         *
  ***************************************************************************/
 
+#ifdef _WIN32
+#include <windows.h>
+#endif
 #include "starcomponent.h"
 
 #include <kglobal.h>
diff -Nru kstars-4.7.0.orig/kstars/skyglpainter.cpp kstars-4.7.0/kstars/skyglpainter.cpp
--- kstars-4.7.0.orig/kstars/skyglpainter.cpp	2011-06-22 16:48:32.000000000 +0200
+++ kstars-4.7.0/kstars/skyglpainter.cpp	2011-08-22 18:09:38.008019400 +0200
@@ -17,6 +17,9 @@
 
 */
 
+#ifdef _WIN32
+#include <windows.h>
+#endif
 #include "skyglpainter.h"
 
 #include <cstddef>
diff -Nru kstars-4.7.0.orig/kstars/skymap.cpp kstars-4.7.0/kstars/skymap.cpp
--- kstars-4.7.0.orig/kstars/skymap.cpp	2011-05-20 22:34:17.000000000 +0200
+++ kstars-4.7.0/kstars/skymap.cpp	2011-08-22 18:09:38.012019600 +0200
@@ -15,6 +15,9 @@
  *                                                                         *
  ***************************************************************************/
 
+#ifdef _WIN32
+#include <windows.h>
+#endif
 #include "skymap.h"
 
 #include <QCursor>
diff -Nru kstars-4.7.0.orig/kstars/skymapgldraw.cpp kstars-4.7.0/kstars/skymapgldraw.cpp
--- kstars-4.7.0.orig/kstars/skymapgldraw.cpp	2011-05-20 22:34:17.000000000 +0200
+++ kstars-4.7.0/kstars/skymapgldraw.cpp	2011-08-22 18:09:38.015019800 +0200
@@ -14,6 +14,10 @@
  *                                                                         *
  ***************************************************************************/
 
+#ifdef _WIN32
+#include <windows.h>
+#endif
+
 #include "texturemanager.h"
 #include "skymapcomposite.h"
 #include "skyglpainter.h"
diff -Nru kstars-4.7.0.orig/kstars/skyobjects/skypoint.cpp kstars-4.7.0/kstars/skyobjects/skypoint.cpp
--- kstars-4.7.0.orig/kstars/skyobjects/skypoint.cpp	2011-05-20 22:34:17.000000000 +0200
+++ kstars-4.7.0/kstars/skyobjects/skypoint.cpp	2011-08-22 18:21:54.611150700 +0200
@@ -31,6 +31,7 @@
 #include "Options.h"
 #include "skycomponents/skymapcomposite.h"
 
+const double SkyPoint::altCrit = -1;
 SkyPoint::SkyPoint() {
     // Default constructor. Set nonsense values
     RA0.setD(-1); // RA >= 0 always :-)
diff -Nru kstars-4.7.0.orig/kstars/skyobjects/skypoint.h kstars-4.7.0/kstars/skyobjects/skypoint.h
--- kstars-4.7.0.orig/kstars/skyobjects/skypoint.h	2011-05-20 22:34:17.000000000 +0200
+++ kstars-4.7.0/kstars/skyobjects/skypoint.h	2011-08-22 18:10:05.557595200 +0200
@@ -486,7 +486,7 @@
     /** Remove refraction correction. */
     static dms unrefract(dms h);
 
-    static const double altCrit  = -1;
+    static const double altCrit;
 
 protected:
     /**Precess this SkyPoint's catalog coordinates to the epoch described by the
