diff -Nru -x '*~' kstars-4.10.1.orig/kstars/skyobjects/ksmoon.cpp kstars-4.10.1/kstars/skyobjects/ksmoon.cpp
--- kstars-4.10.1.orig/kstars/skyobjects/ksmoon.cpp	2013-03-01 07:45:57.384215000 +0100
+++ kstars-4.10.1/kstars/skyobjects/ksmoon.cpp	2013-03-06 00:41:59.583346500 +0100
@@ -18,7 +18,7 @@
 #include "ksmoon.h"
 
 #include <stdlib.h>
-#include <math.h>
+#include <cmath>
 
 #include <QFile>
 #include <QTextStream>
@@ -31,6 +31,8 @@
 #include "skycomponents/skymapcomposite.h"
 #include "texturemanager.h"
 
+using namespace std;
+
 namespace {
     // Convert degrees to radians and put it into [0,2*pi] range
     double degToRad(double x) {
