diff -Nru kstars-4.8.0.orig/kstars/kstarsactions.cpp kstars-4.8.0/kstars/kstarsactions.cpp
--- kstars-4.8.0.orig/kstars/kstarsactions.cpp	2011-08-22 15:31:31.000000000 +0200
+++ kstars-4.8.0/kstars/kstarsactions.cpp	2012-01-25 23:02:25.240600000 +0100
@@ -16,6 +16,11 @@
  ***************************************************************************/
 
 //needed in slotRunScript() for chmod() syscall (remote script downloaded to temp file)
+
+#ifdef _WIN32
+#include <windows.h>
+#undef interface
+#endif
 #include <sys/stat.h>
 
 #include <QCheckBox>
diff -Nru kstars-4.8.0.orig/kstars/projections/azimuthalequidistantprojector.cpp kstars-4.8.0/kstars/projections/azimuthalequidistantprojector.cpp
--- kstars-4.8.0.orig/kstars/projections/azimuthalequidistantprojector.cpp	2011-07-27 20:41:02.000000000 +0200
+++ kstars-4.8.0/kstars/projections/azimuthalequidistantprojector.cpp	2012-01-25 23:02:20.261000000 +0100
@@ -18,6 +18,10 @@
 
 */
 
+#ifdef _WIN32
+#include <windows.h>
+#endif
+
 #include "azimuthalequidistantprojector.h"
 
 AzimuthalEquidistantProjector::AzimuthalEquidistantProjector(const ViewParams& p)
diff -Nru kstars-4.8.0.orig/kstars/skycomponents/deepstarcomponent.cpp kstars-4.8.0/kstars/skycomponents/deepstarcomponent.cpp
--- kstars-4.8.0.orig/kstars/skycomponents/deepstarcomponent.cpp	2011-11-04 15:35:06.000000000 +0100
+++ kstars-4.8.0/kstars/skycomponents/deepstarcomponent.cpp	2012-01-25 23:02:25.240600000 +0100
@@ -15,6 +15,9 @@
  *                                                                         *
  ***************************************************************************/
 
+#ifdef _WIN32
+#include <windows.h>
+#endif
 #include "deepstarcomponent.h"
 
 #include <QPixmap>
diff -Nru kstars-4.8.0.orig/kstars/skycomponents/starcomponent.cpp kstars-4.8.0/kstars/skycomponents/starcomponent.cpp
--- kstars-4.8.0.orig/kstars/skycomponents/starcomponent.cpp	2011-11-04 15:35:06.000000000 +0100
+++ kstars-4.8.0/kstars/skycomponents/starcomponent.cpp	2012-01-25 23:02:25.240600000 +0100
@@ -15,6 +15,9 @@
  *                                                                         *
  ***************************************************************************/
 
+#ifdef _WIN32
+#include <windows.h>
+#endif
 #include "starcomponent.h"
 
 #include <kglobal.h>
diff -Nru kstars-4.8.0.orig/kstars/skyglpainter.cpp kstars-4.8.0/kstars/skyglpainter.cpp
--- kstars-4.8.0.orig/kstars/skyglpainter.cpp	2011-08-22 15:31:31.000000000 +0200
+++ kstars-4.8.0/kstars/skyglpainter.cpp	2012-01-25 23:02:25.256200000 +0100
@@ -17,6 +17,9 @@
 
 */
 
+#ifdef _WIN32
+#include <windows.h>
+#endif
 #include "skyglpainter.h"
 
 #include <cstddef>
diff -Nru kstars-4.8.0.orig/kstars/skymap.cpp kstars-4.8.0/kstars/skymap.cpp
--- kstars-4.8.0.orig/kstars/skymap.cpp	2011-09-26 11:44:52.000000000 +0200
+++ kstars-4.8.0/kstars/skymap.cpp	2012-01-25 23:02:25.256200000 +0100
@@ -15,6 +15,9 @@
  *                                                                         *
  ***************************************************************************/
 
+#ifdef _WIN32
+#include <windows.h>
+#endif
 #include "skymap.h"
 
 #include <QCursor>
diff -Nru kstars-4.8.0.orig/kstars/skymapgldraw.cpp kstars-4.8.0/kstars/skymapgldraw.cpp
--- kstars-4.8.0.orig/kstars/skymapgldraw.cpp	2011-07-27 20:41:02.000000000 +0200
+++ kstars-4.8.0/kstars/skymapgldraw.cpp	2012-01-25 23:02:25.256200000 +0100
@@ -14,6 +14,10 @@
  *                                                                         *
  ***************************************************************************/
 
+#ifdef _WIN32
+#include <windows.h>
+#endif
+
 #include "texturemanager.h"
 #include "skymapcomposite.h"
 #include "skyglpainter.h"
