diff -Nru cantor-4.8.0.orig/src/pagebreakentry.cpp cantor-4.8.0/src/pagebreakentry.cpp
--- cantor-4.8.0.orig/src/pagebreakentry.cpp	2011-07-27 20:37:50.000000000 +0200
+++ cantor-4.8.0/src/pagebreakentry.cpp	2012-01-25 17:14:57.689200000 +0100
@@ -25,6 +25,7 @@
 #include <kmenu.h>
 #include <kicon.h>
 #include <klocale.h>
+#include <kzip.h>
 
 PageBreakEntry::PageBreakEntry(QTextCursor position, Worksheet* parent) : 
     WorksheetEntry( position, parent )
@@ -166,7 +167,7 @@
     cursor.setPosition(m_frame->lastPosition(), QTextCursor::KeepAnchor);
     cursor.removeSelectedText();
 
-    if (not m_worksheet->isPrinting())
+    if (!m_worksheet->isPrinting())
     {
 	QTextBlockFormat block(cursor.blockFormat());
 	block.setAlignment(Qt::AlignCenter);
