diff -Nru amarok-2.3.2.orig/src/core-impl/collections/sqlcollection/mysqlecollection/MySqlEmbeddedStorage.cpp amarok-2.3.2/src/core-impl/collections/sqlcollection/mysqlecollection/MySqlEmbeddedStorage.cpp
--- amarok-2.3.2.orig/src/core-impl/collections/sqlcollection/mysqlecollection/MySqlEmbeddedStorage.cpp	2010-09-15 23:24:11.000000000 +0200
+++ amarok-2.3.2/src/core-impl/collections/sqlcollection/mysqlecollection/MySqlEmbeddedStorage.cpp	2011-01-10 02:15:25.906894700 +0100
@@ -79,8 +79,14 @@
         dir.mkpath( "." );
     }
 
-    setenv( "MYSQL_HOME", Amarok::saveLocation().toAscii().data(), 1 );
-    if( mysql_library_init( 0 , 0, 0 ) != 0 )
+    QByteArray defaultsFileOption( "--defaults-file=" );
+    defaultsFileOption.append( defaultsFile.toAscii() );
+
+    static char *server_options[] = { "amarok", defaultsFileOption.data() };
+    int num_elements = sizeof(server_options) / sizeof(char *);
+    static char *server_groups[] = { "embedded", NULL };
+    
+    if( mysql_library_init( num_elements, server_options, server_groups ) != 0 )
     {
         error() << "MySQL library initialization failed.";
         reportError( "init" );
