diff -Nru -x '*~' amarok-2.5.0.orig/src/core-impl/collections/db/sql/mysqlecollection/CMakeLists.txt amarok-2.5.0/src/core-impl/collections/db/sql/mysqlecollection/CMakeLists.txt
--- amarok-2.5.0.orig/src/core-impl/collections/db/sql/mysqlecollection/CMakeLists.txt	2011-12-15 08:36:48.000000000 +0100
+++ amarok-2.5.0/src/core-impl/collections/db/sql/mysqlecollection/CMakeLists.txt	2011-12-22 00:20:08.875768200 +0100
@@ -20,6 +20,9 @@
  )
 
 remove_definitions(-DMAKE_AMAROK_DATABASECOLLECTION_LIB)
+if(MINGW)
+    add_definitions(-DMAKE_AMAROK_SQLCOLLECTION_LIB)
+endif(MINGW)
 kde4_add_plugin(amarok_collection-mysqlecollection ${amarok_collection-mysqlecollection_PART_SRCS})
 
 STRING(REPLACE "-Wl,--fatal-warnings" "" CMAKE_SHARED_LINKER_FLAGS_NOFATALWARN "${CMAKE_SHARED_LINKER_FLAGS}")
