diff -Nru kdebase-workspace-4.3.95.orig/plasma/generic/containmentactions/minimalcontextmenu/CMakeLists.txt kdebase-workspace-4.3.95/plasma/generic/containmentactions/minimalcontextmenu/CMakeLists.txt
--- kdebase-workspace-4.3.95.orig/plasma/generic/containmentactions/minimalcontextmenu/CMakeLists.txt	2010-01-06 18:00:18.000000000 +0100
+++ kdebase-workspace-4.3.95/plasma/generic/containmentactions/minimalcontextmenu/CMakeLists.txt	2010-01-12 16:28:22.225200000 +0100
@@ -4,12 +4,12 @@
     menu.cpp
 )
 
-kde4_add_plugin(plasma_containmentactions_minimalcontextmenu ${contextmenu_SRCS})
+kde4_add_plugin(p ${contextmenu_SRCS})
 if(WIN32)
-target_link_libraries(plasma_containmentactions_minimalcontextmenu ${KDE4_PLASMA_LIBS} ${KDE4_KIO_LIBS})
+target_link_libraries(p ${KDE4_PLASMA_LIBS} ${KDE4_KIO_LIBS})
 else(WIN32)
-target_link_libraries(plasma_containmentactions_minimalcontextmenu ${KDE4_PLASMA_LIBS} ${KDE4_KIO_LIBS})
+target_link_libraries(p ${KDE4_PLASMA_LIBS} ${KDE4_KIO_LIBS})
 endif(WIN32)
-
-install(TARGETS plasma_containmentactions_minimalcontextmenu DESTINATION ${PLUGIN_INSTALL_DIR})
+set_target_properties(p PROPERTIES OUTPUT_NAME "plasma_containmentactions_minimalcontextmenu")
+install(TARGETS p DESTINATION ${PLUGIN_INSTALL_DIR})
 install(FILES plasma-containmentactions-minimalcontextmenu.desktop DESTINATION ${SERVICES_INSTALL_DIR})
