diff -Nru libsamplerate-0.1.7.orig/CMakeLists.txt libsamplerate-0.1.7/CMakeLists.txt
--- libsamplerate-0.1.7.orig/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ libsamplerate-0.1.7/CMakeLists.txt	2009-10-02 00:00:42.485966400 +0200
@@ -0,0 +1,108 @@
+# a CMake build system for libsamplerate
+# copyright Patrick Spendrin <ps_ml@gmx.de> 2009
+#           Ralf Habacker <ralf.habacker@freenet.de> 2009
+# this file can be distributed under exactly the same conditions as the libsamplerate
+
+project(libsamplerate)
+
+cmake_minimum_required(VERSION 2.6)
+
+set (VERSION_MAJOR 0)
+set (VERSION_MINOR 1)
+set (VERSION_PATH 7)
+
+# some options
+option(BUILD_EXAMPLES "building of examples" ON)
+option(BUILD_TESTS "building of tests" OFF)
+option(BUILD_SHARED_LIB "build the library as a shared library (DLL)" ON)
+
+# in case of building with tests, use ctest to run the build tests
+# after building your library, simply call 'make test' within your build directory
+if(BUILD_TESTS)
+enable_testing()
+endif(BUILD_TESTS)
+
+# for the 
+if(BUILD_EXAMPLES)
+    find_path(LIBSNDFILE_INCLUDE sndfile.h)
+endif(BUILD_EXAMPLES)
+
+if(NOT LIBSNDFILE_INCLUDE)
+    # we couldn't find the header
+    message(STATUS "Couldn't find sndfile.h, disabling building of examples")
+    set(BUILD_EXAMPLES OFF)
+
+else(NOT LIBSNDFILE_INCLUDE)
+    # try to find the library now
+    find_library(LIBSNDFILE_LIBRARY sndfile sndfile-1 libsndfile libsndfile-1)
+    if(NOT LIBSNDFILE_LIBRARY)
+        message(STATUS "Couldn't find sndfile.h, disabling building of examples")
+        set(BUILD_EXAMPLES OFF)
+    else(NOT LIBSNDFILE_LIBRARY)
+        # include the directory where the header is
+        include_directories(${LIBSNDFILE_INCLUDE})
+        message(STATUS "found libsndfile: ${LIBSNDFILE_LIBRARY}")
+    endif(NOT LIBSNDFILE_LIBRARY)
+endif(NOT LIBSNDFILE_INCLUDE)
+
+
+if(WIN32)
+    include_directories(Win32)
+    if(MSVC)
+        # suppress stupid warning and remove some errors which happen in newer versions of msvc (those all come from libsndfile)
+        add_definitions(-D_CRT_SECURE_NO_WARNINGS -Dinline=__inline -D__int64_t=long)
+    endif(MSVC)
+endif(WIN32)
+
+include_directories(src)
+
+set(samplerate_SRCS
+    src/samplerate.c
+    src/src_linear.c
+    src/src_zoh.c
+    src/src_sinc.c
+)
+
+if(BUILD_SHARED_LIB)
+    if(WIN32)
+        # add the definition file so that exports can be included in the dll
+        set(samplerate_SRCS ${samplerate_SRCS} Win32/libsamplerate-0.def)
+    endif(WIN32)
+    set(SR_SHARED SHARED)
+endif(BUILD_SHARED_LIB)
+
+add_library(samplerate ${SR_SHARED} ${samplerate_SRCS})
+set_target_properties(samplerate PROPERTIES OUTPUT_NAME "libsamplerate-${VERSION_MAJOR}")
+
+install(TARGETS samplerate RUNTIME DESTINATION bin
+                           LIBRARY DESTINATION lib
+                           ARCHIVE DESTINATION lib)
+
+install(FILES src/samplerate.h DESTINATION include)
+                        
+if(BUILD_EXAMPLES)
+    # add example programs
+    macro(add_example name)
+        add_executable(${name} examples/${name}.c ${ARGN})
+        target_link_libraries(${name} ${LIBSNDFILE_LIBRARY} samplerate winmm)
+        install(TARGETS ${name} RUNTIME DESTINATION bin)
+    endmacro(add_example)
+    add_example(sndfile-resample)
+    add_example(varispeed-play examples/audio_out.c)
+endif(BUILD_EXAMPLES)
+
+if(BUILD_TESTS)
+    # compile test program and add it as a unit test
+    macro(add_srtest name)
+        add_executable(${name} tests/${name}.c ${ARGN})
+        target_link_libraries(${name} samplerate)
+        add_test(${name} ${name})
+    endmacro(add_srtest)
+    add_srtest(misc_test tests/util.c)
+    add_srtest(termination_test tests/util.c)
+    add_srtest(simple_test tests/util.c)
+    add_srtest(reset_test tests/util.c)
+    add_srtest(multi_channel_test tests/util.c tests/calc_snr.c)
+    add_srtest(snr_bw_test tests/util.c)
+    add_srtest(throughput_test tests/util.c)
+endif(BUILD_TESTS)
\ No newline at end of file
diff -Nru libsamplerate-0.1.7.orig/Win32/config.h libsamplerate-0.1.7/Win32/config.h
--- libsamplerate-0.1.7.orig/Win32/config.h	2009-02-14 12:42:49.000000000 +0100
+++ libsamplerate-0.1.7/Win32/config.h	2009-10-01 23:55:45.544966400 +0200
@@ -190,19 +190,4 @@
 #define STDC_HEADERS 1
 
 /* Version number of package */
-<<<<<<< diff from left file
-#define VERSION "0.1.4"
-
-
-
-/* Extra Win32 hacks. */
-
-/*
-**	Microsoft's compiler still does not support the 1999 ISO C Standard 
-**	which includes 'inline'.
-*/
-
-#define inline __inline
-========
 #define VERSION "0.1.7"
->>>>>>> diff from right file
