diff -Nru taglib-1.6.3.orig/taglib/ape/apefooter.cpp taglib-1.6.3/taglib/ape/apefooter.cpp
--- taglib-1.6.3.orig/taglib/ape/apefooter.cpp	2008-02-04 16:11:56.000000000 +0100
+++ taglib-1.6.3/taglib/ape/apefooter.cpp	2010-12-13 00:24:39.242623300 +0100
@@ -189,7 +189,7 @@
 
   // Read the flags
 
-  std::bitset<32> flags(data.mid(20, 4).toUInt(false));
+  std::bitset<32> flags(static_cast<__int32>(data.mid(20, 4).toUInt(false)));
 
   d->headerPresent = flags[31];
   d->footerPresent = !flags[30];
diff -Nru taglib-1.6.3.orig/taglib/mpc/mpcproperties.cpp taglib-1.6.3/taglib/mpc/mpcproperties.cpp
--- taglib-1.6.3.orig/taglib/mpc/mpcproperties.cpp	2007-10-10 20:22:58.000000000 +0200
+++ taglib-1.6.3/taglib/mpc/mpcproperties.cpp	2010-12-13 00:23:56.357170400 +0100
@@ -113,7 +113,7 @@
   if(d->version >= 7) {
     frames = d->data.mid(4, 4).toUInt(false);
 
-    std::bitset<32> flags = d->data.mid(8, 4).toUInt(false);
+    std::bitset<32> flags = static_cast<__int32>(d->data.mid(8, 4).toUInt(false));
     d->sampleRate = sftable[flags[17] * 2 + flags[16]];
     d->channels = 2;
   }
diff -Nru taglib-1.6.3.orig/taglib/mpeg/mpegheader.cpp taglib-1.6.3/taglib/mpeg/mpegheader.cpp
--- taglib-1.6.3.orig/taglib/mpeg/mpegheader.cpp	2009-01-16 06:16:48.000000000 +0100
+++ taglib-1.6.3/taglib/mpeg/mpegheader.cpp	2010-12-13 00:22:15.084378000 +0100
@@ -168,7 +168,7 @@
     return;
   }
 
-  std::bitset<32> flags(data.toUInt());
+  std::bitset<32> flags(static_cast<__int32>(data.toUInt()));
 
   // Check for the second byte's part of the MPEG synch
 
