diff -Nru -x '*~' picmi-4.10.2.orig/src/CMakeLists.txt picmi-4.10.2/src/CMakeLists.txt
--- picmi-4.10.2.orig/src/CMakeLists.txt	2013-03-01 08:23:49.517552000 +0100
+++ picmi-4.10.2/src/CMakeLists.txt	2013-06-02 10:46:46.598799200 +0200
@@ -13,7 +13,7 @@
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${KDE4_ENABLE_EXCEPTIONS}")
 
 if (CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-    add_definitions("-Wextra -pedantic")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wextra -pedantic")
 endif ()
 
 include(CheckIncludeFileCXX)
@@ -42,6 +42,7 @@
     systemexception.cpp
 )
 
+kde4_add_app_icon(picmi_SRCS "${CMAKE_SOURCE_DIR}/icons/hi*-apps-picmi.png")
 kde4_add_executable(picmi
     ${picmi_SRCS}
 )
