diff -Nru libxml2-2.7.8.orig/win32/Makefile.mingw libxml2-2.7.8/win32/Makefile.mingw
--- libxml2-2.7.8.orig/win32/Makefile.mingw     2010-10-12 08:25:32.000000000 +0200
+++ libxml2-2.7.8/win32/Makefile.mingw  2011-08-01 14:13:20.487878300 +0200
@@ -69,17 +69,17 @@
 LIBS = 
 ifeq ($(WITH_FTP),1)
 CFLAGS += -D_WINSOCKAPI_
-LIBS += -lwsock32
+LIBS += -lwsock32 -lws2_32
 endif 
 ifeq ($(WITH_HTTP),1)
 CFLAGS += -D_WINSOCKAPI_
-LIBS += -lwsock32
+LIBS += -lwsock32 -lws2_32
 endif 
 ifeq ($(WITH_ICONV),1)
 LIBS += -liconv
 endif 
 ifeq ($(WITH_ZLIB),1)
-LIBS += -lzdll
+LIBS += -lz
 endif
 ifeq ($(WITH_THREADS),posix)
 LIBS += -lpthreadGC
diff -Nru libxml2-2.7.8.orig/win32/Makefile.msvc libxml2-2.7.8/win32/Makefile.msvc
--- libxml2-2.7.8.orig/win32/Makefile.msvc      2010-11-04 15:58:43.000000000 +0100
+++ libxml2-2.7.8/win32/Makefile.msvc   2011-08-01 14:17:02.919600600 +0200
@@ -71,11 +71,11 @@
 !if "$(WITH_ICONV)" == "1"
 LIBS = $(LIBS) iconv.lib
 !endif 
-+!if "$(WITH_ICU)" == "1"
-+LIBS = $(LIBS) icu.lib
-+!endif
+!if "$(WITH_ICU)" == "1"
+LIBS = $(LIBS) icu.lib
+!endif
 !if "$(WITH_ZLIB)" == "1"
-LIBS = $(LIBS) zdll.lib
+LIBS = $(LIBS) zlib.lib
 !endif
 !if "$(WITH_THREADS)" == "posix"
 LIBS = $(LIBS) pthreadVC.lib
@@ -94,7 +94,7 @@
 LDFLAGS = $(LDFLAGS) /DEBUG
 !else
 CFLAGS = $(CFLAGS) /D "NDEBUG" /O2 
-LDFLAGS = $(LDFLAGS) /OPT:NOWIN98
+LDFLAGS = $(LDFLAGS)
 !endif
 
 # Libxml object files.