From 1603677514d93beea3ac6b401c6c37331ce155e3 Mon Sep 17 00:00:00 2001
From: Patrick von Reth <vonreth@kde.org>
Date: Sat, 7 Dec 2013 00:35:40 +0100
Subject: [PATCH] fixed windows x64 build

---
 kleopatra/kleopatraapplication.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kleopatra/kleopatraapplication.cpp b/kleopatra/kleopatraapplication.cpp
index 02c8a42..09a28e5 100644
--- a/kleopatra/kleopatraapplication.cpp
+++ b/kleopatra/kleopatraapplication.cpp
@@ -296,7 +296,7 @@ int KleopatraApplication::newInstance() {
         // Check for Parent Window id
         WId parentId = 0;
         if ( args->isSet( "parent-windowid" ) ) {
-            parentId = args->getOption( "parent-windowid" ).toUInt();
+            parentId = reinterpret_cast<WId>(args->getOption( "parent-windowid" ).toUInt());
         }
 
         // Search for local keys
-- 
1.8.3.msysgit.0

