diff -Nru -x '*~' kdepim-4.12.0.orig/agents/sendlateragent/sendlateragent.cpp kdepim-4.12.0/agents/sendlateragent/sendlateragent.cpp
--- kdepim-4.12.0.orig/agents/sendlateragent/sendlateragent.cpp	2013-12-11 21:11:23.000000000 +0100
+++ kdepim-4.12.0/agents/sendlateragent/sendlateragent.cpp	2014-01-07 17:24:58.582462400 +0100
@@ -108,7 +108,7 @@
 
 void SendLaterAgent::configure( WId windowId )
 {
-    showConfigureDialog(windowId);
+    showConfigureDialog((qlonglong)windowId);
 }
 
 void SendLaterAgent::slotSendNow(Akonadi::Item::Id id)
diff -Nru -x '*~' kdepim-4.12.0.orig/messageviewer/viewer/urlhandlermanager.cpp kdepim-4.12.0/messageviewer/viewer/urlhandlermanager.cpp
--- kdepim-4.12.0.orig/messageviewer/viewer/urlhandlermanager.cpp	2013-12-11 21:11:23.000000000 +0100
+++ kdepim-4.12.0/messageviewer/viewer/urlhandlermanager.cpp	2014-01-07 16:10:11.875582100 +0100
@@ -641,7 +641,7 @@
                           displayName, libName, keyId ) )
       return false;
     QStringList lst;
-    lst << QLatin1String("-parent-windowid") << QString::number( w->viewer()->mainWindow()->winId() ) << QLatin1String("-query") << keyId;
+    lst << QLatin1String("-parent-windowid") << QString::number((qlonglong)w->viewer()->mainWindow()->winId() ) << QLatin1String("-query") << keyId;
     if ( !QProcess::startDetached( QLatin1String("kleopatra"),lst) )
       KMessageBox::error( w->mMainWindow, i18n("Could not start certificate manager. "
                                   "Please check your installation."),
