diff -Nru kdepim-4.7.0.orig/akonadiconsole/CMakeLists.txt kdepim-4.7.0/akonadiconsole/CMakeLists.txt
--- kdepim-4.7.0.orig/akonadiconsole/CMakeLists.txt	2011-05-21 01:10:45.000000000 +0200
+++ kdepim-4.7.0/akonadiconsole/CMakeLists.txt	2011-08-27 13:04:05.296740400 +0200
@@ -81,7 +81,7 @@
 
 kde4_add_kcfg_files(akonadiconsole_bin_SRCS settings.kcfgc)
 
-kde4_add_app_icon(akonadiconsole_bin_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/icons/hi*-apps-akonadiconsole.png")
+kde4_add_app_icon(akonadiconsole_bin_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/icons/hi*-app-akonadiconsole.png")
 
 kde4_add_executable(akonadiconsole_bin ${akonadiconsole_bin_SRCS})
 set_target_properties(akonadiconsole_bin PROPERTIES OUTPUT_NAME akonadiconsole)
diff -Nru kdepim-4.7.0.orig/akregator/src/CMakeLists.txt kdepim-4.7.0/akregator/src/CMakeLists.txt
--- kdepim-4.7.0.orig/akregator/src/CMakeLists.txt	2011-06-22 16:45:19.000000000 +0200
+++ kdepim-4.7.0/akregator/src/CMakeLists.txt	2011-08-27 13:03:36.085069600 +0200
@@ -13,7 +13,7 @@
 
 set(akregator_SRCS main.cpp mainwindow.cpp )
 
-kde4_add_app_icon(akregator_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/icons/hi*-apps-akregator.png")
+kde4_add_app_icon(akregator_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/icons/hi*-app-akregator.png")
 
 kde4_add_executable(akregator ${akregator_SRCS})
 
diff -Nru kdepim-4.7.0.orig/kalarm/autostart/CMakeLists.txt kdepim-4.7.0/kalarm/autostart/CMakeLists.txt
--- kdepim-4.7.0.orig/kalarm/autostart/CMakeLists.txt	2011-05-21 01:10:45.000000000 +0200
+++ kdepim-4.7.0/kalarm/autostart/CMakeLists.txt	2011-08-27 13:06:31.952128600 +0200
@@ -6,7 +6,7 @@
 	autostart.cpp 
 )
 
-kde4_add_app_icon(kalarmautostart_SRCS "../ox*-app-kalarm.png")
+kde4_add_app_icon(kalarmautostart_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../appicons/ox*-app-kalarm.png")
 
 kde4_add_executable(kalarmautostart ${kalarmautostart_SRCS})
 
diff -Nru kdepim-4.7.0.orig/knode/CMakeLists.txt kdepim-4.7.0/knode/CMakeLists.txt
--- kdepim-4.7.0.orig/knode/CMakeLists.txt	2011-05-21 01:10:46.000000000 +0200
+++ kdepim-4.7.0/knode/CMakeLists.txt	2011-08-27 13:02:52.491576100 +0200
@@ -159,7 +159,7 @@
 
 set(knode_SRCS knode.cpp knapplication.cpp main.cpp )
 
-kde4_add_app_icon(knode_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/icons/hi*-app-knode.png")
+kde4_add_app_icon(knode_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/icons/hi*-apps-knode.png")
 
 kde4_add_executable(knode ${knode_SRCS})
 
diff -Nru kdepim-4.7.0.orig/korganizer/korgac/CMakeLists.txt kdepim-4.7.0/korganizer/korgac/CMakeLists.txt
--- kdepim-4.7.0.orig/korganizer/korgac/CMakeLists.txt	2011-05-21 01:10:46.000000000 +0200
+++ kdepim-4.7.0/korganizer/korgac/CMakeLists.txt	2011-08-27 13:00:51.614662400 +0200
@@ -43,7 +43,7 @@
   add_definitions(-DSERIALIZER_PLUGIN_STATIC)
 endif (RUNTIME_PLUGINS_STATIC)
 
-kde4_add_app_icon(korgac_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/pixmap/hi*-apps-korgac.png")
+kde4_add_app_icon(korgac_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/pixmap/hi*-apps-korganizer.png")
 
 if(KORGAC_AKONADI_AGENT)
   if(NOT WINCE)
