diff --git a/kontact/src/mainwindow.cpp b/kontact/src/mainwindow.cpp
index 6e8f09d..49f70b1 100644
--- a/kontact/src/mainwindow.cpp
+++ b/kontact/src/mainwindow.cpp
@@ -391,12 +391,12 @@ void MainWindow::paintAboutScreen( const QString &msg )
 {
   QString location = KStandardDirs::locate( "data", QLatin1String("kontact/about/main.html") );
   QString content = QLatin1String(KPIMUtils::kFileToByteArray( location ));
-  content = content.arg( QLatin1String("file:") + KStandardDirs::locate(
+  content = content.arg( QLatin1String("file:/") + KStandardDirs::locate(
                            "data", QLatin1String("kdeui/about/kde_infopage.css") ) );
   if ( QApplication::isRightToLeft() ) {
     content =
       content.arg( QLatin1String("@import \"%1\";") ).
-              arg( QLatin1String("file:") + KStandardDirs::locate(
+              arg( QLatin1String("file:/") + KStandardDirs::locate(
                      "data", QLatin1String("kdeui/about/kde_infopage_rtl.css") ) );
   } else {
     content = content.arg( QString() );
@@ -1253,21 +1253,21 @@ QString MainWindow::introductionString()
     subs( QLatin1String("exec:/help?kontact") ).
     subs( iconSize ).
     subs( iconSize ).
-    subs( QLatin1String("file:") + handbook_icon_path ).
+    subs( QLatin1String("file:/") + handbook_icon_path ).
     subs( QLatin1String("exec:/help?kontact") ).
     subs( i18nc( "@item:intext", "Read Manual" ) ).
     subs( i18nc( "@item:intext", "Learn more about Kontact and its components" ) ).
     subs( QLatin1String("http://kontact.org") ).
     subs( iconSize ).
     subs( iconSize ).
-    subs( QLatin1String("file:") + html_icon_path ).
+    subs( QLatin1String("file:/") + html_icon_path ).
     subs( QLatin1String("http://kontact.org") ).
     subs( i18nc( "@item:intext", "Visit Kontact Website" ) ).
     subs( i18nc( "@item:intext", "Access online resources and tutorials" ) ).
     subs( QLatin1String("exec:/accountwizard") ).
     subs( iconSize ).
     subs( iconSize ).
-    subs( QLatin1String("file:") + wizard_icon_path ).
+    subs( QLatin1String("file:/") + wizard_icon_path ).
     subs( QLatin1String("exec:/accountwizard") ).
     subs( i18nc( "@item:intext", "Setup your Accounts" ) ).
     subs( i18nc( "@item:intext", "Prepare Kontact for use" ) ).
