Index: mobile/contacts/contact-editor.qml
===================================================================
--- mobile/contacts/contact-editor.qml	(revision 1190370)
+++ mobile/contacts/contact-editor.qml	(working copy)
@@ -95,6 +95,7 @@
         }
       ]
     }
+ /*
     SlideoutPanel {
       anchors.fill: parent
       id: cryptoPanel
@@ -119,6 +120,7 @@
         }
       ]
     }
+    */
     SlideoutPanel {
       anchors.fill: parent
       id: morePanel
