diff --git a/importwizard/CMakeLists.txt b/importwizard/CMakeLists.txt
index 3af4c52..3e941e8 100644
--- a/importwizard/CMakeLists.txt
+++ b/importwizard/CMakeLists.txt
@@ -78,6 +78,7 @@ soprano_add_ontology(importwizard_SRCS
     "trig"
   )
 
+kde4_add_app_icon(importwizard_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/icons/hi*-app-kontact-import-wizard.png")
 
 kde4_add_executable(importwizard ${importwizard_SRCS})
 
diff --git a/kjots/CMakeLists.txt b/kjots/CMakeLists.txt
index cb4228e..0624613 100644
--- a/kjots/CMakeLists.txt
+++ b/kjots/CMakeLists.txt
@@ -106,6 +106,7 @@ set(kjots_SRCS
    KJotsMain.cpp
    ${kjots_common_SRCS}
 )
+kde4_add_app_icon(kjots_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/icons/hi*-apps-kjots.png")
 
 kde4_add_executable(kjots ${kjots_SRCS})
 
diff --git a/ktnef/CMakeLists.txt b/ktnef/CMakeLists.txt
index 6aa395a..819319e 100644
--- a/ktnef/CMakeLists.txt
+++ b/ktnef/CMakeLists.txt
@@ -15,11 +15,14 @@ set(ktnef_SRCS
   qwmf.cpp
 )
 
+kde4_add_app_icon(ktnef_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/pics/hi*-app-ktnef.png")
+
 kde4_add_ui_files(ktnef_SRCS attachpropertydialogbase.ui)
 
 kde4_add_executable(ktnef ${ktnef_SRCS})
 target_link_libraries(ktnef ${KDEPIMLIBS_KTNEF_LIBS} ${KDE4_KIO_LIBS})
 
+
 install(TARGETS ktnef ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 install(PROGRAMS ktnef.desktop DESTINATION ${XDG_APPS_INSTALL_DIR})
diff --git a/mobile/tasks/CMakeLists.txt b/mobile/tasks/CMakeLists.txt
index 6aa395a..819319e 100644
--- a/mobile/tasks/CMakeLists.txt
+++ b/mobile/tasks/CMakeLists.txt
@@ -31,7 +31,7 @@
   add_definitions(-DKCALCORE_SERIALIZER_PLUGIN_STATIC)
 endif (RUNTIME_PLUGINS_STATIC)
 
-kde4_add_app_icon(mobile_tasks_srcs "${CMAKE_CURRENT_SOURCE_DIR}/hi*-apps-tasks-mobile.png")
+kde4_add_app_icon(mobile_tasks_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/hi*-apps-tasks-mobile.png")
 
 kde4_add_executable(tasks-mobile ${mobile_tasks_SRCS} )
 if (WINCE)
