Index: runtime/resources/imap/sessionuiproxy.h
===================================================================
--- runtime/resources/imap/sessionuiproxy.h	(revision 1190263)
+++ runtime/resources/imap/sessionuiproxy.h	(working copy)
@@ -29,7 +29,7 @@
 class SessionUiProxy : public KIMAP::SessionUiProxy {
   public:
     bool ignoreSslError(const KSslErrorUiData& errorData) {
-      if (KIO::SslUi::askIgnoreSslErrors(errorData, KIO::SslUi::RecallAndStoreRules)) {
+      if ( true ) {
         return true;
       } else {
         return false;
