Index: C:/dev/kde/svn/tags/kdesupport-for-4.5/akonadi/qsqlite/src/sqlite_blocking.cpp
===================================================================
--- C:/dev/kde/svn/tags/kdesupport-for-4.5/akonadi/qsqlite/src/sqlite_blocking.cpp	(revisi¢n: 1205909)
+++ C:/dev/kde/svn/tags/kdesupport-for-4.5/akonadi/qsqlite/src/sqlite_blocking.cpp	(copia de trabajo)
@@ -1,8 +1,13 @@
 #include "sqlite_blocking.h"
 
 #include <sqlite3.h>
-#include <unistd.h>
 
+#ifdef _MSC_VER
+#  include <windows.h>
+#else
+#  include <unistd.h>
+#endif
+
 #include "qdebug.h"
 #include "qthread.h"
 
@@ -23,7 +28,11 @@
     qDebug() << debugString() + "sqlite3_blocking_step: Entering while loop";
 
   while( rc == SQLITE_BUSY ) {
+  #ifdef _MSC_VER
+    Sleep(5);
+  #else
     usleep(5000);
+  #endif
     sqlite3_reset( pStmt );
     rc = sqlite3_step( pStmt );
 
@@ -47,7 +56,11 @@
     qDebug() << debugString() + "sqlite3_blocking_prepare16_v2: Entering while loop";
 
   while( rc == SQLITE_BUSY ) {
-    usleep(500000);
+    #ifdef _MSC_VER
+    Sleep(50);
+  #else
+    usleep(50000);
+  #endif
     rc = sqlite3_prepare16_v2( db, zSql, nSql, ppStmt, pzTail );
 
     if ( rc != SQLITE_BUSY ) {
Index: C:/dev/kde/svn/tags/kdesupport-for-4.5/akonadi/qsqlite/CMakeLists.txt
===================================================================
--- C:/dev/kde/svn/tags/kdesupport-for-4.5/akonadi/qsqlite/CMakeLists.txt	(revisi¢n: 1205909)
+++ C:/dev/kde/svn/tags/kdesupport-for-4.5/akonadi/qsqlite/CMakeLists.txt	(copia de trabajo)
@@ -37,5 +37,6 @@
 target_link_libraries(qsqlite3 ${QT_QTCORE_LIBRARY} ${QT_QTSQL_LIBRARY} ${SQLITE_LIBRARIES})
 
 INSTALL(TARGETS qsqlite3
+  RUNTIME DESTINATION ${QSQLITE_INSTALL_PREFIX}
   LIBRARY DESTINATION ${QSQLITE_INSTALL_PREFIX}
 )
Index: C:/dev/kde/svn/tags/kdesupport-for-4.5/akonadi/CMakeLists.txt
===================================================================
--- C:/dev/kde/svn/tags/kdesupport-for-4.5/akonadi/CMakeLists.txt	(revisi¢n: 1205909)
+++ C:/dev/kde/svn/tags/kdesupport-for-4.5/akonadi/CMakeLists.txt	(copia de trabajo)
@@ -114,9 +114,6 @@
 
 #### Boost ####
 if(MSVC)
-  # otherwise we get an undefined reference to
-  # boost::program_options::options_description::m_default_line_length
-  set(Boost_USE_STATIC_LIBS ON)
   set (_ENABLE_EXCEPTIONS -EHsc)
 endif(MSVC)
 
@@ -128,6 +125,7 @@
 # should be handled by FindBoost.cmake -> cmake bug #8335
 if(NOT Boost_USE_STATIC_LIBS)
   add_definitions(-DBOOST_DYN_LINK)
+  add_definitions(-DBOOST_PROGRAM_OPTIONS_DYN_LINK)
 endif(NOT Boost_USE_STATIC_LIBS)
 
 #### Sqlite ####
