diff -Nru -x '*~' nepomuk-core-4.10.0.orig/services/fileindexer/indexer/extractorplugin.h nepomuk-core-4.10.0/services/fileindexer/indexer/extractorplugin.h
--- nepomuk-core-4.10.0.orig/services/fileindexer/indexer/extractorplugin.h	2013-01-23 22:44:47.000000000 +0100
+++ nepomuk-core-4.10.0/services/fileindexer/indexer/extractorplugin.h	2013-02-17 16:40:00.838426100 +0100
@@ -44,7 +44,7 @@
      *
      * \author Vishesh Handa <me@vhanda.in>
      */
-    class NEPOMUK_EXPORT ExtractorPlugin : public QObject
+    class ExtractorPlugin : public QObject
     {
         Q_OBJECT
     public:
