diff --git a/cleaner/CMakeLists.txt b/cleaner/CMakeLists.txt
index af7f94b..a507cf5 100644
--- a/cleaner/CMakeLists.txt
+++ b/cleaner/CMakeLists.txt
@@ -9,6 +9,8 @@ set( cleaner_SRCS
   mainwindow.cpp
 )
 
+kde4_add_app_icon( cleaner_SRCS "${KDE4_INSTALL_DIR}/share/icons/oxygen/*/apps/nepomuk.png")
+
 kde4_add_executable( nepomukcleaner ${cleaner_SRCS} )
 
 target_link_libraries( nepomukcleaner
diff --git a/services/storage/backup/gui/CMakeLists.txt b/services/storage/backup/gui/CMakeLists.txt
index 45a4618..be58961 100644
--- a/services/storage/backup/gui/CMakeLists.txt
+++ b/services/storage/backup/gui/CMakeLists.txt
@@ -37,6 +37,7 @@ kde4_add_ui_files( NepomukBackup_GUI_SRCS
   errorpage.ui
 )
 
+kde4_add_app_icon(NepomukBackup_GUI_SRCS "${KDE4_INSTALL_DIR}/share/icons/oxygen/*/apps/nepomuk.png")
 kde4_add_executable(nepomukbackup ${NepomukBackup_GUI_SRCS})
 
 target_link_libraries(nepomukbackup
