diff -Nru mysql-5.5.20.orig/cmake/ssl.cmake mysql-5.5.20/cmake/ssl.cmake
--- mysql-5.5.20.orig/cmake/ssl.cmake	2011-12-16 20:52:07.000000000 +0100
+++ mysql-5.5.20/cmake/ssl.cmake	2012-01-16 20:04:14.590625000 +0100
@@ -66,9 +66,10 @@
     # Check for system library
     SET(OPENSSL_FIND_QUIETLY TRUE)
     INCLUDE(FindOpenSSL)
-    FIND_LIBRARY(CRYPTO_LIBRARY crypto)
+    FIND_LIBRARY(CRYPTO_LIBRARY NAMES crypto libeay32)
     MARK_AS_ADVANCED(CRYPTO_LIBRARY)
     INCLUDE(CheckSymbolExists)
+    SET(CMAKE_REQUIRED_INCLUDES ${OPENSSL_INCLUDE_DIR})
     CHECK_SYMBOL_EXISTS(SHA512_DIGEST_LENGTH "openssl/sha.h" 
                         HAVE_SHA512_DIGEST_LENGTH)
     IF(OPENSSL_FOUND AND CRYPTO_LIBRARY AND HAVE_SHA512_DIGEST_LENGTH)
diff -Nru mysql-5.5.20.orig/CMakeLists.txt mysql-5.5.20/CMakeLists.txt
--- mysql-5.5.20.orig/CMakeLists.txt	2011-12-16 20:52:07.000000000 +0100
+++ mysql-5.5.20/CMakeLists.txt	2012-01-16 18:35:12.661401400 +0100
@@ -221,13 +221,13 @@
 
 
 # Set commonly used variables
-IF(WIN32)
-  SET(DEFAULT_MYSQL_HOME "C:/Program Files/MySQL/MySQL Server ${MYSQL_BASE_VERSION}" )
-  SET(SHAREDIR share)
-ELSE()
+# IF(WIN32)
+#   SET(DEFAULT_MYSQL_HOME "C:/Program Files/MySQL/MySQL Server ${MYSQL_BASE_VERSION}" )
+#   SET(SHAREDIR share)
+# ELSE()
   SET(DEFAULT_MYSQL_HOME ${CMAKE_INSTALL_PREFIX})
-  SET(SHAREDIR ${DEFAULT_MYSQL_HOME}/${INSTALL_MYSQLSHAREDIR})
-ENDIF()
+  SET(SHAREDIR share)
+# ENDIF()
 
 SET(DEFAULT_BASEDIR "${DEFAULT_MYSQL_HOME}")
 IF(INSTALL_MYSQLDATADIR MATCHES "^/.*")
