diff -Nru -x '*~' libkcddb-4.10.1.orig/libkcddb/musicbrainz/musicbrainzlookup.cpp libkcddb-4.10.1/libkcddb/musicbrainz/musicbrainzlookup.cpp
--- libkcddb-4.10.1.orig/libkcddb/musicbrainz/musicbrainzlookup.cpp	2013-03-01 08:08:12.253637000 +0100
+++ libkcddb-4.10.1/libkcddb/musicbrainz/musicbrainzlookup.cpp	2013-03-10 00:09:51.195475700 +0100
@@ -37,6 +37,7 @@
 #include <musicbrainz5/ArtistCredit.h>
 #include <musicbrainz5/Artist.h>
 #include <musicbrainz5/NameCredit.h>
+#include <musicbrainz5/SecondaryType.h>
 
 namespace KCDDB
 {
