diff -Nru -x '*~' calligra-2.7.90.orig/CMakeLists.txt calligra-2.7.90/CMakeLists.txt
--- calligra-2.7.90.orig/CMakeLists.txt	2013-12-01 16:24:21.000000000 -0200
+++ calligra-2.7.90/CMakeLists.txt	2013-12-31 18:20:37.171479600 -0200
@@ -864,6 +864,25 @@
     ${CMAKE_SOURCE_DIR}/filters/libodfreader
 )
 
+
+#######################################################
+####  Utility macro to remove duplicated includes  ####
+####                                               ####
+####  On Windows duplicated include path may       ####
+####  cause too long command line for the compiler ####
+####  and also for moc.                            ####
+####                                               ####
+####  Other platforms may also benefit from this.  ####
+####                                               ####
+####  CMake >= 2.8.8 do not do this automatically. ####
+#######################################################
+
+macro(remove_duplicated_include_directories)
+    get_directory_property(dupdirs DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} INCLUDE_DIRECTORIES)
+    list(REMOVE_DUPLICATES dupdirs)
+    set_directory_properties(PROPERTIES INCLUDE_DIRECTORIES "${dupdirs}")
+endmacro()
+
  #########################################
 ##########################################
 ## Detect which products can be compile ##
diff -Nru -x '*~' calligra-2.7.90.orig/krita/image/CMakeLists.txt calligra-2.7.90/krita/image/CMakeLists.txt
--- calligra-2.7.90.orig/krita/image/CMakeLists.txt	2013-12-01 16:24:22.000000000 -0200
+++ calligra-2.7.90/krita/image/CMakeLists.txt	2013-12-31 19:00:08.823063500 -0200
@@ -51,6 +51,8 @@
                      ${KOMAIN_INCLUDES}
 )
 
+remove_duplicated_include_directories()
+
 if(HAVE_VC)
   include_directories(${Vc_INCLUDE_DIR})
   ko_compile_for_all_implementations(__per_arch_circle_mask_generator_objs kis_brush_mask_applicator_factories.cpp)
diff -Nru -x '*~' calligra-2.7.90.orig/krita/ui/CMakeLists.txt calligra-2.7.90/krita/ui/CMakeLists.txt
--- calligra-2.7.90.orig/krita/ui/CMakeLists.txt	2013-12-01 16:24:25.000000000 -0200
+++ calligra-2.7.90/krita/ui/CMakeLists.txt	2013-12-31 18:28:36.129919400 -0200
@@ -18,6 +18,8 @@
     ${CMAKE_SOURCE_DIR}/krita/image/metadata
     ${EXIV2_INCLUDE_DIR} )
 
+remove_duplicated_include_directories()
+
 add_subdirectory( tests )
 
 
diff -Nru -x '*~' calligra-2.7.90.orig/sheets/shape/CMakeLists.txt calligra-2.7.90/sheets/shape/CMakeLists.txt
--- calligra-2.7.90.orig/sheets/shape/CMakeLists.txt	2013-12-01 16:24:26.000000000 -0200
+++ calligra-2.7.90/sheets/shape/CMakeLists.txt	2013-12-31 18:40:51.050343100 -0200
@@ -6,6 +6,8 @@
     ${TEXTLAYOUT_INCLUDES}
 	)
 
+remove_duplicated_include_directories()
+
 ########### Flake Plugin library ###############
 
 SET (tableshape_LIB_SRCS
diff -Nru -x '*~' calligra-2.7.90.orig/stage/part/CMakeLists.txt calligra-2.7.90/stage/part/CMakeLists.txt
--- calligra-2.7.90.orig/stage/part/CMakeLists.txt	2013-12-01 16:24:26.000000000 -0200
+++ calligra-2.7.90/stage/part/CMakeLists.txt	2013-12-31 18:21:42.514499900 -0200
@@ -2,6 +2,8 @@
 
 include_directories( ${KOMAIN_INCLUDES} ${KOPAGEAPP_INCLUDES} ${Boost_INCLUDE_DIR})
 
+remove_duplicated_include_directories()
+
 if(KDE4_BUILD_TESTS)
     # only with this definition will the STAGE_TEST_EXPORT macro do something
     add_definitions(-DCOMPILING_TESTS)
