diff -Nru -x '*~' calligra-2.7.90.orig/plugins/formulashape/CMakeLists.txt calligra-2.7.90/plugins/formulashape/CMakeLists.txt
--- calligra-2.7.90.orig/plugins/formulashape/CMakeLists.txt    2013-12-01 16:24:25.000000000 -0200
+++ calligra-2.7.90/plugins/formulashape/CMakeLists.txt 2014-01-15 00:16:39.341657700 -0200
@@ -91,7 +91,7 @@
    FormulaPart.cpp
 )
 
-if (NOT MSVC AND NOT (WIN32 AND "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel"))
+if (NOT WIN32)
 set (formulashape_PART_SRCS
    ${formulashape_PART_SRCS}
    KoM2MMLForumulaTool.cpp
@@ -99,7 +99,7 @@
    itexToMML/lex.yy.c
    itexToMML/y.tab.c
 )
-endif (NOT MSVC AND NOT (WIN32 AND "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel"))
+endif (NOT WIN32)
 
 kde4_add_ui_files( formulashape_PART_SRCS
     FormulaToolWidget.ui
diff -Nru -x '*~' calligra-2.7.90.orig/plugins/formulashape/KoFormulaShapePlugin.cpp calligra-2.7.90/plugins/formulashape/KoFormulaShapePlugin.cpp
--- calligra-2.7.90.orig/plugins/formulashape/KoFormulaShapePlugin.cpp  2013-12-01 16:24:25.000000000 -0200
+++ calligra-2.7.90/plugins/formulashape/KoFormulaShapePlugin.cpp       2014-01-15 00:23:10.895276900 -0200
@@ -25,7 +25,7 @@
 
 #include <kpluginfactory.h>
 
-#ifndef _MSC_VER
+#ifndef _WIN32
 #include "KoM2MMLForumulaTool.h"
 #endif
 
@@ -36,7 +36,7 @@
                     : QObject( parent )
 {
     KoToolRegistry::instance()->add( new KoFormulaToolFactory() );
-#ifndef _MSC_VER
+#ifndef _WIN32
     KoToolRegistry::instance()->add( new KoM2MMLFormulaToolFactory());
 #endif
     KoShapeRegistry::instance()->add( new KoFormulaShapeFactory() );
