diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' p7zip_16.02.orig/CPP/7zip/CMAKE/CMakeLists.txt p7zip_16.02/CPP/7zip/CMAKE/CMakeLists.txt
--- p7zip_16.02.orig/CPP/7zip/CMAKE/CMakeLists.txt	2016-02-17 07:16:08.000000000 +0100
+++ p7zip_16.02/CPP/7zip/CMAKE/CMakeLists.txt	2018-08-16 20:41:47.765554600 +0200
@@ -21,7 +21,11 @@
 
 add_definitions(-DENV_HAVE_GCCVISIBILITYPATCH)
 
-SET(CMAKE_CXX_FLAGS "-fvisibility=hidden -fvisibility-inlines-hidden")
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=hidden -fvisibility-inlines-hidden")
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-c++11-narrowing")
+endif()
 
 add_subdirectory(7za)
 
@@ -45,3 +49,4 @@
 
 
 
+
