--- utils\NTMakefile.orig 2007-03-26 10:51:30.627125000 -0600
+++ utils\NTMakefile 2007-03-26 10:54:23.986500000 -0600
@@ -1,29 +1,39 @@
 !INCLUDE ..\win32\common.mak
 
-sasl_apps=saslpasswd2.exe sasldblistusers2.exe testsuite.exe pluginviewer.exe
-sasl_out=saslpasswd2.pdb sasldblistusers2.pdb testsuite.pdb pluginviewer.pdb
+sasl_apps=saslpasswd2.exe testsuite.exe pluginviewer.exe
+sasl_out=saslpasswd2.pdb testsuite.pdb pluginviewer.pdb
+!IF "$(SASLDB)" == "1"
+sasl_apps = $(sasl_apps) sasldblistusers2.exe
+sasl_out = $(sasl_apps) sasldblistusers2.pdb
+!ENDIF
 
 saslpwd_objs = saslpasswd.obj
 sasldblistusers_objs = sasldblistusers.obj
 testsuite_objs = testsuite.obj
 pluginviewer_objs = pluginviewer.obj
 
 all_objs = $(saslpwd_objs) $(sasldblistusers_objs) $(testsuite_objs) $(pluginviewer_objs)
 all_out = $(sasl_apps) $(sasl_out)
 
+!IF "$(SASLDB)" == "1"
 DB_FLAGS = /I $(DB_INCLUDE) /I "..\sasldb"
+!ENDIF
 CPPFLAGS = /I "..\win32\include" /I "." /I "..\include" $(DB_FLAGS) /D "WIN32" /D "_WINDOWS" /D "_MBCS" /D "_USRDLL"
 CPPFLAGS = $(CPPFLAGS) /DNEED_GETOPT
 
 
+!IF "$(SASLDB)" == "1"
 DB_LIBS=/libpath:$(DB_LIBPATH) $(DB_LIB)
+!ENDIF
 SASL_LIB=/libpath:"..\lib" libsasl.lib
+!IF "$(SASLDB)" == "1"
 SASL_DB_LIB=/libpath:"..\plugins" saslSASLDB.lib
+!ENDIF
 
 # EXTRA_LIBS is automatically included into LINK32EXE_FLAGS/LINK32DLL_FLAGS
 EXTRA_LIBS=$(SASL_LIB) 
 
 # Where to install files from this directory
 bindir = $(prefix)\bin
 
 
