diff -Nru -x '*~' rocs-4.10.1.orig/App/CMakeLists.txt rocs-4.10.1/App/CMakeLists.txt
--- rocs-4.10.1.orig/App/CMakeLists.txt	2013-03-01 07:51:29.444091000 +0100
+++ rocs-4.10.1/App/CMakeLists.txt	2013-03-05 20:10:41.115512800 +0100
@@ -87,7 +87,7 @@
 
 # needed to get icon on windows
 kde4_add_app_icon( rocs_SRCS
-    ${CMAKE_CURRENT_SOURCE_DIR}/../graphics/hi16-app-rocs.png
+    ${CMAKE_CURRENT_SOURCE_DIR}/../graphics/hi*-app-rocs.png
 )
 
 kde4_add_executable( rocs
diff -Nru -x '*~' rocs-4.10.1.orig/RocsCore/RocsCoreExport.h rocs-4.10.1/RocsCore/RocsCoreExport.h
--- rocs-4.10.1.orig/RocsCore/RocsCoreExport.h	2013-03-01 07:51:29.504090000 +0100
+++ rocs-4.10.1/RocsCore/RocsCoreExport.h	2013-03-06 00:01:37.811792000 +0100
@@ -24,7 +24,7 @@
 #include <kdemacros.h>
 
 #ifndef ROCSLIB_EXPORT
-# if defined(MAKE_ROCSLIB_LIB)
+# if defined(MAKE_ROCSCORE_LIB)
 // We are building this library
 #  define ROCSLIB_EXPORT KDE_EXPORT
 # else
diff -Nru -x '*~' rocs-4.10.1.orig/VisualEditor/VisualEditorExport.h rocs-4.10.1/VisualEditor/VisualEditorExport.h
--- rocs-4.10.1.orig/VisualEditor/VisualEditorExport.h	2013-03-01 07:51:29.524090000 +0100
+++ rocs-4.10.1/VisualEditor/VisualEditorExport.h	2013-03-06 00:03:54.336600800 +0100
@@ -23,7 +23,7 @@
 #include <kdemacros.h>
 
 #ifndef VISUALEDITOR_EXPORT
-# if defined(MAKE_VISUALEDITOR_LIB)
+# if defined(MAKE_ROCSVISUALEDITOR_LIB)
 // We are building this library
 #  define VISUALEDITOR_EXPORT KDE_EXPORT
 # else
