diff -Nru -x '*~' libalkimia-4.3.2.orig/libalkimia/CMakeLists.txt libalkimia-4.3.2/libalkimia/CMakeLists.txt
--- libalkimia-4.3.2.orig/libalkimia/CMakeLists.txt	2012-01-20 20:23:14.000000000 +0100
+++ libalkimia-4.3.2/libalkimia/CMakeLists.txt	2013-05-04 23:39:46.036255500 +0200
@@ -38,7 +38,7 @@
 
 include(KDE4Defaults)
 include_directories(${QDBUS_INCLUDE_DIRS} ${KDE4_INCLUDES})
-add_definitions(${QT_DEFINITIONS} ${KDE4_DEFINITIONS} -DMAKE_ALK_LIB)
+add_definitions(${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
 add_definitions(-DQT_USE_FAST_CONCATENATION -DQT_USE_FAST_OPERATOR_PLUS)
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${KDE4_ENABLE_EXCEPTIONS}")
@@ -64,7 +64,7 @@
  
 target_link_libraries(alkimia ${QT_QTCORE_LIBRARY} ${QT_QTDBUS_LIBRARY} ${GMP_LIBRARIES})
 
-set_target_properties(alkimia PROPERTIES VERSION ${ALKIMIA_LIB_VERSION} SOVERSION ${ALKIMIA_LIB_SOVERSION})
+set_target_properties(alkimia PROPERTIES VERSION ${ALKIMIA_LIB_VERSION} SOVERSION ${ALKIMIA_LIB_SOVERSION} DEFINE_SYMBOL MAKE_ALK_LIB)
 
 if (NOT WIN32)
    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/libalkimia.pc.in ${CMAKE_CURRENT_BINARY_DIR}/libalkimia.pc IMMEDIATE @ONLY)
