diff -Nru -x '*~' LibRaw-0.16.0-Beta1.orig/internal/dht_demosaic.cpp LibRaw-0.16.0-Beta1/internal/dht_demosaic.cpp
--- LibRaw-0.16.0-Beta1.orig/internal/dht_demosaic.cpp	2013-11-15 06:06:53.000000000 +0100
+++ LibRaw-0.16.0-Beta1/internal/dht_demosaic.cpp	2013-12-23 16:48:17.416235000 +0100
@@ -812,14 +812,14 @@
 
 void DHT::make_rb() {
 #if defined(LIBRAW_USE_OPENMP)
-#pragma omp barier
+#pragma omp barrier
 #pragma omp parallel for schedule(guided)
 #endif
 	for (int i = 0; i < libraw.imgdata.sizes.iheight; ++i) {
 		make_rbdiag(i);
 	}
 #if defined(LIBRAW_USE_OPENMP)
-#pragma omp barier
+#pragma omp barrier
 #pragma omp parallel for schedule(guided)
 #endif
 	for (int i = 0; i < libraw.imgdata.sizes.iheight; ++i) {
