diff --git a/plugins/generic/skg_monthly/grantlee_filters/CMakeLists.txt b/plugins/generic/skg_monthly/grantlee_filters/CMakeLists.txt
index cc436f4..6b350c7 100644
--- a/plugins/generic/skg_monthly/grantlee_filters/CMakeLists.txt
+++ b/plugins/generic/skg_monthly/grantlee_filters/CMakeLists.txt
@@ -28,8 +28,4 @@ GRANTLEE_ADD_PLUGIN(grantlee_skgfilters skggrantleefilters FILTERS skgdocumentfi
 TARGET_LINK_LIBRARIES(grantlee_skgfilters skgbasemodeler skgbankmodeler skgbasegui)
 
 ########### install files ###############
-IF(WIN32)
-  INSTALL(TARGETS grantlee_skgfilters ARCHIVE DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/grantlee/${Grantlee_VERSION_MAJOR}.${Grantlee_VERSION_MINOR} )
-ELSE(WIN32)
-  INSTALL(TARGETS grantlee_skgfilters LIBRARY DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/grantlee/${Grantlee_VERSION_MAJOR}.${Grantlee_VERSION_MINOR} )
-ENDIF(WIN32)
+INSTALL(TARGETS grantlee_skgfilters LIBRARY DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/grantlee/${Grantlee_VERSION_MAJOR}.${Grantlee_VERSION_MINOR} )
