diff -Nru redland-1.0.8.orig/cmake/config.cpp.in redland-1.0.8/cmake/config.cpp.in
--- redland-1.0.8.orig/cmake/config.cpp.in	1970-01-01 01:00:00.000000000 +0100
+++ redland-1.0.8/cmake/config.cpp.in	2009-09-24 02:48:28.850800000 +0200
@@ -0,0 +1,26 @@
+#include <stdio.h>
+#include <string.h>
+#include <@PROGRAMM@.h>
+
+int main(int argc, char **argv)
+{
+    if(argc != 2)
+        return 1;
+    if(strcmp(argv[1], "--version") == 0) {
+        fprintf(stdout, "%s", @PROGRAMM@_version_string);
+        return 0;
+    }
+    if(strcmp(argv[1], "--version-decimal") == 0) {
+        fprintf(stdout, "%d", @PROGRAMM@_version_decimal);
+        return 0;
+    }
+    if(strcmp(argv[1], "--cflags") == 0) {
+        fprintf(stdout, "\n");
+        return 0;
+    }
+    if(strcmp(argv[1], "--libs") == 0) {
+        fprintf(stdout, "\n");
+        return 0;
+    }
+    return 1;
+}
diff -Nru redland-1.0.8.orig/cmake/FindPCRE.cmake redland-1.0.8/cmake/FindPCRE.cmake
--- redland-1.0.8.orig/cmake/FindPCRE.cmake	1970-01-01 01:00:00.000000000 +0100
+++ redland-1.0.8/cmake/FindPCRE.cmake	2009-09-24 02:48:28.850800000 +0200
@@ -0,0 +1,44 @@
+# - Try to find the PCRE regular expression library
+# Once done this will define
+#
+#  PCRE_FOUND - system has the PCRE library
+#  PCRE_INCLUDE_DIR - the PCRE include directory
+#  PCRE_LIBRARIES - The libraries needed to use PCRE
+
+# Copyright (c) 2006, Alexander Neundorf, <neundorf@kde.org>
+#
+# Redistribution and use is allowed according to the terms of the BSD license.
+# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+
+
+if (PCRE_INCLUDE_DIR AND PCRE_PCREPOSIX_LIBRARY AND PCRE_PCRE_LIBRARY)
+  # Already in cache, be silent
+  set(PCRE_FIND_QUIETLY TRUE)
+endif (PCRE_INCLUDE_DIR AND PCRE_PCREPOSIX_LIBRARY AND PCRE_PCRE_LIBRARY)
+
+
+if (NOT WIN32)
+  # use pkg-config to get the directories and then use these values
+  # in the FIND_PATH() and FIND_LIBRARY() calls
+  find_package(PkgConfig)
+
+  pkg_check_modules(PC_PCRE libpcre)
+
+  set(PCRE_DEFINITIONS ${PC_PCRE_CFLAGS_OTHER})
+
+endif (NOT WIN32)
+
+find_path(PCRE_INCLUDE_DIR pcre.h
+          HINTS ${PC_PCRE_INCLUDEDIR} ${PC_PCRE_INCLUDE_DIRS}
+          PATH_SUFFIXES pcre)
+
+find_library(PCRE_PCRE_LIBRARY NAMES pcre HINTS ${PC_PCRE_LIBDIR} ${PC_PCRE_LIBRARY_DIRS})
+
+find_library(PCRE_PCREPOSIX_LIBRARY NAMES pcreposix HINTS ${PC_PCRE_LIBDIR} ${PC_PCRE_LIBRARY_DIRS})
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(PCRE DEFAULT_MSG PCRE_INCLUDE_DIR PCRE_PCRE_LIBRARY PCRE_PCREPOSIX_LIBRARY )
+
+set(PCRE_LIBRARIES ${PCRE_PCRE_LIBRARY} ${PCRE_PCREPOSIX_LIBRARY})
+
+mark_as_advanced(PCRE_INCLUDE_DIR PCRE_LIBRARIES PCRE_PCREPOSIX_LIBRARY PCRE_PCRE_LIBRARY)
diff -Nru redland-1.0.8.orig/cmake/FindSQLite3.cmake redland-1.0.8/cmake/FindSQLite3.cmake
--- redland-1.0.8.orig/cmake/FindSQLite3.cmake	1970-01-01 01:00:00.000000000 +0100
+++ redland-1.0.8/cmake/FindSQLite3.cmake	2009-09-24 02:48:28.850800000 +0200
@@ -0,0 +1,19 @@
+# - Try to find the SQLite3 library
+# Once done this will define
+#
+#  SQLITE3_FOUND - system has the SQLite3 library
+#  SQLITE3_INCLUDE_DIR - the SQLite3 include directory
+#  SQLITE3_LIBRARIES - The libraries needed to use SQLite3
+
+if (SQLITE3_INCLUDE_DIR AND SQLITE3_LIBRARIES)
+  # Already in cache, be silent
+  set(SQLITE3_FIND_QUIETLY TRUE)
+endif (SQLITE3_INCLUDE_DIR AND SQLITE3_LIBRARIES)
+
+find_path(SQLITE3_INCLUDE_DIR sqlite3.h)
+find_library(SQLITE3_LIBRARIES NAMES sqlite3)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(SQLite3 DEFAULT_MSG SQLITE3_INCLUDE_DIR SQLITE3_LIBRARIES)
+
+mark_as_advanced(SQLITE3_INCLUDE_DIR SQLITE3_LIBRARIES)
diff -Nru redland-1.0.8.orig/CMakeLists.txt redland-1.0.8/CMakeLists.txt
--- redland-1.0.8.orig/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ redland-1.0.8/CMakeLists.txt	2009-09-24 02:48:28.850800000 +0200
@@ -0,0 +1,11 @@
+project(redland)
+cmake_minimum_required(VERSION 2.6.2)
+set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
+
+if(MSVC)
+  add_definitions(-D_CRT_SECURE_NO_DEPRECATE -D_CRT_NONSTDC_NO_DEPRECATE)
+endif(MSVC)
+
+add_subdirectory(librdf)
+add_subdirectory(raptor)
+add_subdirectory(rasqal)
diff -Nru redland-1.0.8.orig/librdf/CMakeLists.txt redland-1.0.8/librdf/CMakeLists.txt
--- redland-1.0.8.orig/librdf/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ redland-1.0.8/librdf/CMakeLists.txt	2009-09-24 02:48:28.866400000 +0200
@@ -0,0 +1,120 @@
+project(librdf)
+
+find_package(SQLite3 REQUIRED)
+find_package(OpenSSL REQUIRED)
+include_directories(${SQLITE3_INCLUDE_DIR})
+include_directories(${OPENSSL_INCLUDE_DIR})
+
+set(librdf_SRCS
+    rdf_avltree.c
+    rdf_concepts.c
+    rdf_digest.c
+    rdf_digest_md5.c
+    rdf_digest_openssl.c
+    rdf_digest_sha1.c
+    rdf_files.c
+    rdf_hash.c
+# Berkley DB backend
+#    rdf_hash_bdb.c
+    rdf_hash_cursor.c
+    rdf_hash_memory.c
+    rdf_heuristics.c
+    rdf_init.c
+    rdf_iterator.c
+    rdf_list.c
+    rdf_log.c
+    rdf_model.c
+    rdf_model_storage.c
+    rdf_node.c
+    rdf_parser.c
+    rdf_parser_raptor.c
+    rdf_query.c
+    rdf_query_rasqal.c
+    rdf_query_results.c
+    rdf_query_triples.c
+    rdf_serializer.c
+    rdf_serializer_raptor.c
+    rdf_statement.c
+    rdf_storage.c
+    rdf_storage_file.c
+    rdf_storage_hashes.c
+    rdf_storage_list.c
+# MySQL backend
+#    rdf_storage_mysql.c
+# PostgreSQL backend
+#    rdf_storage_postgresql.c
+#    rdf_storage_sql.c
+    rdf_storage_sqlite.c
+#    rdf_storage_sql_test.c
+    rdf_storage_trees.c
+#    rdf_storage_tstore.c
+    rdf_stream.c
+    rdf_uri.c
+    rdf_utf8.c
+)
+
+set(librdf_HDRS
+    rdf_avltree_internal.h
+    rdf_concepts_internal.h
+    rdf_digest_internal.h
+    rdf_files.h
+    rdf_hash_internal.h
+    rdf_heuristics.h
+    rdf_init_internal.h
+    rdf_internal.h
+    rdf_iterator.h
+    rdf_iterator_internal.h
+    rdf_list.h
+    rdf_list_internal.h
+    rdf_log_internal.h
+    rdf_model_internal.h
+    rdf_node_internal.h
+    rdf_parser_internal.h
+    rdf_query_internal.h
+    rdf_serializer_internal.h
+    rdf_statement_internal.h
+    rdf_storage_internal.h
+    rdf_stream_internal.h
+    rdf_types.h
+    rdf_uri_internal.h
+    win32_rdf_config.h
+)
+
+set(librdf_HDRS_INST
+    librdf.h
+    rdf_concepts.h
+    rdf_digest.h
+    rdf_hash.h
+    rdf_init.h
+    rdf_iterator.h
+    rdf_log.h
+    rdf_model.h
+    rdf_node.h
+    rdf_query.h
+    rdf_parser.h
+    rdf_serializer.h
+    rdf_statement.h
+    rdf_storage.h
+    rdf_stream.h
+    rdf_uri.h
+    rdf_utf8.h
+    redland.h
+)
+
+include_directories(${CMAKE_CURRENT_SOURCE_DIR})
+include_directories(${CMAKE_SOURCE_DIR}/raptor/src)
+include_directories(${CMAKE_SOURCE_DIR}/rasqal/src)
+add_library(rdf SHARED ${librdf_SRCS} ${librdf_HDRS} ${librdf_HDRS_INST})
+set_target_properties(rdf PROPERTIES DEFINE_SYMBOL LIBRDF_INTERNAL)
+target_link_libraries(rdf raptor rasqal ${SQLITE3_LIBRARIES})
+if(MSVC)
+  set_target_properties(rdf PROPERTIES OUTPUT_NAME librdf)
+endif(MSVC)
+
+set(PROGRAMM librdf)
+configure_file(${CMAKE_SOURCE_DIR}/cmake/config.cpp.in ${CMAKE_CURRENT_BINARY_DIR}/redland-config.cpp)
+add_executable(redland-config ${CMAKE_CURRENT_BINARY_DIR}/redland-config.cpp)
+target_link_libraries(redland-config rdf)
+
+install(TARGETS rdf redland-config RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+install(FILES   ${librdf_HDRS_INST} DESTINATION include/redland)
diff -Nru redland-1.0.8.orig/librdf/win32_rdf_config.h redland-1.0.8/librdf/win32_rdf_config.h
--- redland-1.0.8.orig/librdf/win32_rdf_config.h	2008-07-01 05:10:26.000000000 +0200
+++ redland-1.0.8/librdf/win32_rdf_config.h	2009-09-24 02:48:28.866400000 +0200
@@ -43,7 +43,9 @@
 #define HAVE_STRICMP 1
 
 /* MS names for these functions */
+#ifndef _MSC_VER
 #define vsnprintf _vsnprintf
+#endif
 #define snprintf _snprintf
 #define access _access
 #define stricmp _stricmp
@@ -87,7 +89,7 @@
 #define HAVE_BDB_FD_2_ARGS 1
 
 /* Have BDB hash support */
-#define HAVE_BDB_HASH 1
+//#define HAVE_BDB_HASH 1
 
 /* BDB has open method with 6 args */
 /* #undef HAVE_BDB_OPEN_6_ARGS */
@@ -255,7 +257,7 @@
 #define STORAGE_FILE 1
 
 /* Building MySQL storage */
-#define STORAGE_MYSQL 1
+/*#define STORAGE_MYSQL 1*/
 
 /* Building PostgreSQL storage */
 /*#undef STORAGE_POSTGRESQL*/
diff -Nru redland-1.0.8.orig/raptor/CMakeLists.txt redland-1.0.8/raptor/CMakeLists.txt
--- redland-1.0.8.orig/raptor/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ redland-1.0.8/raptor/CMakeLists.txt	2009-09-24 02:48:28.866400000 +0200
@@ -0,0 +1,100 @@
+project(raptor)
+
+find_package(CURL REQUIRED)
+find_package(LibXml2 REQUIRED)
+find_package(LibXslt REQUIRED)
+
+set(raptor_SRCS
+    src/n3_lexer.c
+    src/n3_parser.c
+    src/ntriples_parse.c
+    src/parsedate.c
+    src/raptor_abbrev.c
+    src/raptor_avltree.c
+#    src/raptor_expat.c
+    src/raptor_feature.c
+    src/raptor_general.c
+    src/raptor_grddl.c
+    src/raptor_guess.c
+    src/raptor_identifier.c
+    src/raptor_iostream.c
+    src/raptor_json_writer.c
+#    src/raptor_librdfa.c
+    src/raptor_libxml.c
+    src/raptor_locator.c
+    src/raptor_namespace.c
+    src/raptor_nfc.c
+    src/raptor_nfc_data.c
+#    src/raptor_nfc_test.c
+    src/raptor_parse.c
+    src/raptor_qname.c
+    src/raptor_rdfxml.c
+    src/raptor_rfc2396.c
+    src/raptor_rss.c
+    src/raptor_rss_common.c
+    src/raptor_sax2.c
+    src/raptor_sequence.c
+    src/raptor_serialize.c
+    src/raptor_serialize_dot.c
+    src/raptor_serialize_json.c
+    src/raptor_serialize_ntriples.c
+    src/raptor_serialize_rdfxml.c
+    src/raptor_serialize_rdfxmla.c
+    src/raptor_serialize_rss.c
+    src/raptor_serialize_simple.c
+    src/raptor_serialize_turtle.c
+    src/raptor_set.c
+    src/raptor_statement.c
+    src/raptor_stringbuffer.c
+    src/raptor_turtle_writer.c
+    src/raptor_uri.c
+    src/raptor_utf8.c
+    src/raptor_win32.c
+    src/raptor_www.c
+    src/raptor_www_curl.c
+    src/raptor_www_libfetch.c
+    src/raptor_www_libxml.c
+#    src/raptor_www_test.c
+    src/raptor_xml.c
+    src/raptor_xml_writer.c
+    src/raptor_xsd.c
+    src/snprintf.c
+    src/strcasecmp.c
+    src/turtle_common.c
+    src/turtle_lexer.c
+    src/turtle_parser.c
+)
+
+set(raptor_HDRS
+    src/n3_common.h
+    src/n3_lexer.h
+    src/n3_parser.h
+    src/raptor.h
+    src/raptor_internal.h
+    src/raptor_nfc.h
+    src/raptor_rss.h
+    src/turtle_common.h
+    src/turtle_lexer.h
+    src/turtle_parser.h
+    src/win32_raptor_config.h
+)
+
+set(raptor_HDRS_INST
+    src/raptor.h
+)
+
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/src)
+include_directories(${CURL_INCLUDE_DIR})
+include_directories(${LIBXML2_INCLUDE_DIR})
+include_directories(${LIBXSLT_INCLUDE_DIR})
+add_library(raptor SHARED ${raptor_SRCS} ${raptor_HDRS} ${raptor_HDRS_INST})
+set_target_properties(raptor PROPERTIES DEFINE_SYMBOL RAPTOR_INTERNAL)
+target_link_libraries(raptor ${CURL_LIBRARIES} ${LIBXML2_LIBRARIES} ${LIBXSLT_LIBRARIES})
+
+set(PROGRAMM raptor)
+configure_file(${CMAKE_SOURCE_DIR}/cmake/config.cpp.in ${CMAKE_CURRENT_BINARY_DIR}/raptor-config.cpp)
+add_executable(raptor-config ${CMAKE_CURRENT_BINARY_DIR}/raptor-config.cpp)
+target_link_libraries(raptor-config raptor)
+
+install(TARGETS raptor raptor-config RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+install(FILES   ${raptor_HDRS_INST} DESTINATION include/redland)
diff -Nru redland-1.0.8.orig/raptor/src/raptor_win32.c redland-1.0.8/raptor/src/raptor_win32.c
--- redland-1.0.8.orig/raptor/src/raptor_win32.c	2008-06-14 05:39:35.000000000 +0200
+++ redland-1.0.8/raptor/src/raptor_win32.c	2009-09-24 02:48:28.866400000 +0200
@@ -33,8 +33,26 @@
 
 
 #ifdef WIN32
+#include <windows.h>
+#include <Winsock2.h>
+
+#define KDE_SECONDS_SINCE_1601	11644473600LL
+#define KDE_USEC_IN_SEC			1000000LL
 
 /* Only on WIN32 systems */
+int gettimeofday(struct timeval *__p, void *__t)
+{
+	union {
+		unsigned long long ns100; /*time since 1 Jan 1601 in 100ns units */
+		FILETIME ft;
+	} now;
+
+	GetSystemTimeAsFileTime (&now.ft);
+	__p->tv_usec = (long) ((now.ns100 / 10LL) % KDE_USEC_IN_SEC);
+	__p->tv_sec  = (long)(((now.ns100 / 10LL ) / KDE_USEC_IN_SEC) - KDE_SECONDS_SINCE_1601);
+
+	return (0);
+}
 
 
 /* DLL entry point */
diff -Nru redland-1.0.8.orig/raptor/src/win32_raptor_config.h redland-1.0.8/raptor/src/win32_raptor_config.h
--- redland-1.0.8.orig/raptor/src/win32_raptor_config.h	2008-04-02 06:50:14.000000000 +0200
+++ redland-1.0.8/raptor/src/win32_raptor_config.h	2009-09-24 02:49:14.143600000 +0200
@@ -69,7 +69,9 @@
 #define HAVE_STRICMP 1
 
 /* MS names for these functions */
+#ifndef _MSC_VER
 #define vsnprintf _vsnprintf
+#endif
 #define snprintf _snprintf
 #define access _access
 #define stricmp _stricmp
@@ -118,6 +120,7 @@
 #define RAPTOR_PARSER_TURTLE 1
 #define RAPTOR_PARSER_NTRIPLES 1
 #define RAPTOR_PARSER_RDFXML 1
+#define RAPTOR_PARSER_TRIG 1
 
 #define RAPTOR_SERIALIZER_ATOM 1
 #define RAPTOR_SERIALIZER_RSS_1_0 1
diff -Nru redland-1.0.8.orig/rasqal/CMakeLists.txt redland-1.0.8/rasqal/CMakeLists.txt
--- redland-1.0.8.orig/rasqal/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ redland-1.0.8/rasqal/CMakeLists.txt	2009-09-24 02:48:28.866400000 +0200
@@ -0,0 +1,63 @@
+project(rasqal)
+
+find_package(PCRE REQUIRED)
+
+set(rasqal_SRCS
+    src/rasqal_datetime.c
+    src/rasqal_decimal.c
+    src/rasqal_engine.c
+    src/rasqal_expr.c
+    src/rasqal_feature.c
+    src/rasqal_general.c
+    src/rasqal_graph_pattern.c
+#    src/rasqal_limit_test.c
+    src/rasqal_literal.c
+    src/rasqal_map.c
+    src/rasqal_query.c
+    src/rasqal_query_results.c
+    src/rasqal_query_test.c
+    src/rasqal_raptor.c
+# only when rasqal links against redland and not the other way round
+#    src/rasqal_redland.c
+    src/rasqal_result_formats.c
+    src/rasqal_rowsource.c
+    src/rasqal_sparql_xml.c
+    src/rasqal_xsd_datatypes.c
+    src/rdql_lexer.c
+    src/rdql_parser.c
+    src/sparql_lexer.c
+    src/sparql_parser.c
+    src/strcasecmp.c
+)
+
+set(rasqal_HDRS
+    src/rasqal_internal.h
+    src/rdql_common.h
+    src/rdql_lexer.h
+    src/rdql_parser.h
+    src/sparql_common.h
+    src/sparql_lexer.h
+    src/sparql_parser.h
+    src/win32_rasqal_config.h
+)
+
+set(rasqal_HDRS_INST
+    src/rasqal.h
+)
+
+add_definitions(-DRASQAL_DECIMAL_NONE)
+include_directories(${CMAKE_SOURCE_DIR}/raptor/src)
+include_directories(${CMAKE_SOURCE_DIR}/librdf)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/src)
+include_directories(${PCRE_INCLUDE_DIR})
+add_library(rasqal SHARED ${rasqal_SRCS})
+set_target_properties(rasqal PROPERTIES DEFINE_SYMBOL RASQAL_INTERNAL)
+target_link_libraries(rasqal raptor ${PCRE_LIBRARIES})
+
+set(PROGRAMM rasqal)
+configure_file(${CMAKE_SOURCE_DIR}/cmake/config.cpp.in ${CMAKE_CURRENT_BINARY_DIR}/rasqal-config.cpp)
+add_executable(rasqal-config ${CMAKE_CURRENT_BINARY_DIR}/rasqal-config.cpp)
+target_link_libraries(rasqal-config rasqal)
+
+install(TARGETS rasqal rasqal-config RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+install(FILES   ${rasqal_HDRS_INST} DESTINATION include/redland)
diff -Nru redland-1.0.8.orig/rasqal/src/sparql_lexer.c redland-1.0.8/rasqal/src/sparql_lexer.c
--- redland-1.0.8.orig/rasqal/src/sparql_lexer.c	2008-06-26 20:52:33.000000000 +0200
+++ redland-1.0.8/rasqal/src/sparql_lexer.c	2009-09-24 02:48:28.882000000 +0200
@@ -1166,7 +1166,7 @@
 
 #include <rasqal.h>
 #include <rasqal_internal.h>
-
+#undef DELETE
 #include <sparql_parser.h>
 
 #include <sparql_common.h>
diff -Nru redland-1.0.8.orig/rasqal/src/sparql_parser.c redland-1.0.8/rasqal/src/sparql_parser.c
--- redland-1.0.8.orig/rasqal/src/sparql_parser.c	2008-06-26 20:52:33.000000000 +0200
+++ redland-1.0.8/rasqal/src/sparql_parser.c	2009-09-24 02:48:28.882000000 +0200
@@ -43,6 +43,14 @@
    define necessary library symbols; they are noted "INFRINGES ON
    USER NAME SPACE" below.  */
 
+#ifdef WIN32
+//undefine some constants which are used below
+#define _WINSOCK2_H
+#include <windows.h>
+#undef DELETE
+#undef OPTIONAL
+#endif
+
 /* Identify Bison output.  */
 #define YYBISON 1
 
diff -Nru redland-1.0.8.orig/rasqal/src/win32_rasqal_config.h redland-1.0.8/rasqal/src/win32_rasqal_config.h
--- redland-1.0.8.orig/rasqal/src/win32_rasqal_config.h	2008-06-27 04:02:09.000000000 +0200
+++ redland-1.0.8/rasqal/src/win32_rasqal_config.h	2009-09-24 02:48:28.882000000 +0200
@@ -40,7 +40,9 @@
 #define HAVE_STRICMP 1
 
 /* MS names for these functions */
+#ifndef _MSC_VER
 #define vsnprintf _vsnprintf
+#endif
 #define snprintf _snprintf
 #define access _access
 #define stricmp _stricmp


diff -Nru orig/cmake/FindOpenSSL.cmake redland-1.0.8/cmake/FindOpenSSL.cmake
--- orig/cmake/FindOpenSSL.cmake        1970-01-01 01:00:00.000000000 +0100
+++ redland-1.0.8/cmake/FindOpenSSL.cmake       2010-04-17 21:42:40.023483300 +0200
@@ -0,0 +1,16 @@
+
+FIND_PATH(OPENSSL_INCLUDE_DIR openssl/ssl.h)
+
+FIND_LIBRARY(OPENSSL_SSL_LIBRARIES NAMES ssl ssleay32 ssleay32MD)
+FIND_LIBRARY(OPENSSL_CRYPTO_LIBRARIES NAMES crypto)
+MARK_AS_ADVANCED(OPENSSL_CRYPTO_LIBRARIES OPENSSL_SSL_LIBRARIES)
+
+SET(OPENSSL_LIBRARIES ${OPENSSL_SSL_LIBRARIES} ${OPENSSL_CRYPTO_LIBRARIES})
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(OpenSSL DEFAULT_MSG
+  OPENSSL_LIBRARIES
+  OPENSSL_INCLUDE_DIR
+)
+
+MARK_AS_ADVANCED(OPENSSL_INCLUDE_DIR OPENSSL_LIBRARIES)

--- redland-1.0.8.orig\librdf\rdf_storage_sqlite.c	2008-07-01 05:10:26.000000000 +0200
+++ redland-1.0.81\librdf\rdf_storage_sqlite.c	2010-04-18 12:00:37.195375100 +0200
@@ -44,8 +44,9 @@
 #include <redland.h>
 #include <rdf_storage.h>
 
-
-#if SQLITE_API == 3
+#undef SQLITE_API
+#define _SQLITE_API_VER  3
+#if _SQLITE_API_VER == 3
 #include <sqlite3.h>
 #define sqlite_DB sqlite3
 #define sqlite_STATEMENT sqlite3_stmt
@@ -57,7 +58,7 @@
 #define sqlite_last_insert_rowid sqlite3_last_insert_rowid
 #endif
 
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
 #include <sqlite.h>
 #define sqlite_DB sqlite
 #define sqlite_STATEMENT sqlite_vm
@@ -68,11 +69,11 @@
 #endif
 
 
-#if SQLITE_API == 3
+#if _SQLITE_API_VER == 3
   #define GET_COLUMN_VALUE_TEXT(vm, col) sqlite3_column_text(vm, col)
   #define GET_COLUMN_VALUE_INT(vm, col) sqlite3_column_int(vm, col)
 #endif
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
   #define GET_COLUMN_VALUE_TEXT(vm, col) (unsigned char*)pazValue[col]
   #define GET_COLUMN_VALUE_INT(vm, col) atoi(pazValue[col])
 #endif
@@ -759,7 +760,7 @@
   librdf_storage_sqlite_context *context=(librdf_storage_sqlite_context*)storage->context;
   int rc=SQLITE_OK;
   char *errmsg=NULL;
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
   int mode=0;
 #endif
   int db_file_exists=0;
@@ -772,13 +773,13 @@
   if(context->is_new && db_file_exists)
     unlink(context->name);
 
-#if SQLITE_API == 3
+#if _SQLITE_API_VER == 3
   context->db=NULL;
   rc=sqlite3_open(context->name, &context->db);
   if(rc != SQLITE_OK)
     errmsg=(char*)sqlite3_errmsg(context->db);
 #endif
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
   context->db=sqlite_open(context->name, mode, &errmsg);
   if(context->db == NULL)
     rc=SQLITE_ERROR;
@@ -787,7 +788,7 @@
     librdf_log(storage->world, 0, LIBRDF_LOG_ERROR, LIBRDF_FROM_STORAGE, NULL,
                "SQLite database %s open failed - %s", 
                context->name, errmsg);
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
     sqlite_FREE(errmsg);
 #endif
     librdf_storage_sqlite_close(storage);
@@ -1252,7 +1253,7 @@
 
   LIBRDF_DEBUG2("SQLite prepare '%s'\n", request);
 
-#if SQLITE_API == 3
+#if _SQLITE_API_VER == 3
   status=sqlite3_prepare(context->db,
                          (const char*)request,
                          raptor_stringbuffer_length(sb),
@@ -1261,7 +1262,7 @@
   if(status != SQLITE_OK)
     errmsg=(char*)sqlite3_errmsg(context->db);
 #endif
-#if SQLITE_API == 2  
+#if _SQLITE_API_VER == 2  
   status=sqlite_compile(context->db,
                         (const char*)request,
                         &scontext->zTail,
@@ -1301,7 +1302,7 @@
                                       librdf_statement **statement,
                                       librdf_node **context_node) {
   int status=SQLITE_BUSY;
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
   int pN;
   const char **pazValue;   /* Column data */
   const char **pazColName; /* Column names and datatypes */
@@ -1315,10 +1316,10 @@
    * SQLITE_MISUSE.
   */
   do {
-#if SQLITE_API == 3
+#if _SQLITE_API_VER == 3
     status=sqlite3_step(vm);
 #endif
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
     status=sqlite_step(vm, &pN, &pazValue, &pazColName);
 #endif
     if(status == SQLITE_BUSY) {
@@ -1450,12 +1451,12 @@
   if(status == SQLITE_ERROR) {
     char *errmsg=NULL;
 
-#if SQLITE_API == 3
+#if _SQLITE_API_VER == 3
     status=sqlite3_finalize(vm);
     if(status != SQLITE_OK)
       errmsg=(char*)sqlite3_errmsg(scontext->db);
 #endif
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
     status=sqlite_finalize(vm, &errmsg);
 #endif
     if(status != SQLITE_OK) {
@@ -1463,7 +1464,7 @@
                  0, LIBRDF_LOG_ERROR, LIBRDF_FROM_STORAGE, NULL,
                  "SQLite database %s finalize failed - %s (%d)", 
                  scontext->name, errmsg, status);
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
       sqlite_FREE(errmsg);
 #endif
     }
@@ -1554,12 +1555,12 @@
     char *errmsg=NULL;
     int status;
     
-#if SQLITE_API == 3
+#if _SQLITE_API_VER == 3
     status=sqlite3_finalize(scontext->vm);
     if(status != SQLITE_OK)
       errmsg=(char*)sqlite3_errmsg(scontext->sqlite_context->db);
 #endif
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
     status=sqlite_finalize(scontext->vm, &errmsg);
 #endif
     if(status != SQLITE_OK) {
@@ -1567,7 +1568,7 @@
                  0, LIBRDF_LOG_ERROR, LIBRDF_FROM_STORAGE, NULL,
                  "SQLite database %s finalize failed - %s (%d)", 
                  scontext->sqlite_context->name, errmsg, status);
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
       sqlite_FREE(errmsg);
 #endif
     }
@@ -1704,7 +1705,7 @@
 
   LIBRDF_DEBUG2("SQLite prepare '%s'\n", request);
 
-#if SQLITE_API == 3
+#if _SQLITE_API_VER == 3
   status=sqlite3_prepare(context->db,
                          (const char*)request,
                          raptor_stringbuffer_length(sb),
@@ -1713,7 +1714,7 @@
   if(status != SQLITE_OK)
     errmsg=(char*)sqlite3_errmsg(context->db);
 #endif
-#if SQLITE_API == 2  
+#if _SQLITE_API_VER == 2  
   status=sqlite_compile(context->db,
                         (const char*)request,
                         &scontext->zTail, 
@@ -1826,12 +1827,12 @@
     char *errmsg=NULL;
     int status;
     
-#if SQLITE_API == 3
+#if _SQLITE_API_VER == 3
     status=sqlite3_finalize(scontext->vm);
     if(status != SQLITE_OK)
       errmsg=(char*)sqlite3_errmsg(scontext->sqlite_context->db);
 #endif
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
     status=sqlite_finalize(scontext->vm, &errmsg);
 #endif
     if(status != SQLITE_OK) {
@@ -1839,7 +1840,7 @@
                  0, LIBRDF_LOG_ERROR, LIBRDF_FROM_STORAGE, NULL,
                  "SQLite database %s finalize failed - %s (%d)", 
                  scontext->sqlite_context->name, errmsg, status);
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
       sqlite_FREE(errmsg);
 #endif
     }
@@ -2148,7 +2149,7 @@
 
   LIBRDF_DEBUG2("SQLite prepare '%s'\n", request);
 
-#if SQLITE_API == 3
+#if _SQLITE_API_VER == 3
   status=sqlite3_prepare(context->db,
                          (const char*)request,
                          raptor_stringbuffer_length(sb),
@@ -2157,7 +2158,7 @@
   if(status != SQLITE_OK)
     errmsg=(char*)sqlite3_errmsg(context->db);
 #endif
-#if SQLITE_API == 2  
+#if _SQLITE_API_VER == 2  
   status=sqlite_compile(context->db,
                         (const char*)request,
                         &scontext->zTail,
@@ -2271,12 +2272,12 @@
     char *errmsg=NULL;
     int status;
     
-#if SQLITE_API == 3
+#if _SQLITE_API_VER == 3
     status=sqlite3_finalize(scontext->vm);
     if(status != SQLITE_OK)
       errmsg=(char*)sqlite3_errmsg(scontext->sqlite_context->db);
 #endif
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
     status=sqlite_finalize(scontext->vm, &errmsg);
 #endif
     if(status != SQLITE_OK) {
@@ -2284,7 +2285,7 @@
                  0, LIBRDF_LOG_ERROR, LIBRDF_FROM_STORAGE, NULL,
                  "SQLite database %s finalize failed - %s (%d)", 
                  scontext->sqlite_context->name, errmsg, status);
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
       sqlite_FREE(errmsg);
 #endif
     }
@@ -2328,7 +2329,7 @@
                                               sqlite_STATEMENT *vm,
                                               librdf_node **context_node) {
   int status=SQLITE_BUSY;
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
   int pN;
   const char **pazValue;   /* Column data */
   const char **pazColName; /* Column names and datatypes */
@@ -2342,10 +2343,10 @@
    * SQLITE_MISUSE.
   */
   do {
-#if SQLITE_API == 3
+#if _SQLITE_API_VER == 3
     status=sqlite3_step(vm);
 #endif
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
     status=sqlite_step(vm, &pN, &pazValue, &pazColName);
 #endif
     if(status == SQLITE_BUSY) {
@@ -2397,12 +2398,12 @@
   if(status == SQLITE_ERROR) {
     char *errmsg=NULL;
 
-#if SQLITE_API == 3
+#if _SQLITE_API_VER == 3
     status=sqlite3_finalize(vm);
     if(status != SQLITE_OK)
       errmsg=(char*)sqlite3_errmsg(scontext->db);
 #endif
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
     status=sqlite_finalize(vm, &errmsg);
 #endif
     if(status != SQLITE_OK) {
@@ -2410,7 +2411,7 @@
                  0, LIBRDF_LOG_ERROR, LIBRDF_FROM_STORAGE, NULL,
                  "SQLite database %s finalize failed - %s (%d)", 
                  scontext->name, errmsg, status);
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
       sqlite_FREE(errmsg);
 #endif
     }
@@ -2509,12 +2510,12 @@
     char *errmsg=NULL;
     int status;
     
-#if SQLITE_API == 3
+#if _SQLITE_API_VER == 3
     status=sqlite3_finalize(icontext->vm);
     if(status != SQLITE_OK)
       errmsg=(char*)sqlite3_errmsg(icontext->sqlite_context->db);
 #endif
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
     status=sqlite_finalize(icontext->vm, &errmsg);
 #endif
     if(status != SQLITE_OK) {
@@ -2522,7 +2523,7 @@
                  0, LIBRDF_LOG_ERROR, LIBRDF_FROM_STORAGE, NULL,
                  "SQLite database %s finalize failed - %s (%d)", 
                  icontext->sqlite_context->name, errmsg, status);
-#if SQLITE_API == 2
+#if _SQLITE_API_VER == 2
       sqlite_FREE(errmsg);
 #endif
     }
@@ -2587,7 +2588,7 @@
 
   LIBRDF_DEBUG2("SQLite prepare '%s'\n", request);
 
-#if SQLITE_API == 3
+#if _SQLITE_API_VER == 3
   status=sqlite3_prepare(context->db,
                          (const char*)request,
                          raptor_stringbuffer_length(sb),
@@ -2596,7 +2597,7 @@
   if(status != SQLITE_OK)
     errmsg=(char*)sqlite3_errmsg(context->db);
 #endif
-#if SQLITE_API == 2  
+#if _SQLITE_API_VER == 2  
   status=sqlite_compile(context->db,
                         (const char*)request,
                         &icontext->zTail,


