diff -Nru -x '*~' phonon-4.5.1.orig/cmake/FindPhononInternal.cmake phonon-4.5.1/cmake/FindPhononInternal.cmake
--- phonon-4.5.1.orig/cmake/FindPhononInternal.cmake	2011-10-21 21:03:49.000000000 +0200
+++ phonon-4.5.1/cmake/FindPhononInternal.cmake	2012-01-01 23:18:11.394781800 +0100
@@ -334,16 +334,16 @@
    endif (CMAKE_SYSTEM_NAME STREQUAL GNU)
 
    # gcc under Windows
-   if (MINGW)
-      set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--export-all-symbols -Wl,--disable-auto-import")
-      set (CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,--export-all-symbols -Wl,--disable-auto-import")
+#   if (MINGW)
+#      set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--export-all-symbols -Wl,--disable-auto-import")
+#      set (CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,--export-all-symbols -Wl,--disable-auto-import")
 
       # we always link against the release version of QT with mingw
       # (even for debug builds). So we need to define QT_NO_DEBUG
       # or else QPluginLoader rejects plugins because it thinks
       # they're built against the wrong QT.
       add_definitions(-DQT_NO_DEBUG)
-   endif (MINGW)
+#   endif (MINGW)
 
    check_cxx_compiler_flag(-fPIE HAVE_FPIE_SUPPORT)
    if(KDE4_ENABLE_FPIE)
