diff -Nru phonon-4.6.0.orig/cmake/FindPhononInternal.cmake phonon-4.6.0/cmake/FindPhononInternal.cmake
--- phonon-4.6.0.orig/cmake/FindPhononInternal.cmake	2011-12-19 21:30:45.000000000 +0100
+++ phonon-4.6.0/cmake/FindPhononInternal.cmake	2012-03-07 11:39:56.071398500 +0100
@@ -223,13 +223,6 @@
       set (CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -pthread")
    endif (CMAKE_SYSTEM_NAME STREQUAL GNU)
 
-   # gcc under Windows
-   if (MINGW)
-      set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--export-all-symbols -Wl,--disable-auto-import")
-      set (CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,--export-all-symbols -Wl,--disable-auto-import")
-
-   endif (MINGW)
-
    check_cxx_compiler_flag(-fPIE HAVE_FPIE_SUPPORT)
    if(KDE4_ENABLE_FPIE)
        if(HAVE_FPIE_SUPPORT)
