--- akonadi-contacts-17.04.1/src/actions/qekigadialer.cpp--
+++ akonadi-contacts-17.04.1/src/actions/qekigadialer.cpp
@@ -32,7 +32,11 @@
 #include <QtDBus/QDBusInterface>
 #include <QtDBus/QDBusReply>
 
+#ifdef WIN32
+#include <windows.h>
+#else
 #include <unistd.h>
+#endif
 #include <QString>
 
 static bool isEkigaServiceRegistered()
@@ -89,7 +93,11 @@ bool QEkigaDialer::initializeEkiga()
         const int runs = 100;
         for (int i = 0; i < runs; ++i) {
             if (!isEkigaServiceRegistered()) {
+#ifdef WIN32
+                Sleep(2000);
+#else
                 ::sleep(2);
+#endif
             } else {
                 break;
             }
--- akonadi-contacts-17.04.1/src/actions/qsflphonedialer.cpp--
+++ akonadi-contacts-17.04.1/src/actions/qsflphonedialer.cpp
@@ -32,7 +32,11 @@
 #include <QtDBus/QDBusInterface>
 #include <QtDBus/QDBusReply>
 
+#ifdef WIN32
+#include <windows.h>
+#else
 #include <unistd.h>
+#endif
 
 static bool isSflPhoneServiceRegistered()
 {
@@ -73,7 +77,11 @@ bool QSflPhoneDialer::initializeSflPhone()
         const int runs = 100;
         for (int i = 0; i < runs; ++i) {
             if (!isSflPhoneServiceRegistered()) {
+#ifdef WIN32
+                Sleep(2000);
+#else
                 ::sleep(2);
+#endif
             } else {
                 return true;
             }
--- akonadi-contacts-17.04.1/src/actions/qskypedialer.cpp--
+++ akonadi-contacts-17.04.1/src/actions/qskypedialer.cpp
@@ -30,7 +30,11 @@
 #include <QtDBus/QDBusReply>
 #include <KLocalizedString>
 
+#ifdef WIN32
+#include <windows.h>
+#else
 #include <unistd.h>
+#endif
 
 static bool isSkypeServiceRegistered()
 {
@@ -91,7 +95,11 @@ bool QSkypeDialer::initializeSkype()
         const int runs = 100;
         for (int i = 0; i < runs; ++i) {
             if (!isSkypeServiceRegistered()) {
+#ifdef WIN32
+                Sleep(2000);
+#else
                 ::sleep(2);
+#endif
             } else {
                 break;
             }
