diff -Nru -x '*~' marble-4.10.0.orig/src/lib/global.h marble-4.10.0/src/lib/global.h
--- marble-4.10.0.orig/src/lib/global.h	2013-01-23 22:54:46.000000000 +0100
+++ marble-4.10.0/src/lib/global.h	2013-03-01 23:27:38.979688800 +0100
@@ -1,2 +1,4 @@
 #include "MarbleGlobal.h"
+#ifdef __GNUC__
 #warning global.h is deprecated. Please #include <marble/MarbleGlobal.h> instead (no further changes needed).
+#endif
