diff -Nru marble-4.8.0.orig/src/lib/MarbleRunnerManager.cpp marble-4.8.0/src/lib/MarbleRunnerManager.cpp
--- marble-4.8.0.orig/src/lib/MarbleRunnerManager.cpp	2012-01-18 19:49:40.000000000 +0100
+++ marble-4.8.0/src/lib/MarbleRunnerManager.cpp	2012-01-25 16:00:58.791000000 +0100
@@ -152,7 +152,7 @@
     QList<RunnerPlugin*> plugins = d->plugins( RunnerPlugin::ReverseGeocoding );
     foreach( RunnerPlugin* plugin, plugins ) {
         ReverseGeocodingTask* task = new ReverseGeocodingTask( plugin, this, d->m_marbleModel, coordinates );
-        mDebug() << "reverse task " << plugin->nameId() << " " << (long)task;
+        mDebug() << "reverse task " << plugin->nameId() << " " << (quintptr)task;
         QThreadPool::globalInstance()->start( task );
     }
 
@@ -241,7 +241,7 @@
         started = true;
         RoutingTask* task = new RoutingTask( plugin, this, d->m_marbleModel, request );
         connect( task, SIGNAL( finished( RunnerTask* ) ), this, SLOT( cleanupRoutingTask( RunnerTask* ) ) );
-        mDebug() << "route task " << plugin->nameId() << " " << (long)task;
+        mDebug() << "route task " << plugin->nameId() << " " << (quintptr)task;
         d->m_routingTasks << task;
         connect( task, SIGNAL( finished( RunnerTask* ) ), this, SLOT( cleanupRoutingTask( RunnerTask* ) ) );
         QThreadPool::globalInstance()->start( task );
@@ -266,7 +266,7 @@
     QList<RunnerPlugin*> plugins = d->plugins( RunnerPlugin::Parsing );
     foreach( RunnerPlugin *plugin, plugins ) {
         ParsingTask *task = new ParsingTask( plugin, this, fileName, role );
-        mDebug() << "parse task " << plugin->nameId() << " " << (long)task;
+        mDebug() << "parse task " << plugin->nameId() << " " << (quintptr)task;
         QThreadPool::globalInstance()->start( task );
     }
 }
