diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' fluidsynth-1.1.6.orig/CMakeLists.txt fluidsynth-1.1.6/CMakeLists.txt
--- fluidsynth-1.1.6.orig/CMakeLists.txt	2012-08-16 06:01:13.000000000 +0200
+++ fluidsynth-1.1.6/CMakeLists.txt	2016-09-08 19:24:04.534782900 +0200
@@ -85,11 +85,11 @@
 endif ( ${CMAKE_SYSTEM} MATCHES "OS2" )
 
 # Initialize the library directory name suffix.
-if ( CMAKE_SIZEOF_VOID_P EQUAL 8 )
+if ( CMAKE_SIZEOF_VOID_P EQUAL 8 AND NOT WIN32)
   set ( _init_lib_suffix "64" )
-else ( CMAKE_SIZEOF_VOID_P EQUAL 8 )
+else ()
   set ( _init_lib_suffix "" )
-endif ( CMAKE_SIZEOF_VOID_P EQUAL 8 )
+endif ()
 set ( LIB_SUFFIX ${_init_lib_suffix} CACHE STRING
       "library directory name suffix (32/64/nothing)" )
 mark_as_advanced ( LIB_SUFFIX )
@@ -281,7 +281,13 @@
 find_package ( PkgConfig REQUIRED )
 
 # Mandatory libraries: glib and gthread
-pkg_check_modules ( GLIB REQUIRED glib-2.0>=2.6.5 gthread-2.0>=2.6.5 )
+if (NOT WIN32)
+    pkg_check_modules ( GLIB REQUIRED glib-2.0>=2.6.5 gthread-2.0>=2.6.5 )
+else()
+    set(LIBFLUID_LIBS "${LIBFLUID_LIBS};glib-2.0;gthread-2.0")
+    set(GLIB_INCLUDEDIR ${CMAKE_INSTALL_PREFIX}/include/glib-2.0)
+    set(GLIB_LIBDIR ${CMAKE_INSTALL_PREFIX}/lib/)
+endif()
 
 include ( UnsetPkgConfig )
 
@@ -487,7 +493,6 @@
 set ( CPACK_SOURCE_STRIP_FILES OFF )
 
 # binary packages
-include ( InstallRequiredSystemLibraries )
 set ( CPACK_GENERATOR STGZ;TGZ;TBZ2;ZIP )
 set ( CPACK_PACKAGE_NAME ${PACKAGE} )
 set ( CPACK_STRIP_FILES ON )
