diff -Nru kdeedu-4.5.4.orig/kalgebra/analitza/analyzer.cpp kdeedu-4.5.4/kalgebra/analitza/analyzer.cpp
--- kdeedu-4.5.4.orig/kalgebra/analitza/analyzer.cpp	2010-06-24 18:29:38.000000000 +0200
+++ kdeedu-4.5.4/kalgebra/analitza/analyzer.cpp	2010-12-15 01:33:38.391963800 +0100
@@ -732,7 +732,7 @@
 			int count=c->countValues();
 			Q_ASSERT(count>0);
 			
-			Object* numbers[count];
+			QVector<Object*> numbers(count);
 			Apply::const_iterator it = c->firstValue(), itEnd=c->constEnd();
 			for(int i=0; it!=itEnd; ++it, ++i)
 				numbers[i]=calc(*it);
