From 4bdbf31af0cfe3b9d28242854060989ef2a9b91e Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@intevation.de>
Date: Thu, 6 Jan 2011 13:26:34 +0000
Subject: [PATCH] Fix build with QT_NO_BEARERMANAGEMENT

---
 src/network/access/qnetworkaccessdatabackend.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/network/access/qnetworkaccessdatabackend.cpp b/src/network/access/qnetworkaccessdatabackend.cpp
index 74aebdb..f9c544d 100644
--- a/src/network/access/qnetworkaccessdatabackend.cpp
+++ b/src/network/access/qnetworkaccessdatabackend.cpp
@@ -124,7 +124,9 @@ bool QNetworkAccessDataBackend::waitForUpstreamBytesWritten(int)
 
 bool QNetworkAccessDataBackend::processRequestSynchronously()
 {
+#ifndef QT_NO_BEARERMANAGEMENT
     start();
+#endif
     return true;
 }
 
-- 
1.7.0.4

