From b6d9c9e127f515e8302554d3190325e9838009f9 Mon Sep 17 00:00:00 2001
From: Andreas Holzammer <andreas.holzammer@kdab.com>
Date: Thu, 6 Jan 2011 15:15:41 +0100
Subject: [PATCH] fix QSortFilterProxyModel

---
 src/gui/itemviews/qsortfilterproxymodel.cpp |   12 ++++++++++++
 1 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/src/gui/itemviews/qsortfilterproxymodel.cpp b/src/gui/itemviews/qsortfilterproxymodel.cpp
index dc8d938..59970b8 100644
--- a/src/gui/itemviews/qsortfilterproxymodel.cpp
+++ b/src/gui/itemviews/qsortfilterproxymodel.cpp
@@ -795,6 +795,18 @@ void QSortFilterProxyModelPrivate::source_items_inserted(
                 sort_source_rows(orthogonal_proxy_to_source, source_parent);
             }
             build_source_to_proxy_mapping(orthogonal_proxy_to_source, orthogonal_source_to_proxy);
+        } else if (orient == Qt::Vertical) {
+
+            const int sourceColumnCount = model->columnCount();
+
+            if (proxy_sort_column < sourceColumnCount) {
+                source_sort_column = proxy_sort_column;
+
+                for (int column = 0; column < proxy_sort_column; ++column) {
+                  if (!q->filterAcceptsColumn(column, source_parent))
+                    ++source_sort_column;
+                }
+            }
         }
     }
 
-- 
1.7.0.2.msysgit.0

