diff -Nru -x '*~' libkgapi-0.4.4.orig/libkgapi/account.h libkgapi-0.4.4/libkgapi/account.h
--- libkgapi-0.4.4.orig/libkgapi/account.h	2012-12-16 19:23:52.000000000 +0100
+++ libkgapi-0.4.4/libkgapi/account.h	2013-03-06 22:59:36.243725400 +0100
@@ -166,6 +166,8 @@
      * will just refresh existing tokens until the scopes are changed again.
      */
     bool m_scopesChanged;
+
+    Account& operator=(const Account& other);
 };
 
 } /* namespace KGAPI */
diff -Nru -x '*~' libkgapi-0.4.4.orig/libkgapi/object.h libkgapi-0.4.4/libkgapi/object.h
--- libkgapi-0.4.4.orig/libkgapi/object.h	2012-12-16 19:23:52.000000000 +0100
+++ libkgapi-0.4.4/libkgapi/object.h	2013-03-06 23:01:35.131534300 +0100
@@ -59,7 +59,8 @@
 
   private:
     QExplicitlySharedDataPointer< ObjectPrivate > d;
-
+    
+    Object& operator=(const Object& other);
 };
 
 } // namespace KGAPI
diff -Nru -x '*~' libkgapi-0.4.4.orig/libkgapi/objects/calendar.h libkgapi-0.4.4/libkgapi/objects/calendar.h
--- libkgapi-0.4.4.orig/libkgapi/objects/calendar.h	2012-12-16 19:23:52.000000000 +0100
+++ libkgapi-0.4.4/libkgapi/objects/calendar.h	2013-03-06 23:02:39.688487700 +0100
@@ -71,6 +71,8 @@
 
     private:
         QExplicitlySharedDataPointer < ReminderData > d;
+        
+        Reminder& operator=(const Reminder& other);
 };
 
 class CalendarData;
