diff --git a/build.sh b/build.sh
index 87cc722..1c0be84 100755
--- a/build.sh
+++ b/build.sh
@@ -119,111 +119,5 @@ pushd app_packages
 
 popd
 
-## Create the distribution
-
-pushd distribution
-
-/usr/local/bin/packagesbuild -v Packages.pkgproj
-
-## Create the disk image
-
-DISKIMAGE_NAME="Packages"
-
-## Convert disk image template to read-write disk image
-
-if [ -f build/"$DISKIMAGE_NAME"_rw.dmg ]
-then 
-	/bin/rm build/"$DISKIMAGE_NAME"_rw.dmg
-fi
-
-/usr/bin/hdiutil convert Template/Template_ro.dmg -format UDRW -o build/"$DISKIMAGE_NAME"_rw.dmg > /dev/null
-
-## Mount the disk image
-
-/usr/bin/hdiutil attach build/"$DISKIMAGE_NAME"_rw.dmg -mountpoint build/diskimage_rw > /dev/null
-
-## Rename the disk image
-
-if [ -f Version ];
-then
-
-	/usr/sbin/diskutil rename "$DISKIMAGE_NAME" "$DISKIMAGE_NAME $VERSION" 
-fi
-
-## Copy the Read Before You Install Packages to the disk image and prevent edition
-
-if [ -f "Documents/Read Before You Install Packages.rtf" ]
-then
-
-	/usr/bin/sed '2 s/^/\\readonlydoc1/' <"Documents/Read Before You Install Packages.rtf" > "build/diskimage_rw/Read Before You Install Packages.rtf"
-
-else
-
-	echo "Missing Read Before You Install Packages"
-fi
-
-## Copy the User Guide webloc
-
-if [ -f "Documents/Packages User Guide.webloc" ]
-then
-
-	/bin/cp "Documents/Packages User Guide.webloc" "build/diskimage_rw/Extras/Packages User Guide.webloc"
-
-else
-
-	echo "Missing Packages User Guide.webloc"
-fi
-
-## Copy the uninstall.sh script
-
-if [ -f "Scripts/uninstall.sh" ]
-then
-
-	/bin/cp Scripts/uninstall.sh "build/diskimage_rw/Extras/uninstall.sh"
-
-else
-
-	echo "Missing uninstall.sh"
-fi
-
-## Copy the distribution package to the disk image
-
-if [ -f build/Packages.pkg ]
-then
-
-	/bin/cp build/Packages.pkg build/diskimage_rw/packages/Packages.pkg
-
-else
-
-	echo "Missing distribution packages"
-fi
-
-## Remove useless files for a disk image
-
-/bin/rm "build/diskimage_rw/Desktop DB"
-/bin/rm "build/diskimage_rw/Desktop DF"
-/bin/rm -r build/diskimage_rw/.fseventsd
-
-## Unmount the disk image
-
-/usr/bin/hdiutil detach build/diskimage_rw > /dev/null
-
-## Convert disk image to read-only
-
-if [ -f artifacts/"$DISKIMAGE_NAME".dmg ]
-then 
-	/bin/rm artifacts/"$DISKIMAGE_NAME".dmg
-fi
-
-/usr/bin/hdiutil convert build/"$DISKIMAGE_NAME"_rw.dmg -format UDZO -o artifacts/"$DISKIMAGE_NAME".dmg > /dev/null
-
-## Remove the temporary disk image
-
-if [ -f build/"$DISKIMAGE_NAME"_rw.dmg ]
-then 
-	/bin/rm build/"$DISKIMAGE_NAME"_rw.dmg
-fi
-
-popd
 
 exit 0
diff --git a/xar b/xar
--- a/xar
+++ b/xar
@@ -1 +1 @@
-Subproject commit 8b9122a44fc9ce30301f83159876ecbd80548fa1
+Subproject commit 8b9122a44fc9ce30301f83159876ecbd80548fa1-dirty
