--- share.orig/cmake-2.8/Modules/FindPNG.cmake.orig	2012-11-27 14:26:32.000000000 +0100
+++ share/cmake-2.8/Modules/FindPNG.cmake	2013-11-10 17:31:30.293450800 +0100
@@ -38,7 +38,7 @@
   /usr/local/include/libpng             # OpenBSD
   )
 
-  set(PNG_NAMES ${PNG_NAMES} png libpng png15 libpng15 png15d libpng15d png14 libpng14 png14d libpng14d png12 libpng12 png12d libpng12d)
+  set(PNG_NAMES ${PNG_NAMES} png libpng png16 libpng16 png16d libpng16d png15 libpng15 png15d libpng15d png14 libpng14 png14d libpng14d png12 libpng12 png12d libpng12d)
   find_library(PNG_LIBRARY NAMES ${PNG_NAMES} )
 
   if (PNG_LIBRARY AND PNG_PNG_INCLUDE_DIR)
