diff -Nrub -x '*~' -x '*\.orig' -x'*\.o' exiv2-0.25.orig/include/exiv2/exv_msvc.h exiv2-0.25/include/exiv2/exv_msvc.h
--- exiv2-0.25.orig/include/exiv2/exv_msvc.h	2015-06-15 17:32:21.000000000 +0200
+++ exiv2-0.25/include/exiv2/exv_msvc.h	2015-12-17 11:03:47.993488200 +0100
@@ -113,9 +113,6 @@
 #define CURL_STATICLIB
 #endif
 
-#pragma comment(lib, "ws2_32.lib")
-#pragma comment(lib, "wldap32.lib")
-
 /* File path separator */
 #define EXV_SEPARATOR_STR "\\"
 #define EXV_SEPARATOR_CHR '\\'
diff -Nrub -x '*~' -x '*\.orig' -x'*\.o' exiv2-0.25.orig/src/CMakeLists.txt exiv2-0.25/src/CMakeLists.txt
--- exiv2-0.25.orig/src/CMakeLists.txt	2015-06-14 11:12:19.000000000 +0200
+++ exiv2-0.25/src/CMakeLists.txt	2015-12-17 11:44:54.490563700 +0100
@@ -257,6 +257,10 @@
     endif()
 endif()
 
+if(WIN32)
+    TARGET_LINK_LIBRARIES( exiv2lib ws2_32 wldap32)
+endif()
+
 if( EXIV2_ENABLE_LIBXMP )
     ADD_DEPENDENCIES( exiv2lib xmp )
     if  ( MSVC )
@@ -269,7 +273,7 @@
     IF( ZLIB_FOUND )
         IF( MSVC )
             if ( EXIV2_ENABLE_SHARED )
-                TARGET_LINK_LIBRARIES( exiv2lib optimized ${ZLIB_LIBRARIES} debug ${ZLIB_LIBRARIES}d )
+                TARGET_LINK_LIBRARIES( exiv2lib optimized ${ZLIB_LIBRARIES} debug ${ZLIB_LIBRARIES} )
             else()
                 TARGET_LINK_LIBRARIES( exiv2lib optimized zlibstatic.lib debug zlibstaticd.lib )
             endif()
diff -Nrub -x '*~' -x '*\.orig' -x'*\.o' exiv2-0.25.orig/src/localtime.c exiv2-0.25/src/localtime.c
--- exiv2-0.25.orig/src/localtime.c	2014-12-25 14:55:57.000000000 +0100
+++ exiv2-0.25/src/localtime.c	2015-12-17 11:03:48.006488900 +0100
@@ -105,6 +105,13 @@
 #define MY_TZNAME_MAX	255
 #endif /* !defined TZNAME_MAX */
 
+
+#if defined(_MSC_VER) && _MSC_VER >= 1900
+#ifndef tzname
+#define tzname _tzname
+#endif
+#endif
+
 struct state {
 	int		leapcnt;
 	int		timecnt;
diff -Nrub -x '*~' -x '*\.orig' -x'*\.o' exiv2-0.25.orig/src/svn_version.h exiv2-0.25/src/svn_version.h
--- exiv2-0.25.orig/src/svn_version.h	1970-01-01 01:00:00.000000000 +0100
+++ exiv2-0.25/src/svn_version.h	2015-12-17 11:04:12.635897600 +0100
@@ -0,0 +1,3 @@
+#ifndef SVN_VERSION
+#define SVN_VERSION 0
+#endif
