diff -Nru gpgme-1.3.0.orig/ConfigureChecks.cmake gpgme-1.3.0/ConfigureChecks.cmake
--- gpgme-1.3.0.orig/ConfigureChecks.cmake	2010-11-21 01:50:53.457698200 +0100
+++ gpgme-1.3.0/ConfigureChecks.cmake	2010-11-21 01:24:58.151739800 +0100
@@ -25,6 +25,9 @@
     if(WINCE)
         set(uintptr_t "unsigned __int32")
     endif(WINCE)
+ 	if(MSVC)
+ 		set(uint64_t "unsigned __int64")
+ 	endif(MSVC)
 endif(WIN32)
 
 check_function_exists("fopencookie" HAVE_FOPENCOOKIE)
diff -Nru gpgme-1.3.0.orig/src/data-compat.c gpgme-1.3.0/src/data-compat.c
--- gpgme-1.3.0.orig/src/data-compat.c	2010-11-21 03:03:55.210048400 +0100
+++ gpgme-1.3.0/src/data-compat.c	2010-11-21 02:56:56.723112300 +0100
@@ -120,10 +120,10 @@
 gpgme_data_new_from_file (gpgme_data_t *r_dh, const char *fname, int copy)
 {
   gpgme_error_t err = -1;
-  TRACE_BEG3 (DEBUG_DATA, "gpgme_data_new_from_filepart", r_dh,
-	      "file_name=%s, copy=%i (%s)", fname, copy, copy ? "yes" : "no");
 #ifndef HAVE_W32CE_SYSTEM
   struct stat statbuf;
+  TRACE_BEG3 (DEBUG_DATA, "gpgme_data_new_from_filepart", r_dh,
+	      "file_name=%s, copy=%i (%s)", fname, copy, copy ? "yes" : "no");
 
   if (!fname || !copy)
     return TRACE_ERR (gpg_error (GPG_ERR_INV_VALUE));
diff -Nru gpgme-1.3.0.orig/src/gpgme.h.in gpgme-1.3.0/src/gpgme.h.in
--- gpgme-1.3.0.orig/src/gpgme.h.in	2010-11-21 01:50:53.502700800 +0100
+++ gpgme-1.3.0/src/gpgme.h.in	2010-11-21 00:45:24.449213700 +0100
@@ -34,7 +34,7 @@
 /* Include stdio.h for the FILE type definition.  */
 #include <stdio.h>
 
-#ifdef _WIN32
+#ifdef _MSC_VER
   typedef long off_t;
   typedef long ssize_t;
   typedef int pid_t;
diff -Nru gpgme-1.3.0.orig/src/util.h gpgme-1.3.0/src/util.h
--- gpgme-1.3.0.orig/src/util.h	2010-11-21 01:50:53.511701300 +0100
+++ gpgme-1.3.0/src/util.h	2010-11-21 01:49:24.775625900 +0100
@@ -24,6 +24,9 @@
 
 #ifdef HAVE_W32CE_SYSTEM
 #include "w32-ce.h"
+#else
+#include <winsock2.h>
+#include <windows.h>
 #endif
 
 /* For pid_t.  */
diff -Nru gpgme-1.3.0.orig/src/w32-util.c gpgme-1.3.0/src/w32-util.c
--- gpgme-1.3.0.orig/src/w32-util.c	2010-11-21 03:03:55.283052600 +0100
+++ gpgme-1.3.0/src/w32-util.c	2010-11-21 03:03:40.763222100 +0100
@@ -23,6 +23,7 @@
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -47,6 +48,10 @@
 #include <fcntl.h>
 #endif
 #include <windows.h>
+
+#if defined(__MINGW32__) && _WIN32_IE < 0x0400
+#define _WIN32_IE 0x0400
+#endif
 #include <shlobj.h>
 #ifndef HAVE_W32CE_SYSTEM
 #include <io.h>
@@ -75,7 +80,6 @@
 #define HAVE_ALLOW_SET_FOREGROUND_WINDOW 1
 #endif
 
-
 DEFINE_STATIC_LOCK (get_path_lock);
 
 
diff -Nru gpgme-1.3.0.orig/src/wait-private.c gpgme-1.3.0/src/wait-private.c
--- gpgme-1.3.0.orig/src/wait-private.c	2009-10-28 15:43:53.000000000 +0100
+++ gpgme-1.3.0/src/wait-private.c	2010-11-21 00:45:24.457214100 +0100
@@ -25,6 +25,9 @@
 #include <assert.h>
 #include <errno.h>
 
+#ifdef __MINGW32__
+# include <winsock2.h>
+#endif
 #include "gpgme.h"
 #include "context.h"
 #include "wait.h"
