--- libkdegames-17.04.0/highscore/khighscore.cpp--
+++ libkdegames-17.04.0/highscore/khighscore.cpp
@@ -32,7 +32,11 @@
 #include <QLockFile>
 #include <QGlobalStatic>
 
+#ifdef WIN32
+#include <windows.h>
+#else
 #include <unistd.h> // sleep
+#endif
 
 #define GROUP "KHighscore"
 
@@ -155,7 +159,13 @@ bool KHighscore::lockForWriting(QWidget *widget)
             item.setText(i18n("Retry"));
             int res = KMessageBox::warningContinueCancel(widget, i18n("Cannot access the highscore file. Another user is probably currently writing to it."), QString(), item, KStandardGuiItem::cancel(), QStringLiteral( "ask_lock_global_highscore_file" ));
             if ( res==KMessageBox::Cancel ) break;
-        } else sleep(1);
+        } else {
+#ifdef WIN32
+            Sleep(1000);
+#else
+            sleep(1);
+#endif
+        }
         first = false;
     }
     return false;
--- libkdegames-17.04.0/libkdegamesprivate/kgame/kgame.cpp--
+++ libkdegames-17.04.0/libkdegamesprivate/kgame/kgame.cpp
@@ -29,7 +29,6 @@
 
 #include "kgamemessage.h"
 
-#include <unistd.h>
 #include <stdio.h>
 #include <assert.h>
 
