diff -Nru kdenetwork-4.5.4.orig/kopete/protocols/jabber/libiris/CMakeLists.txt kdenetwork-4.5.4/kopete/protocols/jabber/libiris/CMakeLists.txt
--- kdenetwork-4.5.4.orig/kopete/protocols/jabber/libiris/CMakeLists.txt	2009-03-10 13:25:37.000000000 +0100
+++ kdenetwork-4.5.4/kopete/protocols/jabber/libiris/CMakeLists.txt	2010-12-13 20:29:41.986301000 +0100
@@ -158,90 +158,90 @@
 
 
 iris/xmpp/base/randrandomnumbergenerator.h
-iris/xmpp/base/randomnumbergenerator.h
-iris/xmpp/jid/jid.h
-iris/xmpp/cutestuff/bytestream.h
-iris/xmpp/cutestuff/socks.h
-iris/xmpp/cutestuff/httpconnect.h
+# iris/xmpp/base/randomnumbergenerator.h
+# iris/xmpp/jid/jid.h
+# iris/xmpp/cutestuff/bytestream.h
+# iris/xmpp/cutestuff/socks.h
+# iris/xmpp/cutestuff/httpconnect.h
 iris/xmpp/cutestuff/bsocket.h
-iris/xmpp/cutestuff/httppoll.h
-iris/xmpp/xmpp-im/xmpp_xdata.h
-iris/xmpp/xmpp-im/xmpp_rosterx.h
-iris/xmpp/xmpp-im/xmpp_tasks.h
-iris/xmpp/xmpp-im/xmpp_discoitem.h
-iris/xmpp/xmpp-im/xmpp_pubsubretraction.h
-iris/xmpp/xmpp-im/xmpp_xmlcommon.h
-iris/xmpp/xmpp-im/xmpp_roster.h
-iris/xmpp/xmpp-im/xmpp_muc.h
-iris/xmpp/xmpp-im/xmpp_htmlelement.h
-iris/xmpp/xmpp-im/xmpp_receipts.h
-iris/xmpp/xmpp-im/xmpp_message.h
-iris/xmpp/xmpp-im/xmpp_discoinfotask.h
-iris/xmpp/xmpp-im/xmpp_ibb.h
-iris/xmpp/xmpp-im/xmpp_liverosteritem.h
-iris/xmpp/xmpp-im/xmpp_httpauthrequest.h
-iris/xmpp/xmpp-im/xmpp_pubsubitem.h
-iris/xmpp/xmpp-im/xmpp_vcard.h
+# iris/xmpp/cutestuff/httppoll.h
+# iris/xmpp/xmpp-im/xmpp_xdata.h
+# iris/xmpp/xmpp-im/xmpp_rosterx.h
+# iris/xmpp/xmpp-im/xmpp_tasks.h
+# iris/xmpp/xmpp-im/xmpp_discoitem.h
+# iris/xmpp/xmpp-im/xmpp_pubsubretraction.h
+# iris/xmpp/xmpp-im/xmpp_xmlcommon.h
+# iris/xmpp/xmpp-im/xmpp_roster.h
+# iris/xmpp/xmpp-im/xmpp_muc.h
+# iris/xmpp/xmpp-im/xmpp_htmlelement.h
+# iris/xmpp/xmpp-im/xmpp_receipts.h
+# iris/xmpp/xmpp-im/xmpp_message.h
+# iris/xmpp/xmpp-im/xmpp_discoinfotask.h
+# iris/xmpp/xmpp-im/xmpp_ibb.h
+# iris/xmpp/xmpp-im/xmpp_liverosteritem.h
+# iris/xmpp/xmpp-im/xmpp_httpauthrequest.h
+# iris/xmpp/xmpp-im/xmpp_pubsubitem.h
+# iris/xmpp/xmpp-im/xmpp_vcard.h
 iris/xmpp/xmpp-im/xmpp_client.h
 iris/xmpp/xmpp-im/s5b.h
-iris/xmpp/xmpp-im/xmpp_url.h
-iris/xmpp/xmpp-im/xmpp_chatstate.h
-iris/xmpp/xmpp-im/xmpp_task.h
-iris/xmpp/xmpp-im/xmpp_address.h
-iris/xmpp/xmpp-im/xmpp_agentitem.h
-iris/xmpp/xmpp-im/filetransfer.h
-iris/xmpp/xmpp-im/xmpp_status.h
-iris/xmpp/xmpp-im/xmpp_resourcelist.h
-iris/xmpp/xmpp-im/xmpp_liveroster.h
-iris/xmpp/xmpp-im/xmpp_resource.h
-iris/xmpp/xmpp-im/xmpp_rosteritem.h
-iris/xmpp/xmpp-im/xmpp_features.h
-iris/xmpp/xmpp-im/im.h
-iris/xmpp/sasl/digestmd5response.h
-iris/xmpp/sasl/digestmd5proplist.h
-iris/xmpp/sasl/plainmessage.h
-iris/xmpp/zlib/common.h
-iris/xmpp/zlib/zlibdecompressor.h
-iris/xmpp/zlib/zlibcompressor.h
-iris/xmpp/base64/base64.h
-iris/xmpp/xmpp-core/xmpp_stanza.h
-iris/xmpp/xmpp-core/simplesasl.h
+# iris/xmpp/xmpp-im/xmpp_url.h
+# iris/xmpp/xmpp-im/xmpp_chatstate.h
+# iris/xmpp/xmpp-im/xmpp_task.h
+# iris/xmpp/xmpp-im/xmpp_address.h
+# iris/xmpp/xmpp-im/xmpp_agentitem.h
+# iris/xmpp/xmpp-im/filetransfer.h
+# iris/xmpp/xmpp-im/xmpp_status.h
+# iris/xmpp/xmpp-im/xmpp_resourcelist.h
+# iris/xmpp/xmpp-im/xmpp_liveroster.h
+# iris/xmpp/xmpp-im/xmpp_resource.h
+# iris/xmpp/xmpp-im/xmpp_rosteritem.h
+# iris/xmpp/xmpp-im/xmpp_features.h
+# iris/xmpp/xmpp-im/im.h
+# iris/xmpp/sasl/digestmd5response.h
+# iris/xmpp/sasl/digestmd5proplist.h
+# iris/xmpp/sasl/plainmessage.h
+# iris/xmpp/zlib/common.h
+# iris/xmpp/zlib/zlibdecompressor.h
+# iris/xmpp/zlib/zlibcompressor.h
+# iris/xmpp/base64/base64.h
+# iris/xmpp/xmpp-core/xmpp_stanza.h
+# iris/xmpp/xmpp-core/simplesasl.h
 iris/xmpp/xmpp-core/securestream.h
-iris/xmpp/xmpp-core/protocol.h
-iris/xmpp/xmpp-core/compressionhandler.h
-iris/xmpp/xmpp-core/xmlprotocol.h
+# iris/xmpp/xmpp-core/protocol.h
+# iris/xmpp/xmpp-core/compressionhandler.h
+# iris/xmpp/xmpp-core/xmlprotocol.h
 iris/xmpp/xmpp-core/xmpp.h
 iris/xmpp/xmpp-core/xmpp_clientstream.h
 iris/xmpp/xmpp-core/xmpp_stream.h
-iris/xmpp/xmpp-core/td.h
-iris/xmpp/xmpp-core/parser.h
+# iris/xmpp/xmpp-core/td.h
+# iris/xmpp/xmpp-core/parser.h
 
 iris/jdns/qjdns.h
 
 
-iris/irisnet/noncore/stuntransaction.h
-iris/irisnet/noncore/stunbinding.h
-iris/irisnet/noncore/legacy/ndns.h
-iris/irisnet/noncore/legacy/safedelete.h
-iris/irisnet/noncore/legacy/servsock.h
-iris/irisnet/noncore/legacy/srvresolver.h
-iris/irisnet/noncore/stunallocate.h
-iris/irisnet/noncore/ice176.h
-iris/irisnet/noncore/processquit.h
-iris/irisnet/noncore/icelocaltransport.h
-iris/irisnet/corelib/irisnetglobal.h
+# iris/irisnet/noncore/stuntransaction.h
+# iris/irisnet/noncore/stunbinding.h
+# iris/irisnet/noncore/legacy/ndns.h
+# iris/irisnet/noncore/legacy/safedelete.h
+# iris/irisnet/noncore/legacy/servsock.h
+# iris/irisnet/noncore/legacy/srvresolver.h
+# iris/irisnet/noncore/stunallocate.h
+# iris/irisnet/noncore/ice176.h
+# iris/irisnet/noncore/processquit.h
+# iris/irisnet/noncore/icelocaltransport.h
+# iris/irisnet/corelib/irisnetglobal.h
 iris/irisnet/corelib/objectsession.h
-iris/irisnet/corelib/netavailability.h
+# iris/irisnet/corelib/netavailability.h
 iris/irisnet/corelib/netnames.h
-iris/irisnet/corelib/irisnetplugin.h
+# iris/irisnet/corelib/irisnetplugin.h
 iris/irisnet/corelib/netinterface.h
 iris/irisnet/corelib/jdnsshared.h
-#iris/irisnet/appledns/qdnssd.h
+# #iris/irisnet/appledns/qdnssd.h
 
-iris/xmpp/jingle/jinglecontent.h
-iris/xmpp/jingle/jinglesession.h
-iris/xmpp/jingle/jinglesessionmanager.h
-iris/xmpp/jingle/jingletasks.h
+# iris/xmpp/jingle/jinglecontent.h
+# iris/xmpp/jingle/jinglesession.h
+# iris/xmpp/jingle/jinglesessionmanager.h
+# iris/xmpp/jingle/jingletasks.h
 
 
  )
