diff -Nru -x '*~' qtsoap-2.7_1-opensource.orig/CMakeLists.txt qtsoap-2.7_1-opensource/CMakeLists.txt
--- qtsoap-2.7_1-opensource.orig/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ qtsoap-2.7_1-opensource/CMakeLists.txt	2013-01-09 14:40:46.361887600 +0100
@@ -0,0 +1,64 @@
+project( qtsoap )
+cmake_minimum_required( VERSION 2.8 )
+
+set(CMAKE_AUTOMOC ON)
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+
+find_package( Qt5Core REQUIRED )
+find_package( Qt5Network REQUIRED )
+find_package( Qt5Xml REQUIRED )
+
+include_directories(
+        ${CMAKE_CURRENT_SOURCE_DIR}  ${CMAKE_CURRENT_BINARY_DIR}
+        ${Qt5Core_INCLUDE_DIRS} ${Qt5Network_INCLUDE_DIR} )
+
+
+if (CMAKE_COMPILER_IS_GNUCXX)
+   set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2 -g")
+   set(CMAKE_CXX_FLAGS_RELEASE        "-O2 -DNDEBUG -DQT_NO_DEBUG")
+   set(CMAKE_CXX_FLAGS_DEBUG          "-g -O2 -fno-reorder-blocks -fno-schedule-insns -fno-inline")
+   set(CMAKE_CXX_FLAGS_DEBUGFULL      "-g3 -fno-inline")
+   set(CMAKE_CXX_FLAGS_PROFILE        "-g3 -fno-inline -ftest-coverage -fprofile-arcs")
+   set(CMAKE_C_FLAGS_RELWITHDEBINFO   "-O2 -g")
+   set(CMAKE_C_FLAGS_RELEASE          "-O2 -DNDEBUG -DQT_NO_DEBUG")
+   set(CMAKE_C_FLAGS_DEBUG            "-g -O2 -fno-reorder-blocks -fno-schedule-insns -fno-inline")
+   set(CMAKE_C_FLAGS_DEBUGFULL        "-g3 -fno-inline")
+   set(CMAKE_C_FLAGS_PROFILE          "-g3 -fno-inline -ftest-coverage -fprofile-arcs")
+
+   if (CMAKE_SYSTEM_NAME MATCHES Linux)
+     set ( CMAKE_C_FLAGS     "${CMAKE_C_FLAGS} -Wno-long-long -std=iso9899:1990 -Wundef -Wcast-align -Werror-implicit-function-declaration -Wchar-subscripts -Wall -W -Wpointer-arith -Wwrite-strings -Wformat-security -Wmissing-format-attribute -fno-common")
+     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wnon-virtual-dtor -Wno-long-long -ansi -Wundef -Wcast-align -Wchar-subscripts -Wall -W -Wpointer-arith -Wformat-security -fno-exceptions -fno-check-new -fno-common")
+     add_definitions (-D_BSD_SOURCE)
+   endif (CMAKE_SYSTEM_NAME MATCHES Linux)
+
+endif(CMAKE_COMPILER_IS_GNUCXX)
+
+set(CMAKE_DEBUG_POSTFIX "d")
+
+
+
+add_library(qtsoap SHARED src/qtsoap.cpp)
+set_target_properties( qtsoap PROPERTIES OUTPUT_NAME "QtSolutions_SOAP-2.7" DEFINE_SYMBOL "QT_QTSOAP_EXPORT" )
+target_link_libraries(qtsoap ${Qt5Core_LIBRARIES} ${Qt5Xml_LIBRARIES} ${Qt5Network_LIBRARIES})
+
+install(TARGETS qtsoap
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin
+  ARCHIVE DESTINATION lib
+)
+
+install(FILES src/qtsoap.h DESTINATION include)
+install(FILES src/QtSoapArray
+              src/QtSoapArrayIterator
+              src/QtSoapHttpTransport
+              src/QtSoapMessage
+              src/QtSoapNamespaces
+              src/QtSoapQName
+              src/QtSoapSimpleType
+              src/QtSoapStruct
+              src/QtSoapStructIterator
+              src/QtSoapType
+              src/QtSoapTypeConstructor
+              src/QtSoapTypeConstructorBase
+              src/QtSoapTypeFactory
+        DESTINATION include/QtSoap)
diff -Nru -x '*~' qtsoap-2.7_1-opensource.orig/src/qtsoap.cpp qtsoap-2.7_1-opensource/src/qtsoap.cpp
--- qtsoap-2.7_1-opensource.orig/src/qtsoap.cpp	2013-01-09 14:42:26.886232400 +0100
+++ qtsoap-2.7_1-opensource/src/qtsoap.cpp	2013-01-09 14:36:34.187920400 +0100
@@ -3157,7 +3157,7 @@
 {
     QNetworkRequest networkReq;
     networkReq.setHeader(QNetworkRequest::ContentTypeHeader, QLatin1String("text/xml;charset=utf-8"));
-    networkReq.setRawHeader("SOAPAction", soapAction.toAscii());
+    networkReq.setRawHeader("SOAPAction", soapAction.toLatin1());
     url.setPath(path);
     networkReq.setUrl(url);
 
