diff -Nru libkdcraw-4.8.0.orig/libraw/src/libraw_datastream.cpp libkdcraw-4.8.0/libraw/src/libraw_datastream.cpp
--- libkdcraw-4.8.0.orig/libraw/src/libraw_datastream.cpp	2012-01-18 19:49:41.000000000 +0100
+++ libkdcraw-4.8.0/libraw/src/libraw_datastream.cpp	2012-01-25 22:08:44.360600000 +0100
@@ -1,3 +1,6 @@
+#ifdef _WIN32
+#include <cmath>
+#endif
 #define LIBRAW_LIBRARY_BUILD
 #include "libraw/libraw_types.h"
 #include "libraw/libraw.h"
