diff -Nru kdevelop-4.1.1.orig/CMakeLists.txt kdevelop-4.1.1/CMakeLists.txt
--- kdevelop-4.1.1.orig/CMakeLists.txt	2010-11-25 22:14:04.000000000 +0100
+++ kdevelop-4.1.1/CMakeLists.txt	2010-12-15 18:43:56.730354800 +0100
@@ -11,6 +11,10 @@
 set(KDE_MIN_VERSION "4.3.0")
 find_package(KDE4 4.3.0 REQUIRED)
 
+if(MSVC)
+    add_definitions(-wd4996)        # remove if you ever want to start developing kdevelop
+endif(MSVC)
+
 if( ${KDE_VERSION} VERSION_GREATER 4.5.63 )
     message(FATAL_ERROR "KDevelop doesn't work (yet) with kdelibs 4.5.64 or later, please use the movingranges branch (http://gitorious.org/+kdevelop-developers/kdevelop/devel-kdevelop)")
 endif(${KDE_VERSION} VERSION_GREATER 4.5.63)
diff -Nru kdevelop-4.1.1.orig/languages/cpp/codecompletion/context.cpp kdevelop-4.1.1/languages/cpp/codecompletion/context.cpp
--- kdevelop-4.1.1.orig/languages/cpp/codecompletion/context.cpp	2010-11-25 22:07:30.000000000 +0100
+++ kdevelop-4.1.1/languages/cpp/codecompletion/context.cpp	2010-12-15 18:48:36.408351400 +0100
@@ -21,6 +21,8 @@
 #include <ktexteditor/document.h>
 #include <klocalizedstring.h>
 
+#include <iterator>
+
 #include <interfaces/idocumentcontroller.h>
 
 #include <language/interfaces/ilanguagesupport.h>
diff -Nru kdevelop-4.1.1.orig/languages/cpp/cppduchain/declarationbuilder.cpp kdevelop-4.1.1/languages/cpp/cppduchain/declarationbuilder.cpp
--- kdevelop-4.1.1.orig/languages/cpp/cppduchain/declarationbuilder.cpp	2010-11-25 22:07:30.000000000 +0100
+++ kdevelop-4.1.1/languages/cpp/cppduchain/declarationbuilder.cpp	2010-12-15 18:30:18.631562200 +0100
@@ -23,6 +23,7 @@
 
 #include <QByteArray>
 #include <typeinfo>
+#include <iterator>
 
 #include "templatedeclaration.h"
 
diff -Nru kdevelop-4.1.1.orig/projectmanagers/cmake/CMakeLists.txt kdevelop-4.1.1/projectmanagers/cmake/CMakeLists.txt
--- kdevelop-4.1.1.orig/projectmanagers/cmake/CMakeLists.txt	2010-11-25 22:07:30.000000000 +0100
+++ kdevelop-4.1.1/projectmanagers/cmake/CMakeLists.txt	2010-12-15 18:34:44.060743900 +0100
@@ -72,6 +72,7 @@
                         ${QT_QTSCRIPT_LIBRARY} ${KDEVPLATFORM_LANGUAGE_LIBRARIES}
                         )
 
+kde4_add_kcfg_files( cmakemanager_SRCS cmakeconfig.kcfgc )
 kde4_add_ui_files( cmakemanager_SRCS ${cmakemanager_UI} )
 kde4_add_plugin( kdevcmakemanager ${cmakemanager_SRCS} )
 target_link_libraries( kdevcmakemanager ${KDE4_KDEUI_LIBS} ${KDE4_KIO_LIBS}
